(window.webpackJsonp=window.webpackJsonp||[]).push([[34],{351:function(e,a,t){"use strict";t.r(a);var i=t(7),n=Object(i.a)({},(function(){var e=this,a=e._self._c;return a("ContentSlotsDistributor",{attrs:{"slot-key":e.$parent.slotKey}},[a("p",[e._v("2024-06-25")]),e._v(" "),a("p",[e._v("Determinants of gastric cancer immune escape identified from non-coding immunelandscape quantitative trait loci")]),e._v(" "),a("p",[e._v("胃癌免疫逃逸的决定因素是从非编码免疫图谱中确定的数量性状位点")]),e._v(" "),a("h3",{attrs:{id:"背景"}},[a("a",{staticClass:"header-anchor",attrs:{href:"#背景"}},[e._v("#")]),e._v(" 背景")]),e._v(" "),a("p",[e._v("eQTL的介绍（来自Kimi）")]),e._v(" "),a("p",[e._v("eQTL，全称为expression quantitative trait loci，即表达数量性状基因座位。它是指基因组中那些能够影响基因表达量的遗传位点，通常是单核苷酸多态性（SNP）。eQTL分析的目的是研究遗传变异（如SNP）与基因表达水平之间的关联，从而揭示基因表达调控的遗传基础。")]),e._v(" "),a("p",[e._v("原理：")]),e._v(" "),a("ol",[a("li",[a("strong",[e._v("遗传变异与基因表达的关联")]),e._v("：eQTL分析通过检测个体的基因型（通常是SNP）和相应基因的表达水平，来寻找两者之间的关联。")]),e._v(" "),a("li",[a("strong",[e._v("线性回归模型")]),e._v("：eQTL分析的核心算法是线性回归，类似于GWAS（全基因组关联分析），但将基因表达量作为因变量，遗传变异（如SNP）作为自变量。")])]),e._v(" "),a("p",[e._v("分类：")]),e._v(" "),a("ul",[a("li",[a("strong",[e._v("cis-eQTL")]),e._v("：顺式作用eQTL，指eQTL位点位于调控基因的附近区域，通常在基因的上下游1Mb内。")]),e._v(" "),a("li",[a("strong",[e._v("trans-eQTL")]),e._v("：反式作用eQTL，指eQTL位点距离调控基因较远，甚至可能位于不同的染色体上。")])]),e._v(" "),a("p",[e._v("作用：")]),e._v(" "),a("ol",[a("li",[a("strong",[e._v("疾病关联研究")]),e._v("：通过eQTL分析，可以识别与疾病相关的基因表达变化背后的遗传变异，有助于理解疾病的分子机制。")]),e._v(" "),a("li",[a("strong",[e._v("药物反应研究")]),e._v("：eQTL可以帮助研究者了解个体对药物反应差异的遗传基础。")]),e._v(" "),a("li",[a("strong",[e._v("遗传学研究")]),e._v("：eQTL为研究基因功能和基因网络提供了重要信息。")]),e._v(" "),a("li",[a("strong",[e._v("生物育种")]),e._v("：在农业生物技术中，eQTL分析有助于识别影响作物产量、抗病性等性状的基因，进而进行分子标记辅助选择。")])]),e._v(" "),a("p",[e._v("数据库和工具：")]),e._v(" "),a("ul",[a("li",[a("strong",[e._v("GTEx数据库")]),e._v("：提供了大量的eQTL数据，可以用于跨组织的eQTL分析。")]),e._v(" "),a("li",[a("strong",[e._v("Matrix eQTL")]),e._v("：一个R包，用于eQTL分析，考虑了基因型、表达水平、基因位置、SNP位置和协变量。")])]),e._v(" "),a("p",[e._v("eQTL分析是一个强大的工具，它帮助科学家们在分子层面上理解遗传变异如何影响复杂的生物学性状和疾病。通过这种分析，可以更好地理解基因表达的调控机制，为遗传病的诊断、治疗和预防提供科学依据。")]),e._v(" "),a("p",[e._v("非编码区突变，预测患者预后的潜力")]),e._v(" "),a("p",[e._v("3'-UTR，种系和体细胞，单核苷酸变异/SNP，短插入-缺失/indels")]),e._v(" "),a("blockquote",[a("p",[e._v("3′-UTR germline and somatic singlenucleotide variants (SNVs) as well as short insertions-deletions (indels)")])]),e._v(" "),a("p",[e._v("QTL，是一种分析？数量性状位点，和数量变化有关的位点？")]),e._v(" "),a("p",[e._v("massively parallel reporter assay (MPRA)，大规模平行报告实验\nMPRA分析："),a("a",{attrs:{href:"https://github.com/ivlachos/3UTR",target:"_blank",rel:"noopener noreferrer"}},[e._v("ivlachos/3UTR (github.com)"),a("OutboundLink")],1)]),e._v(" "),a("p",[e._v("MPRA介绍（来自kimi）\nMassively Parallel Reporter Assay（MPRA）是一种高通量的实验技术，它能够同时评估基因组中数千个调控元件的活性。这项技术主要应用于基因组学研究，特别是在识别和理解基因表达调控元件方面。")]),e._v(" "),a("p",[e._v("实验原理：\nMPRA的原理基于报告基因的表达来衡量调控元件的功能。在MPRA中，研究者将候选的调控序列（如启动子、增强子等）与一个报告基因（通常是荧光素酶基因）结合，并在这些序列上引入具有唯一标识的条形码。这些构建体随后被转染到细胞中。通过比较转染前后（即DNA和mRNA水平）的条形码读数，可以评估每个调控序列的活性。这种方法允许研究者通过二代测序技术来高通量地分析大量调控序列的功能。")]),e._v(" "),a("p",[e._v("应用：")]),e._v(" "),a("ol",[a("li",[a("strong",[e._v("基因组顺式调控元件的鉴定")]),e._v("：MPRA可用于识别具有调控活性的DNA序列，如增强子和抑制子。")]),e._v(" "),a("li",[a("strong",[e._v("功能性变异的鉴定")]),e._v("：通过MPRA可以研究特定遗传变异如何影响基因表达的调控。")]),e._v(" "),a("li",[a("strong",[e._v("转录后调控研究")]),e._v("：MPRA不仅局限于转录前调控，还可以研究转录后调控对表型的影响。")]),e._v(" "),a("li",[a("strong",[e._v("疾病相关基因的识别")]),e._v("：通过分析与疾病相关的调控元件，MPRA有助于理解疾病发生的分子机制。")]),e._v(" "),a("li",[a("strong",[e._v("药物响应研究")]),e._v("：MPRA可以用于研究药物如何影响基因表达的调控。")])]),e._v(" "),a("p",[e._v("PD-L1 3‘-UTR的一个普通的突变，会影响miR570结合，然后导致表达升高")]),e._v(" "),a("blockquote",[a("p",[e._v("For instance, a common somatic mutation in the PD-L1 3′-UTR has been shown to disrupt miR570 binding leading to increased expression")])]),e._v(" "),a("p",[e._v("MPRA技术的一个关键优势是其高通量特性，这使得研究者能够在一个实验中测试大量的调控序列，从而加速了基因调控网络的发现和理解。此外，MPRA的设计和上下文依赖性也受到了系统性评估，这有助于优化实验设计，提高结果的准确性和再现性。")]),e._v(" "),a("p",[e._v("转录组范围内，无偏倚的鉴定出，和肿瘤中顺式基因表达、免疫表型，有关的3'-UTR变化！")]),e._v(" "),a("h3",{attrs:{id:"r1、3-utr-landscape"}},[a("a",{staticClass:"header-anchor",attrs:{href:"#r1、3-utr-landscape"}},[e._v("#")]),e._v(" R1、3‘-UTR landscape")]),e._v(" "),a("p",[e._v("Characterization of the somatic and germline 3′-UTR variant landscape in gastric adenocarcinoma")]),e._v(" "),a("p",[e._v("分析流程"),a("br"),e._v(" "),a("img",{attrs:{src:"https://s2.loli.net/2024/06/25/7SkjaPleT9Rzxwc.png",alt:"image.png|600"}})]),e._v(" "),a("p",[e._v("GATK，Strelka2")]),e._v(" "),a("h3",{attrs:{id:"r2、验证"}},[a("a",{staticClass:"header-anchor",attrs:{href:"#r2、验证"}},[e._v("#")]),e._v(" R2、验证")]),e._v(" "),a("p",[e._v("High-throughput capture of known functional variants in PD-L1 3′-UTR")]),e._v(" "),a("p",[e._v("高哦通量的方法，也捕获到了较少数量3’-UTR SNVs的功能影响（和之前报道过的突变影响一致） -- 携带 alternative allele 的，PD-L1表达量就会高一些（参考理解："),a("a",{attrs:{href:"https://gwaslab.org/2021/08/21/allele-concept/",target:"_blank",rel:"noopener noreferrer"}},[e._v("major/minor/reference/alternative/risk/effect allele 概念解析 – GWASLab – GWAS实验室"),a("OutboundLink")],1),e._v(" ）")]),e._v(" "),a("p",[a("img",{attrs:{src:"https://s2.loli.net/2024/06/25/FAOP32pZGiYeWhD.png",alt:"image.png|400"}})]),e._v(" "),a("p",[e._v("Normalized PD-L1 expression (log2(read count+1)) in patients carrying the alternative allele (Alt) vs patients homozygous for the reference allele (Ref).  "),a("code",[e._v("*P <0.05,**P < 0.01, ***P <0.001,****P <0.0001")])]),e._v(" "),a("h3",{attrs:{id:"r3、重要作用-调控表达"}},[a("a",{staticClass:"header-anchor",attrs:{href:"#r3、重要作用-调控表达"}},[e._v("#")]),e._v(" R3、重要作用（调控表达）")]),e._v(" "),a("p",[e._v("Prioritization of 3′-UTR germline variants and somatic mutations controlling cis gene expression in gastric adenocarcinoma")]),e._v(" "),a("p",[e._v("胃腺癌中控制cis基因表达的3′- UTR种系变异和体细胞突变的优先性")]),e._v(" "),a("p",[e._v("MAF，minor allele frequency，最小等位基因频率")]),e._v(" "),a("p",[e._v("the highest percentage (60%) of all significant "),a("code",[e._v("cis-eQTLs")]),e._v(" in protein-coding genes reside in the 3′-UTR genic region")]),e._v(" "),a("p",[e._v("Comparing the number of 3′-UTR variants to the averagerelativelengthofthe3′-UTR in protein coding genes revealed a significant enrichment")]),e._v(" "),a("p",[a("img",{attrs:{src:"https://s2.loli.net/2024/06/25/1FEd9zgKGveLXym.png",alt:"image.png|500"}})]),e._v(" "),a("p",[e._v("3'-UTR，编码蛋白质的cis-eQTLs数量不仅是最多的，通过碱基长度校正后的数量也是最多的！")]),e._v(" "),a("p",[e._v("alternative，可变的，可选择的，可替代的，不同寻常的；可选择的事物，可供选择的解决办法\nisoform，亚型，异构体")]),e._v(" "),a("p",[e._v("3‘-UTR ceis-eQTLs 和 miRNA，RBP binding sites，重叠比例很大！\n然而，通过对随机采样的3 ' UTR区域进行基于排列的测试，仅预测的miRNA结合位点而不是RBPs被确定为显著富集在eQTL位点中。")]),e._v(" "),a("p",[e._v("最大显著的3′- UTR cis - e QTL的GSEA富集到免疫相关通路，表明3′- UTR变异可能对胃癌免疫表型产生影响。")]),e._v(" "),a("h3",{attrs:{id:"r4、mpra实验验证"}},[a("a",{staticClass:"header-anchor",attrs:{href:"#r4、mpra实验验证"}},[e._v("#")]),e._v(" R4、MPRA实验验证")]),e._v(" "),a("p",[a("code",[e._v("Massively parallel reporter assay")]),e._v(" validation of cis-eQTLs residing in cancer immunoediting genes")]),e._v(" "),a("p",[e._v("癌症免疫编辑基因中，cis-eQTLs的，"),a("code",[e._v("大规模平行报告基因验证")])]),e._v(" "),a("p",[a("img",{attrs:{src:"https://s2.loli.net/2024/06/25/CGMW9jcNInUqgPF.png",alt:"image.png|700"}}),a("br"),e._v("\n一个包含着"),a("code",[e._v("参考条形码")]),e._v("和"),a("code",[e._v("替代等位基因")]),e._v("的"),a("code",[e._v("报告质粒文库")]),e._v("，将749个eQTL突变导入到两个肿瘤细胞系，然后对转染或的细胞进行RNA测序，根据"),a("code",[e._v("参考条形码")]),e._v("的定量来分析对转录后水平的影响。")]),e._v(" "),a("blockquote",[a("p",[e._v("Briefly, "),a("code",[e._v("a reporter plasmid library")]),e._v(" containing "),a("strong",[e._v("barcoded reference")]),e._v(" and "),a("strong",[e._v("alternative alleles")]),e._v(" for the 749 eQTL variants was transfected into AGS and SNU719 cells. The effect of the variant on post-transcriptional expression of the reporter was assessed by barcode quantification from amplicon sequencing of RNA extracted from transfected cells.")])]),e._v(" "),a("p",[e._v("两种细胞系的实验结果是比较相似的，但是"),a("code",[e._v("significant causative effect")]),e._v("，"),a("code",[e._v("causal regulatory variants")]),e._v("，没见过，不理解呢。")]),e._v(" "),a("h3",{attrs:{id:"r5、重要作用-免疫相关"}},[a("a",{staticClass:"header-anchor",attrs:{href:"#r5、重要作用-免疫相关"}},[e._v("#")]),e._v(" R5、重要作用（免疫相关）")]),e._v(" "),a("p",[e._v("Uncovering transcriptome-wide 3′-UTR variants regulating the gastric adenocarcinoma immune landscape")]),e._v(" "),a("p",[e._v("3′- UTR区不仅调控基因的表达，还可以影响mRNA的定位、蛋白质-蛋白质相互作用以及其他转录后、翻译和翻译后的功能。因此，3′- UTR变异体可以 "),a("code",[e._v("独立于")]),e._v(" 其对表达的影响而影响癌症中的免疫表型。")]),e._v(" "),a("p",[e._v("immune landscape (il)QTL analysis，免疫景观的定量特征位点分析")]),e._v(" "),a("p",[e._v("和eQTL的结果类似，大多数的ilQTL在3‘-UTR，然后很大比例的ilQTL是在miRNA/RBP binding regions")]),e._v(" "),a("p",[e._v("permutation analysis，置换分析？？")]),e._v(" "),a("p",[e._v("关注到 CD8+ T cells，其特定的QTLs，富集到了ribosome和ribosomal subunits（核糖体和核糖体亚基）")]),e._v(" "),a("h3",{attrs:{id:"r6、上下游调节"}},[a("a",{staticClass:"header-anchor",attrs:{href:"#r6、上下游调节"}},[e._v("#")]),e._v(" R6、上下游调节")]),e._v(" "),a("p",[e._v("Functional 3′-UTR cis-eQTL and ilQTL variants in ADAR")]),e._v(" "),a("p",[e._v("ADAR中，功能性3’-UTR cis-eQTL和ilQTL突变")]),e._v(" "),a("p",[e._v("聚焦到ADAR -- 在所有的高通量研究中，在ADAR中，鉴定出来特定的功能突变")]),e._v(" "),a("blockquote",[a("p",[e._v("Significant functional variants were identified in all our high throughput investigations (eQTLs, ilQTLs, MPRA validated variants) in the Adenosine Deaminase RNA Specific (ADAR) gene.")])]),e._v(" "),a("p",[e._v("Adenosine Deaminase RNA Specific (ADAR) gene，腺苷脱氨酶RNA特异性基因")]),e._v(" "),a("blockquote",[a("p",[e._v("ADAR编码一种在RNA中催化 A-to-I 编辑的酶，已被证实在多种癌症类型中促进癌症特征，包括乳腺癌、甲状腺和胃恶性肿瘤")])]),e._v(" "),a("p",[e._v("ADAR的下调也被证明可以特异性地诱导胃癌中的炎症信号")]),e._v(" "),a("p",[e._v("这是研究到点突变的效应了呀！")]),e._v(" "),a("p",[e._v("TARDBP，通过结合典型的富含GU的基序或非典型序列来调节基因表达、前体mRNA编辑、mRNA定位和microRNA加工的能力已经被研究，3 ' UTRs是常见的靶位点")]),e._v(" "),a("p",[e._v("TARDBP已被证明在肝癌和白血病细胞系模型中直接调控ADAR1的表达；TCGA的胃癌患者中，两个基因也表现出较强的相关性。")]),e._v(" "),a("h3",{attrs:{id:"r7、临床相关性-预测价值"}},[a("a",{staticClass:"header-anchor",attrs:{href:"#r7、临床相关性-预测价值"}},[e._v("#")]),e._v(" R7、临床相关性 - 预测价值")]),e._v(" "),a("p",[e._v("3′-UTR variants as an effective means for immunotherapy response prediction")]),e._v(" "),a("p",[e._v("3′- UTR变异体作为免疫治疗应答预测的有效手段")]),e._v(" "),a("p",[e._v("training set，test set，orthogonal test set")]),e._v(" "),a("p",[e._v("Polygenic Risk Score (PRS)，多基因风险评分（像是多因素回归分析呢？）")]),e._v(" "),a("p",[a("img",{attrs:{src:"https://s2.loli.net/2024/06/26/sS4XKUrDgydjNxf.png",alt:"image.png|700"}})]),e._v(" "),a("p",[e._v("预测免疫检查点抑制治疗的效果，比PD-L1表达量的预测效果还好（AUC更高）")]),e._v(" "),a("p",[e._v("比 tumor mutational burden (TMB) or microsatellite instability (MSI)，预测效果还好")]),e._v(" "),a("p",[e._v("ps：比传统的指标好，然后合并在一起再预测一次")]),e._v(" "),a("h3",{attrs:{id:"discussion"}},[a("a",{staticClass:"header-anchor",attrs:{href:"#discussion"}},[e._v("#")]),e._v(" Discussion")]),e._v(" "),a("p",[e._v("重新描述了一遍本文所做的工作")]),e._v(" "),a("h3",{attrs:{id:"methods"}},[a("a",{staticClass:"header-anchor",attrs:{href:"#methods"}},[e._v("#")]),e._v(" Methods")]),e._v(" "),a("p",[a("strong",[e._v("RNAseq variant calling")])]),e._v(" "),a("p",[e._v("Raw RNAseq data for 375 TCGA STAD primary cancer and 40 matched normal samples were obtained from "),a("code",[e._v("Genomic Data Commons (GDC)")]),e._v(" following NIH dbGAP approval75. Reads were mapped against the "),a("code",[e._v("human genome (hg38)")]),e._v(" using "),a("code",[e._v("STAR")]),e._v("76. Mapped reads were deduplicated and short variants/indels were called using "),a("code",[e._v("Mutect2")]),e._v(" following "),a("code",[e._v("GATK")]),e._v(" (v4.1.4.0) best practices34. The Mutect2 output was converted to a gVCF format by using region coverage statistics. Since Mutect2 cannot perform genotype calling and does not distinguish between homozygous reference and no-call regions, HaplotypeCaller was also run in parallel by following GATK best practices for RNA77. For samples lacking a Mutect2 call at a specific variant position, HaplotypeCaller was used to distinguish whether the lack of a Mutect2 call was because of no coverage in that region or a homozygous reference genotype. A mutation was characterized as likely somatic by calculating the posterior probability of the event, while using variant call statistics, clonality in tumor samples, matched healthy tissues, and gnomAD variants78 as priors. Calls were filtered using the "),a("code",[e._v("FilterMutect2")]),e._v(" tags “base_qual”, “map_qual”, “n_ratio” and “slippage”.Onlybiallelicvariants present in at least 5 out of the 375 (>1.3%) samples were pursued further. Variants that were present in at least 2 out of the 40 matched normal samples comprised the Panel of Normals (PoN). "),a("code",[e._v("Strelka")]),e._v("236 was also run following the same preprocessing steps as for GATK callers.")]),e._v(" "),a("p",[a("strong",[e._v("Germline variant calling")])]),e._v(" "),a("p",[e._v("Affymetrix SNP array 6.0 data from blood samples for all TCGA STAD patients analyzed in this study were downloaded from GDC. The SNP array data were converted to a VCF format using "),a("code",[e._v("birdseed2vcf")]),e._v(" (https:// github.com/ding-lab/birdseed2vcf), and then whole-genome variant calls were imputed using "),a("code",[e._v("Minimac4")]),e._v(" and the 1000 Genomes Phase 3 project as a reference on the Michigan Imputation Server79. Somatic variants were further intersected against GTEx RNA editing events from REDIportal V2.035.")]),e._v(" "),a("p",[a("strong",[e._v("eQTL analysis")])]),e._v(" "),a("p",[e._v("Gene-level expression in TCGA STAD samples was calculated using "),a("code",[e._v("Salmon v0.91")]),e._v(" and "),a("code",[e._v("Ensembl genome annotation v77")]),e._v("80,81. A linear model was utilized to call eQTLs with "),a("code",[e._v("FastQTL")]),e._v("39 following best practices82. In brief, gene expression across libraries was normalized using trimmed mean of m-values as implemented in "),a("code",[e._v("edgeR")]),e._v("83. Genes were selected based on an expression threshold of 1 read in at least 80% of the samples. An inverse quantile normal transformation was performed on the expression values prior to their inclusion into the linear model. "),a("code",[e._v("Mutect2/HaplotypeCaller alternative allele dosage")]),e._v(" was utilized as genotype input, while age at diagnosis, sex and the top 5 genetic principal components (gPCs) and expression surrogate variables (SVs) were included as covariates39,84, as follows: gene expression "),a("code",[e._v("~Alt Dosage+age +sex+gPC1+gPC2+gPC3+gPC4+gPC5+SV1+SV2+SV3+SV4+SV5")]),e._v(". Genetic PCs were calculated using "),a("code",[e._v("SmartPCA")]),e._v("85 on WES-derived germline variants from the same TCGA STAD patients, obtained from Huang et al.86.")]),e._v(" "),a("p",[e._v("Following the nominal run of FastQTL39,foreachgeneweselected variants that mapped in cis (within the genomic coordinates of the gene) and calculated a per-gene FDR-adjusted q value for each variant. Genes that contained at least one variant with a q value lower than 0.05 were defined as eGenes. A nominal threshold was defined for each eGene based on the highest nominal p value that corresponded to a q value < 0.05. The distribution of the threshold nominal p value in all eGenes is shown in Supplementary Fig. 10. The median threshold nominal p value (1e-7) was applied horizontally, across all eGenes, to describe significant calls (final cutoff: q value < 0.05 and nominal p value < 1e-7). eQTL calls were mapped to transcript annotations ("),a("code",[e._v("Gencode v32")]),e._v(") and relative genomic locations (5′UTR, CDS, 3′-UTR) were assigned using "),a("code",[e._v("annotatr")]),e._v("87.OnlyeQTLs with significant cis effects were retained for further analysis. Variant annotation for potential overlap with post-transcriptional regulatory regions was performed using the "),a("code",[e._v("GenomicRanges package")]),e._v(" in R (v1.38.0)88. Experimentally supported miRNA binding site coordinates were obtained from "),a("code",[e._v("TarBase")]),e._v("47, predicted miRNA binding sites were acquired from "),a("code",[e._v("microTCDS")]),e._v("48, and CLIP-based predictions of RBP binding sites were obtained from the "),a("code",[e._v("POSTAR2 database")]),e._v("49.")]),e._v(" "),a("p",[a("strong",[e._v("ilQTL analysis")])]),e._v(" "),a("p",[e._v("Immune data per sample were obtained from the CRI iAtlas project57. QTL analysis was performed with FastQTL, using the same genotype and covariate data as above, while using quantilenormalized immune profile estimates as phenotypes. ilQTL selection as well as genomic and regulatory annotation were performed as for eQTLs.")]),e._v(" "),a("p",[a("strong",[e._v("MPRA analysis")])]),e._v(" "),a("p",[e._v("MPRA analysis was performed similarly to Mattioli et al.104. Briefly, barcode counts were calculated from raw reads and then normalized per sample based on sequencing depth. For each sample, a barcode RNA to DNA ratio was calculated by dividing the barcode counts in each replicate to that in the plasmid pool library. The RNA to DNA ratios were then log-transformed and quantile normalized across samples. A two-sided Wilcoxon test was performed to compare barcode count ratios between reference and alternative allele oligos in each replicate. To combine replicate p-values, the Stouffer’smethod was used, and Benjamini-Hochberg false discovery rate (FDR) correction was applied. Each oligo was represented by 10 barcodes, so to obtain a per-oligo activity in each sample, the median activity was calculated. Fold-change was defined as the ratio of the alternative to the reference allele median activity.")]),e._v(" "),a("h3",{attrs:{id:"一些想法"}},[a("a",{staticClass:"header-anchor",attrs:{href:"#一些想法"}},[e._v("#")]),e._v(" 一些想法")]),e._v(" "),a("p",[e._v("这篇文章，给我一种老旧的感觉，是不是除了数据“比较旧”之外，画的图也比较“老土”吧，数据量不是那么大，展示的就没有单细胞那么丰富，但是呢，又是研究的基因组“点突变”情况，所以用到了类似IGV这样子的工具，就导致有种，老版软件的感觉。。。不过，人家最近发的NC，从结果上来看，效果还是可以的，有点用处。")]),e._v(" "),a("p",[e._v("嗯，算是看的半懂，似懂非懂的感觉，暂且先放这儿，需要再仔细看看。")])])}),[],!1,null,null,null);a.default=n.exports}}]);