(window.webpackJsonp=window.webpackJsonp||[]).push([[37],{354:function(s,a,e){"use strict";e.r(a);var n=e(7),t=Object(n.a)({},(function(){var s=this,a=s._self._c;return a("ContentSlotsDistributor",{attrs:{"slot-key":s.$parent.slotKey}},[a("p",[a("a",{attrs:{href:"https://zhuanlan.zhihu.com/p/667672976",target:"_blank",rel:"noopener noreferrer"}},[s._v("测序须知"),a("OutboundLink")],1)]),s._v(" "),a("p",[s._v("日后肯定还需要再用的，所以这一部分，还是得单独开一个笔记记录这个过程！")]),s._v(" "),a("p",[s._v("转录组实战")]),s._v(" "),a("p",[s._v("分析流程及所需软件")]),s._v(" "),a("p",[s._v("上有流程：")]),s._v(" "),a("ul",[a("li",[s._v("创建单独的分析环境：miniconda3")]),s._v(" "),a("li",[s._v("数据下载和格式转化：sra-tools（sratoolkit）")]),s._v(" "),a("li",[s._v("质控清洗：fastqc，multiqc，trim-glore（trimmomatic，cutadapt，fastp")]),s._v(" "),a("li",[s._v("比对：hisat2，subread，samtools，salmon（star，bwa，bowtie2，tophat")]),s._v(" "),a("li",[s._v("计数：featureCounts（整合在subread）（htseq，bedtools，deeptools")])]),s._v(" "),a("p",[s._v("下游流程")]),s._v(" "),a("ul",[a("li",[s._v("DEG分析：DESeq2，edgeR，limma")]),s._v(" "),a("li",[s._v("差异基因注释：GO，KEGG")]),s._v(" "),a("li",[s._v("GSEA，基因集富集分析")]),s._v(" "),a("li",[s._v("GSVA，基因集变异分析")]),s._v(" "),a("li",[s._v("PPI，蛋白互作网络")]),s._v(" "),a("li",[s._v("WGCNA，加权基因共表达网络分析")])]),s._v(" "),a("p",[s._v("一部分代码参考")]),s._v(" "),a("h2",{attrs:{id:"数据下载-sra-fastq"}},[a("a",{staticClass:"header-anchor",attrs:{href:"#数据下载-sra-fastq"}},[s._v("#")]),s._v(" 数据下载，SRA，fastq")]),s._v(" "),a("div",{staticClass:"language- line-numbers-mode"},[a("pre",{pre:!0,attrs:{class:"language-text"}},[a("code",[s._v("位置：NCBI，UCSC，Ensembol，GEO\n\ncat ~/idname | while read id;\ndo (prefetch -O ~/outfile $id &) done;\nprefetch SRAXXXXX -O /path/to/sra_file\n\n>>> hohup bash prefetch.sh >log_prefetch 2>&1 &\n\nfastq-dump \n\t--split-3 \n\t--gzip -e 12 path/to/SRA_file.sra \n\t-o ~/path/to/output_fastq_dir    \n--split-3 把双端拆分成两个文件，单端并不会\n-e 设置线程  \n--gzip 输出gz格式, 节省空间,也不会给后续比对软件造成压力\n\nls ~/sra/*.sra | while read id;\n\tdo fasterq-dump -3 -e 12 -O ~/fastq_file;\n\tpigz -p 12 ~/fastq_file/*q;\n\tdone  \n\n-3 把双端拆成两个文件，-e 设置线程 ；-p 显示进度条\n")])]),s._v(" "),a("div",{staticClass:"line-numbers-wrapper"},[a("span",{staticClass:"line-number"},[s._v("1")]),a("br"),a("span",{staticClass:"line-number"},[s._v("2")]),a("br"),a("span",{staticClass:"line-number"},[s._v("3")]),a("br"),a("span",{staticClass:"line-number"},[s._v("4")]),a("br"),a("span",{staticClass:"line-number"},[s._v("5")]),a("br"),a("span",{staticClass:"line-number"},[s._v("6")]),a("br"),a("span",{staticClass:"line-number"},[s._v("7")]),a("br"),a("span",{staticClass:"line-number"},[s._v("8")]),a("br"),a("span",{staticClass:"line-number"},[s._v("9")]),a("br"),a("span",{staticClass:"line-number"},[s._v("10")]),a("br"),a("span",{staticClass:"line-number"},[s._v("11")]),a("br"),a("span",{staticClass:"line-number"},[s._v("12")]),a("br"),a("span",{staticClass:"line-number"},[s._v("13")]),a("br"),a("span",{staticClass:"line-number"},[s._v("14")]),a("br"),a("span",{staticClass:"line-number"},[s._v("15")]),a("br"),a("span",{staticClass:"line-number"},[s._v("16")]),a("br"),a("span",{staticClass:"line-number"},[s._v("17")]),a("br"),a("span",{staticClass:"line-number"},[s._v("18")]),a("br"),a("span",{staticClass:"line-number"},[s._v("19")]),a("br"),a("span",{staticClass:"line-number"},[s._v("20")]),a("br"),a("span",{staticClass:"line-number"},[s._v("21")]),a("br"),a("span",{staticClass:"line-number"},[s._v("22")]),a("br")])]),a("h2",{attrs:{id:"质控"}},[a("a",{staticClass:"header-anchor",attrs:{href:"#质控"}},[s._v("#")]),s._v(" 质控")]),s._v(" "),a("div",{staticClass:"language- line-numbers-mode"},[a("pre",{pre:!0,attrs:{class:"language-text"}},[a("code",[s._v("工具：fastqc，multiqc，trimmomatic\n\nls ~/fastq_file/*.fq | xargs fastqc -t 12 -o ~/qc_file # 全部跑一遍QC\nmultiqc ~/qc_dir/ # 应该是合并qc结果\n\nls ~/fastq/*_1*> 1\nls ~/fastq/*_2* > 2\npaste 1 2 > config\ncat config | while read id\ndo\narr = ($id) fq1 = ${arr[0]} fq2 = ${arr[1]}\ntrim_galore -j 4 -q 25 --phred33 --length 35 --strigency 3 --paired --gzip -o ~/clean/ $fq1 $fq2  \n\nfastp -i in.fq -o out.fq # 单端测序数据; \nfastp -i in.R1.fq.gz -I in.R2.fq.gz -o out.R1.fq.gz -O out.R2.fq.gz # 双端测序数据\n")])]),s._v(" "),a("div",{staticClass:"line-numbers-wrapper"},[a("span",{staticClass:"line-number"},[s._v("1")]),a("br"),a("span",{staticClass:"line-number"},[s._v("2")]),a("br"),a("span",{staticClass:"line-number"},[s._v("3")]),a("br"),a("span",{staticClass:"line-number"},[s._v("4")]),a("br"),a("span",{staticClass:"line-number"},[s._v("5")]),a("br"),a("span",{staticClass:"line-number"},[s._v("6")]),a("br"),a("span",{staticClass:"line-number"},[s._v("7")]),a("br"),a("span",{staticClass:"line-number"},[s._v("8")]),a("br"),a("span",{staticClass:"line-number"},[s._v("9")]),a("br"),a("span",{staticClass:"line-number"},[s._v("10")]),a("br"),a("span",{staticClass:"line-number"},[s._v("11")]),a("br"),a("span",{staticClass:"line-number"},[s._v("12")]),a("br"),a("span",{staticClass:"line-number"},[s._v("13")]),a("br"),a("span",{staticClass:"line-number"},[s._v("14")]),a("br"),a("span",{staticClass:"line-number"},[s._v("15")]),a("br")])]),a("h2",{attrs:{id:"比对和定量"}},[a("a",{staticClass:"header-anchor",attrs:{href:"#比对和定量"}},[s._v("#")]),s._v(" 比对和定量")]),s._v(" "),a("p",[s._v("salmon：对基因进行定量分析，如果不研究新的转录本，这个方法能更快得到定量信息，不经过比对计数步骤，而是直接对基因进行定量,下载参考转录组，建立索引,开始比对和定量\nhisat2：比对+差异分析")]),s._v(" "),a("div",{staticClass:"language- line-numbers-mode"},[a("pre",{pre:!0,attrs:{class:"language-text"}},[a("code",[s._v("# 建立索引\nhisat2-build -p 4 genome.fa genome \n\n# 比对\nhisat2 -p 10  # -p 线程\n\t-x /path/to/refer/hisat2index/genome  # -x索引文件\n\t-1 比对.fastq.gz \n\t-2 比对-2.fastq.gz \n\t-S tmp.hisat2.sam # -S，自己命名的文件\n\ncat tmp.hisat2.sam | head  \n\tless -S tmp.hisat2.sam  \n\nsamtools view tmp.hisat2.bam | less -S  \n  \n# sam转bam：\nsamtools sort -O bam -@ 5 -o output_bam.bam input_sam.sam  \n\n# 定量工具\n\n")])]),s._v(" "),a("div",{staticClass:"line-numbers-wrapper"},[a("span",{staticClass:"line-number"},[s._v("1")]),a("br"),a("span",{staticClass:"line-number"},[s._v("2")]),a("br"),a("span",{staticClass:"line-number"},[s._v("3")]),a("br"),a("span",{staticClass:"line-number"},[s._v("4")]),a("br"),a("span",{staticClass:"line-number"},[s._v("5")]),a("br"),a("span",{staticClass:"line-number"},[s._v("6")]),a("br"),a("span",{staticClass:"line-number"},[s._v("7")]),a("br"),a("span",{staticClass:"line-number"},[s._v("8")]),a("br"),a("span",{staticClass:"line-number"},[s._v("9")]),a("br"),a("span",{staticClass:"line-number"},[s._v("10")]),a("br"),a("span",{staticClass:"line-number"},[s._v("11")]),a("br"),a("span",{staticClass:"line-number"},[s._v("12")]),a("br"),a("span",{staticClass:"line-number"},[s._v("13")]),a("br"),a("span",{staticClass:"line-number"},[s._v("14")]),a("br"),a("span",{staticClass:"line-number"},[s._v("15")]),a("br"),a("span",{staticClass:"line-number"},[s._v("16")]),a("br"),a("span",{staticClass:"line-number"},[s._v("17")]),a("br"),a("span",{staticClass:"line-number"},[s._v("18")]),a("br"),a("span",{staticClass:"line-number"},[s._v("19")]),a("br"),a("span",{staticClass:"line-number"},[s._v("20")]),a("br")])]),a("p",[s._v("另一个比对")]),s._v(" "),a("div",{staticClass:"language-sh line-numbers-mode"},[a("pre",{pre:!0,attrs:{class:"language-sh"}},[a("code",[a("span",{pre:!0,attrs:{class:"token function"}},[s._v("mkdir")]),s._v(" "),a("span",{pre:!0,attrs:{class:"token parameter variable"}},[s._v("-p")]),s._v(" RNA-Seq/aligned  \n"),a("span",{pre:!0,attrs:{class:"token keyword"}},[s._v("for")]),s._v(" "),a("span",{pre:!0,attrs:{class:"token for-or-select variable"}},[s._v("i")]),s._v(" "),a("span",{pre:!0,attrs:{class:"token keyword"}},[s._v("in")]),s._v(" "),a("span",{pre:!0,attrs:{class:"token variable"}},[a("span",{pre:!0,attrs:{class:"token variable"}},[s._v("`")]),a("span",{pre:!0,attrs:{class:"token function"}},[s._v("seq")]),s._v(" "),a("span",{pre:!0,attrs:{class:"token number"}},[s._v("57")]),s._v(" "),a("span",{pre:!0,attrs:{class:"token number"}},[s._v("62")]),a("span",{pre:!0,attrs:{class:"token variable"}},[s._v("`")])]),s._v("  \n"),a("span",{pre:!0,attrs:{class:"token keyword"}},[s._v("do")]),s._v(" hisat2 "),a("span",{pre:!0,attrs:{class:"token parameter variable"}},[s._v("-t")]),s._v(" \n\t"),a("span",{pre:!0,attrs:{class:"token parameter variable"}},[s._v("-x")]),s._v(" reference/index/hg19/genome \n\t"),a("span",{pre:!0,attrs:{class:"token parameter variable"}},[s._v("-1")]),s._v(" RNA-Seq/SRR35899i_1.fastq.gz \n\t"),a("span",{pre:!0,attrs:{class:"token parameter variable"}},[s._v("-2")]),s._v(" RNA-Seq/SRR35899i1.fastq.gz−2RNA−Seq/SRR35899i_1.fastq.gz \n\t"),a("span",{pre:!0,attrs:{class:"token parameter variable"}},[s._v("-2")]),s._v(" RNA-Seq/SRR35899i1​.fastq.gz−2RNA−Seq/SRR35899"),a("span",{pre:!0,attrs:{class:"token punctuation"}},[s._v("{")]),s._v("i"),a("span",{pre:!0,attrs:{class:"token punctuation"}},[s._v("}")]),s._v("_2.fastq.gz \n\t"),a("span",{pre:!0,attrs:{class:"token parameter variable"}},[s._v("-S")]),s._v(" RNA-Seq/aligned/SRR35899"),a("span",{pre:!0,attrs:{class:"token variable"}},[s._v("${i}")]),s._v(".sam \n"),a("span",{pre:!0,attrs:{class:"token operator"}},[s._v("&")]),a("span",{pre:!0,attrs:{class:"token keyword"}},[s._v("done")]),s._v("\n")])]),s._v(" "),a("div",{staticClass:"line-numbers-wrapper"},[a("span",{staticClass:"line-number"},[s._v("1")]),a("br"),a("span",{staticClass:"line-number"},[s._v("2")]),a("br"),a("span",{staticClass:"line-number"},[s._v("3")]),a("br"),a("span",{staticClass:"line-number"},[s._v("4")]),a("br"),a("span",{staticClass:"line-number"},[s._v("5")]),a("br"),a("span",{staticClass:"line-number"},[s._v("6")]),a("br"),a("span",{staticClass:"line-number"},[s._v("7")]),a("br"),a("span",{staticClass:"line-number"},[s._v("8")]),a("br"),a("span",{staticClass:"line-number"},[s._v("9")]),a("br")])]),a("h2",{attrs:{id:"理解参考"}},[a("a",{staticClass:"header-anchor",attrs:{href:"#理解参考"}},[s._v("#")]),s._v(" 理解参考")]),s._v(" "),a("p",[s._v("创建SRA数据下载的脚本文件：00_prefetch.sh")]),s._v(" "),a("div",{staticClass:"language- line-numbers-mode"},[a("pre",{pre:!0,attrs:{class:"language-text"}},[a("code",[s._v("date\necho “prefetch sra ”\ncd/mkdir\ncat idname | while read id;\ndo\nprefetch $id -O ./   #输出文件夹位置\ndone\n")])]),s._v(" "),a("div",{staticClass:"line-numbers-wrapper"},[a("span",{staticClass:"line-number"},[s._v("1")]),a("br"),a("span",{staticClass:"line-number"},[s._v("2")]),a("br"),a("span",{staticClass:"line-number"},[s._v("3")]),a("br"),a("span",{staticClass:"line-number"},[s._v("4")]),a("br"),a("span",{staticClass:"line-number"},[s._v("5")]),a("br"),a("span",{staticClass:"line-number"},[s._v("6")]),a("br"),a("span",{staticClass:"line-number"},[s._v("7")]),a("br")])]),a("p",[s._v("运行脚本：nohup bash 00_prefetch .sh >log_00 2>&1 &\n0：标准输入；1：标准输出；2：标准错误输出\n2>&1：将标准输出重定向到标准输出，“>&”是一个整体符号\n>log2>&1 &：最后一个& 表示把命令后台进行，本来1是输出到屏幕，现在先输出到log，2通过>&又指向1，所以1和2都输出到了log")]),s._v(" "),a("p",[s._v("SRA文件转为FASTQ格式")]),s._v(" "),a("p",[s._v("单端：")]),s._v(" "),a("div",{staticClass:"language- line-numbers-mode"},[a("pre",{pre:!0,attrs:{class:"language-text"}},[a("code",[s._v("ls ~/test/raw/fq/*.f* | while read id\ndo\ntrim_galore -q 25 -j 4 --phred33 --length 35 --stringency 3 \\\n--gzip -o ~/test/clean/ $id\ndone\n")])]),s._v(" "),a("div",{staticClass:"line-numbers-wrapper"},[a("span",{staticClass:"line-number"},[s._v("1")]),a("br"),a("span",{staticClass:"line-number"},[s._v("2")]),a("br"),a("span",{staticClass:"line-number"},[s._v("3")]),a("br"),a("span",{staticClass:"line-number"},[s._v("4")]),a("br"),a("span",{staticClass:"line-number"},[s._v("5")]),a("br")])]),a("p",[s._v("双端：")]),s._v(" "),a("div",{staticClass:"language- line-numbers-mode"},[a("pre",{pre:!0,attrs:{class:"language-text"}},[a("code",[s._v("#先把文件_1,_2的路径文件名分别存储，再合并成两列的格式，存为config\nls ~/test/raw/fq/*_1* >1    \nls ~/test/raw/fq/*_2* >2\n\npaste 1 2 > config\ncat config | while read id\n\ndo\narr=($id)\nfq1=${arr[0]}; fq2=${arr[1]}\ntrim_galore -j 4 -q 25 --phred 33 --length 35 --stringency 3 \\\n\t--paired --gzip -o ~/test/clean/ $fq1 $fq2\n")])]),s._v(" "),a("div",{staticClass:"line-numbers-wrapper"},[a("span",{staticClass:"line-number"},[s._v("1")]),a("br"),a("span",{staticClass:"line-number"},[s._v("2")]),a("br"),a("span",{staticClass:"line-number"},[s._v("3")]),a("br"),a("span",{staticClass:"line-number"},[s._v("4")]),a("br"),a("span",{staticClass:"line-number"},[s._v("5")]),a("br"),a("span",{staticClass:"line-number"},[s._v("6")]),a("br"),a("span",{staticClass:"line-number"},[s._v("7")]),a("br"),a("span",{staticClass:"line-number"},[s._v("8")]),a("br"),a("span",{staticClass:"line-number"},[s._v("9")]),a("br"),a("span",{staticClass:"line-number"},[s._v("10")]),a("br"),a("span",{staticClass:"line-number"},[s._v("11")]),a("br"),a("span",{staticClass:"line-number"},[s._v("12")]),a("br")])]),a("p",[s._v("参考的系列教程：")]),s._v(" "),a("p",[a("a",{attrs:{href:"https://cloud.tencent.com/developer/article/2032035?from=article.detail.2032040",target:"_blank",rel:"noopener noreferrer"}},[s._v("RNA-seq入门实战（一）：上游数据下载、格式转化和质控清洗"),a("OutboundLink")],1)]),s._v(" "),a("p",[a("a",{attrs:{href:"https://cloud.tencent.com/developer/article/2056795",target:"_blank",rel:"noopener noreferrer"}},[s._v("RNA-seq入门实战（二）：上游数据的比对计数——Hisat2+ featureCounts 与 Salmon"),a("OutboundLink")],1)]),s._v(" "),a("p",[a("a",{attrs:{href:"https://cloud.tencent.com/developer/article/2056796",target:"_blank",rel:"noopener noreferrer"}},[s._v("RNA-seq入门实战（三）：在R里面整理表达量counts矩阵"),a("OutboundLink")],1)]),s._v(" "),a("p",[a("a",{attrs:{href:"https://cloud.tencent.com/developer/article/2056797",target:"_blank",rel:"noopener noreferrer"}},[s._v("RNA-seq入门实战（四）：差异分析前的准备——数据检查"),a("OutboundLink")],1)]),s._v(" "),a("p",[a("a",{attrs:{href:"https://cloud.tencent.com/developer/article/2056798",target:"_blank",rel:"noopener noreferrer"}},[s._v("RNA-seq入门实战（五）：差异分析——DESeq2 edgeR limma的使用与比较"),a("OutboundLink")],1)]),s._v(" "),a("p",[a("a",{attrs:{href:"https://zhuanlan.zhihu.com/p/531957349",target:"_blank",rel:"noopener noreferrer"}},[s._v("RNA-seq数据分析 09：DESeq2差异表达分析 - 知乎 (zhihu.com)"),a("OutboundLink")],1)]),s._v(" "),a("p",[a("a",{attrs:{href:"https://cloud.tencent.com/developer/article/2056799?areaId=106001",target:"_blank",rel:"noopener noreferrer"}},[s._v("RNA-seq入门实战（六）：GO、KEGG富集分析与enrichplot超全可视化攻略"),a("OutboundLink")],1)]),s._v(" "),a("p",[s._v("后续还有几个哦！")]),s._v(" "),a("p",[s._v("三大包："),a("a",{attrs:{href:"https://zhuanlan.zhihu.com/p/658948530",target:"_blank",rel:"noopener noreferrer"}},[s._v("DESeq2、edgeR 和 limma"),a("OutboundLink")],1)]),s._v(" "),a("h2",{attrs:{id:"有趣的进一步理解"}},[a("a",{staticClass:"header-anchor",attrs:{href:"#有趣的进一步理解"}},[s._v("#")]),s._v(" 有趣的进一步理解")]),s._v(" "),a("p",[a("a",{attrs:{href:"https://mp.weixin.qq.com/s/_zRJNwLMnI29xpjFy_fReQ?poc_token=HFdeaGajLfzOLk5ggPc5wsQIR_OQoSIT5eFtfbTT",target:"_blank",rel:"noopener noreferrer"}},[s._v("熊读文献｜别再用DEseq2和edgeR进行大样本差异表达基因分析了 (qq.com)"),a("OutboundLink")],1)]),s._v(" "),a("p",[s._v("小样本，两组比较，每组三四个样本的，还是这三个包，但是超过八个样本，还是使用朴实无华的Wilcoxon秩和检验吧！")]),s._v(" "),a("p",[s._v("文中有一部分实现Wilcoxon秩和检验的代码（暂时没看懂呢）")])])}),[],!1,null,null,null);a.default=t.exports}}]);