(window.webpackJsonp=window.webpackJsonp||[]).push([[39],{359:function(a,e,n){"use strict";n.r(e);var s=n(7),t=Object(s.a)({},(function(){var a=this,e=a._self._c;return e("ContentSlotsDistributor",{attrs:{"slot-key":a.$parent.slotKey}},[e("div",{staticClass:"custom-block warning"},[e("p",{staticClass:"custom-block-title"},[a._v("注意")]),a._v(" "),e("p",[a._v("这篇文章主要是自己在阅读一篇文献的时候，一边理解一遍翻译的内容，结构比较混乱，参考意义不大；自己有空可以再看看，复习一下单词而已。")])]),a._v(" "),e("h3",{attrs:{id:"反馈-调整"}},[e("a",{staticClass:"header-anchor",attrs:{href:"#反馈-调整"}},[a._v("#")]),a._v(" 反馈 & 调整")]),a._v(" "),e("p",[a._v("如何定义细胞种类？")]),a._v(" "),e("p",[a._v("定义细胞种类后，如何取出来进行分析？")]),a._v(" "),e("p",[a._v("KEGG，GO，GSEA等等分析，都要详细理解内容后，熟悉API后，才能进行交互呀！！！！")]),a._v(" "),e("p",[a._v("pytorch可以用来做什么？？机器学习，神经网络，等等，python那么牛的代码，可以用来搞什么呢？？")]),a._v(" "),e("p",[e("a",{attrs:{href:"https://blog.sciencenet.cn/blog-3447504-1251830.html",target:"_blank",rel:"noopener noreferrer"}},[a._v("细胞可塑性、细胞分化轨迹、细胞谱系、RNA速率"),e("OutboundLink")],1)]),a._v(" "),e("h2",{attrs:{id:"单细胞概览"}},[e("a",{staticClass:"header-anchor",attrs:{href:"#单细胞概览"}},[a._v("#")]),a._v(" 单细胞概览")]),a._v(" "),e("p",[a._v("通过一篇文献来学习理解单细胞分析\n文献：Best practices for single-cell analysis across modalities\n官网："),e("a",{attrs:{href:"https://www.sc-best-practices.org/introduction/raw_data_processing.html",target:"_blank",rel:"noopener noreferrer"}},[a._v("Single-cell best practices"),e("OutboundLink")],1)]),a._v(" "),e("p",[a._v("官网部分的内容：\nIntroduction，背景知识介绍")]),a._v(" "),e("p",[a._v("Preprocessing and visualization，预处理和可视化")]),a._v(" "),e("ul",[e("li",[a._v("quality control，质控")]),a._v(" "),e("li",[a._v("normalization，标准化")]),a._v(" "),e("li",[a._v("feature selection，选择特征")]),a._v(" "),e("li",[a._v("dimensionality reduction，降维")])]),a._v(" "),e("p",[a._v("Identifying cellular structure，鉴定细胞结构")]),a._v(" "),e("ul",[e("li",[a._v("clustering，聚类")]),a._v(" "),e("li",[a._v("annotation，注释")]),a._v(" "),e("li",[a._v("data integration，数据整合")])]),a._v(" "),e("p",[a._v("Inferring trajectory，推断轨迹")]),a._v(" "),e("ul",[e("li",[a._v("pseudotemporal odering，假想的轨迹")]),a._v(" "),e("li",[a._v("RNA velocity，RNA速率分析")]),a._v(" "),e("li",[a._v("lineage tracing，谱系追踪")])]),a._v(" "),e("p",[a._v("Dealing with conditions，条件分析")]),a._v(" "),e("ul",[e("li",[a._v("differential gene expression analysis，差异基因表达分析")]),a._v(" "),e("li",[a._v("compositional analysis，组成分析")]),a._v(" "),e("li",[a._v("gene set enrichment and pathway analysis，基因集富集和通路分析")]),a._v(" "),e("li",[a._v("perturbation modeling")])]),a._v(" "),e("p",[a._v("Modeling mechanism，机制建模")]),a._v(" "),e("ul",[e("li",[a._v("gene regulatory networks，基因调节网络")]),a._v(" "),e("li",[a._v("cell-cell cummunication，细胞通讯")])]),a._v(" "),e("p",[a._v("Deconvolution")]),a._v(" "),e("ul",[e("li",[a._v("bulk deconvolution")])]),a._v(" "),e("p",[a._v("。。。未完待续")]),a._v(" "),e("h4",{attrs:{id:"背景简介"}},[e("a",{staticClass:"header-anchor",attrs:{href:"#背景简介"}},[a._v("#")]),a._v(" 背景简介")]),a._v(" "),e("p",[a._v("主流的三种单细胞分析工具")]),a._v(" "),e("ul",[e("li",[a._v("Bioconductor：R，生物信息学生态，用户友好的小插图")]),a._v(" "),e("li",[a._v("Seurat：R，单细胞，小插图，用户多")]),a._v(" "),e("li",[a._v("Scverse：python，单细胞分析生态+机器学习")])]),a._v(" "),e("p",[a._v("基本数据结构\nAnnData：cells X genes——obs X var\n横着object--cells，竖着variable--genes\n元数据metada，研究对象的性质，一维或多维的，行/列/表格/稀疏矩阵\n非结构化数据")]),a._v(" "),e("p",[a._v("数据变化流程")]),a._v(" "),e("ol",[e("li",[a._v("初始化AnnData对象，有区别的横纵坐标，只是ids；")]),a._v(" "),e("li",[a._v("添加--对齐--的--元数据，obs和variable；")]),a._v(" "),e("li",[a._v("多维的元数据，可能是计算后的新pandas的dataframe，或者numpy的array，或scipy的sparse matrix")]),a._v(" "),e("li",[a._v("非结构化数据，一般信息的列表或字典")]),a._v(" "),e("li",[a._v("摞起来的层次，layers")]),a._v(" "),e("li",[a._v("导出数据，默认计数矩阵X，可以指定layer，（也可以metadata吧）")]),a._v(" "),e("li",[a._v("读写，特定数据格式，分层数组（HDF5，或Zarr），或自己修改的h5ad")]),a._v(" "),e("li",[a._v('访问数据，adata[1:5,["var_name1","var_name2"]]，数值和字符访问不一样')]),a._v(" "),e("li",[a._v("多模态数据mudata，表达矩阵外同等级的矩阵，相同obs值但不同var的数据，如细胞的RNA表达矩阵，还可以再测细胞表面蛋白数据—— .mod(=modality)")]),a._v(" "),e("li",[a._v("读写和存储，类似于anndata，但是多了一个“mod”，各自存放：path/mod/A.h5ad，还有B.h5ad")])]),a._v(" "),e("div",{staticClass:"language- line-numbers-mode"},[e("pre",{pre:!0,attrs:{class:"language-text"}},[e("code",[a._v('import numpy as np\nimport pandas as pd\nimport anndata as ad\nfrom scipy.sparse import csr_matrix\n\ncounts = csr_matrix(np.random.poisson(1, size=(100, 2000)), dtype=np.float32)\nadata = ad.AnnData(counts)\nadata\nadata.X\n\nadata.obs_names = [f"Cell_{i:d}" for i in range(adata.n_obs)]\nadata.var_names = [f"Gene_{i:d}" for i in range(adata.n_vars)]\nprint(adata.obs_names[:10])\n\n#随机生成细胞类型，并存放到变量ct中\nct = np.random.choice(["B", "T", "Monocyte"], size=(adata.n_obs,))\n#我们将生成的细胞类型赋值到adata.obs列中\n#Categoricals are preferred for efficiency\nadata.obs["cell_type"] = pd.Categorical(ct)  \nadata.obs\nbdata = adata[adata.obs.cell_type == "B"]\nbdata\n\nadata.obsm["X_umap"] = np.random.normal(0, 1, size=(adata.n_obs, 2))\nadata.varm["gene_stuff"] = np.random.normal(0, 1, size=(adata.n_vars, 5))\nadata.obsm\nadata\n\nadata.uns["random"] = [1, 2, 3]\nadata.uns\n\nadata.layers["log_transformed"] = np.log1p(adata.X)\nadata\n\n(adata.X != adata.layers["log_transformed"]).nnz == 0\n\nadata.to_df(layer="log_transformed").head()\n\nadata.write("../../data/my_results.h5ad", compression="gzip")\n!h5ls \'../../data/my_results.h5ad\'\n\nadata_new = ad.read_h5ad("../../data/my_results.h5ad")\nadata_new\n\n\n数据查看和复制\n在造细胞（obs的metadata）\n- time_yr: 代表细胞的时期\n- subject_id: 代表细胞的类别\n- instrument_type: 代表细胞取材的设备\n- site: 代表细胞所属的部位\nobs_meta = pd.DataFrame(\n    {\n        "time_yr": np.random.choice([0, 2, 4, 8], adata.n_obs),\n        "subject_id": np.random.choice(\n            ["subject 1", "subject 2", "subject 4", "subject 8"], adata.n_obs\n        ),\n        "instrument_type": np.random.choice(["type a", "type b"], adata.n_obs),\n        "site": np.random.choice(["site x", "site y"], adata.n_obs),\n    },\n    index=adata.obs.index,  # these are the same IDs of observations as above!\n)\n\n# 注释--造数据表\ndataframe = pd.DataFrame(\n{\n"colname1":list()\n"colname2":list()\n"colname3":list()\n}\n)\n\nadata = ad.AnnData(adata.X, obs=obs_meta, var=adata.var)\nadata\n\n简单看看和复制前五个细胞，1和3基因\nadata_view = adata[:5, ["Gene_1", "Gene_3"]]\nadata_view（查看算是代表原数据，谨慎修改）\n\nadata_subset = adata[:5, ["Gene_1", "Gene_3"]].copy()\nadata_subset（复制后和原数据一样而已）\n\nprint(adata[:3, "Gene_1"].X.toarray().tolist())\nadata[:3, "Gene_1"].X = [0, 0, 0]\nprint(adata[:3, "Gene_1"].X.toarray().tolist())\n\n布尔式切片操作\nadata[adata.obs.time_yr.isin([2, 4])].obs.head()\n\n\nadata = ad.read("../../data/my_results.h5ad", backed="r")\nadata.isbacked\nadata.filename\n\n\n#已经下载\n%pip install mudata\n%conda install -c conda-forge mudata\n\n\nmdata = md.MuData({"A": adata, "B": adata2})\nmdata\nlist(mdata.mod.keys())\nprint(mdata.mod["A"])\nprint(mdata["A"])\n\nprint(mdata.shape)\nprint(mdata.n_obs)\nprint(mdata.n_vars)\n\n修改需要update，添加不需要（mudata是对原始对象的引用）\nadata2.var_names = ["var_ad2_" + e.split("_")[1] for e in adata2.var_names]\nprint(f"Outdated variables names: ...,", ",".join(mdata.var_names[-3:]))\nmdata.update()\nprint(f"Updated variables names: ...,", ", ".join(mdata.var_names[-3:]))\n# Add some unstructured data to the original object\nadata.uns["misc"] = {"adata": True}\n# Access modality A via the .mod attribute\nmdata.mod["A"].uns["misc"]\n\nnp.sum(mdata.obsm["A"]) == np.sum(mdata.obsm["B"]) == n\n\n视图和切片不太懂\n\nmdata.write("../../data/my_mudata.h5mu")\nmdata_r = md.read("../../data/my_mudata.h5mu", backed=True)\nmdata_r\n')])]),a._v(" "),e("div",{staticClass:"line-numbers-wrapper"},[e("span",{staticClass:"line-number"},[a._v("1")]),e("br"),e("span",{staticClass:"line-number"},[a._v("2")]),e("br"),e("span",{staticClass:"line-number"},[a._v("3")]),e("br"),e("span",{staticClass:"line-number"},[a._v("4")]),e("br"),e("span",{staticClass:"line-number"},[a._v("5")]),e("br"),e("span",{staticClass:"line-number"},[a._v("6")]),e("br"),e("span",{staticClass:"line-number"},[a._v("7")]),e("br"),e("span",{staticClass:"line-number"},[a._v("8")]),e("br"),e("span",{staticClass:"line-number"},[a._v("9")]),e("br"),e("span",{staticClass:"line-number"},[a._v("10")]),e("br"),e("span",{staticClass:"line-number"},[a._v("11")]),e("br"),e("span",{staticClass:"line-number"},[a._v("12")]),e("br"),e("span",{staticClass:"line-number"},[a._v("13")]),e("br"),e("span",{staticClass:"line-number"},[a._v("14")]),e("br"),e("span",{staticClass:"line-number"},[a._v("15")]),e("br"),e("span",{staticClass:"line-number"},[a._v("16")]),e("br"),e("span",{staticClass:"line-number"},[a._v("17")]),e("br"),e("span",{staticClass:"line-number"},[a._v("18")]),e("br"),e("span",{staticClass:"line-number"},[a._v("19")]),e("br"),e("span",{staticClass:"line-number"},[a._v("20")]),e("br"),e("span",{staticClass:"line-number"},[a._v("21")]),e("br"),e("span",{staticClass:"line-number"},[a._v("22")]),e("br"),e("span",{staticClass:"line-number"},[a._v("23")]),e("br"),e("span",{staticClass:"line-number"},[a._v("24")]),e("br"),e("span",{staticClass:"line-number"},[a._v("25")]),e("br"),e("span",{staticClass:"line-number"},[a._v("26")]),e("br"),e("span",{staticClass:"line-number"},[a._v("27")]),e("br"),e("span",{staticClass:"line-number"},[a._v("28")]),e("br"),e("span",{staticClass:"line-number"},[a._v("29")]),e("br"),e("span",{staticClass:"line-number"},[a._v("30")]),e("br"),e("span",{staticClass:"line-number"},[a._v("31")]),e("br"),e("span",{staticClass:"line-number"},[a._v("32")]),e("br"),e("span",{staticClass:"line-number"},[a._v("33")]),e("br"),e("span",{staticClass:"line-number"},[a._v("34")]),e("br"),e("span",{staticClass:"line-number"},[a._v("35")]),e("br"),e("span",{staticClass:"line-number"},[a._v("36")]),e("br"),e("span",{staticClass:"line-number"},[a._v("37")]),e("br"),e("span",{staticClass:"line-number"},[a._v("38")]),e("br"),e("span",{staticClass:"line-number"},[a._v("39")]),e("br"),e("span",{staticClass:"line-number"},[a._v("40")]),e("br"),e("span",{staticClass:"line-number"},[a._v("41")]),e("br"),e("span",{staticClass:"line-number"},[a._v("42")]),e("br"),e("span",{staticClass:"line-number"},[a._v("43")]),e("br"),e("span",{staticClass:"line-number"},[a._v("44")]),e("br"),e("span",{staticClass:"line-number"},[a._v("45")]),e("br"),e("span",{staticClass:"line-number"},[a._v("46")]),e("br"),e("span",{staticClass:"line-number"},[a._v("47")]),e("br"),e("span",{staticClass:"line-number"},[a._v("48")]),e("br"),e("span",{staticClass:"line-number"},[a._v("49")]),e("br"),e("span",{staticClass:"line-number"},[a._v("50")]),e("br"),e("span",{staticClass:"line-number"},[a._v("51")]),e("br"),e("span",{staticClass:"line-number"},[a._v("52")]),e("br"),e("span",{staticClass:"line-number"},[a._v("53")]),e("br"),e("span",{staticClass:"line-number"},[a._v("54")]),e("br"),e("span",{staticClass:"line-number"},[a._v("55")]),e("br"),e("span",{staticClass:"line-number"},[a._v("56")]),e("br"),e("span",{staticClass:"line-number"},[a._v("57")]),e("br"),e("span",{staticClass:"line-number"},[a._v("58")]),e("br"),e("span",{staticClass:"line-number"},[a._v("59")]),e("br"),e("span",{staticClass:"line-number"},[a._v("60")]),e("br"),e("span",{staticClass:"line-number"},[a._v("61")]),e("br"),e("span",{staticClass:"line-number"},[a._v("62")]),e("br"),e("span",{staticClass:"line-number"},[a._v("63")]),e("br"),e("span",{staticClass:"line-number"},[a._v("64")]),e("br"),e("span",{staticClass:"line-number"},[a._v("65")]),e("br"),e("span",{staticClass:"line-number"},[a._v("66")]),e("br"),e("span",{staticClass:"line-number"},[a._v("67")]),e("br"),e("span",{staticClass:"line-number"},[a._v("68")]),e("br"),e("span",{staticClass:"line-number"},[a._v("69")]),e("br"),e("span",{staticClass:"line-number"},[a._v("70")]),e("br"),e("span",{staticClass:"line-number"},[a._v("71")]),e("br"),e("span",{staticClass:"line-number"},[a._v("72")]),e("br"),e("span",{staticClass:"line-number"},[a._v("73")]),e("br"),e("span",{staticClass:"line-number"},[a._v("74")]),e("br"),e("span",{staticClass:"line-number"},[a._v("75")]),e("br"),e("span",{staticClass:"line-number"},[a._v("76")]),e("br"),e("span",{staticClass:"line-number"},[a._v("77")]),e("br"),e("span",{staticClass:"line-number"},[a._v("78")]),e("br"),e("span",{staticClass:"line-number"},[a._v("79")]),e("br"),e("span",{staticClass:"line-number"},[a._v("80")]),e("br"),e("span",{staticClass:"line-number"},[a._v("81")]),e("br"),e("span",{staticClass:"line-number"},[a._v("82")]),e("br"),e("span",{staticClass:"line-number"},[a._v("83")]),e("br"),e("span",{staticClass:"line-number"},[a._v("84")]),e("br"),e("span",{staticClass:"line-number"},[a._v("85")]),e("br"),e("span",{staticClass:"line-number"},[a._v("86")]),e("br"),e("span",{staticClass:"line-number"},[a._v("87")]),e("br"),e("span",{staticClass:"line-number"},[a._v("88")]),e("br"),e("span",{staticClass:"line-number"},[a._v("89")]),e("br"),e("span",{staticClass:"line-number"},[a._v("90")]),e("br"),e("span",{staticClass:"line-number"},[a._v("91")]),e("br"),e("span",{staticClass:"line-number"},[a._v("92")]),e("br"),e("span",{staticClass:"line-number"},[a._v("93")]),e("br"),e("span",{staticClass:"line-number"},[a._v("94")]),e("br"),e("span",{staticClass:"line-number"},[a._v("95")]),e("br"),e("span",{staticClass:"line-number"},[a._v("96")]),e("br"),e("span",{staticClass:"line-number"},[a._v("97")]),e("br"),e("span",{staticClass:"line-number"},[a._v("98")]),e("br"),e("span",{staticClass:"line-number"},[a._v("99")]),e("br"),e("span",{staticClass:"line-number"},[a._v("100")]),e("br"),e("span",{staticClass:"line-number"},[a._v("101")]),e("br"),e("span",{staticClass:"line-number"},[a._v("102")]),e("br"),e("span",{staticClass:"line-number"},[a._v("103")]),e("br"),e("span",{staticClass:"line-number"},[a._v("104")]),e("br"),e("span",{staticClass:"line-number"},[a._v("105")]),e("br"),e("span",{staticClass:"line-number"},[a._v("106")]),e("br"),e("span",{staticClass:"line-number"},[a._v("107")]),e("br"),e("span",{staticClass:"line-number"},[a._v("108")]),e("br"),e("span",{staticClass:"line-number"},[a._v("109")]),e("br"),e("span",{staticClass:"line-number"},[a._v("110")]),e("br"),e("span",{staticClass:"line-number"},[a._v("111")]),e("br"),e("span",{staticClass:"line-number"},[a._v("112")]),e("br"),e("span",{staticClass:"line-number"},[a._v("113")]),e("br"),e("span",{staticClass:"line-number"},[a._v("114")]),e("br"),e("span",{staticClass:"line-number"},[a._v("115")]),e("br"),e("span",{staticClass:"line-number"},[a._v("116")]),e("br"),e("span",{staticClass:"line-number"},[a._v("117")]),e("br"),e("span",{staticClass:"line-number"},[a._v("118")]),e("br"),e("span",{staticClass:"line-number"},[a._v("119")]),e("br"),e("span",{staticClass:"line-number"},[a._v("120")]),e("br"),e("span",{staticClass:"line-number"},[a._v("121")]),e("br"),e("span",{staticClass:"line-number"},[a._v("122")]),e("br"),e("span",{staticClass:"line-number"},[a._v("123")]),e("br"),e("span",{staticClass:"line-number"},[a._v("124")]),e("br"),e("span",{staticClass:"line-number"},[a._v("125")]),e("br"),e("span",{staticClass:"line-number"},[a._v("126")]),e("br"),e("span",{staticClass:"line-number"},[a._v("127")]),e("br")])]),e("h4",{attrs:{id:"一般处理"}},[e("a",{staticClass:"header-anchor",attrs:{href:"#一般处理"}},[a._v("#")]),a._v(" 一般处理")]),a._v(" "),e("p",[e("strong",[a._v("（重点在于理解过程和代码）")]),a._v("\n概览：质量控制--归一缩放--高变特征--降维聚类--识别分群\n基础分析")]),a._v(" "),e("ul",[e("li",[a._v("质控QC：去除低质量细胞（过高或过低基因，过高的线粒体基因）")]),a._v(" "),e("li",[a._v("归一和缩放：normalization正态化（细胞可比）；scaling放缩（基因可比）；高变 features")]),a._v(" "),e("li",[a._v("降维聚类：线性PCA，非线性umap、tsne；形成clusters")]),a._v(" "),e("li",[a._v("识别分群：特征基因/高变基因biomarker；标记cell type"),e("br"),a._v("\n高级分析")]),a._v(" "),e("li",[a._v("基因富集分析，GO，KEGG，GSEA，GSVA")]),a._v(" "),e("li",[a._v("拟时序分析，monocle")]),a._v(" "),e("li",[a._v("细胞通讯分析，cellChat\n整合分析")]),a._v(" "),e("li",[a._v("表观组，ATAC，ChIP，")]),a._v(" "),e("li",[a._v("空间转录组，spatial")]),a._v(" "),e("li",[a._v("细胞表面蛋白组，T/B - cell receptor")])]),a._v(" "),e("p",[a._v("一、质控：\n生物学含义：")]),a._v(" "),e("ul",[e("li",[a._v("Drop方法中，空胞，死胞，多胞；环境污染\n计算处理：")]),a._v(" "),e("li",[a._v("UMI/cell表达的数量（计数深度），feature/gene被表达的数量，线粒体比例")]),a._v(" "),e("li",[a._v("液滴/环境/soup，中混入的一些RNA")])]),a._v(" "),e("p",[a._v("二、归一化和缩放：\n移位对数适用于大多数任务，但寻找稀有的细胞考虑进行皮尔森法归一化\n三种特征基因选择方法：")]),a._v(" "),e("ul",[e("li",[a._v("基于基因离散度与平均值阈值，seurat")]),a._v(" "),e("li",[a._v("基于基因归一化方差，seurat-v3")]),a._v(" "),e("li",[a._v("基于基因的皮尔森残差，scanpy")])]),a._v(" "),e("p",[a._v("三、高变特征\n寻找feature，在不同组之间表达差异大的基因")]),a._v(" "),e("p",[a._v("一般2000-3000内的高可变基因数都是能接受的")]),a._v(" "),e("p",[a._v("减少变量后用于以下的降维聚类，因为是有很多细胞的基因表达是类似的（表达量差不多，即，都高或都低）")]),a._v(" "),e("p",[a._v("比较这些就没有意思，有没有对后续分析都没有")]),a._v(" "),e("p",[a._v("四、降维聚类：")]),a._v(" "),e("p",[a._v("维度诅咒：高维数据，包含的信息更多，噪声和冗余也更多")]),a._v(" "),e("p",[a._v("（噪声：不同组间没有差异的数据，冗余：不同组间差异一样的信息）——和上面的处理有何一异同？？")]),a._v(" "),e("p",[a._v("从“稳定性，准确性和计算成本”三者考虑，建议以下分析：")]),a._v(" "),e("p",[a._v("t-SNE，产生了最佳的整体性能。")]),a._v(" "),e("p",[a._v("UMAP，显示出最高的稳定性，并且最好地分离了原始细胞群体。")]),a._v(" "),e("p",[a._v("PCA，使用广泛")]),a._v(" "),e("p",[a._v("使用数据集的归一化值表示进行降维和可视化--使用计数矩阵的位移对数值，应该是log处理")]),a._v(" "),e("p",[a._v("PCA：对原始数据进行正交变换，创建一组新的无关变量，即主成分")]),a._v(" "),e("p",[a._v("方差最大，能够表达最大的差异性；方差递减排列主成分")]),a._v(" "),e("p",[a._v("MS4A1是B细胞的marker基因")]),a._v(" "),e("p",[a._v("六，分群注释")]),a._v(" "),e("p",[a._v("聚类之后要分群呀，知道在相互作用的细胞分别是谁。")]),a._v(" "),e("p",[a._v("胞类别识别——聚类和注释")]),a._v(" "),e("p",[a._v("生物意义：不同的细胞具有相似的特征，比如共同表达别的细胞没有的基因，细胞表达基因量上相似的分布，可以\n统称为，细胞基因表达量上的相似性")]),a._v(" "),e("p",[a._v("根据各个分群的高表达基因，细胞的Marker，来辨别细胞种类")]),a._v(" "),e("p",[a._v("聚类的类型：有监督聚类，无监督聚类")]),a._v(" "),e("p",[a._v("注释：")]),a._v(" "),e("p",[a._v("注释的方法：手动注释，自动注释")]),a._v(" "),e("p",[a._v("注释的深度/维度：细胞主要类型的注释，细胞亚群的注释")]),a._v(" "),e("p",[a._v("注释的顺序：已知标记基因注释，已分群后注释（不同群的差异基因注释）")]),a._v(" "),e("p",[a._v("细胞类型：依据细胞表面特定的抗原（标记基因识别）\n标记基因识别有一定的局限性，单个标记后，一类细胞还有不同的状态，激活或者静息状态的细胞是不一样的，还有，不同病理状态下的B细胞可能也会不一样，或者不表达那个特定的标记基因哦。")]),a._v(" "),e("p",[a._v("（子集的区分不明显，环境改变后身份的确定性不高）")]),a._v(" "),e("p",[a._v("依据：\n流式细胞数据对于RNA类型数据来说不一定准确（一个细胞表面，一个细胞内部；一个在转录后，一个在翻译后；一个是mRNA，一个是蛋白质）\nps：只筛选我们检测到的标志基因，两三个for循环")]),a._v(" "),e("p",[a._v("基于簇特异性marker基因的注释\n已经cluster了，哦，那就找找每群里面特殊的基因，将这些基因进行定义各群细胞\n数据库，注释细胞信息\n细胞亚群注释，一般都是，取出来后，分群再注释")]),a._v(" "),e("p",[a._v("五、整合：\n不同样本的批次效应：")]),a._v(" "),e("p",[a._v("“批次”：不同实验室采取不同的方法处理的样本，整体表达量可能不一样，表达种类也可能不一样")]),a._v(" "),e("p",[a._v("原因：有技术上的，样本上的等等")]),a._v(" "),e("p",[a._v("scRNA-seq中消除批次的步骤：降维选取重要基因，建模消除批次效应，嵌入\n模型的种类：")]),a._v(" "),e("ul",[e("li",[a._v("全局模型，线性嵌入模型，基于图的模型，深度学习模型")])]),a._v(" "),e("p",[a._v("technical biases机器的偏见，batch effect批次效应，ComBat--pyComBat去批次效应")]),a._v(" "),e("p",[a._v("自动注释方法：SCSA，CellTypist（基于深度学习）\n"),e("a",{attrs:{href:"https://github.com/Teichlab/celltypist",target:"_blank",rel:"noopener noreferrer"}},[a._v("celltypist: A tool for semi-automatic cell type classification"),e("OutboundLink")],1),a._v("\n这个自动注释的东西，一些名词不那么理解")]),a._v(" "),e("p",[a._v("MetaTiME 通过整合数百个肿瘤 scRNA-seq 数据中的数百万个单细胞，学习数据驱动、可解释和可重复的基因程序。\n对比手动注释与自动注释的结果，我们可以发现MetaTiME的注释准确率相当地高，其从百万泛癌图谱训练出来也正意味着其具有着较强的泛化能力\n自动注释的准确率完全取决于训练数据的规模以及数据集的匹配程度，在复杂的单细胞测序数据，以及稀有组织中，其注释能力往往较差，这时候我们就需要用手动注释的方式寻找簇的marker来完成细胞类型的注释。")]),a._v(" "),e("p",[a._v("迁移注释\n“学习”已经包含注释信息的数据，给新的数据进行注释，相当于，延续注释原则\n参考注释数据集\n有一个叫cell balst的参考注释数据集，收集了100+高质量单细胞注释数据集；\n用来训练的数据集不能少于30000个细胞，否则训练出来的TOSICA模型效果一般；参考数据集大于100000个细胞，则可随机选出来30000个细胞，效果一样")]),a._v(" "),e("div",{staticClass:"language- line-numbers-mode"},[e("pre",{pre:!0,attrs:{class:"language-text"}},[e("code",[a._v("adata.X.max() #通过计数矩阵最大值判断是否进行了位移对数计算\n\n有意思的一个细胞标记基因字典，不知道对不对\nsmall_marker_dict={\n    'Fibroblast':['ACTA2'],\n    'Endothelium':['PTPRB','PECAM1'],\n    'Epithelium':['KRT5','KRT14'],\n    'Mast cell':['KIT','CD63'],\n    'Neutrophil' :['FCGR3A','ITGAM'],\n    'cDendritic cell':['FCER1A','CST3'],\n    'pDendritic cell':['IL3RA','GZMB','SERPINF1','ITM2C'],\n    'Monocyte' :['CD14','LYZ','S100A8','S100A9','LST1',],\n    'Macrophage':['CSF1R','CD68'],\n    'B cell':['MS4A1','CD79A'],\n    'Plasma cell':['MZB1','IGKC','JCHAIN'],\n    'Proliferative signal':['MKI67','TOP2A','STMN1'],\n    'NK/NKT cell':['GNLY','NKG7','KLRD1'],\n    'T cell':['CD3D','CD3E'],\n}\n")])]),a._v(" "),e("div",{staticClass:"line-numbers-wrapper"},[e("span",{staticClass:"line-number"},[a._v("1")]),e("br"),e("span",{staticClass:"line-number"},[a._v("2")]),e("br"),e("span",{staticClass:"line-number"},[a._v("3")]),e("br"),e("span",{staticClass:"line-number"},[a._v("4")]),e("br"),e("span",{staticClass:"line-number"},[a._v("5")]),e("br"),e("span",{staticClass:"line-number"},[a._v("6")]),e("br"),e("span",{staticClass:"line-number"},[a._v("7")]),e("br"),e("span",{staticClass:"line-number"},[a._v("8")]),e("br"),e("span",{staticClass:"line-number"},[a._v("9")]),e("br"),e("span",{staticClass:"line-number"},[a._v("10")]),e("br"),e("span",{staticClass:"line-number"},[a._v("11")]),e("br"),e("span",{staticClass:"line-number"},[a._v("12")]),e("br"),e("span",{staticClass:"line-number"},[a._v("13")]),e("br"),e("span",{staticClass:"line-number"},[a._v("14")]),e("br"),e("span",{staticClass:"line-number"},[a._v("15")]),e("br"),e("span",{staticClass:"line-number"},[a._v("16")]),e("br"),e("span",{staticClass:"line-number"},[a._v("17")]),e("br"),e("span",{staticClass:"line-number"},[a._v("18")]),e("br"),e("span",{staticClass:"line-number"},[a._v("19")]),e("br")])]),e("h3",{attrs:{id:"best-practice"}},[e("a",{staticClass:"header-anchor",attrs:{href:"#best-practice"}},[a._v("#")]),a._v(" Best practice")]),a._v(" "),e("p",[a._v("（主要目的是学习"),e("strong",[a._v("英文中")]),a._v("基础的概念，以达到可持续化，最新进展的学习）\nhttps://www.sc-best-practices.org/introduction/prior_art.html")]),a._v(" "),e("p",[a._v("OSCA，Orchestrating Single-Cell Analysis with Bioconductor\nhttps://bioconductor.org/books/release/OSCA/\n精心策划，用bioconductor协调单细胞分析\n缺点：没有scATAC-seq")]),a._v(" "),e("p",[a._v("Orchestrating Spatially-Resolved Transcriptomics Analysis with Bioconductor\nhttps://lmweber.org/OSTA-book/\n补上：Spatial transcriptomics")]),a._v(" "),e("p",[a._v("Single cell best practice\n"),e("a",{attrs:{href:"https://www.sc-best-practices.org/introduction/prior_art.html",target:"_blank",rel:"noopener noreferrer"}},[a._v("Single-cell best practices"),e("OutboundLink")],1)]),a._v(" "),e("div",{staticClass:"language- line-numbers-mode"},[e("pre",{pre:!0,attrs:{class:"language-text"}},[e("code",[a._v("benchmarks：基准\nvelocity：速度\nnewbie friendliness：新人友好\nspatial transcriptomics：空间转录组学\nmulti-omics：多组学\nribonucleic acid：核苷酸\nsequencing assays：测序分析\nrespective advanced chapters：各自，高级的，章节\nAll sequencing assays have individual strengths and limitations which must be known by data analysis to be aware of possible biases in the data.\n所有的测序分析都有各自的优点和局限性，必须通过数据分析了解这些优点和局限性才能意识到数据中\n可能存在的偏差。\nresemble a honeycomb：像蜂窝一样\nelaborate understanding：详尽的理解\nterminology：术语\nmisnomer：用词不当\n")])]),a._v(" "),e("div",{staticClass:"line-numbers-wrapper"},[e("span",{staticClass:"line-number"},[a._v("1")]),e("br"),e("span",{staticClass:"line-number"},[a._v("2")]),e("br"),e("span",{staticClass:"line-number"},[a._v("3")]),e("br"),e("span",{staticClass:"line-number"},[a._v("4")]),e("br"),e("span",{staticClass:"line-number"},[a._v("5")]),e("br"),e("span",{staticClass:"line-number"},[a._v("6")]),e("br"),e("span",{staticClass:"line-number"},[a._v("7")]),e("br"),e("span",{staticClass:"line-number"},[a._v("8")]),e("br"),e("span",{staticClass:"line-number"},[a._v("9")]),e("br"),e("span",{staticClass:"line-number"},[a._v("10")]),e("br"),e("span",{staticClass:"line-number"},[a._v("11")]),e("br"),e("span",{staticClass:"line-number"},[a._v("12")]),e("br"),e("span",{staticClass:"line-number"},[a._v("13")]),e("br"),e("span",{staticClass:"line-number"},[a._v("14")]),e("br"),e("span",{staticClass:"line-number"},[a._v("15")]),e("br")])]),e("h4",{attrs:{id:"overview-of-raw-data-processing"}},[e("a",{staticClass:"header-anchor",attrs:{href:"#overview-of-raw-data-processing"}},[a._v("#")]),a._v(" overview of raw data processing")]),a._v(" "),e("p",[e("img",{attrs:{src:"https://cdn.statically.io/gh/hongliangfang/picx-images-hosting@master/study/overview_raw_data_processing.5hpvpz5qwwk0.webp",alt:"overview_raw_data_processing"}})]),a._v(" "),e("ol",[e("li",[a._v("raw data quality control")]),a._v(" "),e("li",[a._v("alignment and mapping")]),a._v(" "),e("li",[a._v("cell barcode identification & correction")]),a._v(" "),e("li",[a._v("UMI resolution")]),a._v(" "),e("li",[a._v("count matrix quality")]),a._v(" "),e("li",[a._v("count data representation")])]),a._v(" "),e("div",{staticClass:"language- line-numbers-mode"},[e("pre",{pre:!0,attrs:{class:"language-text"}},[e("code",[a._v("myriad methods：无数种方法\ninferring cell types：推断细胞类型\ntrajectory：轨迹\npreclude：排除\ndiminish：消失\ndistorte：扭曲\nmethodology：方法\nunique molecular identifiers：独特的分子标识\nprecede：请继续；precedes：先行一步\narbitrary：随心所欲\nnucleotide：核苷酸\ntoggle content：切换内容\nmodule：模组\nexplicitly：明确的\nmetrics：指标\nloci：本地，也指染色体上的位置\nMapping or alignment：映射和排列\ncell barcode：细胞条形码\nmatrices：矩阵\nagnostic：不可知论者\ncumbersome：很麻烦\nsubstantial disk space：大量的磁盘空间\nTo this end：为了这个目的\ndedicated tools：专用工具\nparsing of technical read content：技术性阅读内容的解析\ndemultiplexing and UMI resolution：解复用和UMI解析（比对时识别去掉，识别完后再标上）\nintermediate representations：中间表示\ntrade-offs：权衡利弊\nroughly categorize existing approaches along two axes：按照两个轴对现有的方法进行大致的分类\nputative locus：假定的轨迹\nalignment：校准\n")])]),a._v(" "),e("div",{staticClass:"line-numbers-wrapper"},[e("span",{staticClass:"line-number"},[a._v("1")]),e("br"),e("span",{staticClass:"line-number"},[a._v("2")]),e("br"),e("span",{staticClass:"line-number"},[a._v("3")]),e("br"),e("span",{staticClass:"line-number"},[a._v("4")]),e("br"),e("span",{staticClass:"line-number"},[a._v("5")]),e("br"),e("span",{staticClass:"line-number"},[a._v("6")]),e("br"),e("span",{staticClass:"line-number"},[a._v("7")]),e("br"),e("span",{staticClass:"line-number"},[a._v("8")]),e("br"),e("span",{staticClass:"line-number"},[a._v("9")]),e("br"),e("span",{staticClass:"line-number"},[a._v("10")]),e("br"),e("span",{staticClass:"line-number"},[a._v("11")]),e("br"),e("span",{staticClass:"line-number"},[a._v("12")]),e("br"),e("span",{staticClass:"line-number"},[a._v("13")]),e("br"),e("span",{staticClass:"line-number"},[a._v("14")]),e("br"),e("span",{staticClass:"line-number"},[a._v("15")]),e("br"),e("span",{staticClass:"line-number"},[a._v("16")]),e("br"),e("span",{staticClass:"line-number"},[a._v("17")]),e("br"),e("span",{staticClass:"line-number"},[a._v("18")]),e("br"),e("span",{staticClass:"line-number"},[a._v("19")]),e("br"),e("span",{staticClass:"line-number"},[a._v("20")]),e("br"),e("span",{staticClass:"line-number"},[a._v("21")]),e("br"),e("span",{staticClass:"line-number"},[a._v("22")]),e("br"),e("span",{staticClass:"line-number"},[a._v("23")]),e("br"),e("span",{staticClass:"line-number"},[a._v("24")]),e("br"),e("span",{staticClass:"line-number"},[a._v("25")]),e("br"),e("span",{staticClass:"line-number"},[a._v("26")]),e("br"),e("span",{staticClass:"line-number"},[a._v("27")]),e("br"),e("span",{staticClass:"line-number"},[a._v("28")]),e("br"),e("span",{staticClass:"line-number"},[a._v("29")]),e("br"),e("span",{staticClass:"line-number"},[a._v("30")]),e("br"),e("span",{staticClass:"line-number"},[a._v("31")]),e("br")])]),e("p",[a._v("global alignment：全局比对\nthe entirety of the query and reference sequence are to be aligned\n适用于将整个查询和参考基因组对齐的情况\nlocal alignment：局部比对\nonly a subsequence of the query is expected to match a subsequence of the reference\n只有查询的子序列才能与引用的子序列匹配")]),a._v(" "),e("p",[a._v("heuristic：启发式的，有启发的")]),a._v(" "),e("p",[a._v("mapping algorithm：映射算法")]),a._v(" "),e("p",[a._v("spliced alignment, contiguous alignment, and variations of lightweight mapping：\n拼接对齐、连续对齐和轻量级映射的变体")]),a._v(" "),e("p",[a._v("draw a distinction：区别对待")]),a._v(" "),e("p",[a._v("alignment：校准")]),a._v(" "),e("p",[a._v("soft-clipping of the alignment：对齐的软裁剪")]),a._v(" "),e("p",[a._v("memory-frugal：节省内存的")]),a._v(" "),e("p",[a._v("注释：上游得到count matrix阶段的代码，太烧脑了，不懂的术语太多，想要实践又要下载好多包，软件，需要部署太久了，以后，应该都会给到下游的count matrix数据，就暂时不学那么多了吧")]),a._v(" "),e("h4",{attrs:{id:"analysis-frameworks-and-tools"}},[e("a",{staticClass:"header-anchor",attrs:{href:"#analysis-frameworks-and-tools"}},[a._v("#")]),a._v(" analysis frameworks and tools")]),a._v(" "),e("p",[a._v("the exploratory data analysis：有探索性的数据分析")]),a._v(" "),e("p",[a._v("These differ not only in the used programming languages, but also in the underlying data structures and available specialized analysis tools.")]),a._v(" "),e("p",[a._v("这些不仅在所使用的编程语言中有所不同，而且在底层数据结构和可用的专门分析工具中也有所不同。")]),a._v(" "),e("p",[a._v("三种主流工具的特点：")]),a._v(" "),e("p",[e("code",[a._v("Bioconductor")]),e("br"),a._v("\nfocus on rigorous and reproducible analysis of data for many different biological assays including single-cell"),e("br"),a._v("\nrigorous and reproducible：严格且可重复的")]),a._v(" "),e("p",[a._v("A homogeneous developer and user experience and extensive documentation with user friendly   vignettes are the biggest strengths of Bioconductor.")]),a._v(" "),e("p",[a._v("开发和使用者很像（homogeneous），大量用户友好的documentation，vignettes")]),a._v(" "),e("p",[e("code",[a._v("Seurat")])]),a._v(" "),e("p",[a._v("还提供：multimodal and spatial data多模态和空间数据"),e("br"),a._v("\n#还没学")]),a._v(" "),e("p",[a._v("vignettes：小品，应该也算是tutorial")]),a._v(" "),e("p",[e("code",[a._v("Scverse")])]),a._v(" "),e("p",[a._v("scalability issues：可伸缩性的问题")]),a._v(" "),e("p",[a._v("Scalability, extendability and strong interoperability with the existing Python data and machine learning tooling are some of the advantages of the scverse ecosystem")]),a._v(" "),e("p",[a._v("python中的bioconductor——scverse")]),a._v(" "),e("p",[a._v("可伸缩性，可拓展性，强大的互作行，以及现存的python数据，机器学习工具，是其的优势参与还是陷入？？interoperability of the involved frameworks")]),a._v(" "),e("p",[a._v("interoperability：相互操作性")]),a._v(" "),e("div",{staticClass:"language- line-numbers-mode"},[e("pre",{pre:!0,attrs:{class:"language-text"}},[e("code",[a._v("\nStoring unimodal data with AnnData：使用anndata储存单峰数据\nlatent space embeddings：潜在的空间嵌入\nX：sparse matrixs/dense matrixs\npanda dataframe：文字类型数据表\nannData：\n\tobs = observations = cells’ barcodes\n\tvar = variables = gene identifiers\nmatrices of calculations：计算矩阵：obsm，varm\nGraph like structures which associate cells with cells：obsp，varp\nunstructured data：uns\ncount matrix，normalized，unnormalized：layer\nGraph like structures which associate cells with cells or genes with genes are usually saved in obsp and varp\n将细胞与细胞或基因与基因联系在一起的图形结构通常保存在 obsp 和 varp 中\n")])]),a._v(" "),e("div",{staticClass:"line-numbers-wrapper"},[e("span",{staticClass:"line-number"},[a._v("1")]),e("br"),e("span",{staticClass:"line-number"},[a._v("2")]),e("br"),e("span",{staticClass:"line-number"},[a._v("3")]),e("br"),e("span",{staticClass:"line-number"},[a._v("4")]),e("br"),e("span",{staticClass:"line-number"},[a._v("5")]),e("br"),e("span",{staticClass:"line-number"},[a._v("6")]),e("br"),e("span",{staticClass:"line-number"},[a._v("7")]),e("br"),e("span",{staticClass:"line-number"},[a._v("8")]),e("br"),e("span",{staticClass:"line-number"},[a._v("9")]),e("br"),e("span",{staticClass:"line-number"},[a._v("10")]),e("br"),e("span",{staticClass:"line-number"},[a._v("11")]),e("br"),e("span",{staticClass:"line-number"},[a._v("12")]),e("br"),e("span",{staticClass:"line-number"},[a._v("13")]),e("br"),e("span",{staticClass:"line-number"},[a._v("14")]),e("br")])]),e("p",[a._v("random poisson distributed data.：随机泊松分布数据\n.obsm：")]),a._v(" "),e("ul",[e("li",[a._v("The “array-like” metadata can originate from a Pandas DataFrame, scipy sparse matrix, or numpy dense array.")]),a._v(" "),e("li",[a._v("When using scanpy, their values (columns) are not easily plotted, where instead items from .obs are easily plotted on, e.g., UMAP plots.")])]),a._v(" "),e("p",[a._v("Subsetting the joint data matrix can be important to focus on subsets of variables or observations, or to define train-test splits for a machine learning model.")]),a._v(" "),e("p",[a._v("对observation或variable的联合矩阵，取子集，或者对子集进行设置，或为机器学习定义train-test splits等操作很重要")]),a._v(" "),e("p",[a._v('View和".copy()"，修改不修改')]),a._v(" "),e("p",[a._v("新子集对象只是view，指向同一内存地址；copy后自成一家；")]),a._v(" "),e("p",[a._v("原对象修改就修改了，新子集对象修改后默认copy一份自己去过去。")]),a._v(" "),e("p",[a._v("h5ad：太大，partially read it into memory by using backed mode or with the currently experimental read_elem API")]),a._v(" "),e("p",[a._v("通过使用后备模式或当前的实验 read _ elem API 将其部分读入内存")]),a._v(" "),e("div",{staticClass:"language- line-numbers-mode"},[e("pre",{pre:!0,attrs:{class:"language-text"}},[e("code",[a._v('adata = ad.read("my_results.h5ad", backed="r")\nadata.isbacked #True\nadata.filename #PosixPath(\'my_results.h5ad\')\nadata.file.close() #explicitly close，否则read-only模式下，修改不了\n')])]),a._v(" "),e("div",{staticClass:"line-numbers-wrapper"},[e("span",{staticClass:"line-number"},[a._v("1")]),e("br"),e("span",{staticClass:"line-number"},[a._v("2")]),e("br"),e("span",{staticClass:"line-number"},[a._v("3")]),e("br"),e("span",{staticClass:"line-number"},[a._v("4")]),e("br")])]),e("div",{staticClass:"language- line-numbers-mode"},[e("pre",{pre:!0,attrs:{class:"language-text"}},[e("code",[a._v('\nscverse ecosystem：\nscanpy：general RNA-seq\nsquidpy：spatial transcriptomics\nscirpy：analysis T-cell receptor，B-cell receptor data\nsimulation of gene regulatory networks：模拟基因调控网络\nautomate：自动化（scflow）\ninteractive analyses：交互式分析\nimport scanpy and define defaults for our following quick scanpy demo\n定义默认操作\nWe use scanpy’s setting object to set the Matplotlib plotting defaults for all of scanpy’s plots and finally print scanpy’s header. \nThis header contains the versions of all relevant Python packages in the current environment including Scanpy and AnnData. This output is especially useful when reporting bugs to the scverse team and for reproducibility reasons.\n\nThis gene has been shown to have an inverse correlation with cell health. Especially dead/dying cells have a higher expression of MALAT1\nPCA\nunveil the main axes of variation：解释最大方差的轴\nA fundamental step for any advanced embedding and downstream calculations is the calculating of the neighborhood graph using the PCA representation of the data matrix.\n任何高级嵌入和下游计算的基础，是使用PCA计算的，"领域图"  \nPCA representation of the data matrix：数据矩阵的主成分分析表示\nIt is automatically used for other tools that require it such as the calculation of a UMAP\n\nuse the calculating neighborhood graph，计算的邻域图\nto embed the cells with a UMAP，将细胞嵌入UMAP之中，还是用umap嵌入cell？？\ndimension reduction algorithms\nrefresher：提神\n\nCITE-Seq generate multimodal data by simultaneously measuring RNA and surface proteins\n同时测RNA数据和表面蛋白\nmultimodal，多模态，或者叫多模型\nmudata，相比于anndata数据，在其之上多一维modality，相当于上级多一个文件夹，下级仍然是anndata\n.mod (=modality) attribute，翻译过来，叫”模式“\ntwo AnnData objects (two “modalities”) can then be wrapped into a single MuData object.\n一般.obj_names是一样，但是var_names不一样，相当于”同一个研究对象多一组数据“\n')])]),a._v(" "),e("div",{staticClass:"line-numbers-wrapper"},[e("span",{staticClass:"line-number"},[a._v("1")]),e("br"),e("span",{staticClass:"line-number"},[a._v("2")]),e("br"),e("span",{staticClass:"line-number"},[a._v("3")]),e("br"),e("span",{staticClass:"line-number"},[a._v("4")]),e("br"),e("span",{staticClass:"line-number"},[a._v("5")]),e("br"),e("span",{staticClass:"line-number"},[a._v("6")]),e("br"),e("span",{staticClass:"line-number"},[a._v("7")]),e("br"),e("span",{staticClass:"line-number"},[a._v("8")]),e("br"),e("span",{staticClass:"line-number"},[a._v("9")]),e("br"),e("span",{staticClass:"line-number"},[a._v("10")]),e("br"),e("span",{staticClass:"line-number"},[a._v("11")]),e("br"),e("span",{staticClass:"line-number"},[a._v("12")]),e("br"),e("span",{staticClass:"line-number"},[a._v("13")]),e("br"),e("span",{staticClass:"line-number"},[a._v("14")]),e("br"),e("span",{staticClass:"line-number"},[a._v("15")]),e("br"),e("span",{staticClass:"line-number"},[a._v("16")]),e("br"),e("span",{staticClass:"line-number"},[a._v("17")]),e("br"),e("span",{staticClass:"line-number"},[a._v("18")]),e("br"),e("span",{staticClass:"line-number"},[a._v("19")]),e("br"),e("span",{staticClass:"line-number"},[a._v("20")]),e("br"),e("span",{staticClass:"line-number"},[a._v("21")]),e("br"),e("span",{staticClass:"line-number"},[a._v("22")]),e("br"),e("span",{staticClass:"line-number"},[a._v("23")]),e("br"),e("span",{staticClass:"line-number"},[a._v("24")]),e("br"),e("span",{staticClass:"line-number"},[a._v("25")]),e("br"),e("span",{staticClass:"line-number"},[a._v("26")]),e("br"),e("span",{staticClass:"line-number"},[a._v("27")]),e("br"),e("span",{staticClass:"line-number"},[a._v("28")]),e("br"),e("span",{staticClass:"line-number"},[a._v("29")]),e("br"),e("span",{staticClass:"line-number"},[a._v("30")]),e("br"),e("span",{staticClass:"line-number"},[a._v("31")]),e("br"),e("span",{staticClass:"line-number"},[a._v("32")]),e("br"),e("span",{staticClass:"line-number"},[a._v("33")]),e("br")])]),e("div",{staticClass:"language- line-numbers-mode"},[e("pre",{pre:!0,attrs:{class:"language-text"}},[e("code",[a._v('mdata = md.MuData({"A": adata, "B": adata2})\n#MuData object with n_obs × n_vars = 1000 × 150\n# 2 modalities\n#   A:\t1000 x 100\n#   B:\t1000 x 50\nlist(mdata.mod.keys()) #[\'A\', \'B\']\nprint(mdata.mod["A"])/print(mdata["A"]) #AnnData object with n_obs × n_vars = 1000 × 100\nmdata.update() #需要多来几步更新\n')])]),a._v(" "),e("div",{staticClass:"line-numbers-wrapper"},[e("span",{staticClass:"line-number"},[a._v("1")]),e("br"),e("span",{staticClass:"line-number"},[a._v("2")]),e("br"),e("span",{staticClass:"line-number"},[a._v("3")]),e("br"),e("span",{staticClass:"line-number"},[a._v("4")]),e("br"),e("span",{staticClass:"line-number"},[a._v("5")]),e("br"),e("span",{staticClass:"line-number"},[a._v("6")]),e("br"),e("span",{staticClass:"line-number"},[a._v("7")]),e("br"),e("span",{staticClass:"line-number"},[a._v("8")]),e("br")])]),e("p",[a._v("手动更新！mdata.update()")]),a._v(" "),e("div",{staticClass:"language- line-numbers-mode"},[e("pre",{pre:!0,attrs:{class:"language-text"}},[e("code",[a._v("\nAnalogous to the behavior of AnnData objects：类似于\nslicing MuData objects returns views of the original data：切片返回原始数据\nserialized into：连载\nconcatenate matrices from multiple modalities to perform for example dimensionality reduction\n连接矩阵\na place to store multimodal embeddings: MuData.obsm.\n\nconcatenate：连接\nstacked matrix：叠加矩阵\nhorizontally stack：水平叠加\nAnalogously to：类似于\na quick demo and overview：快速的演示和概述\nGenomics Multiome dataset：基因组多组数据\nrudimentary filtered out cells：最基本的过滤细胞\nPerform rudimentary quality control with muon's ATAC module：\nlatent semantic indexing：潜在语义索引\n")])]),a._v(" "),e("div",{staticClass:"line-numbers-wrapper"},[e("span",{staticClass:"line-number"},[a._v("1")]),e("br"),e("span",{staticClass:"line-number"},[a._v("2")]),e("br"),e("span",{staticClass:"line-number"},[a._v("3")]),e("br"),e("span",{staticClass:"line-number"},[a._v("4")]),e("br"),e("span",{staticClass:"line-number"},[a._v("5")]),e("br"),e("span",{staticClass:"line-number"},[a._v("6")]),e("br"),e("span",{staticClass:"line-number"},[a._v("7")]),e("br"),e("span",{staticClass:"line-number"},[a._v("8")]),e("br"),e("span",{staticClass:"line-number"},[a._v("9")]),e("br"),e("span",{staticClass:"line-number"},[a._v("10")]),e("br"),e("span",{staticClass:"line-number"},[a._v("11")]),e("br"),e("span",{staticClass:"line-number"},[a._v("12")]),e("br"),e("span",{staticClass:"line-number"},[a._v("13")]),e("br"),e("span",{staticClass:"line-number"},[a._v("14")]),e("br"),e("span",{staticClass:"line-number"},[a._v("15")]),e("br"),e("span",{staticClass:"line-number"},[a._v("16")]),e("br"),e("span",{staticClass:"line-number"},[a._v("17")]),e("br")])]),e("h4",{attrs:{id:"preprocessing-and-visualization"}},[e("a",{staticClass:"header-anchor",attrs:{href:"#preprocessing-and-visualization"}},[a._v("#")]),a._v(" preprocessing and visualization")]),a._v(" "),e("p",[a._v("quality control--normalization--feature selection--dimensionality reduction\n质控")]),a._v(" "),e("ul",[e("li",[a._v("去掉死细胞，去掉双细胞\n标准化")]),a._v(" "),e("li",[a._v("可比性\n选择特征")]),a._v(" "),e("li",[a._v("选择有差异的基因\n降维聚类")]),a._v(" "),e("li")]),a._v(" "),e("p",[a._v("dedicated yo：专心致志于\nannotating your cell：注释你的细胞\noutperform other tools：胜过其他工具\nbone marrow mononuclear cells：骨髓单核细胞\nobtain nested batch effects：获得嵌套的批处理效应\naforementioned dataset：上述数据集\nambient RNA correction：环境RNA矫正")]),a._v(" "),e("p",[a._v("测序过程的一个假设：一个drop一个正常细胞\n但是有可能：空胞，多胞，死细胞")]),a._v(" "),e("p",[a._v("数据上来看就是：\nobs/cell测到的基因太少了，测到的的线粒体基因太多了\nvar/gene检测到的太少了，相当于被表达的太少了")]),a._v(" "),e("p",[a._v("a relatively high percentage of "),e("strong",[a._v("mitochondrial")]),a._v(" counts which are often associated with cell degradation\n线粒体含量越高和细胞退化相关")]),a._v(" "),e("p",[a._v("as datasets grow in size this task is becoming more and more time-consuming and it might be worth considering automatic thresholding via MAD (median absolute deviations). The MAD is given by (MAD = median(|X_i - median(X)|)) with (X_i) being the respective QC metric of an observation and describes a robust statistic of the variability of the metric\n数据规模变得太大了之后特别消耗时间，所以通过MAD，中值绝对偏差，来作为阈值来筛掉一部分\n中值绝对偏差，MAD，每个数减去中值后，差值得绝对值，再取其中值；\n对于数据集来说，比标准差更适合排除异常值（贴近中值）\n主要用途：统计离差，检测/排除异常点（经过计算，在几倍的MAD之内能得到占比多少的数据）")]),a._v(" "),e("ul",[e("li",[a._v("方差，S^2，每个数与平均值的差值，平方之和")]),a._v(" "),e("li",[a._v("标准差，S，每个数与平均值的差值，平方之和后再开算数平方根")])]),a._v(" "),e("p",[a._v("Correction of ambient RNA：矫正环境RNA")]),a._v(" "),e("p",[a._v("net effect：净效应（净含量，效应）")]),a._v(" "),e("p",[a._v("background contamination：背景污染")]),a._v(" "),e("p",[a._v("in the dilution：在稀释中")]),a._v(" "),e("p",[a._v("cell lysis：细胞溶解")]),a._v(" "),e("p",[a._v("confounded counts.：混乱的计数")]),a._v(" "),e("p",[a._v("Cell-free mRNA molecules, also known as ambient RNA,环境RNA")]),a._v(" "),e("p",[a._v("partitions (communities) of cells：细胞的分群")]),a._v(" "),e("p",[a._v("Leiden clustering：莱顿聚类（默认的那种聚类方式）")]),a._v(" "),e("p",[a._v("Doublet Detection")]),a._v(" "),e("p",[a._v("A doublet is called homotypic：双胞被称为同型（单个单词--同性恋）")]),a._v(" "),e("p",[a._v("Homotypic doublets are not necessarily identifiable from count matrices and are often considered innocuous as they can be identified with cell hashing or SNPs")]),a._v(" "),e("p",[a._v("Hence, their identification is not the main goal of the doublet detection methods.")]),a._v(" "),e("p",[a._v("Doublets formed from different cell types or states are called heterotypic")]),a._v(" "),e("p",[a._v("不同类型（不同type，不同状态）的细胞才叫双胞，影响较大的那种")]),a._v(" "),e("p",[a._v("Doublets can be either identified through their high number of reads and detected features, or with methods that create artificial doublets and compare these with the cells present in the dataset.")]),a._v(" "),e("p",[a._v("测到太多reads和features，或者比较样本内差距")]),a._v(" "),e("p",[a._v("scDblFinder R package")]),a._v(" "),e("p",[a._v("scDblFinder randomly selects two droplets and creates artificial doublets from those by averaging their gene expression profiles")]),a._v(" "),e("p",[a._v("The doublet score is then defined as the fraction of artificial doublets in the k-nearest neighbor graph of each droplet in the principal component space.")]),a._v(" "),e("p",[a._v("take away points")]),a._v(" "),e("p",[a._v("Filtering of poor-quality cells should be based on median absolute deviations with lenient cutoffs to avoid bias against smaller subpopulations.")]),a._v(" "),e("p",[a._v("Feature-based filtering does not show benefits for downstream tasks.")]),a._v(" "),e("p",[a._v("Doublets can be efficiently detected with tools like scDblFinder.")]),a._v(" "),e("p",[a._v("Doublet detection methods should not be run on aggregated scRNA-seq data representing multiple batches.")]),a._v(" "),e("p",[a._v("Normalization")]),a._v(" "),e("p",[a._v("Gamma-Poisson distribution"),e("br"),a._v("\nA theoretically and empirically established model for UMI data is the Gamma-Poisson distribution which implies a quadratic mean-variance relation with with mean and overdispersion .")]),a._v(" "),e("p",[a._v("基于delta方法的log转换")]),a._v(" "),e("p",[a._v("shifted logarithm which is based on the delta method")]),a._v(" "),e("p",[a._v("size factors：尺寸系数（标准化的数值）")]),a._v(" "),e("p",[a._v("Overdispersion describes the presence of a greater variability in the dataset than one would expect.\nThe shifted logarithm is a fast normalization technique, outperforms other methods for uncovering the latent structure of the dataset (especially when followed by principal component analysis)\nworks beneficial for stabilizing variance for subsequent dimensionality reduction and identification of differentially expressed genes.有助于稳定后续降维的变异和鉴定差异表达的基因。")]),a._v(" "),e("p",[a._v("Analytic Pearson residuals：分析皮尔森残差")]),a._v(" "),e("p",[a._v("Although RNA velocity has been applied successfully to many systems, some model limitations persist. Violated model assumptions may cause erronous result")]),a._v(" "),e("p",[a._v("RNA速度部分比较成功，但是一些模型确限制了。违反了模型假设，可能会造成问题结果。")]),a._v(" "),e("p",[a._v("pitfalls：陷阱")]),a._v(" "),e("p",[a._v("inferred velocity field to infer likely future states of a cell：推断速度场来推断细胞未来可能的状态")]),a._v(" "),e("p",[a._v("reformulate the inference methods：重新定制推理方法")]),a._v(" "),e("p",[a._v("RNA velocity：可能会有一些限制，现在的一些算法更多的采用更原始数据，以防止漏掉信息。")]),a._v(" "),e("p",[a._v("因为，这个速度分析的原理基础是，功能状态变化的过程中，基因表达种类和数量是相似的，更强调相似性，而之前的寻找高变基因降维聚类则大大减少了这类的数据。")]),a._v(" "),e("p",[a._v("technical sampling effects in gene expression：normalization的目的是去掉技术样本效应")]),a._v(" "),e("p",[e("strong",[a._v("feature selection")]),a._v(" which aims to exclude uninformative genes which might not represent meaningful biological variation across samples\n去掉“没有”生物学信息的基因")]),a._v(" "),e("p",[a._v("过去：either compute the coefficient of variation (highly variable genes) or the average expression level (highly expressed genes) of all genes to obtain 500-2000 selected genes")]),a._v(" "),e("p",[a._v("计算所有基因的变异系数(高度可变的基因)或平均表达水平(高度表达的基因)")]),a._v(" "),e("p",[a._v("analysts often add a small pseudo count：分析师常加上一个小小的伪计数")]),a._v(" "),e("p",[a._v("arbitrary：随心所欲")]),a._v(" "),e("p",[a._v("proposes to use deviance for feature selection which works on raw counts\n基于原始计数的特征选择")]),a._v(" "),e("p",[a._v("deviance：越轨行为")]),a._v(" "),e("p",[a._v("multinomial null model：多项式零模型")]),a._v(" "),e("p",[a._v("binomial deviance：二项式偏差")]),a._v(" "),e("p",[a._v("high deviance value which indicates a poor fit by the null model\n高信息的基因的特点：偏离值高，")]),a._v(" "),e("p",[a._v("a poor fit by the null model：空模型模拟不好")]),a._v(" "),e("p",[a._v("dimensionality reduction algorithms：降维算法")]),a._v(" "),e("p",[a._v("to reduce the data complexity and for visualization：降低复杂度是为了可视化")]),a._v(" "),e("p",[a._v("Dimensionality reduction embeds the high-dimensional data into a lower dimensional space：高维数据"),e("strong",[a._v("嵌入")]),a._v("更低维度，这个嵌入就不是那么理解")]),a._v(" "),e("p",[a._v("still captures the underlying structure of the data：仍然保持着高维数据的结构")]),a._v(" "),e("p",[a._v("图例：\n"),e("img",{attrs:{src:"https://cdn.statically.io/gh/hongliangfang/picx-images-hosting@master/study/dimensionality_reduction.4m9u9aasba40.webp",alt:"dimensionality_reduction"}})]),a._v(" "),e("p",[a._v("三种降维聚类的特点（都跑一遍算了）\nt-distributed stochastic neighbor embedding (t-SNE) as it yielded the best overall performance.")]),a._v(" "),e("p",[a._v("Uniform manifold approximation and projection (UMAP) showed the highest stability and separates best the original cell populations.")]),a._v(" "),e("p",[a._v("principal component analysis (PCA) which is still widely used")]),a._v(" "),e("p",[a._v("t-SNE and UMAP are very robust and mostly equivalent if specific choices for the initialization are selected")]),a._v(" "),e("p",[a._v("初始化被选择了之后，就是非常稳健，相似的")]),a._v(" "),e("p",[a._v("We will use a normalized representation of the dataset for dimensionality reduction and visualization, "),e("strong",[a._v("specifically")]),a._v(" the shifted logarithm.")]),a._v(" "),e("p",[a._v("使用规范化的数据用来降维和可视化")]),a._v(" "),e("p",[a._v("spanned by some orthonormal basis：被一些标准正交基")]),a._v(" "),e("p",[a._v("PCA offers the advantage that it is highly interpretable and computationally efficient. However, as scRNA-seq datasets are rather sparse due to dropout events and therefore highly non-linear, visualization with the linear dimensionality reduction technique PCA is not very appropriate. PCA is typically used to select the top 10-50 PCs which are used for downstream analysis tasks.")]),a._v(" "),e("p",[a._v("交互方便，计算高效；")]),a._v(" "),e("p",[a._v("但是数据比较稀疏，高度非线性")]),a._v(" "),e("p",[a._v("PCA是线性降维方法，不太合适")]),a._v(" "),e("p",[a._v("PCA通常是选择前10~50个主成分用于下游分析")]),a._v(" "),e("p",[a._v("t-SNE is a graph based, non-linear dimensionality reduction technique which projects the high dimensional data onto 2D or 3D components. The method defines a Gaussian probability distribution based on the high-dimensional Euclidean distances between data points. Subsequently, a Student t-distribution is used to recreate the probability distribution in a low dimensional space where the embeddings are optimized using gradient descent.")]),a._v(" "),e("p",[a._v("图像依赖的，非线性降维技术，能产生2D或3D")]),a._v(" "),e("p",[a._v("高维有欧式距离的计算分析，低维的有t-分布的检验，算是双重验证了吧")]),a._v(" "),e("p",[a._v("UMAP is a graph based, non-linear dimensionality technique and principally similar to t-SNE. It constructs a high dimensional graph representation of the dataset and optimizes the low-dimensional graph representation to be structurally as similar as possible to the original graph.")]),a._v(" "),e("p",[a._v("图像依赖的，非线性降维，类似于tSNE")]),a._v(" "),e("p",[a._v("Inspecting quality control metrics：检查质量控制指标\ncells with a high doublet score are projected to the same region in the UMAP")]),a._v(" "),e("p",[a._v("具有双倍高分的细胞，被分到同一个UMAP图中")]),a._v(" "),e("p",[a._v("Clustering\nClustering itself is a common unsupervised machine learning problem：聚类本身是一个非监督的机器学习问题")]),a._v(" "),e("p",[a._v("derive clusters by minimizing the intra-cluster distance in the reduced expression space\n在减小的表达空间内，再使cluster内的distance变小，从而产生cluster")]),a._v(" "),e("p",[a._v("topology："),e("strong",[a._v("拓扑学")])]),a._v(" "),e("p",[a._v("Dense regions in the KNN-graph are detected by community detection methods like Leiden and Louvain\nThe "),e("strong",[a._v("Leiden")]),a._v(" algorithm is as an improved version of the Louvain algorithm which outperformed other clustering methods for single-cell RNA-seq data analysis")]),a._v(" "),e("p",[a._v("propose to use the Leiden algorithm on single-cell k-nearest-neighbour (KNN) graphs to cluster single-cell datasets")]),a._v(" "),e("p",[a._v("Leiden creates clusters by taking into account the number of links between cells in a cluster versus the overall expected number of links in the dataset.")]),a._v(" "),e("p",[a._v("考虑细胞在cluster中和数据集中的链接来创造XXX")]),a._v(" "),e("p",[e("img",{attrs:{src:"https://cdn.statically.io/gh/hongliangfang/picx-images-hosting@master/study/clustering.4s1ll0heffy0.webp",alt:"clustering"}})]),a._v(" "),e("p",[a._v("PC--KNN graph--Leiden algorithm\nreduce expression spacing--initial node graph(community)--move and try，refine，that 构成新的 aggregate network(new community)")]),a._v(" "),e("p",[a._v("The Leiden module has a resolution parameter which allows to determine the scale of the partition cluster and therefore the coarseness of the clustering.\n精度越高越多群\n还可以为子群设置参数，相当于区分亚群！！")]),a._v(" "),e("p",[a._v("much coarser：非常粗糙\nWe recommend to not interprete distances between clusters visualized on UMAP embeddings\n我们建议，不要，解释在UMAP上嵌入，cluster之间的距离")]),a._v(" "),e("p",[a._v("Use Leiden community detection on a single-cell KNN graph.\nSub-clustering with different resolution parameters allows the user to focus on more detailed substructures in the dataset to potentially identify finer cell states.")]),a._v(" "),e("p",[a._v("Cell annotation")]),a._v(" "),e("p",[a._v("knowing that there are specific immune cell types in a tumor or unusual hematopoietic stem cells in your bone marrow sample can be a valuable insight into the disease you might be studying.")]),a._v(" "),e("p",[a._v("肿瘤细胞的类型和不同寻常的造血干细胞")]),a._v(" "),e("p",[a._v("Cell types are therefore often further classified into “subtypes” or “cell states” (e.g. activated versus resting) and some researchers use the term “cell identity” to avoid this sometimes arbitrary distinction of cell types, cell subtypes and cell states.")]),a._v(" "),e("p",[a._v("因为可能没有太大的生物学意义，所以，有时候不同的细胞，叫法也会改变")]),a._v(" "),e("p",[a._v("有时候使用细胞识别，而非下面，以")]),a._v(" "),e("p",[a._v("避免随意的分类，，细胞类型，，细胞亚群，，细胞状态")]),a._v(" "),e("p",[a._v("differentiation coordinate：微分坐标\n我觉得细胞群体像是一个社会群体，不同的状态有点像是连续分布的，各种各样子的都会有。细胞内部的RNA也许有很多很多，之间会相互影响的啊；接着哪些细胞做出了什么事情也蛮重要的吧。。")]),a._v(" "),e("p",[a._v("differentiation and cellular trajectories：分化和细胞轨迹\ndeprecation and performance warnings that do not affect our code:：过滤掉不影响代码弃用和性能的警告")]),a._v(" "),e("p",[a._v("transparent：透明的")]),a._v(" "),e("p",[a._v("Clustering enables the detection of cells highly similar in overall gene expression and can therefore account for drop-outs at single cell level.\n聚类可以检测到整体基因表达高度相似的细胞，因此可以解释单个细胞水平的脱落")]),a._v(" "),e("p",[a._v("手动注释步骤：\n定好基因的类型，一个个筛，看cluster中这个基因表达高，符合要求，一个一个定义群")]),a._v(" "),e("p",[a._v("或者，一个一个cluster看，哪些基因表达高，特殊，定义下来哪种细胞")]),a._v(" "),e("p",[a._v("（就是从确定好的基因来看，还是从确定好的簇来看）")]),a._v(" "),e("p",[a._v("FACS数据（细胞的蛋白质数据），不一定适应转录组数据")]),a._v(" "),e("p",[a._v("Ideally a marker set is therefore validated across multiple datasets")]),a._v(" "),e("p",[a._v("和useful to work together with experts: as a bioinformatician, try to team up with a biologist who has more extensive knowledge of the tissue, the biology, the expected cell types and markers etc.")]),a._v(" "),e("p",[a._v("highly deviant genes：高度异常的基因")]),a._v(" "),e("p",[a._v("it is the intersection of those subsets that will tell you where your cell type of interest is.：")]),a._v(" "),e("p",[a._v("将是这些子集的交集，才能告诉你，你感兴趣的细胞在哪里")]),a._v(" "),e("h3",{attrs:{id:"regulation"}},[e("a",{staticClass:"header-anchor",attrs:{href:"#regulation"}},[a._v("#")]),a._v(" regulation")]),a._v(" "),e("p",[a._v("scenic")])])}),[],!1,null,null,null);e.default=t.exports}}]);