(window.webpackJsonp=window.webpackJsonp||[]).push([[60],{378:function(e,t,a){"use strict";a.r(t);var s=a(7),_=Object(s.a)({},(function(){var e=this,t=e._self._c;return t("ContentSlotsDistributor",{attrs:{"slot-key":e.$parent.slotKey}},[t("h3",{attrs:{id:"参考资料"}},[t("a",{staticClass:"header-anchor",attrs:{href:"#参考资料"}},[e._v("#")]),e._v(" 参考资料")]),e._v(" "),t("p",[e._v("原文："),t("a",{attrs:{href:"https://www.nature.com/articles/s41592-023-02117-1",target:"_blank",rel:"noopener noreferrer"}},[e._v("SEVtras delineates small extracellular vesicles at droplet resolution from single-cell transcriptomes | Nature Methods"),t("OutboundLink")],1)]),e._v(" "),t("p",[e._v("源码："),t("a",{attrs:{href:"https://github.com/bioinfo-biols/SEVtras",target:"_blank",rel:"noopener noreferrer"}},[e._v("SEVtras - github.com"),t("OutboundLink")],1)]),e._v(" "),t("p",[e._v("官方教程："),t("a",{attrs:{href:"https://sevtras.readthedocs.io/en/latest/",target:"_blank",rel:"noopener noreferrer"}},[e._v("SEVtras 0.1 documentation"),t("OutboundLink")],1)]),e._v(" "),t("p",[e._v("首先，为啥叫这个名字呢？"),t("br"),e._v("\nSEVtras stands for "),t("strong",[e._v("sEV")]),e._v("-con"),t("strong",[e._v("t")]),e._v("aining d"),t("strong",[e._v("r")]),e._v("oplet identific"),t("strong",[e._v("a")]),e._v("tion in "),t("strong",[e._v("s")]),e._v("cRNA-seq data.")]),e._v(" "),t("p",[e._v("You can freely use SEVtras to explore sEV heterogeneity at single droplet, characterize cell type dynamics in light of sEV activity and unlock diagnostic potential of sEVs in concert with cells.")]),e._v(" "),t("blockquote",[t("p",[e._v("你可以自由的，在单个液滴地水平，使用SEVtras探索 "),t("code",[e._v("sEV")]),e._v(" 的异质性，根据 sEV 的活性，表征不同细胞的分泌活性，还可以与细胞一起，解锁"),t("code",[e._v("sEV")]),e._v("的诊断潜能/价值。")])]),e._v(" "),t("p",[t("img",{attrs:{src:"https://sevtras.readthedocs.io/en/latest/_images/SEVtras_overview.png",alt:"概览"}})]),e._v(" "),t("h3",{attrs:{id:"步骤-功能"}},[t("a",{staticClass:"header-anchor",attrs:{href:"#步骤-功能"}},[e._v("#")]),e._v(" 步骤 & 功能")]),e._v(" "),t("p",[e._v("分析分了三个步骤")]),e._v(" "),t("ol",[t("li",[e._v("识别液滴，找到含有"),t("code",[e._v("小细胞外囊泡")]),e._v("的小液滴")]),e._v(" "),t("li",[e._v("功能分析，分析差异基因的功能")]),e._v(" "),t("li",[e._v("临床应用，把临床信息补充上，进行分析")])]),e._v(" "),t("p",[e._v("然后他能做什么呢？")]),e._v(" "),t("ul",[t("li",[e._v("单个液滴的分辨率下，"),t("code",[e._v("识别sEV")])]),e._v(" "),t("li",[t("code",[e._v("计算")]),e._v("每个样本组织的sEV分泌活性指数--"),t("code",[e._v("ESAI")])]),e._v(" "),t("li",[e._v("通过反卷积，根据EV分泌行为，识别不同的"),t("code",[e._v("细胞类型")]),e._v("，也就是"),t("code",[e._v("ESAI_c")])]),e._v(" "),t("li",[e._v("发现"),t("code",[e._v("临床")]),e._v("潜力")])]),e._v(" "),t("h3",{attrs:{id:"函数参数"}},[t("a",{staticClass:"header-anchor",attrs:{href:"#函数参数"}},[e._v("#")]),e._v(" 函数参数")]),e._v(" "),t("p",[e._v("总共五个，主要用的估计也就两三个")]),e._v(" "),t("p",[e._v("1、识别函数，SEVtras.sEV_recognizer()："),t("br"),e._v(" "),t("code",[e._v("SEVtras.sEV_recognizer(sample_file, out_path, input_path=None, species='Homo', predefine_threads=-2, get_only=False, score_t=None, search_UMI=500, alpha=0.1, dir_origin=True)")])]),e._v(" "),t("ul",[t("li",[t("em",[e._v("sample_file")]),e._v(": 输入文件路径，通向的每个样本按行排列，the path of each sample row by row,")]),e._v(" "),t("li",[t("em",[e._v("out_path")]),e._v(": 输出文件路径，the path for output files,")]),e._v(" "),t("li",[t("em",[e._v("input_path")]),e._v(": 如果所有的输入文件在一个文件夹，我们直接用这个代表路径，if all input files in the same directory, we can use this to represent the path, default is "),t("code",[e._v("None")]),e._v(",")]),e._v(" "),t("li",[t("em",[e._v("species")]),e._v(": 测序物种，人或小鼠的，"),t("code",[e._v("Homo")]),e._v("  or "),t("code",[e._v("Mus")]),e._v("，the species from which the scRNA-seq sample was sequenced, default is "),t("code",[e._v("Homo")]),e._v(". For mouse samples, you can use "),t("code",[e._v("Mus")]),e._v(",")]),e._v(" "),t("li",[t("em",[e._v("predefine_threads")]),e._v(": 预处理的线程（负数），默认是总线程减2，SEVtras uses parallel processing for acceleration, we can define how many cpu cores to use, default is all cpu cores minus two "),t("code",[e._v("-2")]),e._v(",")]),e._v(" "),t("li",[t("em",[e._v("get_only")]),e._v(": 是否读取数据中的蛋白质信息，默认不读，whether to read protein information in the adata, default is "),t("code",[e._v("False")]),e._v(",")]),e._v(" "),t("li",[t("em",[e._v("score_t")]),e._v(": 识别sEV的阈值，找不到可以改小一点，the threshold for SEVtras score to recognize sEVs, default is "),t("code",[e._v("None")]),e._v(". If no sEVs found in the "),t("em",[e._v("sEVs_SEVtras.h5ad")]),e._v(", we can change "),t("em",[e._v("score_t")]),e._v(" to a smaller threshold ("),t("code",[e._v("str")]),e._v("), e.g. ‘10’,")]),e._v(" "),t("li",[t("em",[e._v("search_UMI")]),e._v(": 识别UMI的范围，相当于识别小囊泡UMI的数量最低限度，找不到可以降低“标准”，the UMI range to search for sEVs, default is "),t("code",[e._v("500")]),e._v(", you can use "),t("code",[e._v("200")]),e._v(" for stricter recognization,")]),e._v(" "),t("li",[t("em",[e._v("alpha")]),e._v(": 鉴定每个样本的 sEV 代表性基因的参数，数量不够可以提高点，the parameter for identifying sEV representative genes for each sample, default is "),t("code",[e._v("0.01")]),e._v(". If you cannot detect sEVs in all samples, the parameter can be loosened to a smaller value, e.g. "),t("code",[e._v("0.09")]),e._v(",")]),e._v(" "),t("li",[t("em",[e._v("dir_origin")]),e._v(": （暂时不懂）the path of "),t("em",[e._v("matrix.mtx.gz")]),e._v(", default is "),t("code",[e._v("True")]),e._v(" assuming that the file of "),t("em",[e._v("matrix.mtx.gz")]),e._v(" locates at "),t("code",[e._v("sample/outs/raw_feature_bc_matrix/")]),e._v(". If you set it as "),t("code",[e._v("False")]),e._v(", SEVtras search for "),t("em",[e._v("matrix.mtx.gz")]),e._v(" in the path of each sample in the parameter "),t("em",[e._v("sample_file")]),e._v(".")])]),e._v(" "),t("p",[e._v("2、计算函数，SEVtras.ESAI_calculator()：\n"),t("code",[e._v("SEVtras.ESAI_calculator(adata_ev, adata_cell, out_path, OBSsample='batch', OBScelltype='celltype', OBSev='sEV', OBSMpca='X_pca', cellN=10, Xraw = True, normalW=True, plot_cmp='SEV_builtin', save_plot_prefix='', OBSMumap='X_umap',size=10)")])]),e._v(" "),t("ul",[t("li",[t("em",[e._v("adata_ev")]),e._v(": sEV对象数据路径，the path to sEV-anndata objects,")]),e._v(" "),t("li",[t("em",[e._v("adata_cell")]),e._v(": anndata数据类型细胞对象，the path to cell-anndata objects,")]),e._v(" "),t("li",[t("em",[e._v("out_path")]),e._v(": 输出数据路径，the path for output files,")])]),e._v(" "),t("p",[e._v("下面几个是对应anndata的"),t("code",[e._v("obs")]),e._v("的几个参数名字")]),e._v(" "),t("ul",[t("li",[t("em",[e._v("OBSsample")]),e._v(": the index represents the sample information in the "),t("code",[e._v("obs")]),e._v(" of adata, default is "),t("code",[e._v("batch")]),e._v(",")]),e._v(" "),t("li",[t("em",[e._v("OBScelltype")]),e._v(": the index represents the cell type information in the "),t("code",[e._v("obs")]),e._v(" of adata, default is "),t("code",[e._v("celltype")]),e._v(",")]),e._v(" "),t("li",[t("em",[e._v("OBSev")]),e._v(": the index represents the sEV information in the "),t("code",[e._v("obs")]),e._v(" of adata, default is "),t("code",[e._v("sEV")]),e._v(",")]),e._v(" "),t("li",[t("em",[e._v("OBSMpca")]),e._v(": the index represents the PCA information in the "),t("code",[e._v("obsm")]),e._v(" of adata, default is "),t("code",[e._v("X_pca")]),e._v(",")]),e._v(" "),t("li",[t("em",[e._v("cellN")]),e._v(": 反卷积"),t("code",[e._v("邻居")]),e._v("的数量，the number of neighors used for ESAI deconvolution, default is "),t("code",[e._v("10")]),e._v(",")]),e._v(" "),t("li",[t("em",[e._v("Xraw")]),e._v(": 是否使用原始数据，whether or not to use the raw object in the "),t("code",[e._v("adata_cell")]),e._v(". If "),t("code",[e._v("adata_cell")]),e._v(" has been filtered or normalized, please set "),t("code",[e._v("Xraw=True")]),e._v(", and "),t("code",[e._v("adata_cell.raw")]),e._v(" will be used. Note: save raw "),t("code",[e._v("adata_cell")]),e._v(" as "),t("code",[e._v("adata_cell.raw")]),e._v(" before filtering. Default is "),t("code",[e._v("True")]),e._v(",")]),e._v(" "),t("li",[t("em",[e._v("normalW")]),e._v(": 是否缩放，whether or not to scale "),t("code",[e._v("adata_cell")]),e._v(" in ESAI deconvolution, default is "),t("code",[e._v("True")]),e._v(",")]),e._v(" "),t("li",[t("em",[e._v("plot_cmp")]),e._v(": 调色板，the pallete used for plot different cell types in umap, default is "),t("code",[e._v("SEV_builtin")]),e._v(", you can use other pallete in matplotlib e.g. "),t("code",[e._v("Set2")]),e._v(",")]),e._v(" "),t("li",[t("em",[e._v("save_plot_prefix")]),e._v(": 保存文件前缀名字，默认为空，the prefix name for saved files, default is "),t("code",[e._v("''")]),e._v(",")]),e._v(" "),t("li",[t("em",[e._v("OBSMumap")]),e._v(": UMAP绘图参数，the index represents the umap information in the "),t("code",[e._v("obsm")]),e._v(" of adata, default is "),t("code",[e._v("X_umap")]),e._v(",")]),e._v(" "),t("li",[t("em",[e._v("size")]),e._v(": 绘图点的大小，the size of point in umap plot, default is "),t("code",[e._v("10")]),e._v(".")])]),e._v(" "),t("p",[e._v("3、SEVtras.sEV_imputation()：\n"),t("code",[e._v("SEVtras.sEV_imputation(adata_sEV)")]),e._v("\n此函数用于 sEV 数据插补")]),e._v(" "),t("p",[e._v("4、SEVtras.cellfree_simulator()：\n"),t("code",[e._v("SEVtras.cellfree_simulator(out_path, gene_exp_ev, gene_exp_cell, expect_UMI = [40, 70, 100, 130], sEV_fraction = [0.005, 0.01, 0.05, 0.10], sEV=500)")]),e._v("\n该函数用于模拟，无细胞液滴")]),e._v(" "),t("p",[e._v("5、富集函数，SEVtras.sEV_enrichment(()：\n"),t("code",[e._v("SEVtras.sEV_enrichment(adata_sEV, nBP=15)")]),e._v("\n此函数用于 sEV 数据 GO 富集")]),e._v(" "),t("h3",{attrs:{id:"开始流程"}},[t("a",{staticClass:"header-anchor",attrs:{href:"#开始流程"}},[e._v("#")]),e._v(" 开始流程")]),e._v(" "),t("p",[e._v("下载安装，建议conda下新建个空间")]),e._v(" "),t("div",{staticClass:"language-python line-numbers-mode"},[t("pre",{pre:!0,attrs:{class:"language-python"}},[t("code",[e._v("conda create "),t("span",{pre:!0,attrs:{class:"token operator"}},[e._v("-")]),e._v("y "),t("span",{pre:!0,attrs:{class:"token operator"}},[e._v("-")]),e._v("n SEVtras_env python"),t("span",{pre:!0,attrs:{class:"token operator"}},[e._v("=")]),t("span",{pre:!0,attrs:{class:"token number"}},[e._v("3.7")]),e._v(" "),t("span",{pre:!0,attrs:{class:"token comment"}},[e._v("# 3.7待考虑，后续重建才使第二部的gseapy可行")]),e._v("\nsource activate SEVtras_env\npip install SEVtras\n")])]),e._v(" "),t("div",{staticClass:"line-numbers-wrapper"},[t("span",{staticClass:"line-number"},[e._v("1")]),t("br"),t("span",{staticClass:"line-number"},[e._v("2")]),t("br"),t("span",{staticClass:"line-number"},[e._v("3")]),t("br")])]),t("p",[e._v("接下来下载一下给的官方数据，跑一下~"),t("br"),e._v("\n这里下载原始数据："),t("a",{attrs:{href:"https://github.com/bioinfo-biols/SEVtras",target:"_blank",rel:"noopener noreferrer"}},[e._v("SEVtras (github.com)"),t("OutboundLink")],1),t("br"),e._v("\n它支持输入的数据类型有："),t("code",[e._v("h5ad")]),e._v("，"),t("code",[e._v("10x_mtx")]),e._v("，"),t("code",[e._v("h5")])]),e._v(" "),t("blockquote",[t("p",[e._v("建议，使用原始数据，使用Linux系统，使用多个样本（以获得更多信息）")])]),e._v(" "),t("p",[e._v('1、识别小囊泡，你别说，还真"看到了"自己个小囊泡')]),e._v(" "),t("div",{staticClass:"language-python line-numbers-mode"},[t("pre",{pre:!0,attrs:{class:"language-python"}},[t("code",[t("span",{pre:!0,attrs:{class:"token comment"}},[e._v("##1 if samples locate in one directory")]),e._v("\n"),t("span",{pre:!0,attrs:{class:"token keyword"}},[e._v("import")]),e._v(" SEVtras\nSEVtras"),t("span",{pre:!0,attrs:{class:"token punctuation"}},[e._v(".")]),e._v("sEV_recognizer"),t("span",{pre:!0,attrs:{class:"token punctuation"}},[e._v("(")]),e._v("input_path"),t("span",{pre:!0,attrs:{class:"token operator"}},[e._v("=")]),t("span",{pre:!0,attrs:{class:"token string"}},[e._v("'./tests'")]),t("span",{pre:!0,attrs:{class:"token punctuation"}},[e._v(",")]),e._v(" sample_file"),t("span",{pre:!0,attrs:{class:"token operator"}},[e._v("=")]),t("span",{pre:!0,attrs:{class:"token string"}},[e._v("'./tests/sample_file'")]),t("span",{pre:!0,attrs:{class:"token punctuation"}},[e._v(",")]),e._v(" out_path"),t("span",{pre:!0,attrs:{class:"token operator"}},[e._v("=")]),t("span",{pre:!0,attrs:{class:"token string"}},[e._v("'./outputs'")]),t("span",{pre:!0,attrs:{class:"token punctuation"}},[e._v(",")]),e._v(" species"),t("span",{pre:!0,attrs:{class:"token operator"}},[e._v("=")]),t("span",{pre:!0,attrs:{class:"token string"}},[e._v("'Homo'")]),t("span",{pre:!0,attrs:{class:"token punctuation"}},[e._v(")")]),e._v("\n\n"),t("span",{pre:!0,attrs:{class:"token comment"}},[e._v("##2 if samples locate in differet directories")]),e._v("\n"),t("span",{pre:!0,attrs:{class:"token keyword"}},[e._v("import")]),e._v(" SEVtras\nSEVtras"),t("span",{pre:!0,attrs:{class:"token punctuation"}},[e._v(".")]),e._v("sEV_recognizer"),t("span",{pre:!0,attrs:{class:"token punctuation"}},[e._v("(")]),e._v("sample_file"),t("span",{pre:!0,attrs:{class:"token operator"}},[e._v("=")]),t("span",{pre:!0,attrs:{class:"token string"}},[e._v("'./tests/sample_file'")]),t("span",{pre:!0,attrs:{class:"token punctuation"}},[e._v(",")]),e._v(" out_path"),t("span",{pre:!0,attrs:{class:"token operator"}},[e._v("=")]),t("span",{pre:!0,attrs:{class:"token string"}},[e._v("'./outputs'")]),t("span",{pre:!0,attrs:{class:"token punctuation"}},[e._v(",")]),e._v(" species"),t("span",{pre:!0,attrs:{class:"token operator"}},[e._v("=")]),t("span",{pre:!0,attrs:{class:"token string"}},[e._v("'Homo'")]),t("span",{pre:!0,attrs:{class:"token punctuation"}},[e._v(")")]),e._v("\n")])]),e._v(" "),t("div",{staticClass:"line-numbers-wrapper"},[t("span",{staticClass:"line-number"},[e._v("1")]),t("br"),t("span",{staticClass:"line-number"},[e._v("2")]),t("br"),t("span",{staticClass:"line-number"},[e._v("3")]),t("br"),t("span",{staticClass:"line-number"},[e._v("4")]),t("br"),t("span",{staticClass:"line-number"},[e._v("5")]),t("br"),t("span",{staticClass:"line-number"},[e._v("6")]),t("br"),t("span",{staticClass:"line-number"},[e._v("7")]),t("br")])]),t("p",[e._v("所以最后最少就三个必要参数，"),t("code",[e._v("输入")]),e._v("，"),t("code",[e._v("输出")]),e._v("，"),t("code",[e._v("物种")]),t("br"),e._v("\n然后上边的参数，再解释一下")]),e._v(" "),t("blockquote",[t("p",[t("code",[e._v("input_path")]),e._v("，是要处理的文件在哪个路径，相当于说了在那一层楼上，所有要处理的文件都在里面\n"),t("code",[e._v("sample_file")]),e._v("，是要处理的文件名字，指名道姓的说是几号，这个是确定下来要处理的文件的\n"),t("code",[e._v("dir_origin")]),e._v("，是为了处理"),t("code",[e._v("R10X")]),e._v("，类似的数据文件夹名字，如果要处理mtx那种文件，就要设置一下")])]),e._v(" "),t("p",[e._v("最后输出结果")]),e._v(" "),t("blockquote",[t("p",[e._v("out_path defines the output of SEVtras that is one h5ad file, named raw_SEVtras.h5ad, with SEVtras score (‘score’) and sEV classification (‘sEV’) in the obs for all droplets, and one named sEVs_SEVtras.h5ad with only sEV-containing droplets.\n"),t("code",[e._v("raw_SEVtras.h5ad")]),e._v("，包含细胞和小囊泡\n"),t("code",[e._v("sEVs_SEVtras.h5ad")]),e._v("，只有小囊泡")])]),e._v(" "),t("p",[e._v("根据上一步输出结果，以及细胞类型数据（matrix），还可以计算分泌活性")]),e._v(" "),t("p",[e._v("一直报错没有"),t("code",[e._v("gseapy")]),e._v("模块，然而是安装了的"),t("br"),e._v("\n之后，转移到了"),t("code",[e._v("conda activate scc")]),e._v("，之后再安装才可以的，不知道为什么。最后出来一个很丑的图....")])])}),[],!1,null,null,null);t.default=_.exports}}]);