<!DOCTYPE html>
<html lang="zh-CN">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width,initial-scale=1">
    <title>PDAC biomarkers | 一盏烛光的微笑灿烂人生</title>
    <meta name="generator" content="VuePress 1.9.9">
    <link rel="icon" href="/img/favicon.ico">
    <link rel="stylesheet" href="https://at.alicdn.com/t/font_3129839_xft6cqs5gc.css">
    <noscript><meta http-equiv="refresh" content="0; url=https://www.youngkbt.cn/noscript/"><style>.theme-vdoing-content { display:none }</noscript>
    <meta name="description" content="学习记录博客">
    <meta name="theme-color" content="#11a8cd">
    <meta name="个人博客" content="科研学习，围炉夜话，奇淫巧计">
    
    <link rel="preload" href="/assets/css/0.styles.c9ffc21e.css" as="style"><link rel="preload" href="/assets/js/app.bd488e07.js" as="script"><link rel="preload" href="/assets/js/2.bdf5b8cf.js" as="script"><link rel="preload" href="/assets/js/32.2fcf84e2.js" as="script"><link rel="preload" href="/assets/js/3.365ac406.js" as="script"><link rel="prefetch" href="/assets/js/10.b00b53b4.js"><link rel="prefetch" href="/assets/js/100.1b2ba6dd.js"><link rel="prefetch" href="/assets/js/101.e392eb92.js"><link rel="prefetch" href="/assets/js/102.eb0bb162.js"><link rel="prefetch" href="/assets/js/103.9db14406.js"><link rel="prefetch" href="/assets/js/104.6ae6fd5c.js"><link rel="prefetch" href="/assets/js/105.4d79399b.js"><link rel="prefetch" href="/assets/js/106.4e6ff12d.js"><link rel="prefetch" href="/assets/js/107.bc93ee73.js"><link rel="prefetch" href="/assets/js/108.59a000f0.js"><link rel="prefetch" href="/assets/js/109.f01f4422.js"><link rel="prefetch" href="/assets/js/11.346d31c6.js"><link rel="prefetch" href="/assets/js/110.96a96c44.js"><link rel="prefetch" href="/assets/js/111.87fe63df.js"><link rel="prefetch" href="/assets/js/112.c560a316.js"><link rel="prefetch" href="/assets/js/113.6a371b02.js"><link rel="prefetch" href="/assets/js/114.989a4d3e.js"><link rel="prefetch" href="/assets/js/115.b6baafa2.js"><link rel="prefetch" href="/assets/js/116.d469dc20.js"><link rel="prefetch" href="/assets/js/117.0a6bccf0.js"><link rel="prefetch" href="/assets/js/118.9f1ef088.js"><link rel="prefetch" href="/assets/js/119.059ef806.js"><link rel="prefetch" href="/assets/js/12.8d91a638.js"><link rel="prefetch" href="/assets/js/120.75ed11ab.js"><link rel="prefetch" href="/assets/js/121.1896ee22.js"><link rel="prefetch" href="/assets/js/122.a3664290.js"><link rel="prefetch" href="/assets/js/123.5b849c39.js"><link rel="prefetch" href="/assets/js/124.85737d6b.js"><link rel="prefetch" href="/assets/js/125.a95cd42f.js"><link rel="prefetch" href="/assets/js/126.7b1c1645.js"><link rel="prefetch" href="/assets/js/127.64a16fd4.js"><link rel="prefetch" href="/assets/js/128.a07721ba.js"><link rel="prefetch" href="/assets/js/129.9ba86871.js"><link rel="prefetch" href="/assets/js/13.18ed2aa7.js"><link rel="prefetch" href="/assets/js/130.5cdb8138.js"><link rel="prefetch" href="/assets/js/131.d2f31425.js"><link rel="prefetch" href="/assets/js/132.b2286d65.js"><link rel="prefetch" href="/assets/js/133.1229c9ae.js"><link rel="prefetch" href="/assets/js/134.71fad9a0.js"><link rel="prefetch" href="/assets/js/135.ff682301.js"><link rel="prefetch" href="/assets/js/136.9de1870b.js"><link rel="prefetch" href="/assets/js/137.a58bbdcd.js"><link rel="prefetch" href="/assets/js/138.cc768cb1.js"><link rel="prefetch" href="/assets/js/139.9d52d34a.js"><link rel="prefetch" href="/assets/js/14.4bcf7ee9.js"><link rel="prefetch" href="/assets/js/140.b1790e58.js"><link rel="prefetch" href="/assets/js/141.98bceb9f.js"><link rel="prefetch" href="/assets/js/142.0415d712.js"><link rel="prefetch" href="/assets/js/143.851c7187.js"><link rel="prefetch" href="/assets/js/144.ae79e007.js"><link rel="prefetch" href="/assets/js/145.d5020e58.js"><link rel="prefetch" href="/assets/js/146.c3eafa98.js"><link rel="prefetch" href="/assets/js/147.0a6864ec.js"><link rel="prefetch" href="/assets/js/148.bbc4bd28.js"><link rel="prefetch" href="/assets/js/149.489e8cdb.js"><link rel="prefetch" href="/assets/js/15.368d9d2d.js"><link rel="prefetch" href="/assets/js/150.7778d444.js"><link rel="prefetch" href="/assets/js/16.06f68354.js"><link rel="prefetch" href="/assets/js/17.4f8d6256.js"><link rel="prefetch" href="/assets/js/18.8989b2ff.js"><link rel="prefetch" href="/assets/js/19.abb857ca.js"><link rel="prefetch" href="/assets/js/20.e32420cf.js"><link rel="prefetch" href="/assets/js/21.06441c4d.js"><link rel="prefetch" href="/assets/js/22.7fb78c64.js"><link rel="prefetch" href="/assets/js/23.d9193b83.js"><link rel="prefetch" href="/assets/js/24.6487f43a.js"><link rel="prefetch" href="/assets/js/25.11e66938.js"><link rel="prefetch" href="/assets/js/26.c3e43726.js"><link rel="prefetch" href="/assets/js/27.24117748.js"><link rel="prefetch" href="/assets/js/28.fabf32c7.js"><link rel="prefetch" href="/assets/js/29.3a9331f2.js"><link rel="prefetch" href="/assets/js/30.ac5ca451.js"><link rel="prefetch" href="/assets/js/31.36e6c9c0.js"><link rel="prefetch" href="/assets/js/33.a73adf6b.js"><link rel="prefetch" href="/assets/js/34.49c0fb46.js"><link rel="prefetch" href="/assets/js/35.babda623.js"><link rel="prefetch" href="/assets/js/36.56221bac.js"><link rel="prefetch" href="/assets/js/37.5d6d33aa.js"><link rel="prefetch" href="/assets/js/38.a6b592e0.js"><link rel="prefetch" href="/assets/js/39.14a81539.js"><link rel="prefetch" href="/assets/js/4.a64d5c9f.js"><link rel="prefetch" href="/assets/js/40.e94616b3.js"><link rel="prefetch" href="/assets/js/41.4ad9febc.js"><link rel="prefetch" href="/assets/js/42.4825ff11.js"><link rel="prefetch" href="/assets/js/43.6d554f8a.js"><link rel="prefetch" href="/assets/js/44.ed8f011b.js"><link rel="prefetch" href="/assets/js/45.170e103d.js"><link rel="prefetch" href="/assets/js/46.9cc703a7.js"><link rel="prefetch" href="/assets/js/47.ff6af7ea.js"><link rel="prefetch" href="/assets/js/48.1b2f1a1b.js"><link rel="prefetch" href="/assets/js/49.05223958.js"><link rel="prefetch" href="/assets/js/5.b1fd9f62.js"><link rel="prefetch" href="/assets/js/50.26aa9f07.js"><link rel="prefetch" href="/assets/js/51.65eba0cc.js"><link rel="prefetch" href="/assets/js/52.5b21c37d.js"><link rel="prefetch" href="/assets/js/53.4f209254.js"><link rel="prefetch" href="/assets/js/54.176923ed.js"><link rel="prefetch" href="/assets/js/55.74f6741d.js"><link rel="prefetch" href="/assets/js/56.383e3c0d.js"><link rel="prefetch" href="/assets/js/57.35b21350.js"><link rel="prefetch" href="/assets/js/58.7ae8fd30.js"><link rel="prefetch" href="/assets/js/59.3d7d7dea.js"><link rel="prefetch" href="/assets/js/6.993d7ad0.js"><link rel="prefetch" href="/assets/js/60.67ba2844.js"><link rel="prefetch" href="/assets/js/61.6c5542cd.js"><link rel="prefetch" href="/assets/js/62.a235dad7.js"><link rel="prefetch" href="/assets/js/63.a3a52fe2.js"><link rel="prefetch" href="/assets/js/64.2b6b610e.js"><link rel="prefetch" href="/assets/js/65.8672534a.js"><link rel="prefetch" href="/assets/js/66.c7281353.js"><link rel="prefetch" href="/assets/js/67.e062f6ac.js"><link rel="prefetch" href="/assets/js/68.446f0688.js"><link rel="prefetch" href="/assets/js/69.f03843ed.js"><link rel="prefetch" href="/assets/js/7.afb33a4c.js"><link rel="prefetch" href="/assets/js/70.466d842a.js"><link rel="prefetch" href="/assets/js/71.87fcb26b.js"><link rel="prefetch" href="/assets/js/72.ccf1092d.js"><link rel="prefetch" href="/assets/js/73.3ad6124b.js"><link rel="prefetch" href="/assets/js/74.17a59372.js"><link rel="prefetch" href="/assets/js/75.6c859470.js"><link rel="prefetch" href="/assets/js/76.674a56ad.js"><link rel="prefetch" href="/assets/js/77.5342f3c1.js"><link rel="prefetch" href="/assets/js/78.154ba5e1.js"><link rel="prefetch" href="/assets/js/79.10901b1c.js"><link rel="prefetch" href="/assets/js/8.5bbd7ae9.js"><link rel="prefetch" href="/assets/js/80.0166104e.js"><link rel="prefetch" href="/assets/js/81.4c5a0e6a.js"><link rel="prefetch" href="/assets/js/82.47099701.js"><link rel="prefetch" href="/assets/js/83.7873cd95.js"><link rel="prefetch" href="/assets/js/84.23000f66.js"><link rel="prefetch" href="/assets/js/85.6701c9e4.js"><link rel="prefetch" href="/assets/js/86.188012c3.js"><link rel="prefetch" href="/assets/js/87.e9cde64e.js"><link rel="prefetch" href="/assets/js/88.5ccdea49.js"><link rel="prefetch" href="/assets/js/89.9ab559de.js"><link rel="prefetch" href="/assets/js/9.6699859e.js"><link rel="prefetch" href="/assets/js/90.bc7fb61d.js"><link rel="prefetch" href="/assets/js/91.07537d0a.js"><link rel="prefetch" href="/assets/js/92.b4ed0bad.js"><link rel="prefetch" href="/assets/js/93.b167c2d0.js"><link rel="prefetch" href="/assets/js/94.1a02653c.js"><link rel="prefetch" href="/assets/js/95.69ef19be.js"><link rel="prefetch" href="/assets/js/96.e0b3f5df.js"><link rel="prefetch" href="/assets/js/97.9ea0c268.js"><link rel="prefetch" href="/assets/js/98.074852d9.js"><link rel="prefetch" href="/assets/js/99.708b118b.js">
    <link rel="stylesheet" href="/assets/css/0.styles.c9ffc21e.css">
  </head>
  <body class="theme-mode-light">
    <div id="app" data-server-rendered="true"><div class="theme-container sidebar-open have-rightmenu"><header class="navbar blur"><div title="目录" class="sidebar-button"><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" role="img" viewBox="0 0 448 512" class="icon"><path fill="currentColor" d="M436 124H12c-6.627 0-12-5.373-12-12V80c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12z"></path></svg></div> <a href="/" class="home-link router-link-active"><img src="/img/logo.png" alt="一盏烛光的微笑灿烂人生" class="logo"> <span class="site-name can-hide">一盏烛光的微笑灿烂人生</span></a> <div class="links"><div class="search-box"><input aria-label="Search" autocomplete="off" spellcheck="false" value=""> <!----></div> <nav class="nav-links can-hide"><div class="nav-item"><a href="/" class="nav-link">首页</a></div><div class="nav-item"><div class="dropdown-wrapper"><button type="button" aria-label="科研学习" class="dropdown-title"><a href="/study/" class="link-title">科研学习</a> <span class="title" style="display:none;">科研学习</span> <span class="arrow right"></span></button> <ul class="nav-dropdown" style="display:none;"><li class="dropdown-item"><!----> <a href="/articles/" class="nav-link">阅读记录</a></li><li class="dropdown-item"><!----> <a href="/biotech/" class="nav-link">生信分析</a></li><li class="dropdown-item"><!----> <a href="/basic/" class="nav-link">整理归纳</a></li></ul></div></div><div class="nav-item"><a href="/wlyh/" class="nav-link">围炉夜话</a></div><div class="nav-item"><a href="/technology/" class="nav-link">奇淫巧计</a></div><div class="nav-item"><div class="dropdown-wrapper"><button type="button" aria-label="索引" class="dropdown-title"><a href="/archives/" class="link-title">索引</a> <span class="title" style="display:none;">索引</span> <span class="arrow right"></span></button> <ul class="nav-dropdown" style="display:none;"><li class="dropdown-item"><!----> <a href="/categories/" class="nav-link">分类</a></li><li class="dropdown-item"><!----> <a href="/tags/" class="nav-link">标签</a></li><li class="dropdown-item"><!----> <a href="/archives/" class="nav-link">归档</a></li></ul></div></div><div class="nav-item"><div class="dropdown-wrapper"><button type="button" aria-label="更多" class="dropdown-title"><a href="/more/" class="link-title">更多</a> <span class="title" style="display:none;">更多</span> <span class="arrow right"></span></button> <ul class="nav-dropdown" style="display:none;"><li class="dropdown-item"><!----> <a href="/about/" class="nav-link">关于</a></li><li class="dropdown-item"><!----> <a href="/links/" class="nav-link">一些网站</a></li><li class="dropdown-item"><!----> <a href="/friends/" class="nav-link">友情链接</a></li><li class="dropdown-item"><!----> <a href="/passbyworld/" class="nav-link">路过人间</a></li></ul></div></div> <!----></nav></div></header> <div class="sidebar-mask"></div> <div class="sidebar-hover-trigger"></div> <aside class="sidebar" style="display:none;"><div class="blogger"><img src="https://pica.zhimg.com/v2-f2b30ad5674f652294f0ec6f9b8c9922_720w.webp?source=d16d100b"> <div class="blogger-info"><h3>围炉夜话</h3> <span>知行合一</span></div></div> <nav class="nav-links"><div class="nav-item"><a href="/" class="nav-link">首页</a></div><div class="nav-item"><div class="dropdown-wrapper"><button type="button" aria-label="科研学习" class="dropdown-title"><a href="/study/" class="link-title">科研学习</a> <span class="title" style="display:none;">科研学习</span> <span class="arrow right"></span></button> <ul class="nav-dropdown" style="display:none;"><li class="dropdown-item"><!----> <a href="/articles/" class="nav-link">阅读记录</a></li><li class="dropdown-item"><!----> <a href="/biotech/" class="nav-link">生信分析</a></li><li class="dropdown-item"><!----> <a href="/basic/" class="nav-link">整理归纳</a></li></ul></div></div><div class="nav-item"><a href="/wlyh/" class="nav-link">围炉夜话</a></div><div class="nav-item"><a href="/technology/" class="nav-link">奇淫巧计</a></div><div class="nav-item"><div class="dropdown-wrapper"><button type="button" aria-label="索引" class="dropdown-title"><a href="/archives/" class="link-title">索引</a> <span class="title" style="display:none;">索引</span> <span class="arrow right"></span></button> <ul class="nav-dropdown" style="display:none;"><li class="dropdown-item"><!----> <a href="/categories/" class="nav-link">分类</a></li><li class="dropdown-item"><!----> <a href="/tags/" class="nav-link">标签</a></li><li class="dropdown-item"><!----> <a href="/archives/" class="nav-link">归档</a></li></ul></div></div><div class="nav-item"><div class="dropdown-wrapper"><button type="button" aria-label="更多" class="dropdown-title"><a href="/more/" class="link-title">更多</a> <span class="title" style="display:none;">更多</span> <span class="arrow right"></span></button> <ul class="nav-dropdown" style="display:none;"><li class="dropdown-item"><!----> <a href="/about/" class="nav-link">关于</a></li><li class="dropdown-item"><!----> <a href="/links/" class="nav-link">一些网站</a></li><li class="dropdown-item"><!----> <a href="/friends/" class="nav-link">友情链接</a></li><li class="dropdown-item"><!----> <a href="/passbyworld/" class="nav-link">路过人间</a></li></ul></div></div> <!----></nav>  <ul class="sidebar-links"><li><section class="sidebar-group collapsable depth-0"><p class="sidebar-heading open"><span>阅读记录</span> <span class="arrow down"></span></p> <ul class="sidebar-links sidebar-group-items"><li><a href="/pages/42150e/" class="sidebar-link">阅读积累</a></li><li><a href="/pages/076137/" class="sidebar-link">肿瘤的特征</a></li><li><a href="/pages/509e64/" class="sidebar-link">衰老的特征</a></li><li><a href="/pages/e5759e/" class="sidebar-link">张老师的NK</a></li><li><a href="/pages/234a90/" class="sidebar-link">衰老和癌症的关系</a></li><li><a href="/pages/22093f/" class="sidebar-link">肿瘤进展中的肿瘤微环境</a></li><li><a href="/pages/621f57/" class="sidebar-link">细胞衰老</a></li><li><a href="/pages/a37fec/" class="sidebar-link">眼科综述</a></li><li><a href="/pages/2d9d56/" class="sidebar-link">免疫</a></li><li><a href="/pages/fc684e/" class="sidebar-link">癌症转变过程中的表观遗传调节</a></li><li><a href="/pages/2cc503/" class="sidebar-link">肿瘤代谢</a></li><li><a href="/pages/aee5b7/" class="sidebar-link">肿瘤微环境中的代谢</a></li><li><a href="/pages/c87600/" class="sidebar-link">癌症的研究历程</a></li><li><a href="/pages/8c9af3/" class="sidebar-link">张老师的MC</a></li><li><a href="/pages/58653f/" class="sidebar-link">牛磺酸</a></li><li><a href="/pages/7ea67e/" class="sidebar-link">咽峡炎链球菌</a></li><li><a href="/pages/1c872d/" class="sidebar-link">乳酸</a></li><li><a href="/pages/ba40b8/" class="sidebar-link">胃癌相关的数据挖掘</a></li><li><a href="/pages/1c61f5/" aria-current="page" class="active sidebar-link">PDAC biomarkers</a><ul class="sidebar-sub-headers"><li class="sidebar-sub-header level2"><a href="/pages/1c61f5/#code-availability" class="sidebar-link">Code availability</a></li><li class="sidebar-sub-header level2"><a href="/pages/1c61f5/#abstract" class="sidebar-link">Abstract</a></li><li class="sidebar-sub-header level2"><a href="/pages/1c61f5/#r1-pdac患者" class="sidebar-link">R1，PDAC患者</a></li><li class="sidebar-sub-header level2"><a href="/pages/1c61f5/#r2-临床相关功能蛋白模块" class="sidebar-link">R2，临床相关功能蛋白模块</a></li><li class="sidebar-sub-header level2"><a href="/pages/1c61f5/#r3-预后风险模型" class="sidebar-link">R3，预后风险模型</a></li><li class="sidebar-sub-header level2"><a href="/pages/1c61f5/#r4-反应生物指标" class="sidebar-link">R4，反应生物指标</a></li><li class="sidebar-sub-header level2"><a href="/pages/1c61f5/#r5-外部和内部的验证" class="sidebar-link">R5，外部和内部的验证</a></li><li class="sidebar-sub-header level2"><a href="/pages/1c61f5/#discussion" class="sidebar-link">Discussion</a></li><li class="sidebar-sub-header level2"><a href="/pages/1c61f5/#methods-figures" class="sidebar-link">Methods &amp; Figures</a><ul class="sidebar-sub-headers"><li class="sidebar-sub-header level3"><a href="/pages/1c61f5/#figure-1" class="sidebar-link">Figure 1</a></li><li class="sidebar-sub-header level3"><a href="/pages/1c61f5/#figure-2" class="sidebar-link">Figure 2</a></li><li class="sidebar-sub-header level3"><a href="/pages/1c61f5/#figure-3" class="sidebar-link">Figure 3</a></li><li class="sidebar-sub-header level3"><a href="/pages/1c61f5/#figure-4" class="sidebar-link">Figure 4</a></li><li class="sidebar-sub-header level3"><a href="/pages/1c61f5/#figure-5" class="sidebar-link">Figure 5</a></li></ul></li><li class="sidebar-sub-header level2"><a href="/pages/1c61f5/#感想" class="sidebar-link">感想</a></li></ul></li><li><a href="/pages/0a11dd/" class="sidebar-link">胖子CAF</a></li><li><a href="/pages/f4f4b7/" class="sidebar-link">胃癌免疫逃逸的决定因素</a></li><li><a href="/pages/90dbe8/" class="sidebar-link">APA相关综述</a></li></ul></section></li><li><section class="sidebar-group collapsable depth-0"><p class="sidebar-heading"><span>生信分析</span> <span class="arrow right"></span></p> <!----></section></li><li><section class="sidebar-group collapsable depth-0"><p class="sidebar-heading"><span>实验技术</span> <span class="arrow right"></span></p> <!----></section></li><li><section class="sidebar-group collapsable depth-0"><p class="sidebar-heading"><span>整理归纳</span> <span class="arrow right"></span></p> <!----></section></li></ul> </aside> <div><main class="page"><div class="theme-vdoing-wrapper "><div class="articleInfo-wrap" data-v-06225672><div class="articleInfo" data-v-06225672><ul class="breadcrumbs" data-v-06225672><li data-v-06225672><a href="/" title="首页" class="iconfont icon-home router-link-active" data-v-06225672></a></li> <li data-v-06225672><a href="/study/#科研学习" data-v-06225672>科研学习</a></li><li data-v-06225672><a href="/study/#阅读记录" data-v-06225672>阅读记录</a></li></ul> <div class="info" data-v-06225672><div title="作者" class="author iconfont icon-touxiang" data-v-06225672><a href="javascript:;" data-v-06225672>FHL</a></div> <div title="创建时间" class="date iconfont icon-riqi" data-v-06225672><a href="javascript:;" data-v-06225672>2024-06-01</a></div> <!----></div></div></div> <!----> <div class="content-wrapper"><div class="right-menu-wrapper"><div class="right-menu-margin"><div class="right-menu-title">目录</div> <div class="right-menu-content"></div></div></div> <h1><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAB4AAAAeCAYAAAA7MK6iAAAAAXNSR0IArs4c6QAABH1JREFUSA3tVl1oHFUUPmdmd2ltklqbpJDiNnXFmgbFktho7YMPNiJSSZM0+CAYSkUELVhM6YuwIPpgoOKDqOBDC0XE2CQoNtQXBUFTTcCi+Wlh1V2TQExsUzcltd3M9Tt3ZjZzZ2fT+OJTL8yeM+eee757fmeJbq//KQL8X3DUSFOcfr7cRsRtxNQMWueeVzOkaITIGqQHNg5y8+jNW9ldM7A6nTpAjuolUikAwq7CE3WcM2RRDz+XGVgN3FptU/aUSlvq9Pa3iZ1+sgAqJyyAFqkipd9dqiwHF3P65YycLWc/6sqGrvoEoIp6DOFaX5h6+dnfjkWprwqsPk0dUGq5vySwDImC10KxFHgGL1SWoc92O3eVht09qdXNH11I2SsTsJYqMWzihqGMi+A+Garf3BAuuLI5oGlULyNfyB/HYNujwktOfRrMr5t77NmevqaUopx0grnKAyvVpmwUDB4x6FPXuGvYLTDwWsejwgtgkYKPqRJg8SV6xaiZ3ZTppGneS4yfH5/66fZSDHv+QZci/+h5c5UHtpy67JUqGppM0sh0Nc1dW6/N1W5Yoqat8/TU/VnadmdeW2PLLSyh0cvxBs3KbqTmwYPpxN4do/mzE8nEpvX/UMu2Wbp74zUAK5q6WkHns7V0eWkdPbPzd3rxkTGybadYySumVzhcaJFbs5UrEkQ/+CK8gF5dnh/6ciIZ73gwQ927L1IitoxKLXYP3SjYdOrHHfTZhRRlFyrorafPk20B3HPD1y2G3qKZME5Jcf3t/HUC13/8tSd++vqFveMUTwAUxSUFI1QekR1+bIze3D9MF2aq6cPvG72CgnldWCFqyRw3lwH8ZMerjTD9ElRO7Gv44wNpC90aASqGfVlz/Rx17srQ57/UU26hkhQqUB7dBR71WmzQhHUnblGmVOEw0jhbV1n9OlXUDCIRGaNV5Jp43N516fN7JmnTHdfp7Hgy0luO4aMhtkLL8Bi3bUWYvzh5Mn1dTxrL6QmGuRhGL/TiTTxRoEdTszSaq9GR0NGA3KdkOz3hqSV3MIDhQ5IVX/Ivx3umBti2es2h4eZby7x8br1rkf7Mo90AqC8aQ3sJeNzqFRu+vSANAQe3PL7l0HGOAdwDCeZYvNKeoZp1Qfs6Aipndh86HmFRi0LAnEO47wsqM6cdfjh3jBPUzhZy7nvlUfFsamED1VQt6aISHVymXZ/B2aCtIG8AI8xfobj2d3en1wWVhOeHELKmLQ1s211s88comkv4UCwWyF787mJdYXtNfhKAXVqnKTq8QZvGAGGOfaTo5pGZ/PwbUCr5+DPr/1J92JNHr9aOl/F3iI5+O1nfybsGxoimvZ3ViWSluDITw3P37mypheDIPY0tw7+O/5ApbkYw+zpfaUVu32Pi98+defdUhEpZkRFq0aqyNh9FuL9hpYbEm6iwi0z2REd09ZmyENEbuhjDWzKvZXTqKYaBIr3tt5kuPtQBZFvEUwHt60vfCNu41XsksH9Ij1BMMz1Y0OOunHNShFIP5868g5zeXmuLwL9T4b6Q2+KejgAAAABJRU5ErkJggg==">PDAC biomarkers<!----></h1>  <div class="theme-vdoing-content content__default"><div class="custom-block note"><p class="custom-block-title">笔记</p> <p>2024-06-01，数据，代码给的太全面了，真的想看一看呐！</p></div> <p>话不多说，咱先看一下这篇文章的 <code>Code availability</code></p> <h2 id="code-availability"><a href="#code-availability" class="header-anchor">#</a> Code availability</h2> <p>Source codes used in the study are accessible via the tab ‘CODES &amp; INSTRUCTIONS’ on our website (<a href="http://www.genetictargets.com/PDAC2" target="_blank" rel="noopener noreferrer">http://www.genetictargets.com/PDAC2<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a>), where users can follow detailed step-by-step instructions (<a href="http://www.genetictargets.com/PDAC2BOOKLET/index.html" target="_blank" rel="noopener noreferrer">http://www.genetictargets.com/PDAC2BOOKLET/index.html<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a>) to reproduce all results, including figures and tables, presented in the paper. Codes were also deposited into a public data repository at Figshare (<a href="https://doi.org/10.6084/m9.figshare.24863049" target="_blank" rel="noopener noreferrer">https://doi.org/10.6084/m9.figshare.24863049<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a>). Our commitment to transparency in code and data sharing promotes reproducible research, allowing the broader clinical and research community to fully benefit from our work without any barriers.</p> <p><strong>Prospective observational study on biomarkers of response in pancreatic ductal adenocarcinoma</strong></p> <p>nature medicine，IF：80+</p> <p>胰腺导管腺癌，反应性生物标志物的，前瞻性观察研究</p> <h2 id="abstract"><a href="#abstract" class="header-anchor">#</a> Abstract</h2> <ul><li><code>新辅助化疗</code> 能使切除胰腺导管腺癌的患者受益，但是术后患者的身体状态受损会影响患者的依从性。需要生物标志物来确定及时辅助治疗的候选者。</li> <li>在这项前瞻性观察研究中，纳入了1171例接受胰腺切除术的PDAC患者，并进行了全面的随访。</li> <li>191例患者样本的蛋白质组学分析揭示了临床相关的功能蛋白模块。针对PDAC建立了蛋白质组学水平的预后风险模型，并使用公开的外部队列进一步验证了其实用性。
<ul><li>更重要的是，通过利用<code>临床</code> 和 <code>蛋白质组学</code> 数据集的<code>交互效应回归分析 / interaction effect regression analysis</code>，我们发现了两个生物标志物( NDUFB8和CEMIP2)，表明PDAC患者对辅助化疗的总体敏感性。通过对386例患者的内部队列进行免疫组化验证生物标志物。严格的验证扩展到两个外部多中心队列 -- 一个法国多中心队列( 230例患者)和一个来自中国两所三级甲等医院的队列( 466例患者)，增强了研究结果的稳健性和可推广性。</li> <li>此外，通过功能实验在PDAC细胞系和患者来源的类器官上进行了实验验证。</li></ul></li> <li>总之，我们的队列规模整合了临床和蛋白质组学数据，证明了蛋白质组学指导PDAC预后和生物标志物辅助辅助化疗的潜力。</li></ul> <p>读这篇文章，这么几个想要了解和学习的问题：</p> <ul><li>蛋白质组，一般是如何分析的？</li> <li>最重要的！这个 <code>临床和蛋白质组学数据集的交互效应回归分析 / interaction effect regression analysis</code>，是怎么联系到一起分析的？</li> <li>逐步 <code>缩小研究范围</code> 的方法是哪些？</li> <li>确定自己的研究内容<code>有用</code>，有哪些指标，分别是怎么验证的？</li></ul> <p>治疗方案越来越多，所以个性化治疗也就越来越重要</p> <blockquote><p>With a growing array of regimens available, stratifying patients to identify the best responders for the most effective therapeutic strategies is critical to improve personalized oncology care.</p></blockquote> <h2 id="r1-pdac患者"><a href="#r1-pdac患者" class="header-anchor">#</a> R1，PDAC患者</h2> <p>PDAC患者 → 前瞻性观察性研究 → 最佳的分层策略，<code>RJ-cohort</code>（队列研究，真的是一件，功在当下，利在千秋的事情）</p> <blockquote><p>To elucidate the optimal stratification strategy for patients with PDAC, we conducted a prospective observational study on a Chinese cohort of 1,171 patients with PDAC who were diagnosed and treated by pancreatectomy at Ruijin Hospital, resulting in an entire in-house cohort denoted as ‘RJ-cohort’</p></blockquote> <p>大的队列<code>RJ-cohort</code>，其中进行了蛋白质组学分析的队列叫<code>RJ-cohort1</code>，在多项人口统计学和临床病理信息上，没有统计学差异（demographic，clinicopathological information）</p> <p>有一个实验设计的模式图：
<img src="https://s2.loli.net/2024/06/01/vNbg4sGOJEQlxB2.png" alt="image.png|800"></p> <p>191 treatment-naive tumors and 90 paired <code>tumor adjacent tissues (TATs)</code> for an in-depth proteomic analysis</p> <p>H&amp;E staining on all 191 samples before global proteomic analyses</p> <ul><li>Principal component analyses (PCAs)：a clear separation（Tumor &amp; TATs）</li> <li>Volcano plot：significantly upregulated &amp; downregulated proteins (Tumor VS TATs)</li> <li>Pathway enrichment analysis
<img src="https://s2.loli.net/2024/06/01/b6IOXUagGKZw24L.png" alt="image.png|600"></li></ul> <p>Oncogenic signaling pathways, complement and coagulation cascades and ECMreceptor interactions were active in tumors.<br>
Conversely, metabolic pathways, oxidative phosphorylation and ribosome exhibited significant downregulation in tumor samples.</p> <h2 id="r2-临床相关功能蛋白模块"><a href="#r2-临床相关功能蛋白模块" class="header-anchor">#</a> R2，临床相关功能蛋白模块</h2> <p>Clinically relevant functional protein modules</p> <p>weighted correlation network analysis，WGCNA，加权相关网络分析 → 32 <code>functional modules</code>(??)</p> <p>得到了这么个蛮炫酷的图（ps：我之前做过的一个WGCNA分析结果咋不是这样的呢，感觉比这个丑多了）
<img src="https://s2.loli.net/2024/06/01/HeRo5bvYxB2IjpX.png" alt="image.png|400"></p> <ul><li>a, WGCNA identified 32 functional protein modules (ME01–32) enriched in proteomic data for tumor samples. Each network node represents one protein, color-coded by the different functional modules.</li> <li>b, Overlay of significantly upregulated/downregulated proteins between tumors and TATs onto the network nodes.</li></ul> <p>得到的<code>module</code>是啥呀？
<code>Robustness analysis</code> confirmed the <code>stability</code> of these protein modules -- 不懂这俩词的含义</p> <p>下面这俩图不懂！！</p> <p><img src="https://s2.loli.net/2024/06/01/yCvJetqxmUG7sd8.png" alt="image.png|700"></p> <ul><li>(a) Determination of soft-threshold power in WGCNA. Analysis of the scale-free index for various soft-threshold powers (β) and the mean connectivity for various soft-threshold powers.</li> <li>(b) Dendrogram of proteins based on the measurement of dissimilarity and identification of the 32 modules.</li></ul> <p>下面这三个图，也不懂呢！！！</p> <p><img src="https://s2.loli.net/2024/06/01/q96QkWsKeVJZrDh.png" alt="image.png|800"></p> <ul><li>(c) Evaluating the stability of the 32 protein modules. For each calculation process, 80% of PDAC tumors were randomly selected for the identification of protein modules, employing a sample construction workflow and determining the soft-threshold power for the initial 32 protein modules.
<ul><li>Taking ME01 as an example, the accuracy for ME01 was calculated using the formula [Number of proteins clustered in ME01] / [Total protein number of ME01]. This calculation process was repeated 20 times (n=20 biologically independent calculations) with different random seed, consistently yielding median accuracies exceeding 90%. This underscores the robustness and stability for module identification. In the boxplot, a black line within the box marks the median. The bottom and top of the box are located at the 25th and 75th percentiles, respectively. The bars represent values that are more than 1.5 times the interquartile range from the border of each box.</li></ul></li> <li>(d) Receiver operating characteristic (ROC) curves, illustrating the prediction accuracy of the 32 protein modules in ‘RJ-cohort 1’. A random forest algorithm was employed to build the model, with 80% PDACs used as the training cohort and the remaining 20% as the validation cohort. Area under curve (AUC) values were calculated for each module separately.</li> <li>(e) ROC curves depicting the prediction accuracy of the 32 modules in Cao et al cohort. The model was constructed based on ‘RJ-cohort 1’, and the Cao et al. cohort was used as the validation cohort. (f) Validation of overlay of the significantly up/down-regulated proteins (left) and genes (right) between tumors and TATs on the weighted correlation network nodes from the Cao et al cohort.</li></ul> <p>module （包括几十个基因），这种数据是如何区分，然后做的生存分析？
<img src="https://s21.ax1x.com/2024/06/02/pkGGNOx.png" alt="pkGGNOx.png|400"></p> <h2 id="r3-预后风险模型"><a href="#r3-预后风险模型" class="header-anchor">#</a> R3，预后风险模型</h2> <p>A proteomics-level prognostic risk model for PDAC</p> <p>拿蛋白质进行生存分析，预后的预测了</p> <p>least absolute shrinkage and selection operator (LASSO) – Cox regression analysis</p> <p>还根据<code>LASSO scores</code>，将患者分成低分和高分组，然后能预测预后情况 -- P，HR，CI，AUC，→ an independent prognostic predictor for OS</p> <p>multivariable analysis，多因素分析</p> <p>在别的模型中去预测，判断是一致的，是有用的</p> <p>parallel reaction monitoring (PRM)，平行反应监测 -- a method for targeted proteomic analysis</p> <p>这一步，应该是把几千个蛋白质信息，缩小到了14个modules</p> <h2 id="r4-反应生物指标"><a href="#r4-反应生物指标" class="header-anchor">#</a> R4，反应生物指标</h2> <p>Proteomic biomarkers of response in PDAC</p> <p>Cox proportional hazards models</p> <p>肿瘤中低表达 / 越低化疗越好<br>
NDUFB8, a mitochondrial respiratory chain complex I protein crucial for oxidative energy metabolism（线粒体呼吸链，氧化能量代谢）</p> <p>肿瘤中高表达 / 越高化疗越好<br>
CEMIP2, encoding a cell surface hyaluronidase responsible for breaking down high-molecular-weight hyaluronan into low-molecular-weight hyaluronan（hyaluronidase，透明质酸酶；hyaluronan，透明质酸）</p> <h2 id="r5-外部和内部的验证"><a href="#r5-外部和内部的验证" class="header-anchor">#</a> R5，外部和内部的验证</h2> <p>External and internal validation of two biomarkers in PDAC</p> <p>ad hoc，adj.临时安排的；特别的；专门的</p> <p>RJ-cohort 1，RJ-cohort 3，SH-cohort，FR-cohort</p> <p>好多个HR，CI，P值....</p> <p>没学好流行病学，也没学好统计学，理解，但是不知道在这里是怎么算的...</p> <h2 id="discussion"><a href="#discussion" class="header-anchor">#</a> Discussion</h2> <p>在过去的几十年中，基因组和转录组数据的整合分析揭示了PDAC的分子格局，包括确定关键的基因改变(如KRAS、TP53、CDKN2A、SMAD4等)和基于表达的分子亚型。</p> <p>尽管这些分析大大拓宽了对这种恶性肿瘤的分子基础的认识，但对该疾病的完整理解需要整合全面的蛋白质组学数据。</p> <p>蛋白质组学可以为PDAC的临床、生物学和治疗方面提供新的见解，以更直接的方式弥合基因型和表型之间的差距。在之前的研究中，Cao等人整合了多组学数据来研究基因组和表观基因组扰动对基因和蛋白表达的影响。</p> <p>在此基础上，我们的研究涉及了191个未经治疗的肿瘤和90个配对的TATs的蛋白质组学分析，这些分析来自于在中国( ' RJ-cohort 1 ')进行的大规模、前瞻性、观察性研究。我们的研究结果，基于我们的内部队列和外部队列，为PDAC的临床、生物学和治疗的理解提供了见解。</p> <p>精准医疗！！！</p> <p>RNAseq，预测新辅助化疗的敏感性</p> <ul><li>A transcriptomic signature to predict adjuvant gemcitabine sensitivity in pancreatic adenocarcinoma</li></ul> <p>嗯，文章还蛮短的，结论不多，但是看着觉得用处还蛮大的，挺有用的。分析的方式不熟悉，但是有感觉不是太难，主要是数据比较难获取吧，随访了那么久，又做那么多蛋白质组学，肯定很花钱,,,</p> <h2 id="methods-figures"><a href="#methods-figures" class="header-anchor">#</a> Methods &amp; Figures</h2> <p>文章给的代码这么齐全，怎么不好好利用一下子呢？</p> <p><a href="http://www.genetictargets.com/PDAC2BOOKLET/index.html" target="_blank" rel="noopener noreferrer">PDAC2BOOKLET (genetictargets.com)<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a></p> <h3 id="figure-1"><a href="#figure-1" class="header-anchor">#</a> Figure 1</h3> <p>Volcano plot，limma package，adjust P value - Benjamini &amp; Hochberg (BH) method，| log2FC | ＞0.58（ | FC |＞1.5）</p> <p>3D PCA</p> <div class="language- line-numbers-mode"><pre class="language-text"><code>library(scatterplot3d)
library(tidyverse)
library(openxlsx)
library(factoextra) # Extract and Visualize the Results of Multivariate Data Analyses
color.bin &lt;- c(&quot;#00599F&quot;,&quot;#D01910&quot;)
dir.create(&quot;./results/Figure1&quot;,recursive = T)

#  Step 1: Load Proteomics data
pro  &lt;- readRDS(&quot;./data/proteomics/20230412_PDAC_PRO_exp.rds&quot;) # 4787 protein * 281 samples（4787行蛋白，281列样本）
meta &lt;- readRDS(&quot;./data/proteomics/20230412_PDAC_PRO_meta.rds&quot;) 
identical(rownames(meta),colnames(pro)) # check names  # identical() 是否相同？

#  Step 2: PCA
res.pca.comp &lt;- prcomp(pro, scale = F)

#  Step 3: Plot
plot.data &lt;- as.data.frame(res.pca.comp$rotation[, 1:10])
plot.data &lt;- plot.data %&gt;% 
              mutate(ID = rownames(plot.data),
                     Type = meta$Type,
                     TypeColor = color.bin[as.numeric(as.factor(Type))])
pdf(&quot;./results/Figure1/1B.PRO_PCA3d.pdf&quot;, width = 7, height = 7)
scatterplot3d(x = plot.data$PC2, 
              y = plot.data$PC1, 
              z = plot.data$PC3,
              color = plot.data$TypeColor,
              pch = 16, cex.symbols = 1, # 点的类型和大小
              scale.y = 0.7, angle = 45, # 有轴的比例和角度
              xlab = &quot;PC2&quot;, ylab = &quot;PC1&quot;, zlab = &quot;PC3&quot;,
              main=&quot;3D Scatter Plot of proteomics&quot;,
              col.axis = &quot;#444444&quot;, col.grid = &quot;#CCCCCC&quot; # 坐标轴和网格的颜色
              )
legend(&quot;bottom&quot;, legend = levels(as.factor(meta$Type)),
      col =  color.bin,  pch = 16,
      inset = -0.15, xpd = TRUE, horiz = TRUE)
dev.off()
write.xlsx(plot.data, &quot;./results/Figure1/1B.PRO_PCA3d.xlsx&quot;, overwrite = T)
</code></pre> <div class="line-numbers-wrapper"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br><span class="line-number">6</span><br><span class="line-number">7</span><br><span class="line-number">8</span><br><span class="line-number">9</span><br><span class="line-number">10</span><br><span class="line-number">11</span><br><span class="line-number">12</span><br><span class="line-number">13</span><br><span class="line-number">14</span><br><span class="line-number">15</span><br><span class="line-number">16</span><br><span class="line-number">17</span><br><span class="line-number">18</span><br><span class="line-number">19</span><br><span class="line-number">20</span><br><span class="line-number">21</span><br><span class="line-number">22</span><br><span class="line-number">23</span><br><span class="line-number">24</span><br><span class="line-number">25</span><br><span class="line-number">26</span><br><span class="line-number">27</span><br><span class="line-number">28</span><br><span class="line-number">29</span><br><span class="line-number">30</span><br><span class="line-number">31</span><br><span class="line-number">32</span><br><span class="line-number">33</span><br><span class="line-number">34</span><br><span class="line-number">35</span><br><span class="line-number">36</span><br><span class="line-number">37</span><br></div></div><p>identical，完全相同的，极为相似的</p> <p>tidyverse
<img src="https://s2.loli.net/2024/06/02/7hp8CiGLuZ2fyN5.png" alt="image.png|400"></p> <p>library(tidyverse)
tidyverse_packages()</p> <p><a href="https://www.tidyverse.org/" target="_blank" rel="noopener noreferrer">Tidyverse 官网<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a></p> <p><a href="https://cloud.tencent.com/developer/article/1886955" target="_blank" rel="noopener noreferrer">tidyverse<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a></p> <ul><li>readr：read_，write_</li> <li>tidyr / 整理：gather，spread，seperate，unite，</li> <li>dplyr / 操作：filter，select，arrange，mutate，summarise，slice，rename，relocate，</li> <li>purr：map，map2，pmap，</li></ul> <p><a href="https://lizongzhang.github.io/stat/5dplyr.html" target="_blank" rel="noopener noreferrer">dplyr<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a></p> <ul><li>columns：select，rename，mutate，relocate</li> <li>rows：filter，slice，arrange</li> <li>groups：summarise，count，top_n</li></ul> <p>R 语言教程</p> <ul><li><a href="https://bookdown.org/wangminjie/R4DS/" target="_blank" rel="noopener noreferrer">前言 | 数据科学中的 R 语言 (bookdown.org)<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a></li> <li><a href="https://shalom-lab.github.io/r-book/" target="_blank" rel="noopener noreferrer">R语言入门指南 (shalom-lab.github.io)<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a></li></ul> <div class="language- line-numbers-mode"><pre class="language-text"><code>p &lt;- fviz_eig(res.pca.comp) # variance percent of each PC
var_explained &lt;- get_eig(res.pca.comp)

# var_explained &lt;- res.pca.comp$sdev^2 / sum(res.pca.comp$sdev^2)
ggsave(&quot;./results/Figure1/1B.PRO_PCA_percent.pdf&quot;,p,width = 5, height = 5)
write.xlsx(var_explained, &quot;./results/Figure1/1B.PRO_PCA_percant.xlsx&quot;,rowNames=T, overwrite = T)
</code></pre> <div class="line-numbers-wrapper"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br><span class="line-number">6</span><br></div></div><div class="language- line-numbers-mode"><pre class="language-text"><code>library(limma)
library(tidyverse)
library(openxlsx)
library(ggpubr)
library(ggthemes)


#  Step 1: Load Proteomics data
exp  &lt;- readRDS(&quot;./data/proteomics/20230412_PDAC_PRO_exp.rds&quot;) # 4787 protein * 281 samples
meta &lt;- readRDS(&quot;./data/proteomics/20230412_PDAC_PRO_meta.rds&quot;)
identical(rownames(meta),colnames(exp)) # check names


#  Step 2: limma
meta      &lt;- meta %&gt;% mutate(contrast=as.factor(Type)) 
design    &lt;- model.matrix(~ 0 + contrast , data = meta) # un-paired
fit       &lt;- lmFit(exp, design)
contrast  &lt;- makeContrasts( Tumor_Normal = contrastT - contrastN , levels = design)
fits      &lt;- contrasts.fit(fit, contrast)
ebFit     &lt;- eBayes(fits)
limma.res &lt;- topTable(ebFit, coef = &quot;Tumor_Normal&quot;, adjust.method = 'fdr', number = Inf)
## result
limma.res &lt;- limma.res %&gt;% filter(!is.na(adj.P.Val)) %&gt;% 
              mutate( logP = -log10(adj.P.Val) ) %&gt;%
              mutate( tag = &quot;Tumor -vs- Normal&quot;)%&gt;%
              mutate( Gene = ID)
# cutoff:  FC:1.5   adj.p:0.05
limma.res &lt;- limma.res %&gt;% mutate(group=case_when(
                                  (adj.P.Val &lt; 0.05 &amp; logFC &gt; 0.58) ~ &quot;up&quot;,
                                  (adj.P.Val &lt; 0.05 &amp; logFC &lt; -0.58) ~ &quot;down&quot;,
                                  .default = &quot;not sig&quot;))
table(limma.res$group) # UP:1213 ; DOWN:864 ; not:2710
## output
write.xlsx( limma.res, &quot;./results/Figure1/1C.PRO_Limma_fc1.5.xlsx&quot;, overwrite = T, rowNames = F) 


#  Step 3: Vasualization 
## volcano
limma.res &lt;- limma.res %&gt;% mutate(group=factor(group,levels = c(&quot;up&quot;,&quot;down&quot;,&quot;not sig&quot;)))
my_label &lt;- paste0( &quot;FC:1.5 ; AdjP:0.05 ; &quot;,
                    &quot;Up:&quot;,table(limma.res$group)[1],&quot; ; &quot;,
                    &quot;Down:&quot;,table(limma.res$group)[2])
p &lt;- ggscatter(limma.res,
               x = &quot;logFC&quot;, y = &quot;logP&quot;,
               color = &quot;group&quot;, size = 2,
               main = paste0(&quot;Tumor -vs- TAC&quot;), # ***
               xlab = &quot;log2FoldChange&quot;, ylab = &quot;-log10(adjusted P.value)&quot;,
               palette = c(&quot;#D01910&quot;,&quot;#00599F&quot;,&quot;#CCCCCC&quot;),
               ylim = c(-1, 70),xlim=c(-8,8))+
  theme_base()+
  geom_hline(yintercept = -log10(0.05), linetype=&quot;dashed&quot;, color = &quot;#222222&quot;) +
  geom_vline(xintercept = 0.58 , linetype=&quot;dashed&quot;, color = &quot;#222222&quot;) +
  geom_vline(xintercept = -0.58, linetype=&quot;dashed&quot;, color = &quot;#222222&quot;) +
  labs(subtitle = my_label)+
  theme(plot.background = element_blank())
ggsave(&quot;./results/Figure1/1C.PRO_Limma_fc1.5.pdf&quot;, p, width = 10, height = 10)
</code></pre> <div class="line-numbers-wrapper"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br><span class="line-number">6</span><br><span class="line-number">7</span><br><span class="line-number">8</span><br><span class="line-number">9</span><br><span class="line-number">10</span><br><span class="line-number">11</span><br><span class="line-number">12</span><br><span class="line-number">13</span><br><span class="line-number">14</span><br><span class="line-number">15</span><br><span class="line-number">16</span><br><span class="line-number">17</span><br><span class="line-number">18</span><br><span class="line-number">19</span><br><span class="line-number">20</span><br><span class="line-number">21</span><br><span class="line-number">22</span><br><span class="line-number">23</span><br><span class="line-number">24</span><br><span class="line-number">25</span><br><span class="line-number">26</span><br><span class="line-number">27</span><br><span class="line-number">28</span><br><span class="line-number">29</span><br><span class="line-number">30</span><br><span class="line-number">31</span><br><span class="line-number">32</span><br><span class="line-number">33</span><br><span class="line-number">34</span><br><span class="line-number">35</span><br><span class="line-number">36</span><br><span class="line-number">37</span><br><span class="line-number">38</span><br><span class="line-number">39</span><br><span class="line-number">40</span><br><span class="line-number">41</span><br><span class="line-number">42</span><br><span class="line-number">43</span><br><span class="line-number">44</span><br><span class="line-number">45</span><br><span class="line-number">46</span><br><span class="line-number">47</span><br><span class="line-number">48</span><br><span class="line-number">49</span><br><span class="line-number">50</span><br><span class="line-number">51</span><br><span class="line-number">52</span><br><span class="line-number">53</span><br><span class="line-number">54</span><br><span class="line-number">55</span><br><span class="line-number">56</span><br></div></div><p><code>mutate</code>，<code>case_when</code></p> <div class="language- line-numbers-mode"><pre class="language-text"><code>limma.res &lt;- limma.res %&gt;% mutate(group = case_when(
                                  (adj.P.Val &lt; 0.05 &amp; logFC &gt; 0.58) ~ &quot;up&quot;,
                                  (adj.P.Val &lt; 0.05 &amp; logFC &lt; -0.58) ~ &quot;down&quot;,
                                  .default = &quot;not sig&quot;)
                                  )
</code></pre> <div class="line-numbers-wrapper"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br></div></div><div class="language- line-numbers-mode"><pre class="language-text"><code># data : enriched pathways table
plot.data &lt;- read.xlsx(&quot;./data/Extended Data Table 3.xlsx&quot;, sheet = 2, startRow = 2)
plot.pathway &lt;- c(&quot;GO:0006730~one-carbon metabolic process&quot;,&quot;GO:0006888~ER to Golgi vesicle-mediated transport&quot;,&quot;hsa00020:Citrate cycle (TCA cycle)&quot;,&quot;hsa00071:Fatty acid degradation&quot;,&quot;hsa00190:Oxidative phosphorylation&quot;,&quot;hsa00250:Alanine, aspartate and glutamate metabolism&quot;,&quot;hsa00260:Glycine, serine and threonine metabolism&quot;,&quot;hsa00280:Valine, leucine and isoleucine degradation&quot;,&quot;hsa00480:Glutathione metabolism&quot;,&quot;hsa00520:Amino sugar and nucleotide sugar metabolism&quot;,&quot;hsa00620:Pyruvate metabolism&quot;,&quot;hsa00630:Glyoxylate and dicarboxylate metabolism&quot;,&quot;hsa00640:Propanoate metabolism&quot;,&quot;hsa01100:Metabolic pathways&quot;,&quot;hsa01200:Carbon metabolism&quot;,&quot;hsa01212:Fatty acid metabolism&quot;,&quot;hsa01240:Biosynthesis of cofactors&quot;,&quot;hsa03010:Ribosome&quot;,&quot;hsa03060:Protein export&quot;,&quot;hsa04141:Protein processing in endoplasmic reticulum&quot;,&quot;hsa04972:Pancreatic secretion&quot;,&quot;GO:0001916~positive regulation of T cell mediated cytotoxicity&quot;,&quot;GO:0006096~glycolytic process&quot;,&quot;GO:0007165~signal transduction&quot;,&quot;GO:0007266~Rho protein signal transduction&quot;,&quot;GO:0045087~innate immune response&quot;,&quot;GO:0050778~positive regulation of immune response&quot;,&quot;GO:0050853~B cell receptor signaling pathway&quot;,&quot;GO:0050870~positive regulation of T cell activation&quot;,&quot;GO:0071346~cellular response to interferon-gamma&quot;,&quot;hsa04015:Rap1 signaling pathway&quot;,&quot;hsa04062:Chemokine signaling pathway&quot;,&quot;hsa04066:HIF-1 signaling pathway&quot;,&quot;hsa04151:PI3K-Akt signaling pathway&quot;,&quot;hsa04512:ECM-receptor interaction&quot;,&quot;hsa04610:Complement and coagulation cascades&quot;,&quot;hsa04621:NOD-like receptor signaling pathway&quot;,&quot;hsa04666:Fc gamma R-mediated phagocytosis&quot;)

plot.data &lt;- plot.data %&gt;% filter(Term %in% plot.pathway) %&gt;% 
                    mutate(LogFDR= -log10(FDR))
# plot
color.bin &lt;- c(&quot;#00599F&quot;,&quot;#D01910&quot;)
p &lt;- ggscatter(plot.data, 
			   x = &quot;LogFDR&quot;, y = &quot;Fold.Enrichment&quot;, color = &quot;Type&quot;,
               main = &quot;Enrichment of tumor/TAC protein&quot;,
               size = &quot;Ratio&quot;, shape = 16,
               label = plot.data$Term, palette = color.bin) + theme_base()
p &lt;- p + scale_size(range = c(4,20)) + 
         scale_x_continuous(limit = c(-10, 40)) + 
         theme(plot.background = element_blank())
ggsave(paste0(&quot;./results/Figure1/1D.PRO_deg.tn_enrich.pdf&quot;), p, width = 11, height = 10)
</code></pre> <div class="line-numbers-wrapper"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br><span class="line-number">6</span><br><span class="line-number">7</span><br><span class="line-number">8</span><br><span class="line-number">9</span><br><span class="line-number">10</span><br><span class="line-number">11</span><br><span class="line-number">12</span><br><span class="line-number">13</span><br><span class="line-number">14</span><br><span class="line-number">15</span><br><span class="line-number">16</span><br><span class="line-number">17</span><br></div></div><h3 id="figure-2"><a href="#figure-2" class="header-anchor">#</a> Figure 2</h3> <div class="language- line-numbers-mode"><pre class="language-text"><code>library(WGCNA)
dir.create(&quot;./results/Figure2/WGCNA&quot;,recursive = T)


#  Step 1: Loading expression data and set parameters
protein.nona.tumor &lt;- readRDS(&quot;./data/proteomics/wgcna/20230412_PDAC_PRO_Tumor_exp.rds&quot;)
corType = &quot;pearson&quot;
corFnc = ifelse(corType==&quot;spearman&quot;, cor, bicor)
maxPOutliers = ifelse(corType==&quot;pearson&quot;,1, 0.05)

plot.mat &lt;- t(protein.nona.tumor - rowMeans(protein.nona.tumor)) # row: samples | col: genes # input data normalization


#  Step 2: identification of outlier samples
sampleTree = hclust(dist(plot.mat), method = &quot;ward.D&quot;)
pdf(paste0(&quot;./results/Figure2/WGCNA/1.tree.pdf&quot;), width = 20, height = 9)
plot(sampleTree, main = &quot;Sample clustering to detect outliers&quot;, sub=&quot;&quot;, xlab=&quot;&quot;)
dev.off()
</code></pre> <div class="line-numbers-wrapper"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br><span class="line-number">6</span><br><span class="line-number">7</span><br><span class="line-number">8</span><br><span class="line-number">9</span><br><span class="line-number">10</span><br><span class="line-number">11</span><br><span class="line-number">12</span><br><span class="line-number">13</span><br><span class="line-number">14</span><br><span class="line-number">15</span><br><span class="line-number">16</span><br><span class="line-number">17</span><br><span class="line-number">18</span><br></div></div><p><code>dir.create(&quot;./results/Figure2/WGCNA&quot;,recursive = T)</code>，</p> <ul><li>recursive，回归的，递归的</li> <li>默认FALSE，父目录存在时才创建，TRUE，父目录不存在才会创建</li></ul> <div class="language- line-numbers-mode"><pre class="language-text"><code>#  Step 3: analysis of network topology
powers = c(c(1:10), seq(from = 12, to = 30, by = 2)) # Choose a set of soft-thresholding powers

# Call the network topology analysis function 调用网络拓扑分析函数
sft = pickSoftThreshold(plot.mat, 
						powerVector = powers,  # 包含了用于测试的不同幂律参数值
                        networkType=&quot;unsigned&quot;,  # 指定网络类型为无符号网络（不考虑基因表达正负）
                        verbose=5,   # 日志记录级别参数，越高越详细
                        )
                        
pdf(paste0(&quot;./results/Figure2/WGCNA/2.power.pdf&quot;), width = 12, height = 8)
par(mfrow = c(1,2))
cex1 = 0.9

plot(sft$fitIndices[,1], -sign(sft$fitIndices[,3])*sft$fitIndices[,2],
     xlab=&quot;Soft Threshold (power)&quot;,
     ylab=&quot;Scale Free Topology Model Fit, signed R^2&quot;,type=&quot;n&quot;,
     main = paste(&quot;Scale independence&quot;)
     )
text(sft$fitIndices[,1], -sign(sft$fitIndices[,3])*sft$fitIndices[,2],
     labels=powers,cex=cex1,col=&quot;red&quot;)
abline(h=0.85,col=&quot;red&quot;)

plot(sft$fitIndices[,1], sft$fitIndices[,5],
     xlab=&quot;Soft Threshold (power)&quot;,
     ylab=&quot;Mean Connectivity&quot;, type=&quot;n&quot;,
     main = paste(&quot;Mean connectivity&quot;))
text(sft$fitIndices[,1], sft$fitIndices[,5], labels=powers, 
     cex=cex1, col=&quot;red&quot;)
dev.off()
</code></pre> <div class="line-numbers-wrapper"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br><span class="line-number">6</span><br><span class="line-number">7</span><br><span class="line-number">8</span><br><span class="line-number">9</span><br><span class="line-number">10</span><br><span class="line-number">11</span><br><span class="line-number">12</span><br><span class="line-number">13</span><br><span class="line-number">14</span><br><span class="line-number">15</span><br><span class="line-number">16</span><br><span class="line-number">17</span><br><span class="line-number">18</span><br><span class="line-number">19</span><br><span class="line-number">20</span><br><span class="line-number">21</span><br><span class="line-number">22</span><br><span class="line-number">23</span><br><span class="line-number">24</span><br><span class="line-number">25</span><br><span class="line-number">26</span><br><span class="line-number">27</span><br><span class="line-number">28</span><br><span class="line-number">29</span><br><span class="line-number">30</span><br></div></div><ul><li><code>?pickSoftThreshold</code> 或 <code>help(pickSoftThreshold)</code>，查看<code>pickSoftThreshold</code>用法</li> <li>该函数通过分析不同幂次下的网络拓扑属性来确定最佳幂次，确保网络具有理想的模块化特性</li></ul> <p><img src="http://www.genetictargets.com/PDAC2BOOKLET/_main_files/figure-html/unnamed-chunk-16-1.png" alt="scale independence + mean connectivity |500"></p> <div class="language- line-numbers-mode"><pre class="language-text"><code>#  Step 4: soft threshold : power
power = sft$powerEstimate
power


#  Step 5: One-step network construction and module detection
net = blockwiseModules(plot.mat,  # 一个相似性或距离矩阵，用于构建网络
					   power = power,  # 用于构建网络的幂次参数
					   maxBlockSize = ncol(plot.mat), # 在模块检测过程中使用的最大块大小
                       TOMType = &quot;signed&quot;,  # 拓扑重叠矩阵（TOM）的类型，signed，考虑正负相关性，unsigned，不考虑正负相关性
                       minModuleSize = 20,  # 模块的最小大小
                       reassignThreshold = 0,  # 重新分配基因到其他模块的阈值，这里设置为0，表示不进行重新分配
                       mergeCutHeight = 0.0001,
                       numericLabels = TRUE, 
                       pamRespectsDendro = FALSE,
                       saveTOMs=TRUE, 
                       corType = corType, 
                       maxPOutliers = 1, 
                       loadTOMs=TRUE,
                       randomSeed = 931, # seed
                       saveTOMFileBase = paste0(&quot;./results/Figure2/WGCNA/WGCNA.tom&quot;),
                       verbose = 3, 
                       pearsonFallback = &quot;none&quot;, 
                       deepSplit = 3 )
# module:
table(net$colors) # 0 corresponds to unclassified genes

# Convert [number] labels to [colors] for plotting
moduleLabels = net$colors
moduleColors = labels2colors(moduleLabels)

# plot
pdf(paste0( &quot;./results/Figure2/WGCNA/3.module.pdf&quot;), width = 8, height = 6)
plotDendroAndColors(net$dendrograms[[1]], moduleColors[net$blockGenes[[1]]],
                    &quot;Module colors&quot;,
                    dendroLabels = FALSE, hang = 0.03,
                    addGuide = TRUE, guideHang = 0.05)
dev.off()
</code></pre> <div class="line-numbers-wrapper"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br><span class="line-number">6</span><br><span class="line-number">7</span><br><span class="line-number">8</span><br><span class="line-number">9</span><br><span class="line-number">10</span><br><span class="line-number">11</span><br><span class="line-number">12</span><br><span class="line-number">13</span><br><span class="line-number">14</span><br><span class="line-number">15</span><br><span class="line-number">16</span><br><span class="line-number">17</span><br><span class="line-number">18</span><br><span class="line-number">19</span><br><span class="line-number">20</span><br><span class="line-number">21</span><br><span class="line-number">22</span><br><span class="line-number">23</span><br><span class="line-number">24</span><br><span class="line-number">25</span><br><span class="line-number">26</span><br><span class="line-number">27</span><br><span class="line-number">28</span><br><span class="line-number">29</span><br><span class="line-number">30</span><br><span class="line-number">31</span><br><span class="line-number">32</span><br><span class="line-number">33</span><br><span class="line-number">34</span><br><span class="line-number">35</span><br><span class="line-number">36</span><br><span class="line-number">37</span><br><span class="line-number">38</span><br></div></div><p><img src="http://www.genetictargets.com/PDAC2BOOKLET/_main_files/figure-html/unnamed-chunk-18-1.png" alt="cluster dendrogen|500"></p> <p>！！！很好看的那个圆球</p> <div class="language- line-numbers-mode"><pre class="language-text"><code>#  Step 6: Vasualization
node.data &lt;- readRDS(&quot;./data/proteomics/wgcna/2D-node.data.rds&quot;)
edge.data &lt;- readRDS(&quot;./data/proteomics/wgcna/2D-edge.data.rds&quot;)

color.module &lt;- c(&quot;#CCCCCC&quot;,&quot;#ecb888&quot;,&quot;#af88bb&quot;,&quot;#a032cb&quot;,&quot;#efbed6&quot;,&quot;#fc496a&quot;,&quot;#b6d37f&quot;,&quot;#589336&quot;,&quot;#7fd68e&quot;,&quot;#52c465&quot;,&quot;#3372e0&quot;,&quot;#84d7f6&quot;,&quot;#5394c3&quot;,&quot;#6376b3&quot;,&quot;#7f6cd7&quot;,&quot;#c4ceff&quot;,&quot;#fc9d40&quot;,&quot;#5c95e0&quot;,&quot;#cd7560&quot;,&quot;#ff70e4&quot;, &quot;#ff8738&quot;, &quot;#ffcead&quot;,&quot;#1cbf8b&quot;, &quot;#b76d38&quot;, &quot;#1584ff&quot;, &quot;#7f006d&quot;, &quot;#ffd35f&quot;,&quot;#E66F73&quot;,&quot;#F57F20&quot;,&quot;#1DBB95&quot;,&quot;#9CB79F&quot;,&quot;#F0B8D2&quot;,&quot;#A0485E&quot;,&quot;#A0688E&quot;,&quot;#C7E1DF&quot;,&quot;#51B1DF&quot;,&quot;#6D97D7&quot;,&quot;#5D6193&quot;,&quot;#CEC3E0&quot;,&quot;#A9917E&quot;,&quot;#7C7D80&quot;,&quot;#F4E192&quot;,&quot;#ADD666&quot;)
names(color.module) &lt;- paste0(&quot;ME&quot;, seq(0,length(color.module)-1) %&gt;% str_pad(width = 2,side = &quot;left&quot;,pad = &quot;0&quot;))
## plot network 

gg &lt;- ggplot()
gg &lt;- gg + geom_segment(mapping = aes(x = from.x, y = from.y, xend = to.x, yend = to.y),
                        color = &quot;#CCCCCC&quot;, 
                        size = 0.01, 
                        data = edge.data
                        ) # draw a straight line
gg &lt;- gg + geom_point(mapping = aes(x = pos.x, y = pos.y, color = Module), 
                      size = 2, data = node.data
                      )  # add point
gg &lt;- gg + scale_size(range = c(0, 6) * 2) # specifies the minimum and maximum size 
gg &lt;- gg + theme_void()
gg &lt;- gg + labs(x = &quot;&quot;, y = &quot;&quot;, title = paste0(&quot;PPI&quot;))
gg &lt;- gg + scale_colour_manual(values = color.module)
ggsave(paste0(&quot;./results/Figure2/2A-PRO-modules.png&quot;), gg, width = 10, height = 8)
</code></pre> <div class="line-numbers-wrapper"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br><span class="line-number">6</span><br><span class="line-number">7</span><br><span class="line-number">8</span><br><span class="line-number">9</span><br><span class="line-number">10</span><br><span class="line-number">11</span><br><span class="line-number">12</span><br><span class="line-number">13</span><br><span class="line-number">14</span><br><span class="line-number">15</span><br><span class="line-number">16</span><br><span class="line-number">17</span><br><span class="line-number">18</span><br><span class="line-number">19</span><br><span class="line-number">20</span><br><span class="line-number">21</span><br><span class="line-number">22</span><br></div></div><p><img src="https://s2.loli.net/2024/06/03/ZfIz8lvnt9sHaoW.png" alt="image.png|600"></p> <p>显著差异基因（红蓝色块的蛋白分布圆球）</p> <div class="language- line-numbers-mode"><pre class="language-text"><code>#  Overlay the log2 fold change between tumor and TAC on PRO in the WGCNA network
node.data &lt;- readRDS(&quot;./data/proteomics/wgcna/2D-node.data.rds&quot;) # DEG &amp; module genes
edge.data &lt;- readRDS(&quot;./data/proteomics/wgcna/2D-edge.data.rds&quot;)

## plot
gg &lt;- ggplot()
gg &lt;- gg + geom_segment(mapping = aes(x = from.x, y = from.y, xend = to.x, yend = to.y),
                        color = &quot;#CCCCCC&quot;, size = 0.01, data = edge.data)
gg &lt;- gg + geom_point(mapping = aes(x = pos.x, y = pos.y, color = proSig), size = 1,
                      data = node.data[which(node.data$proSig == &quot;zz&quot;), ])
gg &lt;- gg + geom_point(mapping = aes(x = pos.x, y = pos.y, color = proSig), size = 2.5,
                      data = node.data[which(node.data$proSig != &quot;zz&quot;), ])
gg &lt;- gg + scale_size(range = c(0, 6) * 2)
gg &lt;- gg + theme_void()
gg &lt;- gg + labs(x = &quot;&quot;, y = &quot;&quot;, title = paste0(&quot;color by proSig&quot;))
gg &lt;- gg + scale_colour_manual(values = c(&quot;#00599F&quot;,&quot;#D01910&quot;,&quot;#AAAAAA&quot;))
ggsave(paste0(&quot;./results/Figure2/2B-proSig.png&quot;), gg, width = 8.5, height = 8)
</code></pre> <div class="line-numbers-wrapper"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br><span class="line-number">6</span><br><span class="line-number">7</span><br><span class="line-number">8</span><br><span class="line-number">9</span><br><span class="line-number">10</span><br><span class="line-number">11</span><br><span class="line-number">12</span><br><span class="line-number">13</span><br><span class="line-number">14</span><br><span class="line-number">15</span><br><span class="line-number">16</span><br><span class="line-number">17</span><br></div></div><p><img src="https://s2.loli.net/2024/06/03/o5RBLdaKl4Q8ib3.png" alt="image.png|500"></p> <p><strong>模块富集得分 / 上下条图</strong></p> <div class="language- line-numbers-mode"><pre class="language-text"><code>color.bin &lt;- c(&quot;#00599F&quot;,&quot;#d80700&quot;)
#----------------------------------------------------------------------------------
#  Step 1: Load the RJ-cohort 1 Data
#----------------------------------------------------------------------------------
# Scores of the 32 modules in TACs and PDACs of RJ-cohort 1
plot.data &lt;- read.xlsx(&quot;./data/Extended Data Table 4.xlsx&quot;, sheet = 2, startRow = 2, rowNames = T)
#                       ME01         ME02        ME03       ME04
# RJ-01-0143-N_PRO 0.3474261 -0.114677437 -0.04291552 0.06708347
# RJ-01-0768-N_PRO 0.2532539  0.009051615 -0.03585508 0.03486392
# RJ-01-0697-N_PRO 0.2839048 -0.032966990 -0.02943711 0.05803330
# RJ-01-0609-N_PRO 0.3206755 -0.087559353 -0.05749618 0.10310726
plot.info &lt;- NULL
module.name &lt;- colnames(plot.data)
plot.stat &lt;- data.frame(Module = module.name,
                        Wilcox.P = NA,
                        Wilcox.Padj = NA)
#----------------------------------------------------------------------------------
#  Step 2: wilcox.test
#----------------------------------------------------------------------------------
for (i in 1:ncol(plot.data)) {
  sub &lt;- data.frame(Sample = rownames(plot.data),
                    Score = as.numeric(plot.data[, i]), 
                    ScoreScale = scale(as.numeric(plot.data[, i])),
                    Module = colnames(plot.data)[i],
                    Type = substr(rownames(plot.data), 12, 16))
  plot.stat$Wilcox.P[i] &lt;- wilcox.test(sub$ScoreScale ~ sub$Type)$p.value
  plot.info &lt;- rbind(plot.info, sub)
}
plot.info &lt;- plot.info %&gt;% mutate(Type=factor(as.character(Type), levels = c(&quot;N_PRO&quot;,&quot;T_PRO&quot;))) 
plot.stat &lt;- plot.stat %&gt;% mutate(Wilcox.Padj=p.adjust(Wilcox.P, method = &quot;fdr&quot;))
#----------------------------------------------------------------------------------
#  Step 3: plot
#----------------------------------------------------------------------------------
p &lt;- ggbarplot(plot.info, x = &quot;Module&quot;, y = &quot;ScoreScale&quot;,
               color = &quot;Type&quot;, fill = &quot;Type&quot;,
               palette = color.bin, width = 0.5, size = 0,
               add = c(&quot;mean_se&quot;), add.params = list(width = 0.5),
               order = module.name,
               position = position_dodge(0.6),
               xlab = &quot;&quot;, ylab = &quot;Module Score of Protein&quot;)
p &lt;- p + theme_base() 
p &lt;- p + geom_hline(yintercept = 0, color = &quot;black&quot;)
p &lt;- p + theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5))
p &lt;- p + stat_compare_means(aes(group = Type, label = ..p.format..), size = 1, method = &quot;wilcox.test&quot;, label.y = 1.2)  + theme(plot.background = element_blank())
#----------------------------------------------------------------------------------
#  Step 4: output
#----------------------------------------------------------------------------------
ggsave(paste0(&quot;./results/Figure2/Figure2c-1.pdf&quot;), p, width = 12, height = 4)
write.xlsx(plot.stat, paste0(&quot;./results/Figure2/Figure2c.padj-1.xlsx&quot;))
</code></pre> <div class="line-numbers-wrapper"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br><span class="line-number">6</span><br><span class="line-number">7</span><br><span class="line-number">8</span><br><span class="line-number">9</span><br><span class="line-number">10</span><br><span class="line-number">11</span><br><span class="line-number">12</span><br><span class="line-number">13</span><br><span class="line-number">14</span><br><span class="line-number">15</span><br><span class="line-number">16</span><br><span class="line-number">17</span><br><span class="line-number">18</span><br><span class="line-number">19</span><br><span class="line-number">20</span><br><span class="line-number">21</span><br><span class="line-number">22</span><br><span class="line-number">23</span><br><span class="line-number">24</span><br><span class="line-number">25</span><br><span class="line-number">26</span><br><span class="line-number">27</span><br><span class="line-number">28</span><br><span class="line-number">29</span><br><span class="line-number">30</span><br><span class="line-number">31</span><br><span class="line-number">32</span><br><span class="line-number">33</span><br><span class="line-number">34</span><br><span class="line-number">35</span><br><span class="line-number">36</span><br><span class="line-number">37</span><br><span class="line-number">38</span><br><span class="line-number">39</span><br><span class="line-number">40</span><br><span class="line-number">41</span><br><span class="line-number">42</span><br><span class="line-number">43</span><br><span class="line-number">44</span><br><span class="line-number">45</span><br><span class="line-number">46</span><br><span class="line-number">47</span><br><span class="line-number">48</span><br><span class="line-number">49</span><br></div></div><p><img src="https://s2.loli.net/2024/06/03/qYIMrpDiofeUSzE.png" alt="image.png|400"></p> <ul><li>p.adjust(Wilcox.P, method = &quot;fdr&quot;)</li> <li>Wilcoxon 秩和检验（一种非参数检验方法，常用于比较两个独立样本的中位数差异）</li> <li>用于对一组 P 值进行调整，以控制多重假设检验中的一类错误率（family-wise error rate）</li> <li>&quot;fdr&quot; 指的是 Benjamini-Hochberg 程序的 false discovery rate（FDR）方法，FDR 方法旨在控制错误发现的比例，而不是控制单个检验的错误率</li></ul> <p><strong>Association</strong> of the enrichment score of the 32 modules with clinical data and RNA-seq subtype.</p> <div class="language- line-numbers-mode"><pre class="language-text"><code>library(survival)
#----------------------------------------------------------------------------------
#  Step 1: Loading expression data and set parameters
#----------------------------------------------------------------------------------
# (1) Sample information and clinical characteristics of the 191 PDAC patients (the RJ-cohort 1)
rj1.cohort &lt;- read.xlsx(&quot;./data/Extended Data Table 2.xlsx&quot;, startRow = 2)
#           ID     Proteomic_ID        RNAseq_ID Age Gender      BMI DM Smoking
# 1 RJ-01-0020 RJ-01-0020-T_PRO RJ-01-0020-T_RNA  61 Female 19.63000  1       0
# 2 RJ-01-0038 RJ-01-0038-T_PRO RJ-01-0038-T_RNA  73   Male 22.46003  0       0
# 3 RJ-01-0050 RJ-01-0050-T_PRO RJ-01-0050-T_RNA  69 Female 17.63085  0       0
# 4 RJ-01-0069 RJ-01-0069-T_PRO RJ-01-0069-T_RNA  69 Female 17.48179  1       0
# 5 RJ-01-0070 RJ-01-0070-T_PRO RJ-01-0070-T_RNA  65   Male 23.43750  0       0
# 6 RJ-01-0074 RJ-01-0074-T_PRO RJ-01-0074-T_RNA  61   Male 23.66144  0       0

# (2) Scores of the 32 modules in TACs and PDACs of RJ-cohort 1. 
module.ssgsea.pro.all &lt;- read.xlsx(&quot;./data/Extended Data Table 4.xlsx&quot;, sheet = 2, startRow = 2, rowNames = T)
module.ssgsea.pro.all &lt;- t(module.ssgsea.pro.all)
#      RJ-01-0143-N_PRO RJ-01-0768-N_PRO RJ-01-0697-N_PRO RJ-01-0609-N_PRO RJ-01-1055-N_PRO
# ME01       0.34742611      0.253253916       0.28390477       0.32067554      0.325797851
# ME02      -0.11467744      0.009051615      -0.03296699      -0.08755935     -0.096947986
# ME03      -0.04291552     -0.035855079      -0.02943711      -0.05749618      0.016639009
# ME04       0.06708347      0.034863920       0.05803330       0.10310726      0.007498371
# ME05       0.19099018      0.100153070       0.14246697       0.16675071      0.127932887
# ME06       0.03893163      0.240865770       0.14178269       0.07618613      0.038306853

# (3) Lasso Score in RJ-cohort 1 RNA-seq
rj1.rnaseq &lt;- read.xlsx(&quot;./data/RNA/rj1.rnaseq.lasso.score.xlsx&quot;)
#           ID     Proteomic_ID        RNAseq_ID  LassoScore LassoLevel LassoScoreRNA LassoLevelRNA
# 1 RJ-01-0020 RJ-01-0020-T_PRO RJ-01-0020-T_RNA  0.22945083       High     -2.394123          High
# 2 RJ-01-0038 RJ-01-0038-T_PRO RJ-01-0038-T_RNA -0.05725430       High     -2.561299           Low
# 3 RJ-01-0050 RJ-01-0050-T_PRO RJ-01-0050-T_RNA -0.10574547       High     -2.720116           Low
# 4 RJ-01-0069 RJ-01-0069-T_PRO RJ-01-0069-T_RNA -0.69376332        Low     -2.595342           Low
# 5 RJ-01-0070 RJ-01-0070-T_PRO RJ-01-0070-T_RNA -0.03992765       High     -2.581288           Low
# 6 RJ-01-0074 RJ-01-0074-T_PRO RJ-01-0074-T_RNA  0.17797893       High     -2.572467           Low

#----------------------------------------------------------------------------------
#  Step 2: Module score ~ OS/DFS (coxph, prognosis)
#----------------------------------------------------------------------------------
module.name &lt;- rownames(module.ssgsea.pro.all)

plot.info &lt;- NULL
plot.data &lt;- rj1.cohort
plot.info.os &lt;- NULL

## (1) Module score ~ OS
for (i in 1:length(module.name)) {
  # data:  module score -&gt; Low/High level (median cutoff)
  plot.data &lt;- plot.data %&gt;% 
              mutate(ME_score = module.ssgsea.pro.all[module.name[i], plot.data$Proteomic_ID] %&gt;% as.numeric()) %&gt;% 
              mutate(ME_level=ifelse(ME_score &gt;= median(ME_score),&quot;High&quot;,&quot;Low&quot;)) %&gt;% 
              mutate(ME_level=factor(ME_level,levels = c(&quot;Low&quot;,&quot;High&quot;)))
  # coxph
  fit &lt;- coxph(Surv(OS_month, OS_status) ~ ME_level, data = plot.data)
  fit.info &lt;- summary(fit)
  fit.p &lt;- survdiff(Surv(OS_month, OS_status) ~ ME_level, data = plot.data)$pvalue
  fit.info.out &lt;- c(module.name[i], &quot;OS&quot;, fit.info$conf.int[1, 1], fit.info$coefficients[1,5], &quot;Log-rank test&quot; )
  plot.info.os &lt;- rbind(plot.info.os, fit.info.out)
}
plot.info.os &lt;- as.data.frame(plot.info.os)
colnames(plot.info.os) &lt;- c(&quot;Module&quot;, &quot;Factor&quot;, &quot;Type&quot;,&quot;P&quot;, &quot;Test&quot;)
# result
plot.info.os &lt;- plot.info.os %&gt;% 
                    mutate(Type=as.numeric(Type),P=as.numeric(P),
                           Padj= p.adjust(P, method = &quot;fdr&quot;)) %&gt;% 
                    mutate(Tag=case_when(Type&gt;1&amp;P&lt;0.05 ~ &quot;Unfavorable&quot;,
                                         Type&lt;1&amp;P&lt;0.05 ~ &quot;Favorable&quot;,
                                         .default=&quot;notSig&quot; ))
## (1) Module score ~ DFS                    
plot.data &lt;- rj1.cohort
plot.info.dfs &lt;- NULL
for (i in 1:length(module.name)) {
  # data:  module score -&gt; Low/High level (median cutoff)
  plot.data &lt;- plot.data %&gt;% 
              mutate(ME_score = module.ssgsea.pro.all[module.name[i], plot.data$Proteomic_ID] %&gt;% as.numeric()) %&gt;% 
              mutate(ME_level=ifelse(ME_score &gt;= median(ME_score),&quot;High&quot;,&quot;Low&quot;)) %&gt;% 
              mutate(ME_level=factor(ME_level,levels = c(&quot;Low&quot;,&quot;High&quot;)))
  # coxph  
  fit &lt;- coxph(Surv(DFS_month, DFS_status) ~ ME_level, data = plot.data)
  fit.info &lt;- summary(fit)
  fit.p &lt;- survdiff(Surv(DFS_month, DFS_status) ~ ME_level, data = plot.data)$pvalue
  fit.info.out &lt;- c(module.name[i], &quot;DFS&quot;, fit.info$conf.int[1, 1], fit.info$coefficients[1,5], &quot;Log-rank test&quot; )
  plot.info.dfs &lt;- rbind(plot.info.dfs, fit.info.out)
}
plot.info.dfs &lt;- as.data.frame(plot.info.dfs)
colnames(plot.info.dfs) &lt;- c(&quot;Module&quot;, &quot;Factor&quot;, &quot;Type&quot;, &quot;P&quot;, &quot;Test&quot;)
# result
plot.info.dfs &lt;- plot.info.dfs %&gt;% 
                    mutate(Type=as.numeric(Type),P=as.numeric(P),
                           Padj= p.adjust(P, method = &quot;fdr&quot;)) %&gt;% 
                    mutate(Tag=case_when(Type&gt;1&amp;P&lt;0.05 ~ &quot;Unfavorable&quot;,
                                         Type&lt;1&amp;P&lt;0.05 ~ &quot;Favorable&quot;,
                                         .default=&quot;notSig&quot; ))
#----------------------------------------------------------------------------------
#  Step 3: Module score ~ continuous type clinical trait (cor)
#----------------------------------------------------------------------------------
plot.name &lt;- c(&quot;Age&quot;,&quot;BMI&quot;,&quot;WBC&quot;,&quot;Hb&quot;,&quot;PLT&quot;,&quot;TBIL&quot;,&quot;DBIL&quot;,&quot;ALB&quot;,&quot;Glu&quot;,&quot;CA125&quot;,&quot;CA19_9&quot;,&quot;CEA&quot;,&quot;AFP&quot;,&quot;CA724&quot;,&quot;Diameter_tumor&quot;,&quot;Num_Pos_LN&quot;,&quot;Num_total_LN&quot;)
plot.name[! plot.name %in% colnames(rj1.cohort)]
plot.info.cat &lt;- NULL
for (kkk in 1:length(plot.name)) {
  plot.data &lt;- rj1.cohort
  plot.data$Plot &lt;- plot.data[, which(colnames(plot.data) == plot.name[kkk])]
  plot.data &lt;- plot.data[which(!is.na(plot.data$Plot)), ]
  plot.info.cat.sub &lt;- NULL
  for (i in 1:length(module.name)) {
    plot.data$ME_score &lt;- module.ssgsea.pro.all[module.name[i], plot.data$Proteomic_ID] %&gt;% as.numeric()
    sub &lt;- c(module.name[i], plot.name[kkk], cor(plot.data$ME_score, plot.data$Plot, method = &quot;spearman&quot;), 
             cor.test(plot.data$ME_score, plot.data$Plot, method = &quot;spearman&quot;)$p.value, &quot;Spearman's correlation&quot;)
    plot.info.cat.sub &lt;- rbind(plot.info.cat.sub, sub)
  }
  plot.info.cat.sub &lt;- as.data.frame(plot.info.cat.sub)
  colnames(plot.info.cat.sub) &lt;- c(&quot;Module&quot;, &quot;Factor&quot;, &quot;Type&quot;,&quot;P&quot;, &quot;Test&quot;)
  plot.info.cat.sub &lt;- plot.info.cat.sub %&gt;% 
                            mutate(Type=as.numeric(Type),
                                   P=as.numeric(P),
                                   Padj=p.adjust(P,method=&quot;fdr&quot;)) 
  plot.info.cat &lt;- rbind(plot.info.cat, plot.info.cat.sub)
}
plot.info.cat &lt;- plot.info.cat %&gt;% 
                            mutate(Tag=case_when(Type&lt;0&amp;P&lt;0.05 ~ &quot;Negative&quot;,
                                                 Type&gt;0&amp;P&lt;0.05 ~ &quot;Positive&quot;,
                                                .default=&quot;notSig&quot; ))
#----------------------------------------------------------------------------------
#  Step 4: Module score ~ category type clinical trait (Wilcoxon rank-sum test or ANOVA)
#----------------------------------------------------------------------------------
plot.name &lt;- c(&quot;Gender&quot;,&quot;Drinking&quot;,&quot;Smoking&quot;,&quot;Ninv&quot;,&quot;T&quot;,&quot;N&quot;,&quot;Stage&quot;,&quot;Grade&quot;)
plot.name[! plot.name %in% colnames(rj1.cohort)]
plot.info.con &lt;- NULL
for (kkk in 1:length(plot.name)) {
  plot.data &lt;- rj1.cohort
  plot.data$Plot &lt;- plot.data[, which(colnames(plot.data) == plot.name[kkk])]
  plot.data &lt;- plot.data[which(!is.na(plot.data$Plot)), ]
  plot.info.con.sub &lt;- NULL
  if (length(unique(plot.data$Plot)) == 2) { # 2 types category -&gt; wilcox
    for (i in 1:length(module.name)) {
      plot.data$ME_score &lt;- module.ssgsea.pro.all[module.name[i], plot.data$Proteomic_ID] %&gt;% as.numeric()
      sub &lt;- c(module.name[i], plot.name[kkk], NA, wilcox.test(plot.data$ME_score ~ as.factor(plot.data$Plot) )$p.value, &quot;Wilcoxon rank-sum test&quot;)
      plot.info.con.sub &lt;- rbind(plot.info.con.sub, sub)
    } 
  } else { # more than 2 types category -&gt; anova
    for (i in 1:length(module.name)) {
      plot.data$ME_score &lt;- module.ssgsea.pro.all[module.name[i], plot.data$Proteomic_ID] %&gt;% as.numeric()
      sub &lt;- c(module.name[i], plot.name[kkk], NA, unlist(summary(aov(plot.data$ME_score ~ as.factor(plot.data$Plot) )))[9], &quot;ANOVA&quot;)
      plot.info.con.sub &lt;- rbind(plot.info.con.sub, sub)
    }
  }
  plot.info.con.sub &lt;- as.data.frame(plot.info.con.sub)
  colnames(plot.info.con.sub) &lt;- c(&quot;Module&quot;, &quot;Factor&quot;, &quot;Type&quot;, &quot;P&quot;, &quot;Test&quot;)
  plot.info.con.sub &lt;- plot.info.con.sub %&gt;% 
                            mutate(Type=as.numeric(Type),
                                   P=as.numeric(P),
                                   Padj=p.adjust(P,method=&quot;fdr&quot;)) 
  plot.info.con &lt;- rbind(plot.info.con, plot.info.con.sub)
}
plot.info.con &lt;- plot.info.con %&gt;% mutate(Tag=ifelse(P&lt;0.05,&quot;Sig&quot;,&quot;notSig&quot;))
#----------------------------------------------------------------------------------
#  Step 5: Module score ~ RNAseq subtype (Wilcoxon rank-sum test)
#----------------------------------------------------------------------------------
plot.name &lt;- c(&quot;RNAseq_subtype&quot;)
plot.name[! plot.name %in% colnames(rj1.rnaseq)]
plot.info.rna &lt;- NULL
rj1.rnaseq$RNAseq_subtype &lt;- rj1.cohort$RNAseq_subtype[match(rj1.rnaseq$ID, rj1.cohort$ID)]
plot.data &lt;- rj1.rnaseq
for (i in 1:length(module.name)) {
  plot.data$ME_score &lt;- module.ssgsea.pro.all[module.name[i], plot.data$Proteomic_ID] %&gt;% as.numeric()
  sub &lt;- c(module.name[i], &quot;RNAseq_subtype&quot;, 
           mean(plot.data$ME_score[which(plot.data$RNAseq_subtype == &quot;Basal&quot;)]) - mean(plot.data$ME_score[which(plot.data$RNAseq_subtype == &quot;Classical&quot;)]) , 
           wilcox.test(plot.data$ME_score ~ as.factor(plot.data$RNAseq_subtype) )$p.value, &quot;Wilcoxon rank-sum test&quot;)
  plot.info.rna &lt;- rbind(plot.info.rna, sub)
}
plot.info.rna &lt;- as.data.frame(plot.info.rna)
colnames(plot.info.rna) &lt;- c(&quot;Module&quot;, &quot;Factor&quot;, &quot;Type&quot;, &quot;P&quot;, &quot;Test&quot;)
plot.info.rna &lt;- plot.info.rna %&gt;% 
                    mutate(Type=as.numeric(Type),P=as.numeric(P),
                           Padj= p.adjust(P, method = &quot;fdr&quot;)) %&gt;% 
                    mutate(Tag=case_when(Type&gt;0&amp;P&lt;0.05 ~ &quot;Basal&quot;,
                                         Type&lt;0&amp;P&lt;0.05 ~ &quot;Classical&quot;,
                                         .default=&quot;notSig&quot; ))
#----------------------------------------------------------------------------------
#  Step 6: plot
#----------------------------------------------------------------------------------
plot.info &lt;- rbind(plot.info.os, plot.info.dfs, plot.info.cat, plot.info.con, plot.info.rna)
plot.info &lt;- plot.info %&gt;% mutate(LogP=-log10(P),
                                  Module=factor(as.character(Module), levels = rev(module.name)),
                                  Factor=factor(as.character(Factor), levels = c(&quot;OS&quot;,&quot;DFS&quot;,&quot;Age&quot;,&quot;BMI&quot;,&quot;WBC&quot;,&quot;Hb&quot;,&quot;PLT&quot;,&quot;TBIL&quot;,&quot;DBIL&quot;,&quot;ALB&quot;,&quot;Glu&quot;,&quot;CA125&quot;,&quot;CA19_9&quot;,&quot;CEA&quot;,&quot;AFP&quot;,&quot;CA724&quot;,&quot;Diameter_tumor&quot;,&quot;Num_Pos_LN&quot;,&quot;Num_total_LN&quot;,&quot;Gender&quot;,&quot;Drinking&quot;,&quot;Smoking&quot;,&quot;Ninv&quot;,&quot;T&quot;,&quot;N&quot;,&quot;Stage&quot;,&quot;Grade&quot;,&quot;RNAseq_subtype&quot;)))

p &lt;- ggplot(plot.info, aes(Factor, Module)) +
  geom_point(aes(size = LogP, colour = Tag, fill = Tag), shape = 16) + 
  xlab(&quot;&quot;) + ylab(&quot;&quot;)
p &lt;- p + theme_base() + scale_size(range = c(0,10))
p &lt;- p + theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5, size = 15))
p &lt;- p + theme(axis.text.y = element_text(size = 20))
p &lt;- p + scale_colour_manual(values = c(Favorable = &quot;#00599F&quot;, Unfavorable = &quot;#D01910&quot;, Negative = &quot;#009632&quot;, Positive = &quot;#8f00b7&quot;, Sig = &quot;#ed7a00&quot;, Basal = &quot;#ea6c59&quot;, Classical = &quot;#1288a5&quot;, notSig = &quot;#CCCCCC&quot;)) + theme(plot.background = element_blank())
ggsave(paste0(&quot;./results/Figure2/Figure2d.pdf&quot;), p, width = 12, height = 11)
write.xlsx(plot.info, &quot;./results/Figure2/Figure2d.xlsx&quot;)
</code></pre> <div class="line-numbers-wrapper"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br><span class="line-number">6</span><br><span class="line-number">7</span><br><span class="line-number">8</span><br><span class="line-number">9</span><br><span class="line-number">10</span><br><span class="line-number">11</span><br><span class="line-number">12</span><br><span class="line-number">13</span><br><span class="line-number">14</span><br><span class="line-number">15</span><br><span class="line-number">16</span><br><span class="line-number">17</span><br><span class="line-number">18</span><br><span class="line-number">19</span><br><span class="line-number">20</span><br><span class="line-number">21</span><br><span class="line-number">22</span><br><span class="line-number">23</span><br><span class="line-number">24</span><br><span class="line-number">25</span><br><span class="line-number">26</span><br><span class="line-number">27</span><br><span class="line-number">28</span><br><span class="line-number">29</span><br><span class="line-number">30</span><br><span class="line-number">31</span><br><span class="line-number">32</span><br><span class="line-number">33</span><br><span class="line-number">34</span><br><span class="line-number">35</span><br><span class="line-number">36</span><br><span class="line-number">37</span><br><span class="line-number">38</span><br><span class="line-number">39</span><br><span class="line-number">40</span><br><span class="line-number">41</span><br><span class="line-number">42</span><br><span class="line-number">43</span><br><span class="line-number">44</span><br><span class="line-number">45</span><br><span class="line-number">46</span><br><span class="line-number">47</span><br><span class="line-number">48</span><br><span class="line-number">49</span><br><span class="line-number">50</span><br><span class="line-number">51</span><br><span class="line-number">52</span><br><span class="line-number">53</span><br><span class="line-number">54</span><br><span class="line-number">55</span><br><span class="line-number">56</span><br><span class="line-number">57</span><br><span class="line-number">58</span><br><span class="line-number">59</span><br><span class="line-number">60</span><br><span class="line-number">61</span><br><span class="line-number">62</span><br><span class="line-number">63</span><br><span class="line-number">64</span><br><span class="line-number">65</span><br><span class="line-number">66</span><br><span class="line-number">67</span><br><span class="line-number">68</span><br><span class="line-number">69</span><br><span class="line-number">70</span><br><span class="line-number">71</span><br><span class="line-number">72</span><br><span class="line-number">73</span><br><span class="line-number">74</span><br><span class="line-number">75</span><br><span class="line-number">76</span><br><span class="line-number">77</span><br><span class="line-number">78</span><br><span class="line-number">79</span><br><span class="line-number">80</span><br><span class="line-number">81</span><br><span class="line-number">82</span><br><span class="line-number">83</span><br><span class="line-number">84</span><br><span class="line-number">85</span><br><span class="line-number">86</span><br><span class="line-number">87</span><br><span class="line-number">88</span><br><span class="line-number">89</span><br><span class="line-number">90</span><br><span class="line-number">91</span><br><span class="line-number">92</span><br><span class="line-number">93</span><br><span class="line-number">94</span><br><span class="line-number">95</span><br><span class="line-number">96</span><br><span class="line-number">97</span><br><span class="line-number">98</span><br><span class="line-number">99</span><br><span class="line-number">100</span><br><span class="line-number">101</span><br><span class="line-number">102</span><br><span class="line-number">103</span><br><span class="line-number">104</span><br><span class="line-number">105</span><br><span class="line-number">106</span><br><span class="line-number">107</span><br><span class="line-number">108</span><br><span class="line-number">109</span><br><span class="line-number">110</span><br><span class="line-number">111</span><br><span class="line-number">112</span><br><span class="line-number">113</span><br><span class="line-number">114</span><br><span class="line-number">115</span><br><span class="line-number">116</span><br><span class="line-number">117</span><br><span class="line-number">118</span><br><span class="line-number">119</span><br><span class="line-number">120</span><br><span class="line-number">121</span><br><span class="line-number">122</span><br><span class="line-number">123</span><br><span class="line-number">124</span><br><span class="line-number">125</span><br><span class="line-number">126</span><br><span class="line-number">127</span><br><span class="line-number">128</span><br><span class="line-number">129</span><br><span class="line-number">130</span><br><span class="line-number">131</span><br><span class="line-number">132</span><br><span class="line-number">133</span><br><span class="line-number">134</span><br><span class="line-number">135</span><br><span class="line-number">136</span><br><span class="line-number">137</span><br><span class="line-number">138</span><br><span class="line-number">139</span><br><span class="line-number">140</span><br><span class="line-number">141</span><br><span class="line-number">142</span><br><span class="line-number">143</span><br><span class="line-number">144</span><br><span class="line-number">145</span><br><span class="line-number">146</span><br><span class="line-number">147</span><br><span class="line-number">148</span><br><span class="line-number">149</span><br><span class="line-number">150</span><br><span class="line-number">151</span><br><span class="line-number">152</span><br><span class="line-number">153</span><br><span class="line-number">154</span><br><span class="line-number">155</span><br><span class="line-number">156</span><br><span class="line-number">157</span><br><span class="line-number">158</span><br><span class="line-number">159</span><br><span class="line-number">160</span><br><span class="line-number">161</span><br><span class="line-number">162</span><br><span class="line-number">163</span><br><span class="line-number">164</span><br><span class="line-number">165</span><br><span class="line-number">166</span><br><span class="line-number">167</span><br><span class="line-number">168</span><br><span class="line-number">169</span><br><span class="line-number">170</span><br><span class="line-number">171</span><br><span class="line-number">172</span><br><span class="line-number">173</span><br><span class="line-number">174</span><br><span class="line-number">175</span><br><span class="line-number">176</span><br><span class="line-number">177</span><br><span class="line-number">178</span><br><span class="line-number">179</span><br><span class="line-number">180</span><br><span class="line-number">181</span><br><span class="line-number">182</span><br><span class="line-number">183</span><br><span class="line-number">184</span><br><span class="line-number">185</span><br><span class="line-number">186</span><br><span class="line-number">187</span><br><span class="line-number">188</span><br><span class="line-number">189</span><br><span class="line-number">190</span><br><span class="line-number">191</span><br><span class="line-number">192</span><br><span class="line-number">193</span><br><span class="line-number">194</span><br></div></div><p><img src="https://s2.loli.net/2024/06/03/u62s14tWBDlrq89.png" alt="image.png|600"></p> <ul><li><code>coxph(Surv(OS_month, OS_status) ~ ME_level, data = plot.data)</code> 这行代码的作用是创建一个Cox比例风险模型，用来分析 <code>ME_level</code> 变量如何影响生存时间 <code>OS_month</code>，其中 <code>OS_status</code> 表示生存事件的状态，这个模型会考虑数据框 <code>plot.data</code> 中的数据。</li></ul> <p>这两行代码习惯值得学习（<code>命名方法</code>，<code>排序方法</code>）</p> <div class="language- line-numbers-mode"><pre class="language-text"><code>plot.info &lt;- rbind(plot.info.os, plot.info.dfs, plot.info.cat, plot.info.con, plot.info.rna)

plot.info &lt;- plot.info %&gt;% mutate(LogP=-log10(P),
                                  Module=factor(as.character(Module), levels = rev(module.name)),
                                  Factor=factor(as.character(Factor), levels = c(&quot;OS&quot;,&quot;DFS&quot;,&quot;Age&quot;,&quot;BMI&quot;,&quot;WBC&quot;,&quot;Hb&quot;,&quot;PLT&quot;,&quot;TBIL&quot;,&quot;DBIL&quot;,&quot;ALB&quot;,&quot;Glu&quot;,&quot;CA125&quot;,&quot;CA19_9&quot;,&quot;CEA&quot;,&quot;AFP&quot;,&quot;CA724&quot;,&quot;Diameter_tumor&quot;,&quot;Num_Pos_LN&quot;,&quot;Num_total_LN&quot;,&quot;Gender&quot;,&quot;Drinking&quot;,&quot;Smoking&quot;,&quot;Ninv&quot;,&quot;T&quot;,&quot;N&quot;,&quot;Stage&quot;,&quot;Grade&quot;,&quot;RNAseq_subtype&quot;)))
</code></pre> <div class="line-numbers-wrapper"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br></div></div><p>生存分析图</p> <div class="language- line-numbers-mode"><pre class="language-text"><code>library(survival)
library(survminer)
#----------------------------------------------------------------------------------
#  Step 1: Load the data
#----------------------------------------------------------------------------------
# (1) Sample information and clinical characteristics of the 191 PDAC patients (the RJ-cohort 1)
rj1.cohort &lt;- read.xlsx(&quot;./data/Extended Data Table 2.xlsx&quot;, startRow = 2)
#           ID     Proteomic_ID        RNAseq_ID Age Gender      BMI
# 1 RJ-01-0020 RJ-01-0020-T_PRO RJ-01-0020-T_RNA  61 Female 19.63000
# 2 RJ-01-0038 RJ-01-0038-T_PRO RJ-01-0038-T_RNA  73   Male 22.46003
# 3 RJ-01-0050 RJ-01-0050-T_PRO RJ-01-0050-T_RNA  69 Female 17.63085
# 4 RJ-01-0069 RJ-01-0069-T_PRO RJ-01-0069-T_RNA  69 Female 17.48179

# (2) Scores of the 32 modules in TACs and PDACs of RJ-cohort 1. 
module.ssgsea.pro.all &lt;- read.xlsx(&quot;./data/Extended Data Table 4.xlsx&quot;, sheet = 2, startRow = 2, rowNames = T)
module.ssgsea.pro.all &lt;- t(module.ssgsea.pro.all)
#      RJ-01-0143-N_PRO RJ-01-0768-N_PRO RJ-01-0697-N_PRO RJ-01-0609-N_PRO RJ-01-1055-N_PRO
# ME01       0.34742611      0.253253916       0.28390477       0.32067554      0.325797851
# ME02      -0.11467744      0.009051615      -0.03296699      -0.08755935     -0.096947986
# ME03      -0.04291552     -0.035855079      -0.02943711      -0.05749618      0.016639009
# ME04       0.06708347      0.034863920       0.05803330       0.10310726      0.007498371
# ME05       0.19099018      0.100153070       0.14246697       0.16675071      0.127932887
# ME06       0.03893163      0.240865770       0.14178269       0.07618613      0.038306853
#----------------------------------------------------------------------------------
#  Step 2: Overall survival stratified by ME11
#----------------------------------------------------------------------------------
plot.cohort &lt;- rj1.cohort
plot.cohort &lt;- plot.cohort %&gt;% 
                    mutate(ME11_pro = as.numeric(module.ssgsea.pro.all[&quot;ME11&quot;, Proteomic_ID]),
                           ME11_pro_l = ifelse(ME11_pro&gt;=median(ME11_pro),&quot;High&quot;,&quot;Low&quot;),
                           ME11_pro_l = factor(as.character(ME11_pro_l), levels = c(&quot;Low&quot;,&quot;High&quot;))) %&gt;% 
                    mutate(Censoring_chemo_rev = Censoring_chemo,
                           Censoring_chemo = factor(as.character(Censoring_chemo), levels = c(&quot;Chem&quot;,&quot;NoChem&quot;)),
                           Censoring_chemo_rev = factor(as.character(Censoring_chemo_rev), levels = c(&quot;NoChem&quot;,&quot;Chem&quot;)))
# coxph
info &lt;- summary(coxph(Surv(OS_month, OS_status) ~ ME11_pro_l, data = plot.cohort))
anno.text &lt;- &quot;&quot;
for (i in 1:nrow(info$conf.int)) {
  anno.text &lt;- paste0(anno.text, &quot;\n&quot;, paste0(rownames(info$conf.int)[i], &quot; HR=&quot;, round(info$conf.int[i, 1], 3), &quot; CI=&quot;, round(info$conf.int[i, 3], 3), &quot;-&quot;, round(info$conf.int[i, 4], 3), &quot; P=&quot;, signif(info$coefficients[i, 5], 4) ))
}
anno.text &lt;- paste0(anno.text, &quot;\nKaplan-Meier P=&quot;, signif(survdiff(Surv(OS_month, OS_status) ~ ME11_pro_l, data = plot.cohort)$pvalue, 4) )
anno.text &lt;- str_replace_all(anno.text, &quot;ME11_pro_l&quot;, &quot;&quot;)
fit &lt;- survfit(Surv(OS_month, OS_status) ~ ME11_pro_l, data = plot.cohort)
p1 &lt;- ggsurvplot(fit, 
                 data = plot.cohort,
                 xlab = 'Time (Months)',
                 pval = TRUE,
                 risk.table = TRUE, 
                 risk.table.height = 0.28,
                 conf.int.alpha = 0.05,
                 conf.int = TRUE, 
                 palette = c(&quot;#00599F&quot;,&quot;#d80700&quot;),
                 axes.offset = TRUE,
                 break.time.by = 12,  xlim = c(0, 48),
                 title= paste0(&quot;OS ME11_pro_l \n&quot;, anno.text))
#----------------------------------------------------------------------------------
#  Step 2: KM curve in adjuvant therapy patients stratified by ME11
#----------------------------------------------------------------------------------
plot.cohort.sub &lt;- plot.cohort %&gt;% filter(Censoring_chemo == &quot;Chem&quot;) %&gt;% 
                          mutate(ME11_pro_l = ifelse(ME11_pro&gt;= median(ME11_pro),&quot;High&quot;,&quot;Low&quot;),
                                 ME11_pro_l = factor(as.character(ME11_pro_l), levels = c(&quot;Low&quot;,&quot;High&quot;)))
info &lt;- summary(coxph(Surv(OS_month, OS_status) ~ ME11_pro_l, data = plot.cohort.sub))
anno.text &lt;- &quot;&quot;
for (i in 1:nrow(info$conf.int)) {
  anno.text &lt;- paste0(anno.text, &quot;\n&quot;, paste0(rownames(info$conf.int)[i], &quot; HR=&quot;, round(info$conf.int[i, 1], 3), &quot; CI=&quot;, round(info$conf.int[i, 3], 3), &quot;-&quot;, round(info$conf.int[i, 4], 3), &quot; P=&quot;, signif(info$coefficients[i, 5], 4) ))
}
anno.text &lt;- paste0(anno.text, &quot;\nKaplan-Meier P=&quot;, signif(survdiff(Surv(OS_month, OS_status) ~ ME11_pro_l, data = plot.cohort.sub)$pvalue, 4) )
anno.text &lt;- str_replace_all(anno.text, &quot;ME11_pro_l&quot;, &quot;&quot;)
fit &lt;- survfit(Surv(OS_month, OS_status) ~ ME11_pro_l, data = plot.cohort.sub)
p2 &lt;- ggsurvplot(fit, 
                 data = plot.cohort.sub,
                 xlab = 'Time (Months)',
                 pval = TRUE,
                 risk.table = TRUE, 
                 risk.table.height = 0.28,
                 conf.int.alpha = 0.05,
                 conf.int = TRUE, 
                 palette = c(&quot;#00599F&quot;,&quot;#d80700&quot;),
                 axes.offset = TRUE,
                 break.time.by = 12,  xlim = c(0, 48),
                 title= paste0(&quot;OS ME11_pro_l Chem \n&quot;, anno.text))
#----------------------------------------------------------------------------------
#  Step 2: KM curve in no adjuvant therapy patients stratified by ME11
#----------------------------------------------------------------------------------
plot.cohort.sub &lt;- plot.cohort %&gt;% filter(Censoring_chemo == &quot;NoChem&quot;) %&gt;% 
                      mutate(ME11_pro_l = ifelse(ME11_pro &gt;= median(ME11_pro),&quot;High&quot;,&quot;Low&quot;),
                             ME11_pro_l = factor(as.character(ME11_pro_l), levels = c(&quot;Low&quot;,&quot;High&quot;)))
info &lt;- summary(coxph(Surv(OS_month, OS_status) ~ ME11_pro_l, data = plot.cohort.sub))
anno.text &lt;- &quot;&quot;
for (i in 1:nrow(info$conf.int)) {
  anno.text &lt;- paste0(anno.text, &quot;\n&quot;, paste0(rownames(info$conf.int)[i], &quot; HR=&quot;, round(info$conf.int[i, 1], 3), &quot; CI=&quot;, round(info$conf.int[i, 3], 3), &quot;-&quot;, round(info$conf.int[i, 4], 3), &quot; P=&quot;, signif(info$coefficients[i, 5], 4) ))
}
anno.text &lt;- paste0(anno.text, &quot;\nKaplan-Meier P=&quot;, signif(survdiff(Surv(OS_month, OS_status) ~ ME11_pro_l, data = plot.cohort.sub)$pvalue, 4) )
anno.text &lt;- str_replace_all(anno.text, &quot;ME11_pro_l&quot;, &quot;&quot;)
fit &lt;- survfit(Surv(OS_month, OS_status) ~ ME11_pro_l, data = plot.cohort.sub)
p3 &lt;- ggsurvplot(fit, 
                 data = plot.cohort.sub,
                 xlab = 'Time (Months)',
                 pval = TRUE,
                 risk.table = TRUE, 
                 risk.table.height = 0.28,
                 conf.int.alpha = 0.05,
                 conf.int = TRUE, 
                 palette = c(&quot;#00599F&quot;,&quot;#d80700&quot;),
                 axes.offset = TRUE,
                 break.time.by = 12,  xlim = c(0, 48),
                 title= paste0(&quot;OS ME11_pro_l NoChem \n&quot;, anno.text))
## output
p &lt;- arrange_ggsurvplots(list(p1, p2, p3), ncol = 3, nrow = 1, print = FALSE)
ggsave(paste0(&quot;./results/Figure2/Figure2e.pdf&quot;), p, width = 18, height = 8)
</code></pre> <div class="line-numbers-wrapper"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br><span class="line-number">6</span><br><span class="line-number">7</span><br><span class="line-number">8</span><br><span class="line-number">9</span><br><span class="line-number">10</span><br><span class="line-number">11</span><br><span class="line-number">12</span><br><span class="line-number">13</span><br><span class="line-number">14</span><br><span class="line-number">15</span><br><span class="line-number">16</span><br><span class="line-number">17</span><br><span class="line-number">18</span><br><span class="line-number">19</span><br><span class="line-number">20</span><br><span class="line-number">21</span><br><span class="line-number">22</span><br><span class="line-number">23</span><br><span class="line-number">24</span><br><span class="line-number">25</span><br><span class="line-number">26</span><br><span class="line-number">27</span><br><span class="line-number">28</span><br><span class="line-number">29</span><br><span class="line-number">30</span><br><span class="line-number">31</span><br><span class="line-number">32</span><br><span class="line-number">33</span><br><span class="line-number">34</span><br><span class="line-number">35</span><br><span class="line-number">36</span><br><span class="line-number">37</span><br><span class="line-number">38</span><br><span class="line-number">39</span><br><span class="line-number">40</span><br><span class="line-number">41</span><br><span class="line-number">42</span><br><span class="line-number">43</span><br><span class="line-number">44</span><br><span class="line-number">45</span><br><span class="line-number">46</span><br><span class="line-number">47</span><br><span class="line-number">48</span><br><span class="line-number">49</span><br><span class="line-number">50</span><br><span class="line-number">51</span><br><span class="line-number">52</span><br><span class="line-number">53</span><br><span class="line-number">54</span><br><span class="line-number">55</span><br><span class="line-number">56</span><br><span class="line-number">57</span><br><span class="line-number">58</span><br><span class="line-number">59</span><br><span class="line-number">60</span><br><span class="line-number">61</span><br><span class="line-number">62</span><br><span class="line-number">63</span><br><span class="line-number">64</span><br><span class="line-number">65</span><br><span class="line-number">66</span><br><span class="line-number">67</span><br><span class="line-number">68</span><br><span class="line-number">69</span><br><span class="line-number">70</span><br><span class="line-number">71</span><br><span class="line-number">72</span><br><span class="line-number">73</span><br><span class="line-number">74</span><br><span class="line-number">75</span><br><span class="line-number">76</span><br><span class="line-number">77</span><br><span class="line-number">78</span><br><span class="line-number">79</span><br><span class="line-number">80</span><br><span class="line-number">81</span><br><span class="line-number">82</span><br><span class="line-number">83</span><br><span class="line-number">84</span><br><span class="line-number">85</span><br><span class="line-number">86</span><br><span class="line-number">87</span><br><span class="line-number">88</span><br><span class="line-number">89</span><br><span class="line-number">90</span><br><span class="line-number">91</span><br><span class="line-number">92</span><br><span class="line-number">93</span><br><span class="line-number">94</span><br><span class="line-number">95</span><br><span class="line-number">96</span><br><span class="line-number">97</span><br><span class="line-number">98</span><br><span class="line-number">99</span><br><span class="line-number">100</span><br><span class="line-number">101</span><br><span class="line-number">102</span><br><span class="line-number">103</span><br><span class="line-number">104</span><br><span class="line-number">105</span><br><span class="line-number">106</span><br><span class="line-number">107</span><br><span class="line-number">108</span><br><span class="line-number">109</span><br><span class="line-number">110</span><br></div></div><p>最后结果还挺好看的，值得一学（已经整理到文件夹里，跑的通）
<img src="https://s2.loli.net/2024/06/04/bqpAlMemHPRkQLr.png" alt="image.png|700"></p> <ul><li>另外，生存分析用<code>library(survival)</code>，<code>library(survminer)</code>跑，数据最少只要两列（一列时间，一列是否存活），后边用上<code>coxph</code>，很方便去做风险因素回归！</li></ul> <h3 id="figure-3"><a href="#figure-3" class="header-anchor">#</a> Figure 3</h3> <p><strong>Establishing an optimal proteomics-level prognostic risk model using the LASSO-Cox regression</strong></p> <p>利用LASSO-Cox回归，建立最佳蛋白质组学预后风险模型</p> <div class="language- line-numbers-mode"><pre class="language-text"><code>library(glmnet)
library(openxlsx)
library(survminer)
library(survival)
library(tidyverse)
library(RColorBrewer)
library(ComplexHeatmap)
library(forestmodel)
library(ROCR)
dir.create(&quot;./results/Figure3/lasso&quot;,recursive = T)
#----------------------------------------------------------------------------------
#  Step 1: load data
#----------------------------------------------------------------------------------
## PDAC Tumor proteomics
pdac.pro  &lt;- readRDS('./data/proteomics/20230412_PDAC_PRO_exp.rds') 
pdac.pro  &lt;- pdac.pro [,grep('_T_',colnames(pdac.pro))]   %&gt;% t() %&gt;% as.data.frame() # 191 tumor samples * 4787 protein
## WGCNA protein
wgcna &lt;- readRDS(&quot;./data/proteomics/wgcna/2D-node.data.rds&quot;)
pdac.pro &lt;- pdac.pro [,intersect(wgcna$Gene,colnames(pdac.pro))] #  191 samples * 3906 protein
## PDAC clinical data
cli.pdac &lt;- readRDS('./data/clinical/20230412_PDAC_191_patients_clinical.rds')
rownames(cli.pdac) &lt;- cli.pdac$ID
## Lasso data
y.train &lt;- data.frame(time=cli.pdac$Time_OS,
                      status=cli.pdac$Censoring_OS,
                      row.names = row.names(cli.pdac)) %&gt;% 
           na.omit(y.train) %&gt;% 
           as.matrix() 
x.train &lt;- pdac.pro[row.names(y.train),] %&gt;% as.matrix() %&gt;% na.omit()
y.train &lt;- y.train[row.names(x.train),]
#----------------------------------------------------------------------------------
#  Step 2: Fit the Lasso Regression Model
#----------------------------------------------------------------------------------
set.seed(1000) 
fit &lt;-glmnet(x=x.train,
             y=y.train, 
             family = &quot;cox&quot;,  # cox : cox , binomial : logistic
             alpha  = 1)      # 1 : lasso penalty, 0 : ridge penalty.
pdf('./results/Figure3/lasso/0.Coefficients.pdf',width = 5,height = 4)
plot(fit,xvar=&quot;lambda&quot;,label=F) 
dev.off()
# Cross-validation
cvfit &lt;- cv.glmnet(x=x.train,
                   y=y.train,
                   family = &quot;cox&quot;,
                   type.measure=&quot;deviance&quot;,  
                   alpha  = 1,
                   nfolds = 10)
png(paste0('./results/Figure3/lasso/S5A-Coeff.png'), width = 5,height = 5,res=1200,units=&quot;in&quot;)
plot(fit,xvar=&quot;lambda&quot;,label=F) 
abline(v = log(cvfit$lambda.min), col = &quot;black&quot;,lty=2)
text(-4,-4, labels = paste0(&quot;lambda.min:  &quot;, round(cvfit$lambda.min,4)),col = &quot;black&quot;,cex = 0.75)
dev.off()
</code></pre> <div class="line-numbers-wrapper"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br><span class="line-number">6</span><br><span class="line-number">7</span><br><span class="line-number">8</span><br><span class="line-number">9</span><br><span class="line-number">10</span><br><span class="line-number">11</span><br><span class="line-number">12</span><br><span class="line-number">13</span><br><span class="line-number">14</span><br><span class="line-number">15</span><br><span class="line-number">16</span><br><span class="line-number">17</span><br><span class="line-number">18</span><br><span class="line-number">19</span><br><span class="line-number">20</span><br><span class="line-number">21</span><br><span class="line-number">22</span><br><span class="line-number">23</span><br><span class="line-number">24</span><br><span class="line-number">25</span><br><span class="line-number">26</span><br><span class="line-number">27</span><br><span class="line-number">28</span><br><span class="line-number">29</span><br><span class="line-number">30</span><br><span class="line-number">31</span><br><span class="line-number">32</span><br><span class="line-number">33</span><br><span class="line-number">34</span><br><span class="line-number">35</span><br><span class="line-number">36</span><br><span class="line-number">37</span><br><span class="line-number">38</span><br><span class="line-number">39</span><br><span class="line-number">40</span><br><span class="line-number">41</span><br><span class="line-number">42</span><br><span class="line-number">43</span><br><span class="line-number">44</span><br><span class="line-number">45</span><br><span class="line-number">46</span><br><span class="line-number">47</span><br><span class="line-number">48</span><br><span class="line-number">49</span><br><span class="line-number">50</span><br><span class="line-number">51</span><br><span class="line-number">52</span><br><span class="line-number">53</span><br></div></div><ul><li>library(glmnet) -- glmnet()，cv.glmnet()，glmnet()，</li> <li>library(forestmodel) -- forest_model()
<img src="https://s2.loli.net/2024/06/04/JTl9VghPoKzfavc.png" alt="image.png|400"></li> <li>library(ROCR) -- predition()，performance()，</li></ul> <h3 id="figure-4"><a href="#figure-4" class="header-anchor">#</a> Figure 4</h3> <p>Chemotherapy biomarker -- Cox proportional-hazards models in terms of OS</p> <div class="language- line-numbers-mode"><pre class="language-text"><code>library(openxlsx)
library(tidyverse)
library(stringr)
library(survival)
library(survminer)
library(pbapply)
library(ggthemes)
dir.create(&quot;./results/Figure4&quot;)
#----------------------------------------------------------------------------------
#  Step 1: Load the Data
#----------------------------------------------------------------------------------
deg &lt;- readRDS(&quot;./data/proteomics/wgcna/2D-node.data.rds&quot;)  # DEG &amp; module genes
pro &lt;- readRDS(&quot;./data/proteomics/20230412_PDAC_PRO_exp.rds&quot;)
cli &lt;- readRDS(&quot;./data/clinical/20230412_PDAC_191_patients_clinical.rds&quot;)
#----------------------------------------------------------------------------------
#  Step 2: Coxph data
#----------------------------------------------------------------------------------
# clinical data + DEG genes' expression data
deg &lt;- deg %&gt;% filter(proSig !=&quot;zz&quot;) %&gt;% pull(Gene) # 1860 DEG genes
rownames(cli) &lt;- cli$ID
id.inter &lt;- intersect(rownames(cli),colnames(pro)) # 191 patients
pro &lt;- pro[deg,id.inter] %&gt;% t()
identical(rownames(cli),rownames(pro))
biomarker_data &lt;- cbind(cli %&gt;% dplyr::select(OS_Time,OS_Status,Censoring_chemo),pro)
# category attribute: chemotherapy treatment
biomarker_data &lt;- biomarker_data %&gt;% mutate(chem=factor(Censoring_chemo,levels = c(&quot;0&quot;,&quot;1&quot;)))
#----------------------------------------------------------------------------------
#  Step 3: Coxph
#----------------------------------------------------------------------------------
# Coxph : OS ~ chemotherapy*gene
ls_re &lt;- pbapply::pblapply(colnames(pro),FUN=function(g){
    data &lt;- biomarker_data %&gt;% select(OS_Time,OS_Status,chem) %&gt;%
            mutate(Gene=biomarker_data%&gt;%pull(g))
    fit  &lt;- coxph( Surv(OS_Time,OS_Status) ~ Gene*chem ,data=data) ## interaction term
    broom::tidy(fit) %&gt;% mutate(HR=exp(estimate)) %&gt;% select(p.value,HR) %&gt;% t() %&gt;% as.vector() 
})
# result
res &lt;- do.call(rbind,ls_re)
colnames(res) &lt;- c(&quot;biomarker_p&quot;,&quot;biomarker_HR&quot;,&quot;chemotherapy_p&quot;,
                    &quot;chemotherapy_HR&quot;,&quot;interaction_p&quot;,&quot;interaction_HR&quot;)
## predictive biomarker
res &lt;- res %&gt;% as_tibble() %&gt;% mutate(Gene=colnames(pro)) %&gt;% 
            arrange(interaction_p) %&gt;%  
            dplyr::select(Gene,everything())
# cutoff 
res[which(res$interaction_HR&gt;1 &amp; res$interaction_p &lt;0.05),&quot;group&quot;] = &quot;bad_biomarker&quot;
res[which(res$interaction_HR&lt;1 &amp; res$interaction_p &lt;0.05),&quot;group&quot;] = &quot;good_biomarker&quot;
table(res$group)
#----------------------------------------------------------------------------------
#  Step 4: Visualization
#----------------------------------------------------------------------------------
color.tri &lt;- c(&quot;#D01910&quot;,&quot;#00599F&quot;,&quot;#444444&quot;)  # up ,down ,not
## volcano
res &lt;- res %&gt;% mutate(group=factor(group,levels = c(&quot;bad_biomarker&quot;,&quot;good_biomarker&quot;)),
                      Label=Gene,
                      Label=replace(Label,is.na(group),&quot;&quot;))
res &lt;- res %&gt;% mutate(log2HR=log2(interaction_HR)) %&gt;% 
        mutate(logP= -log10(interaction_p))
p &lt;- ggscatter(res,
               x = &quot;log2HR&quot;, y = &quot;logP&quot;,
               color = &quot;group&quot;, size = 1,
               main = paste0(&quot;Predictive biomarker for chemotherapy&quot;),
               xlab = &quot;log2(HR)&quot;, ylab = &quot;-log10(Pvalue)&quot;,
               palette = color.tri,
               label = res$Label,font.label = 6, , repel = T)+
               theme_base()+
               geom_hline(yintercept = -log10(0.05), linetype=&quot;dashed&quot;, color = &quot;#222222&quot;) +
      theme(plot.background = element_blank())
ggsave(&quot;./results/Figure4/4A-Predictive-biomarker.pdf&quot;, p, width = 10, height = 8)
write.xlsx(as.data.frame(res),&quot;./results/Figure4/4A-Predictive-biomarker.xlsx&quot;,colNames=T,rowNames=F,overwrite = T)
</code></pre> <div class="line-numbers-wrapper"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br><span class="line-number">6</span><br><span class="line-number">7</span><br><span class="line-number">8</span><br><span class="line-number">9</span><br><span class="line-number">10</span><br><span class="line-number">11</span><br><span class="line-number">12</span><br><span class="line-number">13</span><br><span class="line-number">14</span><br><span class="line-number">15</span><br><span class="line-number">16</span><br><span class="line-number">17</span><br><span class="line-number">18</span><br><span class="line-number">19</span><br><span class="line-number">20</span><br><span class="line-number">21</span><br><span class="line-number">22</span><br><span class="line-number">23</span><br><span class="line-number">24</span><br><span class="line-number">25</span><br><span class="line-number">26</span><br><span class="line-number">27</span><br><span class="line-number">28</span><br><span class="line-number">29</span><br><span class="line-number">30</span><br><span class="line-number">31</span><br><span class="line-number">32</span><br><span class="line-number">33</span><br><span class="line-number">34</span><br><span class="line-number">35</span><br><span class="line-number">36</span><br><span class="line-number">37</span><br><span class="line-number">38</span><br><span class="line-number">39</span><br><span class="line-number">40</span><br><span class="line-number">41</span><br><span class="line-number">42</span><br><span class="line-number">43</span><br><span class="line-number">44</span><br><span class="line-number">45</span><br><span class="line-number">46</span><br><span class="line-number">47</span><br><span class="line-number">48</span><br><span class="line-number">49</span><br><span class="line-number">50</span><br><span class="line-number">51</span><br><span class="line-number">52</span><br><span class="line-number">53</span><br><span class="line-number">54</span><br><span class="line-number">55</span><br><span class="line-number">56</span><br><span class="line-number">57</span><br><span class="line-number">58</span><br><span class="line-number">59</span><br><span class="line-number">60</span><br><span class="line-number">61</span><br><span class="line-number">62</span><br><span class="line-number">63</span><br><span class="line-number">64</span><br><span class="line-number">65</span><br><span class="line-number">66</span><br><span class="line-number">67</span><br><span class="line-number">68</span><br><span class="line-number">69</span><br><span class="line-number">70</span><br></div></div><ul><li>x = &quot;log2HR&quot;, y = &quot;logP&quot;，火山图常见，但是横轴为风险指标的可不常见哦！！
<img src="https://s2.loli.net/2024/06/04/uMAV1iJ6XtOHolm.png" alt="image.png|500"></li></ul> <div class="language- line-numbers-mode"><pre class="language-text"><code>library(forestmodel)
library(RColorBrewer)
#----------------------------------------------------------------------------------
#  Step 1: Load the Data
#----------------------------------------------------------------------------------
# Sample information and clinical characteristics of the 191 PDAC patients (the RJ-cohort 1)
rj1 &lt;- read.xlsx(&quot;./data/Extended Data Table 2.xlsx&quot;, startRow = 2)
rj1$`LassoLevel (High vs. Low)` &lt;- ifelse(rj1$LassoLevel == 'High', 1, 0)
rj1$`Adjuvant_therapy (Yes vs. No)` &lt;- ifelse(rj1$Censoring_chemo == 'Chem', 1, 0)
rj1$`Stage (III vs. I_II)` &lt;- ifelse(rj1$Stage == 'III', 1, 0)
rj1$`NDUFB8 (High vs. Low)` &lt;- ifelse(rj1$NDUFB8_l == 'High', 1, 0)
rj1$`CEMIP2 (High vs. Low)` &lt;- ifelse(rj1$CEMIP2_l == 'High', 1, 0)
rj1$`Sex (Male vs. Female)` &lt;- ifelse(rj1$Gender == 'Male', 1, 0)
rj1$`Margin (R1 vs. R0)` &lt;- ifelse(rj1$Margin == 'R1', 1, 0)
rj1$`Smoking (Yes vs. No)` &lt;- rj1$Smoking
rj1$`Drinking (Yes vs. No)` &lt;- rj1$Drinking
#----------------------------------------------------------------------------------
#  Step 2: plot for NDUFB8 * Adjuvant_therapy
#----------------------------------------------------------------------------------
# custom panels for column: Adjust the font style, thickness, and column width of variables in the graph 
forest.panels &lt;- list(
  list(width = 0.01),
  list(width = 0.1, display = ~variable, 
       fontface = &quot;bold&quot;, heading = &quot;Variable&quot;),
  list(width = 0.1, display = ~level),
  list(width = 0.05, display = ~n, hjust = 1, heading = &quot;N&quot;),
  list(width = 0.03, item = &quot;vline&quot;, hjust = 0.5),
  list(width = 0.55, item = &quot;forest&quot;, hjust = 0.5, 
       heading = &quot;Hazard ratio&quot;, linetype = &quot;dashed&quot;,line_x = 0),
  list(width = 0.03, item = &quot;vline&quot;, hjust = 0.5),
  list(width = 0.12, display = ~ ifelse(reference, &quot;Reference&quot;, sprintf(&quot;%0.3f (%0.3f, %0.3f)&quot;,
                                                                        trans(estimate), trans(conf.low), trans(conf.high))), display_na = NA),
  list(width = 0.03,
       display = ~ ifelse(reference, &quot;&quot;, format.pval(p.value, digits = 1, eps = 0.001) ),
       display_na = NA, hjust = 1, heading = &quot;p&quot;),
  list(width = 0.03))

Interaction &lt;- rj1[[&quot;NDUFB8&quot;]] * rj1[['Adjuvant_therapy (Yes vs. No)']]

pdf('./results/Figure4/Figure4b-1.pdf', width = 10, height = 6)
print(forest_model(coxph(as.formula(paste0('Surv(OS_month, OS_status) ~', '`', &quot;NDUFB8&quot;, '`', '+ `Adjuvant_therapy (Yes vs. No)` + Interaction + Age + `Sex (Male vs. Female)` + `Stage (III vs. I_II)` + `Margin (R1 vs. R0)` + CA19_9 + `Smoking (Yes vs. No)` + `Drinking (Yes vs. No)`')), data = rj1), panels = forest.panels, factor_separate_line = F))
dev.off()
</code></pre> <div class="line-numbers-wrapper"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br><span class="line-number">6</span><br><span class="line-number">7</span><br><span class="line-number">8</span><br><span class="line-number">9</span><br><span class="line-number">10</span><br><span class="line-number">11</span><br><span class="line-number">12</span><br><span class="line-number">13</span><br><span class="line-number">14</span><br><span class="line-number">15</span><br><span class="line-number">16</span><br><span class="line-number">17</span><br><span class="line-number">18</span><br><span class="line-number">19</span><br><span class="line-number">20</span><br><span class="line-number">21</span><br><span class="line-number">22</span><br><span class="line-number">23</span><br><span class="line-number">24</span><br><span class="line-number">25</span><br><span class="line-number">26</span><br><span class="line-number">27</span><br><span class="line-number">28</span><br><span class="line-number">29</span><br><span class="line-number">30</span><br><span class="line-number">31</span><br><span class="line-number">32</span><br><span class="line-number">33</span><br><span class="line-number">34</span><br><span class="line-number">35</span><br><span class="line-number">36</span><br><span class="line-number">37</span><br><span class="line-number">38</span><br><span class="line-number">39</span><br><span class="line-number">40</span><br><span class="line-number">41</span><br><span class="line-number">42</span><br></div></div><p>风险森林图！
<img src="https://s2.loli.net/2024/06/04/jPOhMDsG1ZJmuik.png" alt="image.png|400"></p> <p>上面这一步代码，大多数都是用于准备，下面这个代码是出图的主要函数</p> <div class="language- line-numbers-mode"><pre class="language-text"><code>#----------------------------------------------------------------------------------
#  Step 3: plot for CEMIP2 * Adjuvant_therapy
#----------------------------------------------------------------------------------
Interaction &lt;- rj1[[&quot;CEMIP2&quot;]] * rj1[['Adjuvant_therapy (Yes vs. No)']]
pdf('./results/Figure4/Figure4b-2.pdf', width = 10, height = 6)
print(forest_model(coxph(as.formula(paste0('Surv(OS_month, OS_status) ~', '`', &quot;CEMIP2&quot;, '`', '+ `Adjuvant_therapy (Yes vs. No)` + Interaction + Age + `Sex (Male vs. Female)` + `Stage (III vs. I_II)` + `Margin (R1 vs. R0)` + CA19_9 + `Smoking (Yes vs. No)` + `Drinking (Yes vs. No)`')), data = rj1), panels = forest.panels, factor_separate_line = F))
dev.off()
</code></pre> <div class="line-numbers-wrapper"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br><span class="line-number">6</span><br><span class="line-number">7</span><br></div></div><h3 id="figure-5"><a href="#figure-5" class="header-anchor">#</a> Figure 5</h3> <p>后边的只更换了队列数据，一样绘制了几个生存曲线图，验证其筛选出来的分子是有用的，原理和前面前面的类似。</p> <h2 id="感想"><a href="#感想" class="header-anchor">#</a> 感想</h2> <p>读了这篇文章，主要的感觉就是 -- 比较短，五六页吧，就结束了，但是意义还蛮大的，发现了很有用的（差异显著的）预测化疗结果的指标，感觉的不难，但是不认真整理的话，其中的<a href="/01.科研学习/02.生信分析/24.WGCNA.html">WGCNA</a>，<a href="/01.科研学习/02.生信分析/23.临床相关性.html">Survival analysis</a>，相关性分析，还是做不好的。</p> <p>但是这篇文章，为啥能发到这个上面呢？比别的用太多的WGCNA文章好在哪里呢？我觉得主要是因为研究内容（蛋白质组数据）比较新吧，再结合着临床的队列，是一篇临床应用价值比较大的文章呢！</p> <p>只不过，我稍稍查了一下，现在一个蛋白组还是接近一个<code>达不溜</code>呢，190+个患者的蛋白组信息，还有转录组打分呢，多多少少需要两百吧，还需要人力物力进行随访一千多人呢，上限就不清楚了，咱做不起做不起。</p></div></div>  <div class="page-edit"><!----> <!----> <!----></div> <div class="page-nav-wapper"><div class="page-nav-centre-wrap"><a href="/pages/ba40b8/" class="page-nav-centre page-nav-centre-prev"><div class="tooltip">胃癌相关的数据挖掘</div></a> <a href="/pages/0a11dd/" class="page-nav-centre page-nav-centre-next"><div class="tooltip">胖子CAF</div></a></div> <div class="page-nav"><p class="inner"><span class="prev">
        ←
        <a href="/pages/ba40b8/" class="prev">胃癌相关的数据挖掘</a></span> <span class="next"><a href="/pages/0a11dd/">胖子CAF</a>→
      </span></p></div></div></div> <div class="article-list"><div class="article-title"><a href="/archives/" class="iconfont icon-bi">最近更新</a></div> <div class="article-wrapper"><dl><dd>01</dd> <dt><a href="/pages/90dbe8/"><div>
            APA相关综述
            <!----></div></a> <span class="date">06-27</span></dt></dl><dl><dd>02</dd> <dt><a href="/pages/f4f4b7/"><div>
            胃癌免疫逃逸的决定因素
            <!----></div></a> <span class="date">06-25</span></dt></dl><dl><dd>03</dd> <dt><a href="/pages/0a11dd/"><div>
            胖子CAF
            <!----></div></a> <span class="date">06-21</span></dt></dl> <dl><dd></dd> <dt><a href="/archives/" class="more">更多文章&gt;</a></dt></dl></div></div></main></div> <div class="footer"><div class="icons"><a href="mailto:2590643635@qq.com" title="发邮件" target="_blank" class="iconfont icon-youjian"></a><a href="https://github.com/hongliangfang" title="GitHub" target="_blank" class="iconfont icon-github"></a><a href="https://music.163.com/#/playlist?id=755597173" title="听音乐" target="_blank" class="iconfont icon-erji"></a></div> 
  Theme by
  <a href="https://github.com/xugaoyi/vuepress-theme-vdoing" target="_blank" title="本站主题">Vdoing</a> 
    | Copyright © 2023-2024
    <span>FHL | <a href="https://github.com/xugaoyi/vuepress-theme-vdoing/blob/master/LICENSE" target="_blank">MIT License</a></span></div> <div class="buttons"><div title="返回顶部" class="button blur go-to-top iconfont icon-fanhuidingbu" style="display:none;"></div> <div title="去评论" class="button blur go-to-comment iconfont icon-pinglun" style="display:none;"></div> <div title="主题模式" class="button blur theme-mode-but iconfont icon-zhuti"><ul class="select-box" style="display:none;"><li class="iconfont icon-zidong">
          跟随系统
        </li><li class="iconfont icon-rijianmoshi">
          浅色模式
        </li><li class="iconfont icon-yejianmoshi">
          深色模式
        </li><li class="iconfont icon-yuedu">
          阅读模式
        </li></ul></div></div> <!----> <!----> <!----></div><div class="global-ui"><div></div></div></div>
    <script src="/assets/js/app.bd488e07.js" defer></script><script src="/assets/js/2.bdf5b8cf.js" defer></script><script src="/assets/js/32.2fcf84e2.js" defer></script><script src="/assets/js/3.365ac406.js" defer></script>
  </body>
</html>
