<!DOCTYPE html>
<html lang="zh-CN">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width,initial-scale=1">
    <title>NASH复现记录 | 一盏烛光的微笑灿烂人生</title>
    <meta name="generator" content="VuePress 1.9.9">
    <link rel="icon" href="/img/favicon.ico">
    <link rel="stylesheet" href="https://at.alicdn.com/t/font_3129839_xft6cqs5gc.css">
    <noscript><meta http-equiv="refresh" content="0; url=https://www.youngkbt.cn/noscript/"><style>.theme-vdoing-content { display:none }</noscript>
    <meta name="description" content="学习记录博客">
    <meta name="theme-color" content="#11a8cd">
    <meta name="个人博客" content="科研学习，围炉夜话，奇淫巧计">
    
    <link rel="preload" href="/assets/css/0.styles.c9ffc21e.css" as="style"><link rel="preload" href="/assets/js/app.bd488e07.js" as="script"><link rel="preload" href="/assets/js/2.bdf5b8cf.js" as="script"><link rel="preload" href="/assets/js/136.9de1870b.js" as="script"><link rel="preload" href="/assets/js/3.365ac406.js" as="script"><link rel="prefetch" href="/assets/js/10.b00b53b4.js"><link rel="prefetch" href="/assets/js/100.1b2ba6dd.js"><link rel="prefetch" href="/assets/js/101.e392eb92.js"><link rel="prefetch" href="/assets/js/102.eb0bb162.js"><link rel="prefetch" href="/assets/js/103.9db14406.js"><link rel="prefetch" href="/assets/js/104.6ae6fd5c.js"><link rel="prefetch" href="/assets/js/105.4d79399b.js"><link rel="prefetch" href="/assets/js/106.4e6ff12d.js"><link rel="prefetch" href="/assets/js/107.bc93ee73.js"><link rel="prefetch" href="/assets/js/108.59a000f0.js"><link rel="prefetch" href="/assets/js/109.f01f4422.js"><link rel="prefetch" href="/assets/js/11.346d31c6.js"><link rel="prefetch" href="/assets/js/110.96a96c44.js"><link rel="prefetch" href="/assets/js/111.87fe63df.js"><link rel="prefetch" href="/assets/js/112.c560a316.js"><link rel="prefetch" href="/assets/js/113.6a371b02.js"><link rel="prefetch" href="/assets/js/114.989a4d3e.js"><link rel="prefetch" href="/assets/js/115.b6baafa2.js"><link rel="prefetch" href="/assets/js/116.d469dc20.js"><link rel="prefetch" href="/assets/js/117.0a6bccf0.js"><link rel="prefetch" href="/assets/js/118.9f1ef088.js"><link rel="prefetch" href="/assets/js/119.059ef806.js"><link rel="prefetch" href="/assets/js/12.8d91a638.js"><link rel="prefetch" href="/assets/js/120.75ed11ab.js"><link rel="prefetch" href="/assets/js/121.1896ee22.js"><link rel="prefetch" href="/assets/js/122.a3664290.js"><link rel="prefetch" href="/assets/js/123.5b849c39.js"><link rel="prefetch" href="/assets/js/124.85737d6b.js"><link rel="prefetch" href="/assets/js/125.a95cd42f.js"><link rel="prefetch" href="/assets/js/126.7b1c1645.js"><link rel="prefetch" href="/assets/js/127.64a16fd4.js"><link rel="prefetch" href="/assets/js/128.a07721ba.js"><link rel="prefetch" href="/assets/js/129.9ba86871.js"><link rel="prefetch" href="/assets/js/13.18ed2aa7.js"><link rel="prefetch" href="/assets/js/130.5cdb8138.js"><link rel="prefetch" href="/assets/js/131.d2f31425.js"><link rel="prefetch" href="/assets/js/132.b2286d65.js"><link rel="prefetch" href="/assets/js/133.1229c9ae.js"><link rel="prefetch" href="/assets/js/134.71fad9a0.js"><link rel="prefetch" href="/assets/js/135.ff682301.js"><link rel="prefetch" href="/assets/js/137.a58bbdcd.js"><link rel="prefetch" href="/assets/js/138.cc768cb1.js"><link rel="prefetch" href="/assets/js/139.9d52d34a.js"><link rel="prefetch" href="/assets/js/14.4bcf7ee9.js"><link rel="prefetch" href="/assets/js/140.b1790e58.js"><link rel="prefetch" href="/assets/js/141.98bceb9f.js"><link rel="prefetch" href="/assets/js/142.0415d712.js"><link rel="prefetch" href="/assets/js/143.851c7187.js"><link rel="prefetch" href="/assets/js/144.ae79e007.js"><link rel="prefetch" href="/assets/js/145.d5020e58.js"><link rel="prefetch" href="/assets/js/146.c3eafa98.js"><link rel="prefetch" href="/assets/js/147.0a6864ec.js"><link rel="prefetch" href="/assets/js/148.bbc4bd28.js"><link rel="prefetch" href="/assets/js/149.489e8cdb.js"><link rel="prefetch" href="/assets/js/15.368d9d2d.js"><link rel="prefetch" href="/assets/js/150.7778d444.js"><link rel="prefetch" href="/assets/js/16.06f68354.js"><link rel="prefetch" href="/assets/js/17.4f8d6256.js"><link rel="prefetch" href="/assets/js/18.8989b2ff.js"><link rel="prefetch" href="/assets/js/19.abb857ca.js"><link rel="prefetch" href="/assets/js/20.e32420cf.js"><link rel="prefetch" href="/assets/js/21.06441c4d.js"><link rel="prefetch" href="/assets/js/22.7fb78c64.js"><link rel="prefetch" href="/assets/js/23.d9193b83.js"><link rel="prefetch" href="/assets/js/24.6487f43a.js"><link rel="prefetch" href="/assets/js/25.11e66938.js"><link rel="prefetch" href="/assets/js/26.c3e43726.js"><link rel="prefetch" href="/assets/js/27.24117748.js"><link rel="prefetch" href="/assets/js/28.fabf32c7.js"><link rel="prefetch" href="/assets/js/29.3a9331f2.js"><link rel="prefetch" href="/assets/js/30.ac5ca451.js"><link rel="prefetch" href="/assets/js/31.36e6c9c0.js"><link rel="prefetch" href="/assets/js/32.2fcf84e2.js"><link rel="prefetch" href="/assets/js/33.a73adf6b.js"><link rel="prefetch" href="/assets/js/34.49c0fb46.js"><link rel="prefetch" href="/assets/js/35.babda623.js"><link rel="prefetch" href="/assets/js/36.56221bac.js"><link rel="prefetch" href="/assets/js/37.5d6d33aa.js"><link rel="prefetch" href="/assets/js/38.a6b592e0.js"><link rel="prefetch" href="/assets/js/39.14a81539.js"><link rel="prefetch" href="/assets/js/4.a64d5c9f.js"><link rel="prefetch" href="/assets/js/40.e94616b3.js"><link rel="prefetch" href="/assets/js/41.4ad9febc.js"><link rel="prefetch" href="/assets/js/42.4825ff11.js"><link rel="prefetch" href="/assets/js/43.6d554f8a.js"><link rel="prefetch" href="/assets/js/44.ed8f011b.js"><link rel="prefetch" href="/assets/js/45.170e103d.js"><link rel="prefetch" href="/assets/js/46.9cc703a7.js"><link rel="prefetch" href="/assets/js/47.ff6af7ea.js"><link rel="prefetch" href="/assets/js/48.1b2f1a1b.js"><link rel="prefetch" href="/assets/js/49.05223958.js"><link rel="prefetch" href="/assets/js/5.b1fd9f62.js"><link rel="prefetch" href="/assets/js/50.26aa9f07.js"><link rel="prefetch" href="/assets/js/51.65eba0cc.js"><link rel="prefetch" href="/assets/js/52.5b21c37d.js"><link rel="prefetch" href="/assets/js/53.4f209254.js"><link rel="prefetch" href="/assets/js/54.176923ed.js"><link rel="prefetch" href="/assets/js/55.74f6741d.js"><link rel="prefetch" href="/assets/js/56.383e3c0d.js"><link rel="prefetch" href="/assets/js/57.35b21350.js"><link rel="prefetch" href="/assets/js/58.7ae8fd30.js"><link rel="prefetch" href="/assets/js/59.3d7d7dea.js"><link rel="prefetch" href="/assets/js/6.993d7ad0.js"><link rel="prefetch" href="/assets/js/60.67ba2844.js"><link rel="prefetch" href="/assets/js/61.6c5542cd.js"><link rel="prefetch" href="/assets/js/62.a235dad7.js"><link rel="prefetch" href="/assets/js/63.a3a52fe2.js"><link rel="prefetch" href="/assets/js/64.2b6b610e.js"><link rel="prefetch" href="/assets/js/65.8672534a.js"><link rel="prefetch" href="/assets/js/66.c7281353.js"><link rel="prefetch" href="/assets/js/67.e062f6ac.js"><link rel="prefetch" href="/assets/js/68.446f0688.js"><link rel="prefetch" href="/assets/js/69.f03843ed.js"><link rel="prefetch" href="/assets/js/7.afb33a4c.js"><link rel="prefetch" href="/assets/js/70.466d842a.js"><link rel="prefetch" href="/assets/js/71.87fcb26b.js"><link rel="prefetch" href="/assets/js/72.ccf1092d.js"><link rel="prefetch" href="/assets/js/73.3ad6124b.js"><link rel="prefetch" href="/assets/js/74.17a59372.js"><link rel="prefetch" href="/assets/js/75.6c859470.js"><link rel="prefetch" href="/assets/js/76.674a56ad.js"><link rel="prefetch" href="/assets/js/77.5342f3c1.js"><link rel="prefetch" href="/assets/js/78.154ba5e1.js"><link rel="prefetch" href="/assets/js/79.10901b1c.js"><link rel="prefetch" href="/assets/js/8.5bbd7ae9.js"><link rel="prefetch" href="/assets/js/80.0166104e.js"><link rel="prefetch" href="/assets/js/81.4c5a0e6a.js"><link rel="prefetch" href="/assets/js/82.47099701.js"><link rel="prefetch" href="/assets/js/83.7873cd95.js"><link rel="prefetch" href="/assets/js/84.23000f66.js"><link rel="prefetch" href="/assets/js/85.6701c9e4.js"><link rel="prefetch" href="/assets/js/86.188012c3.js"><link rel="prefetch" href="/assets/js/87.e9cde64e.js"><link rel="prefetch" href="/assets/js/88.5ccdea49.js"><link rel="prefetch" href="/assets/js/89.9ab559de.js"><link rel="prefetch" href="/assets/js/9.6699859e.js"><link rel="prefetch" href="/assets/js/90.bc7fb61d.js"><link rel="prefetch" href="/assets/js/91.07537d0a.js"><link rel="prefetch" href="/assets/js/92.b4ed0bad.js"><link rel="prefetch" href="/assets/js/93.b167c2d0.js"><link rel="prefetch" href="/assets/js/94.1a02653c.js"><link rel="prefetch" href="/assets/js/95.69ef19be.js"><link rel="prefetch" href="/assets/js/96.e0b3f5df.js"><link rel="prefetch" href="/assets/js/97.9ea0c268.js"><link rel="prefetch" href="/assets/js/98.074852d9.js"><link rel="prefetch" href="/assets/js/99.708b118b.js">
    <link rel="stylesheet" href="/assets/css/0.styles.c9ffc21e.css">
  </head>
  <body class="theme-mode-light">
    <div id="app" data-server-rendered="true"><div class="theme-container sidebar-open have-rightmenu"><header class="navbar blur"><div title="目录" class="sidebar-button"><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" role="img" viewBox="0 0 448 512" class="icon"><path fill="currentColor" d="M436 124H12c-6.627 0-12-5.373-12-12V80c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12z"></path></svg></div> <a href="/" class="home-link router-link-active"><img src="/img/logo.png" alt="一盏烛光的微笑灿烂人生" class="logo"> <span class="site-name can-hide">一盏烛光的微笑灿烂人生</span></a> <div class="links"><div class="search-box"><input aria-label="Search" autocomplete="off" spellcheck="false" value=""> <!----></div> <nav class="nav-links can-hide"><div class="nav-item"><a href="/" class="nav-link">首页</a></div><div class="nav-item"><div class="dropdown-wrapper"><button type="button" aria-label="科研学习" class="dropdown-title"><a href="/study/" class="link-title">科研学习</a> <span class="title" style="display:none;">科研学习</span> <span class="arrow right"></span></button> <ul class="nav-dropdown" style="display:none;"><li class="dropdown-item"><!----> <a href="/articles/" class="nav-link">阅读记录</a></li><li class="dropdown-item"><!----> <a href="/biotech/" class="nav-link">生信分析</a></li><li class="dropdown-item"><!----> <a href="/basic/" class="nav-link">整理归纳</a></li></ul></div></div><div class="nav-item"><a href="/wlyh/" class="nav-link">围炉夜话</a></div><div class="nav-item"><a href="/technology/" class="nav-link">奇淫巧计</a></div><div class="nav-item"><div class="dropdown-wrapper"><button type="button" aria-label="索引" class="dropdown-title"><a href="/archives/" class="link-title">索引</a> <span class="title" style="display:none;">索引</span> <span class="arrow right"></span></button> <ul class="nav-dropdown" style="display:none;"><li class="dropdown-item"><!----> <a href="/categories/" class="nav-link">分类</a></li><li class="dropdown-item"><!----> <a href="/tags/" class="nav-link">标签</a></li><li class="dropdown-item"><!----> <a href="/archives/" class="nav-link">归档</a></li></ul></div></div><div class="nav-item"><div class="dropdown-wrapper"><button type="button" aria-label="更多" class="dropdown-title"><a href="/more/" class="link-title">更多</a> <span class="title" style="display:none;">更多</span> <span class="arrow right"></span></button> <ul class="nav-dropdown" style="display:none;"><li class="dropdown-item"><!----> <a href="/about/" class="nav-link">关于</a></li><li class="dropdown-item"><!----> <a href="/links/" class="nav-link">一些网站</a></li><li class="dropdown-item"><!----> <a href="/friends/" class="nav-link">友情链接</a></li><li class="dropdown-item"><!----> <a href="/passbyworld/" class="nav-link">路过人间</a></li></ul></div></div> <!----></nav></div></header> <div class="sidebar-mask"></div> <div class="sidebar-hover-trigger"></div> <aside class="sidebar" style="display:none;"><div class="blogger"><img src="https://pica.zhimg.com/v2-f2b30ad5674f652294f0ec6f9b8c9922_720w.webp?source=d16d100b"> <div class="blogger-info"><h3>围炉夜话</h3> <span>知行合一</span></div></div> <nav class="nav-links"><div class="nav-item"><a href="/" class="nav-link">首页</a></div><div class="nav-item"><div class="dropdown-wrapper"><button type="button" aria-label="科研学习" class="dropdown-title"><a href="/study/" class="link-title">科研学习</a> <span class="title" style="display:none;">科研学习</span> <span class="arrow right"></span></button> <ul class="nav-dropdown" style="display:none;"><li class="dropdown-item"><!----> <a href="/articles/" class="nav-link">阅读记录</a></li><li class="dropdown-item"><!----> <a href="/biotech/" class="nav-link">生信分析</a></li><li class="dropdown-item"><!----> <a href="/basic/" class="nav-link">整理归纳</a></li></ul></div></div><div class="nav-item"><a href="/wlyh/" class="nav-link">围炉夜话</a></div><div class="nav-item"><a href="/technology/" class="nav-link">奇淫巧计</a></div><div class="nav-item"><div class="dropdown-wrapper"><button type="button" aria-label="索引" class="dropdown-title"><a href="/archives/" class="link-title">索引</a> <span class="title" style="display:none;">索引</span> <span class="arrow right"></span></button> <ul class="nav-dropdown" style="display:none;"><li class="dropdown-item"><!----> <a href="/categories/" class="nav-link">分类</a></li><li class="dropdown-item"><!----> <a href="/tags/" class="nav-link">标签</a></li><li class="dropdown-item"><!----> <a href="/archives/" class="nav-link">归档</a></li></ul></div></div><div class="nav-item"><div class="dropdown-wrapper"><button type="button" aria-label="更多" class="dropdown-title"><a href="/more/" class="link-title">更多</a> <span class="title" style="display:none;">更多</span> <span class="arrow right"></span></button> <ul class="nav-dropdown" style="display:none;"><li class="dropdown-item"><!----> <a href="/about/" class="nav-link">关于</a></li><li class="dropdown-item"><!----> <a href="/links/" class="nav-link">一些网站</a></li><li class="dropdown-item"><!----> <a href="/friends/" class="nav-link">友情链接</a></li><li class="dropdown-item"><!----> <a href="/passbyworld/" class="nav-link">路过人间</a></li></ul></div></div> <!----></nav>  <ul class="sidebar-links"><li><section class="sidebar-group depth-0"><p class="sidebar-heading open"><span>NASH复现记录</span> <!----></p> <ul class="sidebar-links sidebar-group-items"><li><a href="/pages/929375/#文章思路" class="sidebar-link">文章思路</a><ul class="sidebar-sub-headers"></ul></li><li><a href="/pages/929375/#使用的工具" class="sidebar-link">使用的工具</a><ul class="sidebar-sub-headers"></ul></li><li><a href="/pages/929375/#尝试处理" class="sidebar-link">尝试处理</a><ul class="sidebar-sub-headers"></ul></li><li><a href="/pages/929375/#小结" class="sidebar-link">小结</a><ul class="sidebar-sub-headers"></ul></li></ul></section></li></ul> </aside> <div><main class="page"><div class="theme-vdoing-wrapper "><div class="articleInfo-wrap" data-v-06225672><div class="articleInfo" data-v-06225672><!----> <div class="info" data-v-06225672><div title="作者" class="author iconfont icon-touxiang" data-v-06225672><a href="javascript:;" data-v-06225672>FHL</a></div> <div title="创建时间" class="date iconfont icon-riqi" data-v-06225672><a href="javascript:;" data-v-06225672>2024-03-04</a></div> <div title="分类" class="date iconfont icon-wenjian" data-v-06225672><a href="/categories/?category=%E9%9A%8F%E7%AC%94" data-v-06225672>随笔 </a></div></div></div></div> <!----> <div class="content-wrapper"><div class="right-menu-wrapper"><div class="right-menu-margin"><div class="right-menu-title">目录</div> <div class="right-menu-content"></div></div></div> <h1><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAB4AAAAeCAYAAAA7MK6iAAAAAXNSR0IArs4c6QAABKFJREFUSA3tVl1oFVcQnrMbrak3QUgkya1akpJYcrUtIqW1JvFBE9LiQ5v6JmJpolbMg32rVrhgoYK0QiMY6i9Y6EMaW5D+xFJaTYItIuK2Kr3+BJNwkxBj05sQY3b3nM6cs2dv9t7NT/vQJw/sndk5M/PNzJkzewGerP+pAmy+ON8lLzUJgA8ZYxYIYZmGYRnctDaWvJJAmTtfP1pvXsBCCPP8QFcCaRkZYACgDZFO4stNIcBCajEOlmmC9XpJ9bAGCaPaPmzPl32dvLSVu3BWCTQs0XQQ6g0DYgwLIoAZbBCdW/i+781o1VVlm/410mw4h06Y7bIPHNyWDyL4FHkX03Q8SrzNhZTZriieckWt7cL6MM85YcLpsi/7O9/iXFT6MswI0DmmpkSaJ0qLxFIm3+i1THHB3zmBH3PYx9CcykcLOeQVVa7QtdxTgQgEleX2AjHYfwA+2ddV77ruGoJUbhGDI09YSNXyMpUt5ylOzxgbUmtOp7NmbNt8v3arjTBfYELmLUV+M+nSawNNAUqpT3ClJWg5I3BLT+cGW/DXNGCa6tx1aakCGEigArTn4TDIPdrXXYKCZNrHLMCOEPvHBlLQ99s9eHB7EB6NTki73CVPQ2F5MSx/uRQixfmq7rK0wYD8w8E905bnPDfwoWs/rfv93NWN/ZfvwsLIU7A09gxECyISeGJkHAau98L97tuw7NXnoPyNF8FcYGLGKsOs0mN3OEyec9esGW/ZEl945dTP34wlR2FZVQWU1q0Cw8Tr7p+hgLLNL0FPxx/Q35mA8aEUrH6nCgwEl0tn7wUiZYJnNRh6DK4UH/k0lfyrsBKdPVv/AriGIQcEDQZ65LBAGe2Rzui9Ybjz7XUppz1/uKBbyVPGkN3ZAeC6hr0x7Nr38N5+EqkoOm17xpoqR9ohQF55ERSvr4Dkr3chNfC3DMzGJlNBElW8w9nsGQvhNGIzDkXzCg8cLK951xHsFBlTJspJNi3ZFIMF2AeDV3q8DNOB+YHi6QTrChDIWDBRi5U5f+ZMfJLu3ccrqxtdxk4SKH336LFxSmkqefwU5T8fhdSdQf9IVKD6aNiwI/hnmcAZ91isYMJIaCUCx9W098+LgruikeTqzqqxKPUwqJyCPJiyemVVZBOijDGjD38Os0jOiSPL1z3SPjXNANbiNPXAdzTfukjjuknNBbyz3nwgTd3AVFqUJ5hpHlq9MveLnWwttUfoygBmvVjuikxND3znrhsELnZk7k+OjIGxeNEkomyLVta0xxn+HZhjBc4YZ/AFjHjz9u3xRZl2BN4aq9nFwWh16IrQ1aHHEd3j1+4/dB9OtH4e29A2H1DyHQRmOSfQZ1Fy7MHBTGB6J/Djq6p3OxyO2cB+4Car7v/o3GXgfAkj23+x9ID1Teoamo/SXcbvSf2PX7Vc8DdCmE1vN9di+32P9/5YR3vLnhCVGUWBjEkr3yh4H8v9CzmsbdhzOKzsJKM90iFdaTMjRPhGVsakRvOaRidljo6H6G7j+ctrJpsP+4COhDIl0La2+FS4+5mlocBaXY5QnGZysIBYoeSsl5qQzrSj/cgNrfuEzlWBfwA+EjrZyWUvpAAAAABJRU5ErkJggg==">NASH复现记录<!----></h1>  <div class="theme-vdoing-content content__default"><p>标题：Hepatocytes demarcated by EphB2 contribute to the progression of nonalcoholic steatohepatitis</p> <p>开始时间：2024-03-04</p> <p>复现的整体思路：</p> <ol><li>首先再次阅读文献，整理分析的流程（使用的工具，详细的参数）；</li> <li>其次，下载数据，调整数据，调包进行处理；</li> <li>最后，整理输出数据，进行统计分析，绘制重要的结果图。</li></ol> <h2 id="文章思路"><a href="#文章思路" class="header-anchor">#</a> 文章思路</h2> <p><strong>Introduction</strong></p> <p>nonalcoholic fatty liver disease (NAFLD)；nonalcoholic fatty liver (NAFL)；nonalcoholic steatohepatitis (NASH)</p> <p>NASH病理生理过程涉及多种细胞，scRNA技术对肝细胞、以及肝脏特定细胞的覆盖率不高，对该疾病的进展过程研究也不多，相对来说局限于终末期这个时间点，我们用snRNA-seq，snATAC-seq技术能够有效的捕获人类和小鼠，NASH疾病过程中的，肝细胞核其它类型的细胞。最后我们鉴定出来一个和在人类肝脏中一致的小鼠干细胞亚群，在NASH阶段还表达EphB2。单独激活EphB2就足够引起细胞自身炎症，抑制就能减缓NASH模型小鼠中的干细胞纤维化。因此EphB2定义的干细胞对NASH过程有贡献。</p> <p><strong>R1：characterization of mouse models of NASH progression</strong></p> <p>Gubra amylin NASH (GAN) diet，American lifestyle–induced obesity syndrome (ALIOS) diet</p> <ul><li>3 months → hepatomegaly（肝肿大）；</li> <li>9 months → NASH；</li></ul> <p>肝肿大，体重升高，中度葡萄糖耐受不良，肝组织病理切片 ——→ 证明 NAFL &amp; NASH 小鼠模型成功</p> <p><strong>R2：Global transcriptomic feture shifts across all major mouse liver cell types during NASH progression</strong></p> <p>Sequencing samples / Condition groups / 四种样本：</p> <ul><li>3、9 months ALIOS = 3moALIOS、9moALIOS；</li> <li>3、9 months normal chow = 3moNC、9moNC；</li></ul> <p>Studying / focusing cells / 研究关注的细胞类型：</p> <ul><li>hepatocytes &amp; nonparenchymal cells（NPCs）</li></ul> <blockquote><p>NPCs 包括：macrophage，endothelial cells（ECs），stellate cells，cholangiocytes</p></blockquote> <p>comparable recovery：线粒体比例，基因数量</p> <div class="language- line-numbers-mode"><pre class="language-text"><code>## 描述统计比例的变化，相关基因的丰度变化
liver sinusoid endothelial markers Stab2
adhesion molecule PECAM1
contribute to fibrosis and extracellular matrix assembly such as Col1a1
</code></pre> <div class="line-numbers-wrapper"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br></div></div><p>描述细胞比例的变化，以及重要基因的丰度变化，与前人研究<code>一致</code>，表明小鼠模型的有效性，对人NASH疾病进展的代表性。</p> <p><strong>R3：The transcriptomic signature of hepatocytes switches from macronutrient processing to cell migration during NASH progression in mice</strong></p> <p>hepatocyte</p> <ul><li>pericentral zone hepatocyte cluster (mPC-Hep),</li> <li>intermediate zone hepatocyte cluster (mInt-Hep),</li> <li>periportal zone hepatocyte cluster (mPP-Hep)</li></ul> <blockquote><p>distinct expression patterns of liver zonation markers：Glul, Aldh3a2, and Gls2</p></blockquote> <p>more shared changes：不同状态的差异Venn图，可以展示这个信息（交集越小，差异越大，越个性，specific）；</p> <p>组间比较细胞比例：细胞的比例变化，也和疾病进展有很大的关系；</p> <p>hepatokine fibrinogen-like protein 1 (Fgl1),
phospholipid-binding protein annexin A2 protein (Anxa2), senescence marker Cdkn1a (P21)</p> <p>Elovl5 and Insig2，supporting an adaptive en</p> <p>hancement of both de novo lipogenesis and fatty acid oxidation to process excessive nutrients in NAFL</p> <p><strong>R4：Transcriptomic signatures of hepatocytes in human NASH resemble those of mouse</strong></p> <p>首先说明，数据质量较好、可比较：线粒体比例，基因数量（展示形式为：violin plot、UMAP plot）</p> <p>All major liver cell types were identified by snRNA-seq, including hepatocytes, ECs, stellate cells, cholangiocytes, vascular smooth muscle cells, and monocytes/macrophages</p> <p>肝细胞 ↓，NPCs ↑</p> <blockquote><p>specific zonation genes / zonation landmarks GLUL, SLCO1B3, and HSD17B13
lineage markers of hepatocytes, ACSL1, ALDH1A2, HNF4A</p></blockquote> <p>gluconeogenesis FOXO1 and glucose transporter SLC2A2 (GLUT2) -- 糖异生，葡萄糖转运体</p> <p>对人的单细胞数据，也用monocle2进行分析，得到与小鼠分化相似的轨迹，然后做差异基因的功能研究。</p> <p><strong>R5：EphB2 demarcates a subset of NASH hepatocytes engaged in intercellular cross-talk and trajectory bifurcation</strong></p> <p>与重要细胞间的，crosstalk，筛选出来了重要的细胞类型，在重要的细胞类型中，发现重要的受体/配体（receptor / ligand）</p> <p>Ephb2 RNA was expressed in a subset of hepatocytes labeled with Hnf4a</p> <p>another diet-induced NASH model, the cholinedeficient L-amino acid–defined high-fat diet (CDAHFD) administered for 2 months</p> <p>Ephb2-positive cells abundantly expressed pericentral marker Aldh3a2 but lacked periportal marker Gls2</p> <p>根据基因的作用，重新聚类 / cluster，然后以一个新的角度，分析差异基因（新cluster，新比较角度，新差异基因）</p> <p><strong>R6：Notch induces EphB2 expression in hepatocytes</strong></p> <p>肝细胞的表观遗传情况 / landscape → snATAC-seq</p> <p>transferred cell clustering labels</p> <blockquote><p>scATAC-seq如何进行降维聚类，以及注释的？得到的信息都是开放染色质信息呀？是不是开放的染色质信息转变成基因表达信息，再据此 基因/marker 定义不同的细胞类型？</p></blockquote> <p>不懂：表观调控关系，以及信号通路的关系，所以这段的后半部分看不懂。</p> <p><strong>R7：EphB2 is sufficient to induce cell-autonomous inflammation in mouse and human hepatocytes</strong></p> <p>判断 EphB2（+）hepatocytes的功能 → gain-of-function study</p> <p>Forced expression of EphB2 in hepatocytes did not cause apparent cell apoptosis → DAPI，TUNEL（无则表明无凋亡）</p> <p>之后许多流式细胞分析，免疫荧光（共定位判断），验证了EphB2的作用；</p> <p>还用腺病毒载体，让细胞高表达这个EphB2，然后检测炎症指标。</p> <p><strong>R8：Disease-activate EphB2 in hepatocytes contributes to NASH progression</strong></p> <p>next performed a loss-of-function study using an AAV-based CasRx-mediated RNA targeting system → to interrogate the role of EphB2</p> <p>a potent inducer：强效诱导剂</p> <p>Sirius Red：天狼星红</p> <p><strong>Discussion</strong></p> <p>总结全文的工作，讨论应用的意义，研究的不足，后续的展望。</p> <h2 id="使用的工具"><a href="#使用的工具" class="header-anchor">#</a> 使用的工具</h2> <p><strong>snRNA-seq</strong></p> <p>The nuclei isolation for snRNA-seq and snATAC-seq was adapted from the 10X Genomics nuclei demonstration protocol.</p> <p>snRNA-seq，raw fastq files，GRCH38（mm10，hg38），CellRanger（3.0.2）</p> <p>Seurat（3.2.0）<br>
nFeature_RNA &gt; 500 &amp; nFeature_RNA &lt; 8000 &amp; percent.mt &lt; 2 for mice；<br>
nFeature_RNA &gt; 500 &amp; nFeature_RNA &lt; 8000 &amp; percent.mt &lt; 10 for humans；</p> <p>merge and correct batch effect by harmony</p> <p>Unsupervised clustering，first 40 principal components，visualized in UMAP plot</p> <p>Cluster identities were annotated based on prior knowledge</p> <p>The doublets with mixed cell features were manually removed</p> <p>Pseudotime analysis of hepatocytes was performed with Monocle2 with default parameters</p> <ul><li>mouse: all hepatocytes;</li> <li>human: randomly selected 2000 hepatocytes from each hepatocyte cluster</li></ul> <p>The genes determined the bifurcation of hepatocyte trajectories were defined with q value &lt; 1e-20</p> <p>GO analysis -- Metascape (http://metascape.org)</p> <p>Cell-cell communication was analyzed with CellChat</p> <p>To analyze correlation between Ephb2 expression and the expression of inflammatory genes in hepatocytes, the expression value was transformed using <strong>MAGIC</strong> due to the sparsity of single-cell dataset</p> <p><strong>snATAC-seq</strong></p> <p>raw fastq files，mapped to GRCm38（mm10） using Cellranger-atac（1.2.0）</p> <p>fragment count matrix → Signac（1.8.0）</p> <blockquote><p>The nuclei with TSS enrichment score &lt; 2 &amp; pct_reads_in_peaks &lt; 40 &amp; nucleosome_signal &gt; 4 &amp; blacklist_ratio &gt; 0.025 were removed
The chromatin assays were integrated using FindIntegrationAnchors with dimension 2:30</p></blockquote> <p>ChromVAR，JASPAR 2018 core motifs (Homo sapiens)，STREAM pseudotime analysis，</p> <p><strong>Integration of snRNA-seq and snATAC-seq</strong></p> <p>Signac for integrating</p> <p>The <strong>peak to gene links</strong> were defined by the <strong>correlation between peak accessibility</strong> (snATAC-seq) and <strong>gene expression</strong> (transferred snRNA-seq information) in the matched cells.</p> <p>Signaling pathway enrichment analysis for candidate TFs regulating EphB2 was performed with R package clusterProfiler using all human genes as background and filtered by signaling transduction pathway annotation</p> <p><strong>Bulk-seq</strong></p> <p>raw fastq mapped to Mus musculus assembly GRCm38 (mm10) using STAR (2.5.4)</p> <p>R package DEseq2 to quantify gene expression</p> <blockquote><p>Significantly change genes → Genes with padj&lt;0.1 and Log2(FC) &gt; 0.5 or &lt; -0.5</p></blockquote> <p>Transcript-specific quantification of GSE119340 and GSE130970 were analyzed by Salmon</p> <h2 id="尝试处理"><a href="#尝试处理" class="header-anchor">#</a> 尝试处理</h2> <p>数据早就下载好了，主要是下游的表达矩阵数据，但是还有几个工具不太会用，包括monocle2，CellChat，Signac；先Seurat 跑出来，注释好一部分内容后，再稍稍学习一下这三个工具吧</p> <p>后续进行R包的学习记录位置
<a href="/01.科研学习/02.生信分析/41.Seurat.html">monocle2</a> <a href="/01.科研学习/02.生信分析/45.CellPhoneDB.html">CellPhoneDB</a> <a href="/01.科研学习/02.生信分析/04.单细胞-ATAC.html">Signac</a></p> <p>数据的名称和位置调整 -- 循环改名字和调整位置</p> <p>跑到cluster之后，但是注释给整不会了，不知道marker，难道要一个个找嘛？</p> <p>去看问文献的注释参考文献，5，6，7，10</p> <p>小鼠的
5：<a href="https://www.nature.com/articles/s41591-021-01344-3" target="_blank" rel="noopener noreferrer">XCR1+ type 1 conventional dendritic cells drive liver pathology in non-alcoholic steatohepatitis | Nature Medicine<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a><br>
6：<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7262680/" target="_blank" rel="noopener noreferrer">Landscape of intercellular crosstalk in healthy and NASH liver revealed by single-cell secretome gene analysis - PMC (nih.gov)<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a><br>
7：<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7305990/" target="_blank" rel="noopener noreferrer">Niche-Specific Re-Programming of Epigenetic Landscapes Drives Myeloid Cell Diversity in Nonalcoholic Steatohepatitis - PMC (nih.gov)<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a></p> <p>人的
10：<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6876711/" target="_blank" rel="noopener noreferrer">Resolving the fibrotic niche of human liver cirrhosis at single cell level - PMC (nih.gov)<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a></p> <p>找Marker找的好辛苦啊....... hepatocyt，cholangiocyte，EC cell，Stellate，T cell，B cell，Macrophage，</p> <p><strong>Hepatocyte / 肝细胞（PC，Int，PP，NASH-Hep1，NASH-Hep2）：</strong><br> <strong>Cholangiocyte / 胆管细胞：</strong><br> <strong>EC cell（EC，NASH-EC）：</strong><br> <strong>Stellate（Stellate，NASH-Stellate）：</strong><br> <strong>Macrophage（Mac1，Mac2）：</strong><br> <strong>T cell：</strong><br> <strong>B cell：</strong></p> <p>leucocyte -- CD45+；<br>
fluorescence activated cell sorting，荧光激活细胞分选法<br>
Kupffer cells -- KCs</p> <p>最后还是按照原文来吧：</p> <div class="language- line-numbers-mode"><pre class="language-text"><code>参考来源：Figure.S2
0：mNASH-EC
1：mInt-Hep
2：mNASH-Hep2
3：mInt-Hep
4：mPC-Hep
5：mNASH-Hep2
6：Stellate
7：mHASH-Mac2
8：mNASH-Hep1
9：mNASH-Stellate
10：Mac
11：mPP-Hep
12：mNASH-Cholangiocyte
13：mInt-Hep
14：EC
15：mNASH-Hep2
16：T cell
17：mNASH-Mac1
18：B cell
19：EC
20：Meso.cell
21：EC
</code></pre> <div class="line-numbers-wrapper"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br><span class="line-number">6</span><br><span class="line-number">7</span><br><span class="line-number">8</span><br><span class="line-number">9</span><br><span class="line-number">10</span><br><span class="line-number">11</span><br><span class="line-number">12</span><br><span class="line-number">13</span><br><span class="line-number">14</span><br><span class="line-number">15</span><br><span class="line-number">16</span><br><span class="line-number">17</span><br><span class="line-number">18</span><br><span class="line-number">19</span><br><span class="line-number">20</span><br><span class="line-number">21</span><br><span class="line-number">22</span><br><span class="line-number">23</span><br></div></div><p>看到几个蛮好的教程：</p> <p><a href="https://www.zhihu.com/people/just-1-89/posts" target="_blank" rel="noopener noreferrer">立源 - 知乎 (zhihu.com)<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a></p> <p>原文：<a href="https://github.com/hbctraining/scRNA-seq/blob/master/lessons/01_intro_to_scRNA-seq.md" target="_blank" rel="noopener noreferrer">scRNA-seq/lessons/01_intro_to_scRNA-seq.md (github.com)<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a></p> <p><a href="https://zhuanlan.zhihu.com/p/343032046" target="_blank" rel="noopener noreferrer">第一章 RNA-seq入门 - 知乎 (zhihu.com)<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a></p> <p>原文：<a href="https://github.com/hbctraining/Intro-to-rnaseq-hpc-salmon-flipped/blob/main/lessons/01_intro-to-RNAseq.md" target="_blank" rel="noopener noreferrer">Intro-to-rnaseq/01_intro-to-RNAseq.md at main · hbctraining(github.com)<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a></p> <p>2024-03-07，19:02，遇到一个很大的难题，就是Seurat分群之后，分的群合不合适？我不知道如何判断，这个cluster的数量是可以很方便的根据前面的参数调整的，但是如何调整到符合生物学意义，就不好判断了。</p> <p>方便判断细胞类型：
<a href="https://zhuanlan.zhihu.com/p/384840499" target="_blank" rel="noopener noreferrer">多个单细胞亚群合并 - 知乎 (zhihu.com)<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a></p> <p>其中一些值得借鉴的代码
<a href="https://mp.weixin.qq.com/s/e9Js1gLIVYgzPkOaamsgIg" target="_blank" rel="noopener noreferrer">单细胞 | 多样本整合-harmony去批次<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a></p> <div class="language-R line-numbers-mode"><pre class="language-r"><code>scobj <span class="token operator">&lt;-</span> merge<span class="token punctuation">(</span>x<span class="token operator">=</span>data<span class="token punctuation">[</span><span class="token punctuation">[</span><span class="token number">1</span><span class="token punctuation">]</span><span class="token punctuation">]</span><span class="token punctuation">,</span> y <span class="token operator">=</span> data<span class="token punctuation">[</span><span class="token operator">-</span><span class="token number">1</span><span class="token punctuation">]</span><span class="token punctuation">,</span> project <span class="token operator">=</span> <span class="token string">&quot;pbmc&quot;</span><span class="token punctuation">)</span>

scobj <span class="token operator">&lt;-</span> RunHarmony<span class="token punctuation">(</span>scobj<span class="token punctuation">,</span> 
	reduction <span class="token operator">=</span> <span class="token string">&quot;pca&quot;</span><span class="token punctuation">,</span>                       <span class="token comment"># 默认reduction = &quot;pca&quot;,</span>
	group.by.vars <span class="token operator">=</span> <span class="token string">&quot;group&quot;</span><span class="token punctuation">,</span>            <span class="token comment"># group.by.vars 参数输入的是批次信息,</span>
	reduction.save <span class="token operator">=</span> <span class="token string">&quot;harmony&quot;</span>       <span class="token comment"># reduction.save 是结果保存的名称  </span>
	<span class="token punctuation">)</span>  
scobj <span class="token operator">&lt;-</span> RunUMAP<span class="token punctuation">(</span>scobj<span class="token punctuation">,</span> reduction <span class="token operator">=</span> <span class="token string">&quot;harmony&quot;</span><span class="token punctuation">,</span> dims <span class="token operator">=</span> <span class="token number">1</span><span class="token operator">:</span><span class="token number">30</span><span class="token punctuation">,</span>reduction.name <span class="token operator">=</span> <span class="token string">&quot;umap&quot;</span><span class="token punctuation">)</span>

scobj <span class="token operator">&lt;-</span> FindNeighbors<span class="token punctuation">(</span>scobj<span class="token punctuation">,</span> reduction <span class="token operator">=</span> <span class="token string">&quot;harmony&quot;</span><span class="token punctuation">,</span> dims <span class="token operator">=</span> <span class="token number">1</span><span class="token operator">:</span><span class="token number">30</span><span class="token punctuation">)</span>
scobj <span class="token operator">&lt;-</span> FindClusters<span class="token punctuation">(</span>scobj<span class="token punctuation">,</span> resolution <span class="token operator">=</span> seq<span class="token punctuation">(</span><span class="token number">0.2</span><span class="token punctuation">,</span><span class="token number">1.2</span><span class="token punctuation">,</span><span class="token number">0.1</span><span class="token punctuation">)</span><span class="token punctuation">)</span>  <span class="token comment"># 设置多个resolution选择合适的resolution</span>
library<span class="token punctuation">(</span>clustree<span class="token punctuation">)</span>   <span class="token comment"># （PCA那一步，**ElbowPlot()** ）</span>
<span class="token operator">*</span><span class="token operator">*</span>clustree<span class="token punctuation">(</span>scobj<span class="token punctuation">)</span><span class="token operator">*</span><span class="token operator">*</span>   <span class="token comment"># 判断 cluster 这一步之后，需要多少分辨率！！</span>
scobj<span class="token operator">@</span>meta.data<span class="token operator">$</span>seurat_clusters <span class="token operator">&lt;-</span> scobj<span class="token operator">@</span>meta.data<span class="token operator">$</span>RNA_snn_res.<span class="token number">0.5</span>   <span class="token comment"># 更改seurat_clusters 选定分辨率0.5作为分群 </span>
Idents<span class="token punctuation">(</span>scobj<span class="token punctuation">)</span> <span class="token operator">&lt;-</span> <span class="token string">&quot;seurat_clusters&quot;</span>   <span class="token comment"># 更改active.ident</span>

DimPlot<span class="token punctuation">(</span>scobj<span class="token punctuation">,</span> reduction <span class="token operator">=</span> <span class="token string">&quot;umap&quot;</span><span class="token punctuation">,</span> label <span class="token operator">=</span> T<span class="token punctuation">)</span>  <span class="token comment"># 作图展示  </span>
DimPlot<span class="token punctuation">(</span>scobj<span class="token punctuation">,</span> reduction <span class="token operator">=</span> <span class="token string">&quot;umap&quot;</span><span class="token punctuation">,</span> group.by <span class="token operator">=</span> <span class="token string">&quot;group&quot;</span><span class="token punctuation">)</span>   <span class="token comment"># group.by, 在一张图中展示多个信息 </span>
DimPlot<span class="token punctuation">(</span>scobj<span class="token punctuation">,</span> reduction <span class="token operator">=</span> <span class="token string">&quot;umap&quot;</span><span class="token punctuation">,</span> split.by <span class="token operator">=</span> <span class="token string">&quot;group&quot;</span><span class="token punctuation">)</span>      <span class="token comment"># split.by, 分成多个信息来展示  </span>
</code></pre> <div class="line-numbers-wrapper"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br><span class="line-number">6</span><br><span class="line-number">7</span><br><span class="line-number">8</span><br><span class="line-number">9</span><br><span class="line-number">10</span><br><span class="line-number">11</span><br><span class="line-number">12</span><br><span class="line-number">13</span><br><span class="line-number">14</span><br><span class="line-number">15</span><br><span class="line-number">16</span><br><span class="line-number">17</span><br><span class="line-number">18</span><br><span class="line-number">19</span><br></div></div><p>2024-03-10，22:42，R中处理是真慢....于是，转换到了python中处理，seurat对象可以直接变成anndata对象</p> <div class="language- line-numbers-mode"><pre class="language-text"><code>library(sceasy)
sceasy::convertFormat(seuratobject,
  from=&quot;seurat&quot;,
  to=&quot;anndata&quot;,
  outFile='mergeHarmony_annotation.h5ad')
</code></pre> <div class="line-numbers-wrapper"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br></div></div><p>python中处理的风格与R有很大的不同，特别是他的leiden算法，感觉结果有一小部分的分群是和seurat是不一样的，导致注释的时候，标记基因的marker表达量不太一致</p> <p>2024-03-14，人类的marker小提琴图，对其注释</p> <div class="language- line-numbers-mode"><pre class="language-text"><code>hNASH-Hep
hPP-Hep
hInt-Hep
hPC-Hep
hNASH-Stellate1
Mono/Macs
EC1
hNASH-Cholangiocyte1
hPC-Hep
hPC-Hep
hNASH-Cholangiocyte1
hNASH-VSMC
hNASH-Stellate2
hPC-Hep
hNASH-Cholangiocyte2
Stellate
hNASH-EC
EC2
</code></pre> <div class="line-numbers-wrapper"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br><span class="line-number">6</span><br><span class="line-number">7</span><br><span class="line-number">8</span><br><span class="line-number">9</span><br><span class="line-number">10</span><br><span class="line-number">11</span><br><span class="line-number">12</span><br><span class="line-number">13</span><br><span class="line-number">14</span><br><span class="line-number">15</span><br><span class="line-number">16</span><br><span class="line-number">17</span><br><span class="line-number">18</span><br></div></div><p>小提琴图，UMAP聚类图，一起判断</p> <p>hPC-Hep，hInt-Hep，hPP-Hep，hNASH-Hep
hNASH-EC，EC1，EC2
Stellate，hNASH-Stellate1，hNASH-Stellate2
Mono/Macs，hNASH-VSMC
hNASH-Cholangiocyte1，hNASH-Cholangiocyte2</p> <p>得到注释的结果之后，就比较简单了，是各种样子的展示数据方式了。</p> <h2 id="小结"><a href="#小结" class="header-anchor">#</a> 小结</h2> <p>2024-03-17，小结</p> <p>从最开始，2024-03-04，开始重读文献，下载数据，接着运行，试着复现文献，到现在大概有十四天，也就是两周的时间，总体上不是很满意现在的结果或状态。</p> <p>第一周，主要是比较熟悉的seurat进行处理，降维聚类，之后注释，算是把基本细胞分类确定了，但是那个差异基因的热图却没有理解，不知道咋做出来的；</p> <p>第二周，实际上没做啥图出来，主要就是学习了cellchat2，monocle2，这俩包而已，得到了流程中的一些图而已，而作者用的那个signac包，因为作者没给全所需的数据，所以也没有进一步学下去，进一步准备选择ArchR这个包，但是还没到这一步，所以暂时还没学到。</p> <p>为啥感觉速度那么慢呢？？两周时间，还是有不少时间是在上课的，课上偶尔在试着跑代码，课下的时间，不多，不少时间用在了理解新的内容，等待运行结果上面，之后应该能注意到不少地方，会省下检索的时间。同时，晚自习的一些比较完整的时间去跑步了，这短时间不考虑转到分析中。课上也有不少内容，需要及时记录诶，省的最后复习太狼狈。</p> <p>同时，为啥会感到比较混乱呢？嗯，有几个工具包，是没有那么熟悉的，所以重新学习和了解花了不少时间，没有完整的时间，容易来回混乱的进行学习，导致不连贯，没有一个比较完整的，连贯的思路。还有不少教程，需要去辨别，来回参考和调整，也是较为混乱的。seurat的参数，monocle2，cellchat2的参数，特别是后两个的，不是很熟悉，所以需要多次调整。</p> <br> <p><strong>得到的几点收获</strong></p> <p>在计划复现过程的时候，详细的分析过程没有安排好，应该以数据的变化来安排操作记录，seurat → cellchat → monocle → signac/archr，从counts矩阵到展示基因相互关系，到基因的变化细节分析，再加上表观数据的展示，嗯。</p> <p>然后，那个<strong>数据的内容 &amp; 格式变化</strong>，特别是这几个包处理的过程中，不同的函数对数据处理后的变化，需要仔细了解，这样才能在变化中，及时根据目的及时调整。</p> <p>有一个文件处理的方式很不错，值得后续借鉴</p> <div class="language- line-numbers-mode"><pre class="language-text"><code># 建文件夹 - 改名字 - 移动文件
allfiles &lt;- list.files('./','.gz')

for (file in allfiles){
  dir_name &lt;- sub(&quot;^(.*?)_.*&quot;, &quot;\\1&quot;, file)
  new_name &lt;- sub(&quot;^.*_(.*)&quot;, &quot;\\1&quot;, file) 
  
  dir.create(dir_name)
  file.rename(file, file.path(dir_name,new_name))
} 

</code></pre> <div class="line-numbers-wrapper"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br><span class="line-number">6</span><br><span class="line-number">7</span><br><span class="line-number">8</span><br><span class="line-number">9</span><br><span class="line-number">10</span><br><span class="line-number">11</span><br></div></div><p>有那么几个小<strong>问题</strong>，最后结束的时候，还是不能有把握的理解：</p> <ol><li>14种细胞，一般要分成几个cluster？</li> <li>注释的方法有哪些？（细化到代码，表达图的类型）</li></ol></div></div>  <div class="page-edit"><!----> <!----> <!----></div> <div class="page-nav-wapper"><!----> <!----></div></div> <div class="article-list"><div class="article-title"><a href="/archives/" class="iconfont icon-bi">最近更新</a></div> <div class="article-wrapper"><dl><dd>01</dd> <dt><a href="/pages/90dbe8/"><div>
            APA相关综述
            <!----></div></a> <span class="date">06-27</span></dt></dl><dl><dd>02</dd> <dt><a href="/pages/f4f4b7/"><div>
            胃癌免疫逃逸的决定因素
            <!----></div></a> <span class="date">06-25</span></dt></dl><dl><dd>03</dd> <dt><a href="/pages/0a11dd/"><div>
            胖子CAF
            <!----></div></a> <span class="date">06-21</span></dt></dl> <dl><dd></dd> <dt><a href="/archives/" class="more">更多文章&gt;</a></dt></dl></div></div></main></div> <div class="footer"><div class="icons"><a href="mailto:2590643635@qq.com" title="发邮件" target="_blank" class="iconfont icon-youjian"></a><a href="https://github.com/hongliangfang" title="GitHub" target="_blank" class="iconfont icon-github"></a><a href="https://music.163.com/#/playlist?id=755597173" title="听音乐" target="_blank" class="iconfont icon-erji"></a></div> 
  Theme by
  <a href="https://github.com/xugaoyi/vuepress-theme-vdoing" target="_blank" title="本站主题">Vdoing</a> 
    | Copyright © 2023-2024
    <span>FHL | <a href="https://github.com/xugaoyi/vuepress-theme-vdoing/blob/master/LICENSE" target="_blank">MIT License</a></span></div> <div class="buttons"><div title="返回顶部" class="button blur go-to-top iconfont icon-fanhuidingbu" style="display:none;"></div> <div title="去评论" class="button blur go-to-comment iconfont icon-pinglun" style="display:none;"></div> <div title="主题模式" class="button blur theme-mode-but iconfont icon-zhuti"><ul class="select-box" style="display:none;"><li class="iconfont icon-zidong">
          跟随系统
        </li><li class="iconfont icon-rijianmoshi">
          浅色模式
        </li><li class="iconfont icon-yejianmoshi">
          深色模式
        </li><li class="iconfont icon-yuedu">
          阅读模式
        </li></ul></div></div> <!----> <!----> <!----></div><div class="global-ui"><div></div></div></div>
    <script src="/assets/js/app.bd488e07.js" defer></script><script src="/assets/js/2.bdf5b8cf.js" defer></script><script src="/assets/js/136.9de1870b.js" defer></script><script src="/assets/js/3.365ac406.js" defer></script>
  </body>
</html>
