<!DOCTYPE html>
<html lang="zh-CN">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width,initial-scale=1">
    <title>一次生信实战 | 一盏烛光的微笑灿烂人生</title>
    <meta name="generator" content="VuePress 1.9.9">
    <link rel="icon" href="/img/favicon.ico">
    <link rel="stylesheet" href="https://at.alicdn.com/t/font_3129839_xft6cqs5gc.css">
    <noscript><meta http-equiv="refresh" content="0; url=https://www.youngkbt.cn/noscript/"><style>.theme-vdoing-content { display:none }</noscript>
    <meta name="description" content="学习记录博客">
    <meta name="theme-color" content="#11a8cd">
    <meta name="个人博客" content="科研学习，围炉夜话，奇淫巧计">
    
    <link rel="preload" href="/assets/css/0.styles.c9ffc21e.css" as="style"><link rel="preload" href="/assets/js/app.bd488e07.js" as="script"><link rel="preload" href="/assets/js/2.bdf5b8cf.js" as="script"><link rel="preload" href="/assets/js/142.0415d712.js" as="script"><link rel="preload" href="/assets/js/3.365ac406.js" as="script"><link rel="prefetch" href="/assets/js/10.b00b53b4.js"><link rel="prefetch" href="/assets/js/100.1b2ba6dd.js"><link rel="prefetch" href="/assets/js/101.e392eb92.js"><link rel="prefetch" href="/assets/js/102.eb0bb162.js"><link rel="prefetch" href="/assets/js/103.9db14406.js"><link rel="prefetch" href="/assets/js/104.6ae6fd5c.js"><link rel="prefetch" href="/assets/js/105.4d79399b.js"><link rel="prefetch" href="/assets/js/106.4e6ff12d.js"><link rel="prefetch" href="/assets/js/107.bc93ee73.js"><link rel="prefetch" href="/assets/js/108.59a000f0.js"><link rel="prefetch" href="/assets/js/109.f01f4422.js"><link rel="prefetch" href="/assets/js/11.346d31c6.js"><link rel="prefetch" href="/assets/js/110.96a96c44.js"><link rel="prefetch" href="/assets/js/111.87fe63df.js"><link rel="prefetch" href="/assets/js/112.c560a316.js"><link rel="prefetch" href="/assets/js/113.6a371b02.js"><link rel="prefetch" href="/assets/js/114.989a4d3e.js"><link rel="prefetch" href="/assets/js/115.b6baafa2.js"><link rel="prefetch" href="/assets/js/116.d469dc20.js"><link rel="prefetch" href="/assets/js/117.0a6bccf0.js"><link rel="prefetch" href="/assets/js/118.9f1ef088.js"><link rel="prefetch" href="/assets/js/119.059ef806.js"><link rel="prefetch" href="/assets/js/12.8d91a638.js"><link rel="prefetch" href="/assets/js/120.75ed11ab.js"><link rel="prefetch" href="/assets/js/121.1896ee22.js"><link rel="prefetch" href="/assets/js/122.a3664290.js"><link rel="prefetch" href="/assets/js/123.5b849c39.js"><link rel="prefetch" href="/assets/js/124.85737d6b.js"><link rel="prefetch" href="/assets/js/125.a95cd42f.js"><link rel="prefetch" href="/assets/js/126.7b1c1645.js"><link rel="prefetch" href="/assets/js/127.64a16fd4.js"><link rel="prefetch" href="/assets/js/128.a07721ba.js"><link rel="prefetch" href="/assets/js/129.9ba86871.js"><link rel="prefetch" href="/assets/js/13.18ed2aa7.js"><link rel="prefetch" href="/assets/js/130.5cdb8138.js"><link rel="prefetch" href="/assets/js/131.d2f31425.js"><link rel="prefetch" href="/assets/js/132.b2286d65.js"><link rel="prefetch" href="/assets/js/133.1229c9ae.js"><link rel="prefetch" href="/assets/js/134.71fad9a0.js"><link rel="prefetch" href="/assets/js/135.ff682301.js"><link rel="prefetch" href="/assets/js/136.9de1870b.js"><link rel="prefetch" href="/assets/js/137.a58bbdcd.js"><link rel="prefetch" href="/assets/js/138.cc768cb1.js"><link rel="prefetch" href="/assets/js/139.9d52d34a.js"><link rel="prefetch" href="/assets/js/14.4bcf7ee9.js"><link rel="prefetch" href="/assets/js/140.b1790e58.js"><link rel="prefetch" href="/assets/js/141.98bceb9f.js"><link rel="prefetch" href="/assets/js/143.851c7187.js"><link rel="prefetch" href="/assets/js/144.ae79e007.js"><link rel="prefetch" href="/assets/js/145.d5020e58.js"><link rel="prefetch" href="/assets/js/146.c3eafa98.js"><link rel="prefetch" href="/assets/js/147.0a6864ec.js"><link rel="prefetch" href="/assets/js/148.bbc4bd28.js"><link rel="prefetch" href="/assets/js/149.489e8cdb.js"><link rel="prefetch" href="/assets/js/15.368d9d2d.js"><link rel="prefetch" href="/assets/js/150.7778d444.js"><link rel="prefetch" href="/assets/js/16.06f68354.js"><link rel="prefetch" href="/assets/js/17.4f8d6256.js"><link rel="prefetch" href="/assets/js/18.8989b2ff.js"><link rel="prefetch" href="/assets/js/19.abb857ca.js"><link rel="prefetch" href="/assets/js/20.e32420cf.js"><link rel="prefetch" href="/assets/js/21.06441c4d.js"><link rel="prefetch" href="/assets/js/22.7fb78c64.js"><link rel="prefetch" href="/assets/js/23.d9193b83.js"><link rel="prefetch" href="/assets/js/24.6487f43a.js"><link rel="prefetch" href="/assets/js/25.11e66938.js"><link rel="prefetch" href="/assets/js/26.c3e43726.js"><link rel="prefetch" href="/assets/js/27.24117748.js"><link rel="prefetch" href="/assets/js/28.fabf32c7.js"><link rel="prefetch" href="/assets/js/29.3a9331f2.js"><link rel="prefetch" href="/assets/js/30.ac5ca451.js"><link rel="prefetch" href="/assets/js/31.36e6c9c0.js"><link rel="prefetch" href="/assets/js/32.2fcf84e2.js"><link rel="prefetch" href="/assets/js/33.a73adf6b.js"><link rel="prefetch" href="/assets/js/34.49c0fb46.js"><link rel="prefetch" href="/assets/js/35.babda623.js"><link rel="prefetch" href="/assets/js/36.56221bac.js"><link rel="prefetch" href="/assets/js/37.5d6d33aa.js"><link rel="prefetch" href="/assets/js/38.a6b592e0.js"><link rel="prefetch" href="/assets/js/39.14a81539.js"><link rel="prefetch" href="/assets/js/4.a64d5c9f.js"><link rel="prefetch" href="/assets/js/40.e94616b3.js"><link rel="prefetch" href="/assets/js/41.4ad9febc.js"><link rel="prefetch" href="/assets/js/42.4825ff11.js"><link rel="prefetch" href="/assets/js/43.6d554f8a.js"><link rel="prefetch" href="/assets/js/44.ed8f011b.js"><link rel="prefetch" href="/assets/js/45.170e103d.js"><link rel="prefetch" href="/assets/js/46.9cc703a7.js"><link rel="prefetch" href="/assets/js/47.ff6af7ea.js"><link rel="prefetch" href="/assets/js/48.1b2f1a1b.js"><link rel="prefetch" href="/assets/js/49.05223958.js"><link rel="prefetch" href="/assets/js/5.b1fd9f62.js"><link rel="prefetch" href="/assets/js/50.26aa9f07.js"><link rel="prefetch" href="/assets/js/51.65eba0cc.js"><link rel="prefetch" href="/assets/js/52.5b21c37d.js"><link rel="prefetch" href="/assets/js/53.4f209254.js"><link rel="prefetch" href="/assets/js/54.176923ed.js"><link rel="prefetch" href="/assets/js/55.74f6741d.js"><link rel="prefetch" href="/assets/js/56.383e3c0d.js"><link rel="prefetch" href="/assets/js/57.35b21350.js"><link rel="prefetch" href="/assets/js/58.7ae8fd30.js"><link rel="prefetch" href="/assets/js/59.3d7d7dea.js"><link rel="prefetch" href="/assets/js/6.993d7ad0.js"><link rel="prefetch" href="/assets/js/60.67ba2844.js"><link rel="prefetch" href="/assets/js/61.6c5542cd.js"><link rel="prefetch" href="/assets/js/62.a235dad7.js"><link rel="prefetch" href="/assets/js/63.a3a52fe2.js"><link rel="prefetch" href="/assets/js/64.2b6b610e.js"><link rel="prefetch" href="/assets/js/65.8672534a.js"><link rel="prefetch" href="/assets/js/66.c7281353.js"><link rel="prefetch" href="/assets/js/67.e062f6ac.js"><link rel="prefetch" href="/assets/js/68.446f0688.js"><link rel="prefetch" href="/assets/js/69.f03843ed.js"><link rel="prefetch" href="/assets/js/7.afb33a4c.js"><link rel="prefetch" href="/assets/js/70.466d842a.js"><link rel="prefetch" href="/assets/js/71.87fcb26b.js"><link rel="prefetch" href="/assets/js/72.ccf1092d.js"><link rel="prefetch" href="/assets/js/73.3ad6124b.js"><link rel="prefetch" href="/assets/js/74.17a59372.js"><link rel="prefetch" href="/assets/js/75.6c859470.js"><link rel="prefetch" href="/assets/js/76.674a56ad.js"><link rel="prefetch" href="/assets/js/77.5342f3c1.js"><link rel="prefetch" href="/assets/js/78.154ba5e1.js"><link rel="prefetch" href="/assets/js/79.10901b1c.js"><link rel="prefetch" href="/assets/js/8.5bbd7ae9.js"><link rel="prefetch" href="/assets/js/80.0166104e.js"><link rel="prefetch" href="/assets/js/81.4c5a0e6a.js"><link rel="prefetch" href="/assets/js/82.47099701.js"><link rel="prefetch" href="/assets/js/83.7873cd95.js"><link rel="prefetch" href="/assets/js/84.23000f66.js"><link rel="prefetch" href="/assets/js/85.6701c9e4.js"><link rel="prefetch" href="/assets/js/86.188012c3.js"><link rel="prefetch" href="/assets/js/87.e9cde64e.js"><link rel="prefetch" href="/assets/js/88.5ccdea49.js"><link rel="prefetch" href="/assets/js/89.9ab559de.js"><link rel="prefetch" href="/assets/js/9.6699859e.js"><link rel="prefetch" href="/assets/js/90.bc7fb61d.js"><link rel="prefetch" href="/assets/js/91.07537d0a.js"><link rel="prefetch" href="/assets/js/92.b4ed0bad.js"><link rel="prefetch" href="/assets/js/93.b167c2d0.js"><link rel="prefetch" href="/assets/js/94.1a02653c.js"><link rel="prefetch" href="/assets/js/95.69ef19be.js"><link rel="prefetch" href="/assets/js/96.e0b3f5df.js"><link rel="prefetch" href="/assets/js/97.9ea0c268.js"><link rel="prefetch" href="/assets/js/98.074852d9.js"><link rel="prefetch" href="/assets/js/99.708b118b.js">
    <link rel="stylesheet" href="/assets/css/0.styles.c9ffc21e.css">
  </head>
  <body class="theme-mode-light">
    <div id="app" data-server-rendered="true"><div class="theme-container sidebar-open have-rightmenu"><header class="navbar blur"><div title="目录" class="sidebar-button"><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" role="img" viewBox="0 0 448 512" class="icon"><path fill="currentColor" d="M436 124H12c-6.627 0-12-5.373-12-12V80c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12z"></path></svg></div> <a href="/" class="home-link router-link-active"><img src="/img/logo.png" alt="一盏烛光的微笑灿烂人生" class="logo"> <span class="site-name can-hide">一盏烛光的微笑灿烂人生</span></a> <div class="links"><div class="search-box"><input aria-label="Search" autocomplete="off" spellcheck="false" value=""> <!----></div> <nav class="nav-links can-hide"><div class="nav-item"><a href="/" class="nav-link">首页</a></div><div class="nav-item"><div class="dropdown-wrapper"><button type="button" aria-label="科研学习" class="dropdown-title"><a href="/study/" class="link-title">科研学习</a> <span class="title" style="display:none;">科研学习</span> <span class="arrow right"></span></button> <ul class="nav-dropdown" style="display:none;"><li class="dropdown-item"><!----> <a href="/articles/" class="nav-link">阅读记录</a></li><li class="dropdown-item"><!----> <a href="/biotech/" class="nav-link">生信分析</a></li><li class="dropdown-item"><!----> <a href="/basic/" class="nav-link">整理归纳</a></li></ul></div></div><div class="nav-item"><a href="/wlyh/" class="nav-link">围炉夜话</a></div><div class="nav-item"><a href="/technology/" class="nav-link">奇淫巧计</a></div><div class="nav-item"><div class="dropdown-wrapper"><button type="button" aria-label="索引" class="dropdown-title"><a href="/archives/" class="link-title">索引</a> <span class="title" style="display:none;">索引</span> <span class="arrow right"></span></button> <ul class="nav-dropdown" style="display:none;"><li class="dropdown-item"><!----> <a href="/categories/" class="nav-link">分类</a></li><li class="dropdown-item"><!----> <a href="/tags/" class="nav-link">标签</a></li><li class="dropdown-item"><!----> <a href="/archives/" class="nav-link">归档</a></li></ul></div></div><div class="nav-item"><div class="dropdown-wrapper"><button type="button" aria-label="更多" class="dropdown-title"><a href="/more/" class="link-title">更多</a> <span class="title" style="display:none;">更多</span> <span class="arrow right"></span></button> <ul class="nav-dropdown" style="display:none;"><li class="dropdown-item"><!----> <a href="/about/" class="nav-link">关于</a></li><li class="dropdown-item"><!----> <a href="/links/" class="nav-link">一些网站</a></li><li class="dropdown-item"><!----> <a href="/friends/" class="nav-link">友情链接</a></li><li class="dropdown-item"><!----> <a href="/passbyworld/" class="nav-link">路过人间</a></li></ul></div></div> <!----></nav></div></header> <div class="sidebar-mask"></div> <div class="sidebar-hover-trigger"></div> <aside class="sidebar" style="display:none;"><div class="blogger"><img src="https://pica.zhimg.com/v2-f2b30ad5674f652294f0ec6f9b8c9922_720w.webp?source=d16d100b"> <div class="blogger-info"><h3>围炉夜话</h3> <span>知行合一</span></div></div> <nav class="nav-links"><div class="nav-item"><a href="/" class="nav-link">首页</a></div><div class="nav-item"><div class="dropdown-wrapper"><button type="button" aria-label="科研学习" class="dropdown-title"><a href="/study/" class="link-title">科研学习</a> <span class="title" style="display:none;">科研学习</span> <span class="arrow right"></span></button> <ul class="nav-dropdown" style="display:none;"><li class="dropdown-item"><!----> <a href="/articles/" class="nav-link">阅读记录</a></li><li class="dropdown-item"><!----> <a href="/biotech/" class="nav-link">生信分析</a></li><li class="dropdown-item"><!----> <a href="/basic/" class="nav-link">整理归纳</a></li></ul></div></div><div class="nav-item"><a href="/wlyh/" class="nav-link">围炉夜话</a></div><div class="nav-item"><a href="/technology/" class="nav-link">奇淫巧计</a></div><div class="nav-item"><div class="dropdown-wrapper"><button type="button" aria-label="索引" class="dropdown-title"><a href="/archives/" class="link-title">索引</a> <span class="title" style="display:none;">索引</span> <span class="arrow right"></span></button> <ul class="nav-dropdown" style="display:none;"><li class="dropdown-item"><!----> <a href="/categories/" class="nav-link">分类</a></li><li class="dropdown-item"><!----> <a href="/tags/" class="nav-link">标签</a></li><li class="dropdown-item"><!----> <a href="/archives/" class="nav-link">归档</a></li></ul></div></div><div class="nav-item"><div class="dropdown-wrapper"><button type="button" aria-label="更多" class="dropdown-title"><a href="/more/" class="link-title">更多</a> <span class="title" style="display:none;">更多</span> <span class="arrow right"></span></button> <ul class="nav-dropdown" style="display:none;"><li class="dropdown-item"><!----> <a href="/about/" class="nav-link">关于</a></li><li class="dropdown-item"><!----> <a href="/links/" class="nav-link">一些网站</a></li><li class="dropdown-item"><!----> <a href="/friends/" class="nav-link">友情链接</a></li><li class="dropdown-item"><!----> <a href="/passbyworld/" class="nav-link">路过人间</a></li></ul></div></div> <!----></nav>  <ul class="sidebar-links"><li><section class="sidebar-group depth-0"><p class="sidebar-heading open"><span>一次生信实战</span> <!----></p> <!----></section></li></ul> </aside> <div><main class="page"><div class="theme-vdoing-wrapper "><div class="articleInfo-wrap" data-v-06225672><div class="articleInfo" data-v-06225672><!----> <div class="info" data-v-06225672><div title="作者" class="author iconfont icon-touxiang" data-v-06225672><a href="javascript:;" data-v-06225672>FHL</a></div> <div title="创建时间" class="date iconfont icon-riqi" data-v-06225672><a href="javascript:;" data-v-06225672>2023-12-01</a></div> <div title="分类" class="date iconfont icon-wenjian" data-v-06225672><a href="/categories/?category=%E7%A7%91%E7%A0%94%E5%AD%A6%E4%B9%A0" data-v-06225672>科研学习 </a><a href="/categories/?category=%E7%94%9F%E4%BF%A1%E5%88%86%E6%9E%90" data-v-06225672>生信分析 </a></div></div></div></div> <!----> <div class="content-wrapper"><div class="right-menu-wrapper"><div class="right-menu-margin"><div class="right-menu-title">目录</div> <div class="right-menu-content"></div></div></div> <h1><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAB4AAAAeCAYAAAA7MK6iAAAAAXNSR0IArs4c6QAABGpJREFUSA3tVVtoXFUU3fvOI53UlmCaKIFmwEhsE7QK0ipFEdHEKpXaZGrp15SINsXUWvBDpBgQRKi0+KKoFeJHfZA+ED9KKoIU2gYD9UejTW4rVIzm0VSTziPzuNu1z507dibTTjL4U/DAzLn3nL3X2o91ziX6f9wMFdh6Jvbm9nNSV0msViVO6tN1Rm7NMu2OpeJ9lWBUTDxrJbYTS0hInuwciu9eLHlFxCLCZEk3MegsJmZ5K/JD6t7FkFdEvGUo1g7qJoG3MHImqRIn8/nzY1K9UPKKiJmtnUqHVE3Gbuay6vJE/N2FEmuxFjW2nUuE0yQXRRxLiTUAzs36zhZvOXJPdX850EVnnLZkB8prodQoM5JGj7Xk2mvC7JB8tG04Ef5PiXtG0UtxupRQSfTnBoCy554x18yJHI6I+G5Eru4LHmPJZEQsrvPUbMiA8G/WgMK7w7I+ez7++o2ANfbrjvaOl1tFMs+htG3IrZH9/hDX1Pr8Tc0UvH8tcX29KzAgIGcEkINyW5BF9x891hw6VYqgJHEk0huccS7vh3C6gTiODL+26huuBtbct8eZnqLML8PkxGYpuPZBqtqwkSjgc4mB5gbgig5i+y0UDK35LMxXisn9xQtK+nd26gTIHsHe/oblK/b29fUmN/8Y+9jAQrnBp56m1LcDlDp9irKTExSKduXJVWSqdBMA08pEJnEIOB3FPPMybu/oeV8zFeYN3xx576Q6RH+VmplE4ncQV5v+5rzSoyOU7PuEAg8g803PwBJ0CExno/jcMbN8tONYeOmHiuUNryvm3fRUy4tMPVLdAGkUhNWuggGrJcXPv+ouCjz0MKUHz1J2/E8IC9nqTabcxgaBYM0hPhD5Y65FsbxRQKxCQrDjDctW7PUM3HuZunFyifSAqEfuzCp48Il24luWUWZoyJCaPR82jE0+kFA643wRFVni4RYSq3ohJO2pZ7B5dO4xkDWbEpossJPLSrPjYID8rS2UHTlvyNxqIGsg674XJJ7vnh5L7PNwC4hh2sjCI96mzszOTpxLF0T7l88Yz7lAuK6OnL8gXLOnTvpzSb22YG8W7us3jSebFHeeqnXRG1vt+MoUM84LQIBmMsCTAcOauTh0T0l0neQK7m2bLMt2mGxU3HYssS0J2cdv5wljlPsrIuZLAG/2DOZIXgCYT8uMGZN+e2kSirfxZOPCsC0f24nTZzspnVn9VePS1Z5vubmAGGXG8ZFno9Hel0yfA5ZPhF7Dh972BQJ2qCpgH67lmWtBYbvk6sz02wjky2vXyz0XErP/kFB619js1BtwfOV4OPRqOQBjy3Qbk18vigUPPSD5ceHnwck7W9bhAqZdd7SuG7w4/P2F/GaJh8c7e9qgow+Q7cGBo+98WsLkuktFqiZabtXuQTu/Y5ETbR0v7tNSFnvrmu6pjdoan2KjMu8q/Hmj1EfCO2ZGfEIbIXKUlw8qaX9/b2oeSJmFksSeT/Fn0V3nSypChh4Gjh74ybO9aeZ/AN2dwciu2/MhAAAAAElFTkSuQmCC">一次生信实战<!----></h1>  <div class="theme-vdoing-content content__default"><p>2023-09-22</p> <p><a href="https://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247521923&amp;idx=1&amp;sn=8161dadb6ad8ff8d1616137e61b35f1c&amp;scene=21#wechat_redirect" target="_blank" rel="noopener noreferrer">单细胞+芯片+转录组测序的数据挖掘文章一比一复现 <span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a></p> <p>文章：
Frontier，Integration of Single-Cell RNA Sequencing and Bulk RNA Sequencing Data to Establish and Validate a Prognostic Model for Patients With <strong>Lung Adenocarcinoma</strong></p> <p>数据</p> <h3 id="分析思路"><a href="#分析思路" class="header-anchor">#</a> 分析思路</h3> <ul><li>单细胞分群，功能分析，拟时序，细胞通讯</li> <li>TCGA--DEG，GO，KEGG，</li> <li>WGCNA--hub genes，两者交集</li> <li>单因素COX？？，非负矩阵分解？？cluster生存分析？？免疫浸润？？</li> <li>许多评分？？应该是想证明临床相关</li></ul> <div class="language-R line-numbers-mode"><pre class="language-r"><code>ls<span class="token punctuation">(</span><span class="token punctuation">)</span> <span class="token comment">#列出所有对象，函数/变量/数据集等</span>
ls<span class="token punctuation">(</span>pos <span class="token operator">=</span> <span class="token string">&quot;package:dplyr&quot;</span><span class="token punctuation">)</span> <span class="token comment">#看看加载包的所有函数</span>
help<span class="token punctuation">(</span>package <span class="token operator">=</span> <span class="token string">&quot;dplyr&quot;</span><span class="token punctuation">)</span> <span class="token comment">#look help file</span>
</code></pre> <div class="line-numbers-wrapper"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br></div></div><p>具体代码和功能注释</p> <h3 id="合并数据"><a href="#合并数据" class="header-anchor">#</a> 合并数据</h3> <div class="language- line-numbers-mode"><pre class="language-text"><code>library(Seurat)  
library(harmony)  #去批次
library(dplyr)    #处理数据框
library(stringr)  #处理字符串
rm(list = ls())  

# 多个数据读取与合并  
rawdata_path &lt;- './rawdata'  
filename &lt;- list.files(rawdata_path)  
rawdata_path &lt;- paste(rawdata_path,filename,sep = '/')  
sceList &lt;- lapply(rawdata_path, function(x){  
	obj &lt;- CreateSeuratObject(counts = Read10X(x),  
                             project = str_split(x,'/')[[1]][3])})  
names(sceList) &lt;- filename
</code></pre> <div class="line-numbers-wrapper"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br><span class="line-number">6</span><br><span class="line-number">7</span><br><span class="line-number">8</span><br><span class="line-number">9</span><br><span class="line-number">10</span><br><span class="line-number">11</span><br><span class="line-number">12</span><br><span class="line-number">13</span><br><span class="line-number">14</span><br></div></div><p>rawdata_path，操作出来批量的样本名文件夹（下面就是R10X的三个数据）
sceList，操作出来若干seurat对象
project和name？？</p> <div class="language- line-numbers-mode"><pre class="language-text"><code>#合并数据，疾病和正常两组
sce &lt;- merge(sceList[[1]],sceList[-1],add.cell.ids = names(sceList),project = 'luad')  
sce@meta.data$group &lt;- str_split(sce@meta.data$orig.ident,'_',simplify = T)[,2]
</code></pre> <div class="line-numbers-wrapper"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br></div></div><p>两个中括号内的1&amp;一个中括号的-1</p> <div class="language- line-numbers-mode"><pre class="language-text"><code># 查看线粒体(MT开头)、核糖体(RPS/RPL开头)、血红细胞所占比例  

'^RP[SL]'可以匹配那些以&quot;RP&quot;开头并后面跟着字母&quot;S&quot;或&quot;L&quot;的字符串
'^HB[^(P)]'可以匹配那些以&quot;HB&quot;开头并后面跟着不是字母&quot;P&quot;的字符串
value默认为False返回索引位置；True返回匹配到的实际字符串

grep('^MT',x=rownames(sce@assays$RNA@data),value = T)  
grep('^RP[SL]',x=rownames(sce@assays$RNA@data),value = T)  
grep('^HB[^(P)]',x=rownames(sce@assays$RNA@data),value = T)  
  
sce &lt;- PercentageFeatureSet(sce,'^MT',col.name = 'percent_MT')  
sce &lt;- PercentageFeatureSet(sce,'^RP[SL]',col.name = 'percent_RP')  
sce &lt;- PercentageFeatureSet(sce,'^HB[^(P)]',col.name = 'percent_HB')  
VlnPlot(sce,features = &quot;nCount_RNA&quot;,pt.size = 0,y.max = 10000)  
  
VlnPlot(sce,features = &quot;nFeature_RNA&quot;,pt.size = 0,y.max = 2500)  
VlnPlot(sce,features = &quot;percent_MT&quot;,pt.size = 0)  
VlnPlot(sce,features = &quot;percent_RP&quot;,pt.size = 0)  
VlnPlot(sce,features = &quot;percent_HB&quot;,pt.size = 0,y.max = 0.1)  
VlnPlot(sce,features = c(&quot;nCount_RNA&quot;,&quot;nFeature_RNA&quot;,&quot;percent_MT&quot;),pt.size = 0,group.by = 'orig.ident')
</code></pre> <div class="line-numbers-wrapper"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br><span class="line-number">6</span><br><span class="line-number">7</span><br><span class="line-number">8</span><br><span class="line-number">9</span><br><span class="line-number">10</span><br><span class="line-number">11</span><br><span class="line-number">12</span><br><span class="line-number">13</span><br><span class="line-number">14</span><br><span class="line-number">15</span><br><span class="line-number">16</span><br><span class="line-number">17</span><br><span class="line-number">18</span><br><span class="line-number">19</span><br><span class="line-number">20</span><br></div></div><p>grep返回（行为基因）的字符串，返回之后给谁跑到哪里去了？？
grep（可能）搜到并返回线粒体/核糖体/红细胞基因
PercentFeatureSet给细胞再设置上metadata之线粒体/核糖体/血红蛋白基因比例</p> <div class="language- line-numbers-mode"><pre class="language-text"><code># 过滤细胞  
sce &lt;- subset(sce,subset = nCount_RNA&gt;1000 &amp; nFeature_RNA&gt;300 &amp; percent_MT&lt;25)  
# 过滤基因  
sce &lt;- sce[rowSums(sce@assays$RNA@counts&gt;0)&gt;3,]  
# 过滤线粒体、核糖体、血红细胞进占比高的细胞  
sce &lt;- subset(sce,subset = percent_MT&lt;25 &amp; percent_RP&lt;30 &amp; percent_HB&lt;0.1)
# 查看多少基因（行）多少细胞（列）
dim(sce)
</code></pre> <div class="line-numbers-wrapper"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br><span class="line-number">6</span><br><span class="line-number">7</span><br><span class="line-number">8</span><br></div></div><p>细胞，测出太多数量count的，太多种feature的，太多线粒体比例的都不要
基因，逻辑值--和大于3个细胞--的行切片保留，是筛选基因</p> <div class="language- line-numbers-mode"><pre class="language-text"><code># 细胞周期评分
# S.Score较高的为S期，G2M.Score较高的为G2M期，都比较低的为G1期
# DNA合成前期（G1期）、DNA合成期（S期）与DNA合成后期（G2期）、分裂就是M期（又分前中后末）、G2M，一起，意思一般是G2合成后紧贴着M期
s_feature &lt;- cc.genes.updated.2019$s.genes    #s期的标记基因
g2m_feature &lt;- cc.genes.updated.2019$g2m.genes  
sce &lt;- CellCycleScoring(sce,  
                        s.features = s_feature,  
                        g2m.features = g2m_feature,  
                        set.ident = T)  
VlnPlot(sce,features = c('S.Score','G2M.Score'),group.by = 'orig.ident',pt.size = 0)  
saveRDS(sce,'sce_qc.rds')
</code></pre> <div class="line-numbers-wrapper"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br><span class="line-number">6</span><br><span class="line-number">7</span><br><span class="line-number">8</span><br><span class="line-number">9</span><br><span class="line-number">10</span><br><span class="line-number">11</span><br></div></div><p>cc.genes.updated.2019$s.genes，seurat包中的数据集合</p> <p>CellCycleScoring细胞周期评分</p> <p>VlnPlot可视化看看分数</p> <h3 id="去批次效应"><a href="#去批次效应" class="header-anchor">#</a> 去批次效应</h3> <div class="language- line-numbers-mode"><pre class="language-text"><code># harmony去除批次效应
library(harmony)  
library(dplyr)  
library(Seurat)  
library(clustree)  
rm(list = ls())  

# harmony之前看一次效果
sce &lt;- readRDS('sce_qc.rds')  
sce &lt;- NormalizeData(sce,  
                     normalization.method = 'LogNormalize',  
                     scale.factor = 10000)  
sce &lt;- FindVariableFeatures(sce,  
                            selection.method = &quot;vst&quot;,  
                            nfeatures = 2000)  
sce &lt;- ScaleData(sce)      # 默认用variableFeature做scale  
sce &lt;- RunPCA(sce,features = VariableFeatures(sce))  
DimPlot(sce,reduction = 'pca',group.by = 'group')


# harmony之后再看一次效果
sce &lt;- RunHarmony(sce,group.by.vars = 'orig.ident')

ElbowPlot(sce,reduction = 'harmony')
sce &lt;- RunUMAP(sce,dims = 1:10,reduction = 'harmony')
sce &lt;- RunTSNE(sce,dims = 1:10,reduction = 'harmony')

DimPlot(sce,reduction = 'umap',label = T,group.by = 'group')
DimPlot(sce,reduction = 'tsne',label = T,group.by = 'group')

</code></pre> <div class="line-numbers-wrapper"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br><span class="line-number">6</span><br><span class="line-number">7</span><br><span class="line-number">8</span><br><span class="line-number">9</span><br><span class="line-number">10</span><br><span class="line-number">11</span><br><span class="line-number">12</span><br><span class="line-number">13</span><br><span class="line-number">14</span><br><span class="line-number">15</span><br><span class="line-number">16</span><br><span class="line-number">17</span><br><span class="line-number">18</span><br><span class="line-number">19</span><br><span class="line-number">20</span><br><span class="line-number">21</span><br><span class="line-number">22</span><br><span class="line-number">23</span><br><span class="line-number">24</span><br><span class="line-number">25</span><br><span class="line-number">26</span><br><span class="line-number">27</span><br><span class="line-number">28</span><br><span class="line-number">29</span><br><span class="line-number">30</span><br></div></div><div class="language- line-numbers-mode"><pre class="language-text"><code>sce &lt;- FindNeighbors(sce,reduction = 'harmony',dims = 1:10)  
sce_res &lt;- FindClusters(sce_res,resolution = 0.8)  
# 设置不同的分辨率查看分组情况  
sce_res &lt;- sce  
for (i in c(0.01, 0.05, 0.1, 0.2, 0.3, 0.5,0.8,1)){  
  sce_res &lt;- FindClusters(sce_res,resolution = i)  
}  # 这个循环不错诶，可以看看哪种分辨率最合适
clustree(sce_res,prefix = 'RNA_snn_res.')
</code></pre> <div class="line-numbers-wrapper"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br><span class="line-number">6</span><br><span class="line-number">7</span><br><span class="line-number">8</span><br></div></div><p>FindNeighbors计算细胞间距离</p> <p>FindClusters确定分群</p> <div class="language- line-numbers-mode"><pre class="language-text"><code># 确定分群分辨率为0.5
sce &lt;- FindClusters(sce,resolution = 0.5)  
saveRDS(sce,file = 'step2_harmony.rds')  
  
DimPlot(sce,reduction = 'umap',group.by = 'orig.ident')  
DimPlot(sce,reduction = 'umap',group.by = 'seurat_clusters')

sce@meta.data[1:3,] # meta.data相当于anndata的obs.m、对obs的各种统计信息
</code></pre> <div class="line-numbers-wrapper"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br><span class="line-number">6</span><br><span class="line-number">7</span><br><span class="line-number">8</span><br></div></div><h3 id="细胞分群注释"><a href="#细胞分群注释" class="header-anchor">#</a> 细胞分群注释</h3> <div class="language- line-numbers-mode"><pre class="language-text"><code># 注释细胞
library(SingleR)  
library(celldex)  
library(Seurat)  
library(dplyr)  
library(stringr)  
library(pheatmap)  
library(ReactomeGSA)  
library(monocle)  
library(limma)  
library(ggplot2)  
library(msigdbr)  
library(singleseqgset)  
  
rm(list = ls())  
gc()  
  
sce &lt;- readRDS('step2_harmony.rds')  

# singleR注释  
hpca.se &lt;- HumanPrimaryCellAtlasData() # 获取注释信息（可以先下载rds，load出来）
bpe.se &lt;- BlueprintEncodeData()  
anno &lt;- SingleR(sce@assays$RNA@data,  
                ref = list(BP=bpe.se,HPCA=hpca.se),  
                labels = list(bpe.se$label.main,hpca.se$label.main),  
                clusters = sce@meta.data$seurat_clusters  
                )
table(ann%labels)
plotScoreHeatmap(anno,clusters = anno@rownames,show_colnames = T)

sce@meta.data$singleR_label &lt;- unlist(lapply(sce@meta.data$seurat_clusters, function(x{anno$labels[x]})) # 不知道是否有错误？？？

DimPlot(sce,reduction = 'umap',group.by = 'singleR_label',label = T)
DimPlot(sce,reduction = 'tsne',group.by = 'singleR_label',label = T)

</code></pre> <div class="line-numbers-wrapper"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br><span class="line-number">6</span><br><span class="line-number">7</span><br><span class="line-number">8</span><br><span class="line-number">9</span><br><span class="line-number">10</span><br><span class="line-number">11</span><br><span class="line-number">12</span><br><span class="line-number">13</span><br><span class="line-number">14</span><br><span class="line-number">15</span><br><span class="line-number">16</span><br><span class="line-number">17</span><br><span class="line-number">18</span><br><span class="line-number">19</span><br><span class="line-number">20</span><br><span class="line-number">21</span><br><span class="line-number">22</span><br><span class="line-number">23</span><br><span class="line-number">24</span><br><span class="line-number">25</span><br><span class="line-number">26</span><br><span class="line-number">27</span><br><span class="line-number">28</span><br><span class="line-number">29</span><br><span class="line-number">30</span><br><span class="line-number">31</span><br><span class="line-number">32</span><br><span class="line-number">33</span><br><span class="line-number">34</span><br><span class="line-number">35</span><br></div></div><p>HumanPrimaryCellAtlasData和BluePrintEncodeData，两个SingleR自带的数据库
ann储存注释信息
若想继续进行亚群注释，取sce子集再降维分群，SingleR注释</p> <div class="language- line-numbers-mode"><pre class="language-text"><code># 鉴定关键细胞亚群
cell_list &lt;- split(colnames(sce), sce$singleR_label)  
  
deg &lt;- c()  
for (i in names(cell_list)){  
  sce_temp &lt;- sce[,colnames(sce) %in% cell_list[[i]]]  
  sce_temp &lt;- CreateSeuratObject(counts = sce_temp@assays$RNA@counts,  
                                 meta.data = sce_temp@meta.data,  
                                 min.cells = 3,  
                                 min.features = 200)  
  sce_temp &lt;- NormalizeData(sce_temp)  
  sce_temp &lt;- FindVariableFeatures(sce_temp)  
  sce_temp &lt;- ScaleData(sce_temp,vars.to.regress = c('nFeature_RNA','&quot;percent_MT&quot;'))  
  Idents(sce_temp) &lt;- sce_temp$group  
  sce_temp_markers &lt;- FindAllMarkers(object=sce_temp,  
                                     only.pos = T,  
                                     logfc.threshold = 2,  
                                     min.pct = 0.1)  
  sce_temp_markers &lt;- sce_temp_markers[order(sce_temp_markers$cluster,sce_temp_markers$avg_log2FC),]  
  deg_gene &lt;- sce_temp_markers[sce_temp_markers$p_val_adj&lt;0.05,'gene']  
  deg &lt;- c(deg,length(deg_gene))  
}  
  
df &lt;- data.frame(cell_type = names(cell_list),deg_gene = deg)
df
</code></pre> <div class="line-numbers-wrapper"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br><span class="line-number">6</span><br><span class="line-number">7</span><br><span class="line-number">8</span><br><span class="line-number">9</span><br><span class="line-number">10</span><br><span class="line-number">11</span><br><span class="line-number">12</span><br><span class="line-number">13</span><br><span class="line-number">14</span><br><span class="line-number">15</span><br><span class="line-number">16</span><br><span class="line-number">17</span><br><span class="line-number">18</span><br><span class="line-number">19</span><br><span class="line-number">20</span><br><span class="line-number">21</span><br><span class="line-number">22</span><br><span class="line-number">23</span><br><span class="line-number">24</span><br><span class="line-number">25</span><br></div></div><p>没看太懂，大致是这种意思，把每种类型的细胞分别拿出来，用FindAllMarkers找出正常和肿瘤之间的差异基因</p> <div class="language- line-numbers-mode"><pre class="language-text"><code># 单细胞通路富集
temp &lt;- sce  
Idents(temp) &lt;- temp@meta.data$singleR_label  
reactome_obj &lt;- analyse_sc_clusters(temp,create_reactome_visualization=T,use_interactors=F)  
path_ways &lt;- pathways(reactome_obj)  
path_ways$diff &lt;- rowMax(as.matrix(path_ways[,2:ncol(path_ways)])) - rowMin(as.matrix(path_ways[,2:ncol(path_ways)]))  
path_ways &lt;- path_ways[order(path_ways$diff,decreasing = T),]  
  
plot_gsva_heatmap(reactome_obj,rownames(path_ways)[1:10],margins = c(10,20))  
  
saveRDS(sce,file = 'step3_sce.rds')
</code></pre> <div class="line-numbers-wrapper"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br><span class="line-number">6</span><br><span class="line-number">7</span><br><span class="line-number">8</span><br><span class="line-number">9</span><br><span class="line-number">10</span><br><span class="line-number">11</span><br></div></div><p>没看懂，有些包是不知道干啥的；估计是开始那一坨包里面的函数
介绍说是，“用pathway()提取富集出来的结果，每行是一个通路，每列是一种细胞，计算每个通路在7种细胞中表达差异的diff，根据diff排序，画出前10个通路”</p> <div class="language- line-numbers-mode"><pre class="language-text"><code># 拟时序分析
library(Seurat)
library(monocle)

sce &lt;- readRDS('step3_sce.rds')
# 1. 将seurat对象转化为monocle的CDS对象sparse_data &lt;- as(as.matrix(sce@assays$RNA@counts),'sparseMatrix')
mdata &lt;- new('AnnotatedDataFrame',data=sce@meta.data) # 行为cellf
Data &lt;- data.frame(gene_short_name=row.names(sparse_data),row.names = row.names(sparse_data))
fd &lt;- new('AnnotatedDataFrame',data=fData) # 行为gene 列为gene description
monocle_cds &lt;- newCellDataSet(cellData = sparse_data,
	phenoData = mdata,                              
	featureData = fd,                              
	lowerDetectionLimit = 0.5,                              
	expressionFamily = negbinomial.size())

# 2. 计算size factor和离散度
monocle_cds &lt;- estimateSizeFactors(monocle_cds)
monocle_cds &lt;- estimateDispersions(monocle_cds)

# 3.过滤低质量细胞
monocle_cds &lt;- detectGenes(monocle_cds,min_expr = 0.1)

# 4. 找高变基因、降维分群
# 用seurat找的高变基因 
sce_var_gene &lt;- VariableFeatures(sce)
monocle_cds &lt;- setOrderingFilter(monocle_cds,sce_var_gene)
monocle_cds &lt;- reduceDimension(monocle_cds,num_dim=10,norm_method = 'log',reduction_method = 'tSNE')
monocle_cds &lt;- clusterCells(monocle_cds,num_clusters = 10)

plot_cell_clusters(monocle_cds,color_by = 'singleR_label')

diff_test_gene &lt;- differentialGeneTest(monocle_cds[sce_var_gene,],fullModelFormulaStr = '~singleR_label')
diff_gene &lt;- row.names(subset(diff_test_gene,qval&lt;0.01))
monocle_cds &lt;- setOrderingFilter(monocle_cds,diff_gene)

# 5. 用reversed graph embedding降维monocle_cds &lt;- reduceDimension(monocle_cds,reduction_method = 'DDRTree')

# 6. 计算细胞拟时间monocle_cds &lt;- orderCells(monocle_cds)
</code></pre> <div class="line-numbers-wrapper"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br><span class="line-number">6</span><br><span class="line-number">7</span><br><span class="line-number">8</span><br><span class="line-number">9</span><br><span class="line-number">10</span><br><span class="line-number">11</span><br><span class="line-number">12</span><br><span class="line-number">13</span><br><span class="line-number">14</span><br><span class="line-number">15</span><br><span class="line-number">16</span><br><span class="line-number">17</span><br><span class="line-number">18</span><br><span class="line-number">19</span><br><span class="line-number">20</span><br><span class="line-number">21</span><br><span class="line-number">22</span><br><span class="line-number">23</span><br><span class="line-number">24</span><br><span class="line-number">25</span><br><span class="line-number">26</span><br><span class="line-number">27</span><br><span class="line-number">28</span><br><span class="line-number">29</span><br><span class="line-number">30</span><br><span class="line-number">31</span><br><span class="line-number">32</span><br><span class="line-number">33</span><br><span class="line-number">34</span><br><span class="line-number">35</span><br><span class="line-number">36</span><br><span class="line-number">37</span><br><span class="line-number">38</span><br></div></div><p>第六步报错找公众号上的解决方案（已解决，后续可以直接运行）</p> <div class="language- line-numbers-mode"><pre class="language-text"><code># 6. 计算细胞拟时间  
source('order_cells.R')  
library('igraph')  
my_ordercell &lt;- function(cds){  
  root_state = NULL  
  num_paths = NULL  
  reverse = NULL  
  root_cell &lt;- select_root_cell(cds, root_state, reverse)  
  cds@auxOrderingData &lt;- new.env(hash = TRUE)  
    
  if (cds@dim_reduce_type == &quot;DDRTree&quot;) {  
    if (is.null(num_paths) == FALSE) {  
      message(&quot;Warning: num_paths only valid for method 'ICA' in reduceDimension()&quot;)  
    }  
    cc_ordering &lt;- extract_ddrtree_ordering(cds, root_cell)  
    pData(cds)$Pseudotime &lt;- cc_ordering[row.names(pData(cds)), ]$pseudo_time  
    K_old &lt;- reducedDimK(cds)  
    old_dp &lt;- cellPairwiseDistances(cds)  
    old_mst &lt;- minSpanningTree(cds)  
    old_A &lt;- reducedDimA(cds)  
    old_W &lt;- reducedDimW(cds)  
    cds &lt;- project2MST(cds, project_point_to_line_segment)  
    minSpanningTree(cds) &lt;- cds@auxOrderingData[[cds@dim_reduce_type]]$pr_graph_cell_proj_tree  
    root_cell_idx &lt;- which(V(old_mst)$name == root_cell, arr.ind = T)  
    cells_mapped_to_graph_root &lt;- which(cds@auxOrderingData[[&quot;DDRTree&quot;]]$pr_graph_cell_proj_closest_vertex == root_cell_idx)  
    if (length(cells_mapped_to_graph_root) == 0) {  
      cells_mapped_to_graph_root &lt;- root_cell_idx  
    }  
    cells_mapped_to_graph_root &lt;- V(minSpanningTree(cds))[cells_mapped_to_graph_root]$name  
    tip_leaves &lt;- names(which(degree(minSpanningTree(cds)) == 1))  
    root_cell &lt;- cells_mapped_to_graph_root[cells_mapped_to_graph_root %in% tip_leaves][1]  
    if (is.na(root_cell)) {  
      root_cell &lt;- select_root_cell(cds, root_state, reverse)  
    }  
    cds@auxOrderingData[[cds@dim_reduce_type]]$root_cell &lt;- root_cell  
    cc_ordering_new_pseudotime &lt;- extract_ddrtree_ordering(cds, root_cell)  
    pData(cds)$Pseudotime &lt;- cc_ordering_new_pseudotime[row.names(pData(cds)), ]$pseudo_time  
    if (is.null(root_state) == TRUE) {  
      closest_vertex &lt;- cds@auxOrderingData[[&quot;DDRTree&quot;]]$pr_graph_cell_proj_closest_vertex  
      pData(cds)$State &lt;- cc_ordering[closest_vertex[, 1], ]$cell_state  
    }  
  }  
  cds  
} 

monocle_cds &lt;- my_ordercell(monocle_cds)
</code></pre> <div class="line-numbers-wrapper"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br><span class="line-number">6</span><br><span class="line-number">7</span><br><span class="line-number">8</span><br><span class="line-number">9</span><br><span class="line-number">10</span><br><span class="line-number">11</span><br><span class="line-number">12</span><br><span class="line-number">13</span><br><span class="line-number">14</span><br><span class="line-number">15</span><br><span class="line-number">16</span><br><span class="line-number">17</span><br><span class="line-number">18</span><br><span class="line-number">19</span><br><span class="line-number">20</span><br><span class="line-number">21</span><br><span class="line-number">22</span><br><span class="line-number">23</span><br><span class="line-number">24</span><br><span class="line-number">25</span><br><span class="line-number">26</span><br><span class="line-number">27</span><br><span class="line-number">28</span><br><span class="line-number">29</span><br><span class="line-number">30</span><br><span class="line-number">31</span><br><span class="line-number">32</span><br><span class="line-number">33</span><br><span class="line-number">34</span><br><span class="line-number">35</span><br><span class="line-number">36</span><br><span class="line-number">37</span><br><span class="line-number">38</span><br><span class="line-number">39</span><br><span class="line-number">40</span><br><span class="line-number">41</span><br><span class="line-number">42</span><br><span class="line-number">43</span><br><span class="line-number">44</span><br><span class="line-number">45</span><br><span class="line-number">46</span><br></div></div><div class="language- line-numbers-mode"><pre class="language-text"><code># 可以画轨迹的图了  
plot_cell_trajectory(monocle_cds,color_by = 'Pseudotime')  
plot_cell_trajectory(monocle_cds,color_by = 'State')  
plot_cell_trajectory(monocle_cds,color_by = 'singleR_label')  
  
plot_cell_trajectory(monocle_cds,color_by = 'singleR_label')+facet_wrap(~singleR_label,nrow = 3)  
  
saveRDS(monocle_cds,file = 'monocle.rds')
</code></pre> <div class="line-numbers-wrapper"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br><span class="line-number">6</span><br><span class="line-number">7</span><br><span class="line-number">8</span><br></div></div><div class="language- line-numbers-mode"><pre class="language-text"><code>细胞间通讯
library(CellChat)  
library(patchwork)  
library(Seurat)  
library(dplyr)  
library(ggalluvial)  
library(NMF)  
options(stringsAsFactors = FALSE)  
rm(list = ls())  
gc()  
  
sce &lt;- readRDS('step3_sce.rds')  
  
# Step1.创建cellchat对象  
data.input &lt;- sce@assays$RNA@data  
meta.data &lt;- sce@meta.data  
  
cellchat &lt;- createCellChat(object=data.input,  
                           meta = meta.data,  
                           group.by='singleR_label')  
cellchat &lt;- addMeta(cellchat,meta = meta.data)  
  
# Step2.加载CellChatDB数据库  
cellchatDB &lt;- CellChatDB.human  
cellchat@DB &lt;- cellchatDB  
  
# Step3.处理表达数据  
cellchat &lt;- subsetData(cellchat)  
cellchat &lt;- identifyOverExpressedGenes(cellchat)  
cellchat &lt;- identifyOverExpressedInteractions(cellchat)  
  
# Step4.计算通讯概率，推断细胞通讯网络  这一步很耗时  
cellchat &lt;- computeCommunProb(cellchat,population.size = F)  
cellchat &lt;- filterCommunication(cellchat,min.cells = 10)  
  
# Step5. 提取预测的细胞通讯网络为data frame  
df.net &lt;- subsetCommunication(cellchat)  
df.pathway &lt;- subsetCommunication(cellchat,slot.name = 'netP')  
  
# Step6. 在信号通路水平推断细胞通讯  
cellchat &lt;- computeCommunProbPathway(cellchat)  
  
# Step7. 计算加和的cell-cell通讯网络  
cellchat &lt;- aggregateNet(cellchat)  
par(mfrow = c(1,2))  
netVisual_circle(cellchat@net$count,vertex.weight = groupSize,  
                 weight.scale = T,label.edge = F,  
                 title.name = 'number of Interaction')  
netVisual_circle(cellchat@net$weight,vertex.weight = groupSize,  
                 weight.scale = T,label.edge = F,  
                 title.name = 'Interaction Weight')  
  
mat &lt;- cellchat@net$weight  
par(mfrow = c(3,3),xpd=T)  
for (i in 1:nrow(mat)){  
  mat2 &lt;- matrix(0,nrow = nrow(mat),ncol = ncol(mat),  
                 dimnames = dimnames(mat))  
  mat2[i,] &lt;- mat[i,]  
  netVisual_circle(mat2,vertex.weight = groupSize,  
                   weight.scale = T,edge.weight.max = max(mat),  
                   title.name = rownames(mat)[i])
</code></pre> <div class="line-numbers-wrapper"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br><span class="line-number">6</span><br><span class="line-number">7</span><br><span class="line-number">8</span><br><span class="line-number">9</span><br><span class="line-number">10</span><br><span class="line-number">11</span><br><span class="line-number">12</span><br><span class="line-number">13</span><br><span class="line-number">14</span><br><span class="line-number">15</span><br><span class="line-number">16</span><br><span class="line-number">17</span><br><span class="line-number">18</span><br><span class="line-number">19</span><br><span class="line-number">20</span><br><span class="line-number">21</span><br><span class="line-number">22</span><br><span class="line-number">23</span><br><span class="line-number">24</span><br><span class="line-number">25</span><br><span class="line-number">26</span><br><span class="line-number">27</span><br><span class="line-number">28</span><br><span class="line-number">29</span><br><span class="line-number">30</span><br><span class="line-number">31</span><br><span class="line-number">32</span><br><span class="line-number">33</span><br><span class="line-number">34</span><br><span class="line-number">35</span><br><span class="line-number">36</span><br><span class="line-number">37</span><br><span class="line-number">38</span><br><span class="line-number">39</span><br><span class="line-number">40</span><br><span class="line-number">41</span><br><span class="line-number">42</span><br><span class="line-number">43</span><br><span class="line-number">44</span><br><span class="line-number">45</span><br><span class="line-number">46</span><br><span class="line-number">47</span><br><span class="line-number">48</span><br><span class="line-number">49</span><br><span class="line-number">50</span><br><span class="line-number">51</span><br><span class="line-number">52</span><br><span class="line-number">53</span><br><span class="line-number">54</span><br><span class="line-number">55</span><br><span class="line-number">56</span><br><span class="line-number">57</span><br><span class="line-number">58</span><br><span class="line-number">59</span><br><span class="line-number">60</span><br><span class="line-number">61</span><br></div></div><div class="language- line-numbers-mode"><pre class="language-text"><code># Step8. 使用层次图（Hierarchical plot），圆圈图（Circle plot）或和弦图（Chord diagram）可视化每个信号通路  
cellchat@netP$pathways  
pathway.show &lt;- c('SEMA4')  
vertex.receiver &lt;- c(1,2,3,4)  
netVisual_aggregate(cellchat,signaling = pathway.show,  
                    vertex.receiver = vertex.receiver,  
                    layout = 'hierarchy')

par(mfrow=c(1,2),xpd=T)  
netVisual_aggregate(cellchat,signaling = pathway.show,  
                    layout = 'circle')  
netVisual_aggregate(cellchat,signaling = pathway.show,  
                    layout = 'circle',label.edge=T)

pairLR.CXCL &lt;- extractEnrichedLR(cellchat,signaling = pathway.show,  
                                 geneLR.return = F)  
LR.show &lt;- pairLR.CXCL[1,]  
netVisual_individual(cellchat,signaling = pathway.show,  
                     pairLR.use = 'SEMA4A_PLXNB2',  
                     layout = 'circle')  
  
netVisual_aggregate(cellchat,signaling = pathway.show,layout = 'chord',  
                    title.name = &quot;Chord diagram  1&quot;)
</code></pre> <div class="line-numbers-wrapper"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br><span class="line-number">6</span><br><span class="line-number">7</span><br><span class="line-number">8</span><br><span class="line-number">9</span><br><span class="line-number">10</span><br><span class="line-number">11</span><br><span class="line-number">12</span><br><span class="line-number">13</span><br><span class="line-number">14</span><br><span class="line-number">15</span><br><span class="line-number">16</span><br><span class="line-number">17</span><br><span class="line-number">18</span><br><span class="line-number">19</span><br><span class="line-number">20</span><br><span class="line-number">21</span><br><span class="line-number">22</span><br><span class="line-number">23</span><br></div></div><p>额按，啊吧，啊吧吧，后边的分析有TCGA的生存分析，差异基因还理解，但是再往后的免疫浸润，Cox回归等等就不太理解了。。。</p> <p>算是学到了整合分析吧，一些关于monocle的包还要单独去学学的。</p></div></div>  <div class="page-edit"><!----> <!----> <!----></div> <div class="page-nav-wapper"><!----> <!----></div></div> <div class="article-list"><div class="article-title"><a href="/archives/" class="iconfont icon-bi">最近更新</a></div> <div class="article-wrapper"><dl><dd>01</dd> <dt><a href="/pages/90dbe8/"><div>
            APA相关综述
            <!----></div></a> <span class="date">06-27</span></dt></dl><dl><dd>02</dd> <dt><a href="/pages/f4f4b7/"><div>
            胃癌免疫逃逸的决定因素
            <!----></div></a> <span class="date">06-25</span></dt></dl><dl><dd>03</dd> <dt><a href="/pages/0a11dd/"><div>
            胖子CAF
            <!----></div></a> <span class="date">06-21</span></dt></dl> <dl><dd></dd> <dt><a href="/archives/" class="more">更多文章&gt;</a></dt></dl></div></div></main></div> <div class="footer"><div class="icons"><a href="mailto:2590643635@qq.com" title="发邮件" target="_blank" class="iconfont icon-youjian"></a><a href="https://github.com/hongliangfang" title="GitHub" target="_blank" class="iconfont icon-github"></a><a href="https://music.163.com/#/playlist?id=755597173" title="听音乐" target="_blank" class="iconfont icon-erji"></a></div> 
  Theme by
  <a href="https://github.com/xugaoyi/vuepress-theme-vdoing" target="_blank" title="本站主题">Vdoing</a> 
    | Copyright © 2023-2024
    <span>FHL | <a href="https://github.com/xugaoyi/vuepress-theme-vdoing/blob/master/LICENSE" target="_blank">MIT License</a></span></div> <div class="buttons"><div title="返回顶部" class="button blur go-to-top iconfont icon-fanhuidingbu" style="display:none;"></div> <div title="去评论" class="button blur go-to-comment iconfont icon-pinglun" style="display:none;"></div> <div title="主题模式" class="button blur theme-mode-but iconfont icon-zhuti"><ul class="select-box" style="display:none;"><li class="iconfont icon-zidong">
          跟随系统
        </li><li class="iconfont icon-rijianmoshi">
          浅色模式
        </li><li class="iconfont icon-yejianmoshi">
          深色模式
        </li><li class="iconfont icon-yuedu">
          阅读模式
        </li></ul></div></div> <!----> <!----> <!----></div><div class="global-ui"><div></div></div></div>
    <script src="/assets/js/app.bd488e07.js" defer></script><script src="/assets/js/2.bdf5b8cf.js" defer></script><script src="/assets/js/142.0415d712.js" defer></script><script src="/assets/js/3.365ac406.js" defer></script>
  </body>
</html>
