<!DOCTYPE html>
<html lang="zh-CN">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width,initial-scale=1">
    <title>Monocle2 | 一盏烛光的微笑灿烂人生</title>
    <meta name="generator" content="VuePress 1.9.9">
    <link rel="icon" href="/img/favicon.ico">
    <link rel="stylesheet" href="https://at.alicdn.com/t/font_3129839_xft6cqs5gc.css">
    <noscript><meta http-equiv="refresh" content="0; url=https://www.youngkbt.cn/noscript/"><style>.theme-vdoing-content { display:none }</noscript>
    <meta name="description" content="学习记录博客">
    <meta name="theme-color" content="#11a8cd">
    <meta name="个人博客" content="科研学习，围炉夜话，奇淫巧计">
    
    <link rel="preload" href="/assets/css/0.styles.c9ffc21e.css" as="style"><link rel="preload" href="/assets/js/app.bd488e07.js" as="script"><link rel="preload" href="/assets/js/2.bdf5b8cf.js" as="script"><link rel="preload" href="/assets/js/53.4f209254.js" as="script"><link rel="preload" href="/assets/js/3.365ac406.js" as="script"><link rel="prefetch" href="/assets/js/10.b00b53b4.js"><link rel="prefetch" href="/assets/js/100.1b2ba6dd.js"><link rel="prefetch" href="/assets/js/101.e392eb92.js"><link rel="prefetch" href="/assets/js/102.eb0bb162.js"><link rel="prefetch" href="/assets/js/103.9db14406.js"><link rel="prefetch" href="/assets/js/104.6ae6fd5c.js"><link rel="prefetch" href="/assets/js/105.4d79399b.js"><link rel="prefetch" href="/assets/js/106.4e6ff12d.js"><link rel="prefetch" href="/assets/js/107.bc93ee73.js"><link rel="prefetch" href="/assets/js/108.59a000f0.js"><link rel="prefetch" href="/assets/js/109.f01f4422.js"><link rel="prefetch" href="/assets/js/11.346d31c6.js"><link rel="prefetch" href="/assets/js/110.96a96c44.js"><link rel="prefetch" href="/assets/js/111.87fe63df.js"><link rel="prefetch" href="/assets/js/112.c560a316.js"><link rel="prefetch" href="/assets/js/113.6a371b02.js"><link rel="prefetch" href="/assets/js/114.989a4d3e.js"><link rel="prefetch" href="/assets/js/115.b6baafa2.js"><link rel="prefetch" href="/assets/js/116.d469dc20.js"><link rel="prefetch" href="/assets/js/117.0a6bccf0.js"><link rel="prefetch" href="/assets/js/118.9f1ef088.js"><link rel="prefetch" href="/assets/js/119.059ef806.js"><link rel="prefetch" href="/assets/js/12.8d91a638.js"><link rel="prefetch" href="/assets/js/120.75ed11ab.js"><link rel="prefetch" href="/assets/js/121.1896ee22.js"><link rel="prefetch" href="/assets/js/122.a3664290.js"><link rel="prefetch" href="/assets/js/123.5b849c39.js"><link rel="prefetch" href="/assets/js/124.85737d6b.js"><link rel="prefetch" href="/assets/js/125.a95cd42f.js"><link rel="prefetch" href="/assets/js/126.7b1c1645.js"><link rel="prefetch" href="/assets/js/127.64a16fd4.js"><link rel="prefetch" href="/assets/js/128.a07721ba.js"><link rel="prefetch" href="/assets/js/129.9ba86871.js"><link rel="prefetch" href="/assets/js/13.18ed2aa7.js"><link rel="prefetch" href="/assets/js/130.5cdb8138.js"><link rel="prefetch" href="/assets/js/131.d2f31425.js"><link rel="prefetch" href="/assets/js/132.b2286d65.js"><link rel="prefetch" href="/assets/js/133.1229c9ae.js"><link rel="prefetch" href="/assets/js/134.71fad9a0.js"><link rel="prefetch" href="/assets/js/135.ff682301.js"><link rel="prefetch" href="/assets/js/136.9de1870b.js"><link rel="prefetch" href="/assets/js/137.a58bbdcd.js"><link rel="prefetch" href="/assets/js/138.cc768cb1.js"><link rel="prefetch" href="/assets/js/139.9d52d34a.js"><link rel="prefetch" href="/assets/js/14.4bcf7ee9.js"><link rel="prefetch" href="/assets/js/140.b1790e58.js"><link rel="prefetch" href="/assets/js/141.98bceb9f.js"><link rel="prefetch" href="/assets/js/142.0415d712.js"><link rel="prefetch" href="/assets/js/143.851c7187.js"><link rel="prefetch" href="/assets/js/144.ae79e007.js"><link rel="prefetch" href="/assets/js/145.d5020e58.js"><link rel="prefetch" href="/assets/js/146.c3eafa98.js"><link rel="prefetch" href="/assets/js/147.0a6864ec.js"><link rel="prefetch" href="/assets/js/148.bbc4bd28.js"><link rel="prefetch" href="/assets/js/149.489e8cdb.js"><link rel="prefetch" href="/assets/js/15.368d9d2d.js"><link rel="prefetch" href="/assets/js/150.7778d444.js"><link rel="prefetch" href="/assets/js/16.06f68354.js"><link rel="prefetch" href="/assets/js/17.4f8d6256.js"><link rel="prefetch" href="/assets/js/18.8989b2ff.js"><link rel="prefetch" href="/assets/js/19.abb857ca.js"><link rel="prefetch" href="/assets/js/20.e32420cf.js"><link rel="prefetch" href="/assets/js/21.06441c4d.js"><link rel="prefetch" href="/assets/js/22.7fb78c64.js"><link rel="prefetch" href="/assets/js/23.d9193b83.js"><link rel="prefetch" href="/assets/js/24.6487f43a.js"><link rel="prefetch" href="/assets/js/25.11e66938.js"><link rel="prefetch" href="/assets/js/26.c3e43726.js"><link rel="prefetch" href="/assets/js/27.24117748.js"><link rel="prefetch" href="/assets/js/28.fabf32c7.js"><link rel="prefetch" href="/assets/js/29.3a9331f2.js"><link rel="prefetch" href="/assets/js/30.ac5ca451.js"><link rel="prefetch" href="/assets/js/31.36e6c9c0.js"><link rel="prefetch" href="/assets/js/32.2fcf84e2.js"><link rel="prefetch" href="/assets/js/33.a73adf6b.js"><link rel="prefetch" href="/assets/js/34.49c0fb46.js"><link rel="prefetch" href="/assets/js/35.babda623.js"><link rel="prefetch" href="/assets/js/36.56221bac.js"><link rel="prefetch" href="/assets/js/37.5d6d33aa.js"><link rel="prefetch" href="/assets/js/38.a6b592e0.js"><link rel="prefetch" href="/assets/js/39.14a81539.js"><link rel="prefetch" href="/assets/js/4.a64d5c9f.js"><link rel="prefetch" href="/assets/js/40.e94616b3.js"><link rel="prefetch" href="/assets/js/41.4ad9febc.js"><link rel="prefetch" href="/assets/js/42.4825ff11.js"><link rel="prefetch" href="/assets/js/43.6d554f8a.js"><link rel="prefetch" href="/assets/js/44.ed8f011b.js"><link rel="prefetch" href="/assets/js/45.170e103d.js"><link rel="prefetch" href="/assets/js/46.9cc703a7.js"><link rel="prefetch" href="/assets/js/47.ff6af7ea.js"><link rel="prefetch" href="/assets/js/48.1b2f1a1b.js"><link rel="prefetch" href="/assets/js/49.05223958.js"><link rel="prefetch" href="/assets/js/5.b1fd9f62.js"><link rel="prefetch" href="/assets/js/50.26aa9f07.js"><link rel="prefetch" href="/assets/js/51.65eba0cc.js"><link rel="prefetch" href="/assets/js/52.5b21c37d.js"><link rel="prefetch" href="/assets/js/54.176923ed.js"><link rel="prefetch" href="/assets/js/55.74f6741d.js"><link rel="prefetch" href="/assets/js/56.383e3c0d.js"><link rel="prefetch" href="/assets/js/57.35b21350.js"><link rel="prefetch" href="/assets/js/58.7ae8fd30.js"><link rel="prefetch" href="/assets/js/59.3d7d7dea.js"><link rel="prefetch" href="/assets/js/6.993d7ad0.js"><link rel="prefetch" href="/assets/js/60.67ba2844.js"><link rel="prefetch" href="/assets/js/61.6c5542cd.js"><link rel="prefetch" href="/assets/js/62.a235dad7.js"><link rel="prefetch" href="/assets/js/63.a3a52fe2.js"><link rel="prefetch" href="/assets/js/64.2b6b610e.js"><link rel="prefetch" href="/assets/js/65.8672534a.js"><link rel="prefetch" href="/assets/js/66.c7281353.js"><link rel="prefetch" href="/assets/js/67.e062f6ac.js"><link rel="prefetch" href="/assets/js/68.446f0688.js"><link rel="prefetch" href="/assets/js/69.f03843ed.js"><link rel="prefetch" href="/assets/js/7.afb33a4c.js"><link rel="prefetch" href="/assets/js/70.466d842a.js"><link rel="prefetch" href="/assets/js/71.87fcb26b.js"><link rel="prefetch" href="/assets/js/72.ccf1092d.js"><link rel="prefetch" href="/assets/js/73.3ad6124b.js"><link rel="prefetch" href="/assets/js/74.17a59372.js"><link rel="prefetch" href="/assets/js/75.6c859470.js"><link rel="prefetch" href="/assets/js/76.674a56ad.js"><link rel="prefetch" href="/assets/js/77.5342f3c1.js"><link rel="prefetch" href="/assets/js/78.154ba5e1.js"><link rel="prefetch" href="/assets/js/79.10901b1c.js"><link rel="prefetch" href="/assets/js/8.5bbd7ae9.js"><link rel="prefetch" href="/assets/js/80.0166104e.js"><link rel="prefetch" href="/assets/js/81.4c5a0e6a.js"><link rel="prefetch" href="/assets/js/82.47099701.js"><link rel="prefetch" href="/assets/js/83.7873cd95.js"><link rel="prefetch" href="/assets/js/84.23000f66.js"><link rel="prefetch" href="/assets/js/85.6701c9e4.js"><link rel="prefetch" href="/assets/js/86.188012c3.js"><link rel="prefetch" href="/assets/js/87.e9cde64e.js"><link rel="prefetch" href="/assets/js/88.5ccdea49.js"><link rel="prefetch" href="/assets/js/89.9ab559de.js"><link rel="prefetch" href="/assets/js/9.6699859e.js"><link rel="prefetch" href="/assets/js/90.bc7fb61d.js"><link rel="prefetch" href="/assets/js/91.07537d0a.js"><link rel="prefetch" href="/assets/js/92.b4ed0bad.js"><link rel="prefetch" href="/assets/js/93.b167c2d0.js"><link rel="prefetch" href="/assets/js/94.1a02653c.js"><link rel="prefetch" href="/assets/js/95.69ef19be.js"><link rel="prefetch" href="/assets/js/96.e0b3f5df.js"><link rel="prefetch" href="/assets/js/97.9ea0c268.js"><link rel="prefetch" href="/assets/js/98.074852d9.js"><link rel="prefetch" href="/assets/js/99.708b118b.js">
    <link rel="stylesheet" href="/assets/css/0.styles.c9ffc21e.css">
  </head>
  <body class="theme-mode-light">
    <div id="app" data-server-rendered="true"><div class="theme-container sidebar-open have-rightmenu"><header class="navbar blur"><div title="目录" class="sidebar-button"><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" role="img" viewBox="0 0 448 512" class="icon"><path fill="currentColor" d="M436 124H12c-6.627 0-12-5.373-12-12V80c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12z"></path></svg></div> <a href="/" class="home-link router-link-active"><img src="/img/logo.png" alt="一盏烛光的微笑灿烂人生" class="logo"> <span class="site-name can-hide">一盏烛光的微笑灿烂人生</span></a> <div class="links"><div class="search-box"><input aria-label="Search" autocomplete="off" spellcheck="false" value=""> <!----></div> <nav class="nav-links can-hide"><div class="nav-item"><a href="/" class="nav-link">首页</a></div><div class="nav-item"><div class="dropdown-wrapper"><button type="button" aria-label="科研学习" class="dropdown-title"><a href="/study/" class="link-title">科研学习</a> <span class="title" style="display:none;">科研学习</span> <span class="arrow right"></span></button> <ul class="nav-dropdown" style="display:none;"><li class="dropdown-item"><!----> <a href="/articles/" class="nav-link">阅读记录</a></li><li class="dropdown-item"><!----> <a href="/biotech/" class="nav-link">生信分析</a></li><li class="dropdown-item"><!----> <a href="/basic/" class="nav-link">整理归纳</a></li></ul></div></div><div class="nav-item"><a href="/wlyh/" class="nav-link">围炉夜话</a></div><div class="nav-item"><a href="/technology/" class="nav-link">奇淫巧计</a></div><div class="nav-item"><div class="dropdown-wrapper"><button type="button" aria-label="索引" class="dropdown-title"><a href="/archives/" class="link-title">索引</a> <span class="title" style="display:none;">索引</span> <span class="arrow right"></span></button> <ul class="nav-dropdown" style="display:none;"><li class="dropdown-item"><!----> <a href="/categories/" class="nav-link">分类</a></li><li class="dropdown-item"><!----> <a href="/tags/" class="nav-link">标签</a></li><li class="dropdown-item"><!----> <a href="/archives/" class="nav-link">归档</a></li></ul></div></div><div class="nav-item"><div class="dropdown-wrapper"><button type="button" aria-label="更多" class="dropdown-title"><a href="/more/" class="link-title">更多</a> <span class="title" style="display:none;">更多</span> <span class="arrow right"></span></button> <ul class="nav-dropdown" style="display:none;"><li class="dropdown-item"><!----> <a href="/about/" class="nav-link">关于</a></li><li class="dropdown-item"><!----> <a href="/links/" class="nav-link">一些网站</a></li><li class="dropdown-item"><!----> <a href="/friends/" class="nav-link">友情链接</a></li><li class="dropdown-item"><!----> <a href="/passbyworld/" class="nav-link">路过人间</a></li></ul></div></div> <!----></nav></div></header> <div class="sidebar-mask"></div> <div class="sidebar-hover-trigger"></div> <aside class="sidebar" style="display:none;"><div class="blogger"><img src="https://pica.zhimg.com/v2-f2b30ad5674f652294f0ec6f9b8c9922_720w.webp?source=d16d100b"> <div class="blogger-info"><h3>围炉夜话</h3> <span>知行合一</span></div></div> <nav class="nav-links"><div class="nav-item"><a href="/" class="nav-link">首页</a></div><div class="nav-item"><div class="dropdown-wrapper"><button type="button" aria-label="科研学习" class="dropdown-title"><a href="/study/" class="link-title">科研学习</a> <span class="title" style="display:none;">科研学习</span> <span class="arrow right"></span></button> <ul class="nav-dropdown" style="display:none;"><li class="dropdown-item"><!----> <a href="/articles/" class="nav-link">阅读记录</a></li><li class="dropdown-item"><!----> <a href="/biotech/" class="nav-link">生信分析</a></li><li class="dropdown-item"><!----> <a href="/basic/" class="nav-link">整理归纳</a></li></ul></div></div><div class="nav-item"><a href="/wlyh/" class="nav-link">围炉夜话</a></div><div class="nav-item"><a href="/technology/" class="nav-link">奇淫巧计</a></div><div class="nav-item"><div class="dropdown-wrapper"><button type="button" aria-label="索引" class="dropdown-title"><a href="/archives/" class="link-title">索引</a> <span class="title" style="display:none;">索引</span> <span class="arrow right"></span></button> <ul class="nav-dropdown" style="display:none;"><li class="dropdown-item"><!----> <a href="/categories/" class="nav-link">分类</a></li><li class="dropdown-item"><!----> <a href="/tags/" class="nav-link">标签</a></li><li class="dropdown-item"><!----> <a href="/archives/" class="nav-link">归档</a></li></ul></div></div><div class="nav-item"><div class="dropdown-wrapper"><button type="button" aria-label="更多" class="dropdown-title"><a href="/more/" class="link-title">更多</a> <span class="title" style="display:none;">更多</span> <span class="arrow right"></span></button> <ul class="nav-dropdown" style="display:none;"><li class="dropdown-item"><!----> <a href="/about/" class="nav-link">关于</a></li><li class="dropdown-item"><!----> <a href="/links/" class="nav-link">一些网站</a></li><li class="dropdown-item"><!----> <a href="/friends/" class="nav-link">友情链接</a></li><li class="dropdown-item"><!----> <a href="/passbyworld/" class="nav-link">路过人间</a></li></ul></div></div> <!----></nav>  <ul class="sidebar-links"><li><section class="sidebar-group collapsable depth-0"><p class="sidebar-heading"><span>阅读记录</span> <span class="arrow right"></span></p> <!----></section></li><li><section class="sidebar-group collapsable depth-0"><p class="sidebar-heading open"><span>生信分析</span> <span class="arrow down"></span></p> <ul class="sidebar-links sidebar-group-items"><li><a href="/pages/9fc671/" class="sidebar-link">生信分析</a></li><li><a href="/pages/902bc2/" class="sidebar-link">转录组</a></li><li><a href="/pages/aebaf3/" class="sidebar-link">表观组</a></li><li><a href="/pages/819cd3/" class="sidebar-link">单细胞-概览</a></li><li><a href="/pages/c81394/" class="sidebar-link">单细胞-ATAC</a></li><li><a href="/pages/9033c2/" class="sidebar-link">单细胞-细胞注释</a></li><li><a href="/pages/e0b862/" class="sidebar-link">单细胞-异质性</a></li><li><a href="/pages/1cd666/" class="sidebar-link">Linux</a></li><li><a href="/pages/a8bb10/" class="sidebar-link">R</a></li><li><a href="/pages/d630fc/" class="sidebar-link">python</a></li><li><a href="/pages/73836e/" class="sidebar-link">基因注释</a></li><li><a href="/pages/49f65b/" class="sidebar-link">富集分析</a></li><li><a href="/pages/61f7f0/" class="sidebar-link">临床相关性</a></li><li><a href="/pages/28c669/" class="sidebar-link">WGCNA</a></li><li><a href="/pages/69a308/" class="sidebar-link">Seurat</a></li><li><a href="/pages/784c69/" class="sidebar-link">Scanpy</a></li><li><a href="/pages/44ad24/" aria-current="page" class="active sidebar-link">Monocle2</a><ul class="sidebar-sub-headers"><li class="sidebar-sub-header level2"><a href="/pages/44ad24/#整理理解" class="sidebar-link">整理理解</a></li><li class="sidebar-sub-header level2"><a href="/pages/44ad24/#学习记录" class="sidebar-link">学习记录</a></li></ul></li><li><a href="/pages/893dcf/" class="sidebar-link">单细胞-轨迹分析</a></li><li><a href="/pages/7989c6/" class="sidebar-link">CellPhoneDB</a></li><li><a href="/pages/dbe3de/" class="sidebar-link">CellChat2</a></li><li><a href="/pages/df4df8/" class="sidebar-link">NicheNet</a></li><li><a href="/pages/5f4d84/" class="sidebar-link">SCENIC</a></li><li><a href="/pages/9f91d0/" class="sidebar-link">单细胞-RNA velocity</a></li><li><a href="/pages/d079f5/" class="sidebar-link">细胞外囊泡</a></li><li><a href="/pages/d7b340/" class="sidebar-link">数据分析和数据挖掘</a></li><li><a href="/pages/c7d198/" class="sidebar-link">数据处理</a></li><li><a href="/pages/e50bac/" class="sidebar-link">data.frame</a></li><li><a href="/pages/500bc4/" class="sidebar-link">ggplot2</a></li><li><a href="/pages/937699/" class="sidebar-link">数据可视化-Python</a></li><li><a href="/pages/6cbee5/" class="sidebar-link">绘图</a></li><li><a href="/pages/5666ad/" class="sidebar-link">统计</a></li><li><a href="/pages/e75428/" class="sidebar-link">想法和借鉴</a></li><li><a href="/pages/7f8ebe/" class="sidebar-link">Algorithm</a></li></ul></section></li><li><section class="sidebar-group collapsable depth-0"><p class="sidebar-heading"><span>实验技术</span> <span class="arrow right"></span></p> <!----></section></li><li><section class="sidebar-group collapsable depth-0"><p class="sidebar-heading"><span>整理归纳</span> <span class="arrow right"></span></p> <!----></section></li></ul> </aside> <div><main class="page"><div class="theme-vdoing-wrapper "><div class="articleInfo-wrap" data-v-06225672><div class="articleInfo" data-v-06225672><ul class="breadcrumbs" data-v-06225672><li data-v-06225672><a href="/" title="首页" class="iconfont icon-home router-link-active" data-v-06225672></a></li> <li data-v-06225672><a href="/study/#科研学习" data-v-06225672>科研学习</a></li><li data-v-06225672><a href="/study/#生信分析" data-v-06225672>生信分析</a></li></ul> <div class="info" data-v-06225672><div title="作者" class="author iconfont icon-touxiang" data-v-06225672><a href="javascript:;" data-v-06225672>FHL</a></div> <div title="创建时间" class="date iconfont icon-riqi" data-v-06225672><a href="javascript:;" data-v-06225672>2024-03-17</a></div> <!----></div></div></div> <!----> <div class="content-wrapper"><div class="right-menu-wrapper"><div class="right-menu-margin"><div class="right-menu-title">目录</div> <div class="right-menu-content"></div></div></div> <h1><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAB4AAAAeCAYAAAA7MK6iAAAAAXNSR0IArs4c6QAABH1JREFUSA3tVl1oHFUUPmdmd2ltklqbpJDiNnXFmgbFktho7YMPNiJSSZM0+CAYSkUELVhM6YuwIPpgoOKDqOBDC0XE2CQoNtQXBUFTTcCi+Wlh1V2TQExsUzcltd3M9Tt3ZjZzZ2fT+OJTL8yeM+eee757fmeJbq//KQL8X3DUSFOcfr7cRsRtxNQMWueeVzOkaITIGqQHNg5y8+jNW9ldM7A6nTpAjuolUikAwq7CE3WcM2RRDz+XGVgN3FptU/aUSlvq9Pa3iZ1+sgAqJyyAFqkipd9dqiwHF3P65YycLWc/6sqGrvoEoIp6DOFaX5h6+dnfjkWprwqsPk0dUGq5vySwDImC10KxFHgGL1SWoc92O3eVht09qdXNH11I2SsTsJYqMWzihqGMi+A+Garf3BAuuLI5oGlULyNfyB/HYNujwktOfRrMr5t77NmevqaUopx0grnKAyvVpmwUDB4x6FPXuGvYLTDwWsejwgtgkYKPqRJg8SV6xaiZ3ZTppGneS4yfH5/66fZSDHv+QZci/+h5c5UHtpy67JUqGppM0sh0Nc1dW6/N1W5Yoqat8/TU/VnadmdeW2PLLSyh0cvxBs3KbqTmwYPpxN4do/mzE8nEpvX/UMu2Wbp74zUAK5q6WkHns7V0eWkdPbPzd3rxkTGybadYySumVzhcaJFbs5UrEkQ/+CK8gF5dnh/6ciIZ73gwQ927L1IitoxKLXYP3SjYdOrHHfTZhRRlFyrorafPk20B3HPD1y2G3qKZME5Jcf3t/HUC13/8tSd++vqFveMUTwAUxSUFI1QekR1+bIze3D9MF2aq6cPvG72CgnldWCFqyRw3lwH8ZMerjTD9ElRO7Gv44wNpC90aASqGfVlz/Rx17srQ57/UU26hkhQqUB7dBR71WmzQhHUnblGmVOEw0jhbV1n9OlXUDCIRGaNV5Jp43N516fN7JmnTHdfp7Hgy0luO4aMhtkLL8Bi3bUWYvzh5Mn1dTxrL6QmGuRhGL/TiTTxRoEdTszSaq9GR0NGA3KdkOz3hqSV3MIDhQ5IVX/Ivx3umBti2es2h4eZby7x8br1rkf7Mo90AqC8aQ3sJeNzqFRu+vSANAQe3PL7l0HGOAdwDCeZYvNKeoZp1Qfs6Aipndh86HmFRi0LAnEO47wsqM6cdfjh3jBPUzhZy7nvlUfFsamED1VQt6aISHVymXZ/B2aCtIG8AI8xfobj2d3en1wWVhOeHELKmLQ1s211s88comkv4UCwWyF787mJdYXtNfhKAXVqnKTq8QZvGAGGOfaTo5pGZ/PwbUCr5+DPr/1J92JNHr9aOl/F3iI5+O1nfybsGxoimvZ3ViWSluDITw3P37mypheDIPY0tw7+O/5ApbkYw+zpfaUVu32Pi98+defdUhEpZkRFq0aqyNh9FuL9hpYbEm6iwi0z2REd09ZmyENEbuhjDWzKvZXTqKYaBIr3tt5kuPtQBZFvEUwHt60vfCNu41XsksH9Ij1BMMz1Y0OOunHNShFIP5868g5zeXmuLwL9T4b6Q2+KejgAAAABJRU5ErkJggg==">Monocle2<!----></h1>  <div class="theme-vdoing-content content__default"><div class="custom-block warning"><p class="custom-block-title">注意</p> <p>注意及时修改<code>对象的名称</code>，以及根据目的，进行调整数据的输入，函数的参数！</p></div> <h3 id="理解概述"><a href="#理解概述" class="header-anchor">#</a> 理解概述</h3> <p>参考：<a href="http://cole-trapnell-lab.github.io/monocle-release/docs/" target="_blank" rel="noopener noreferrer">Monocle2官网<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a></p> <p><strong>Pseudotime analysis</strong></p> <p>monocle2能干啥？</p> <ol><li>clustering、classifying、counting cells（分群，定义，计数）</li> <li>constructing single-cell trajectories（构建轨迹）</li> <li>differential expresssion analysis （差异表达分析）</li></ol> <p>monocle2<br>
分析流程<br>
1、CellDataSet class
expression matrix：genes as row，cells as column
phenoData：cells as row，cell attributes as column —— 表型数据/细胞信息
featureData：feature as row，gene sttributes as column —— 特征数据/基因信息</p> <p>2、appropriate distribution
FPKM/TPM ：generally log-normally distributed
UMIs or read counts： better modeled with the negative binomial</p> <p>3、estimate size factors and dispersions
后续分析和计算必须</p> <p>4、classifying and counting cells
by type、without marker genes、using marker genes、imputing cell type</p> <p>下面这几个名词会在不同的函数中用到，但是意思相近，可以参考理解：</p> <ul><li>FeatureData -- gene function -- gene annotation</li> <li>PhenoData -- cell type -- sample annotation</li></ul> <h2 id="整理理解"><a href="#整理理解" class="header-anchor">#</a> 整理理解</h2> <p>最后整理一下主要用到的函数</p> <div class="language- line-numbers-mode"><pre class="language-text"><code>newCellDataSet；

Comparable：
- monocle：estimateSizeFactors；estimateDispersions
- seurat：ScaleData；NormalizeData

QC：
- monocle：detectGenes；
- seurat：PercentageFeatureSet；

DEGs：
- monocle：dispersionTable；differentialGeneTest；
- seurat：VariableFeatures；FindAllMarkers

setOrderingFilter；reduceDimension；orderCells；BEAM；

Plot：
- plot_ordering_genes
- plot_cell_trajectory
- plot_genes_in_pseudotime
- plot_genes_jitter
- plot_genes_violin
- plot_pseudotime_heatmap
- plot_genes_branched_heatmap


Get Information
- pData(HSMM_myo)
- fData(HSMM_myo)
- exprs(HSMM_myo)

</code></pre> <div class="line-numbers-wrapper"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br><span class="line-number">6</span><br><span class="line-number">7</span><br><span class="line-number">8</span><br><span class="line-number">9</span><br><span class="line-number">10</span><br><span class="line-number">11</span><br><span class="line-number">12</span><br><span class="line-number">13</span><br><span class="line-number">14</span><br><span class="line-number">15</span><br><span class="line-number">16</span><br><span class="line-number">17</span><br><span class="line-number">18</span><br><span class="line-number">19</span><br><span class="line-number">20</span><br><span class="line-number">21</span><br><span class="line-number">22</span><br><span class="line-number">23</span><br><span class="line-number">24</span><br><span class="line-number">25</span><br><span class="line-number">26</span><br><span class="line-number">27</span><br><span class="line-number">28</span><br><span class="line-number">29</span><br><span class="line-number">30</span><br><span class="line-number">31</span><br></div></div><p>参考资料：</p> <p><a href="https://www.jianshu.com/p/5d6fd4561bc0" target="_blank" rel="noopener noreferrer">（详细点）单细胞之轨迹分析 - monocle2 原理解读+实操 <span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a></p> <p><a href="https://mp.weixin.qq.com/s?__biz=MzkxNjI0ODk3Mg==&amp;mid=2247486022&amp;idx=1&amp;sn=a42dd2fba68205f83df3668edcdae179&amp;chksm=c1538aa5f62403b3ff0ccbcc3c1193540f8c55ba67da9741e8c96e11518c4a7c9f74e817411e&amp;scene=178&amp;cur_album_id=3025306541085982726#rd" target="_blank" rel="noopener noreferrer">（简洁点）monocle2实战 (qq.com)<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a></p> <h2 id="学习记录"><a href="#学习记录" class="header-anchor">#</a> 学习记录</h2> <p><strong>主要的流程概览</strong></p> <div class="language- line-numbers-mode"><pre class="language-text"><code># 创建对象 -- CellDataSet
pd &lt;- new(&quot;AnnotatedDataFrame&quot;, data = sample_sheet)
fd &lt;- new(&quot;AnnotatedDataFrame&quot;, data = gene_annotation)
cds &lt;- newCellDataSet(expr_matrix, phenoData = pd, featureData = fd)


# 根据已知标记基因鉴定细胞
cth &lt;- newCellTypeHierarchy()
MYF5_id &lt;- row.names(subset(fData(cds), gene_short_name == &quot;MYF5&quot;))
ANPEP_id &lt;- row.names(subset(fData(cds), gene_short_name == &quot;ANPEP&quot;))

cth &lt;- addCellType(cth, &quot;Myoblast&quot;, classify_func = function(x) { x[MYF5_id,] &gt;= 1 })
cth &lt;- addCellType(cth, &quot;Fibroblast&quot;, classify_func = function(x) { x[MYF5_id,] &lt; 1 &amp; x[ANPEP_id,] &gt; 1 } )

cds &lt;- classifyCells(cds, cth, 0.1)


# 聚类细胞
cds &lt;- clusterCells(cds)


# 将细胞按照伪时间规划在轨迹上（order cells in pseudotime along a trajectory）
disp_table &lt;- dispersionTable(cds)
ordering_genes &lt;- subset(disp_table, mean_expression &gt;= 0.1)
cds &lt;- setOrderingFilter(cds, ordering_genes)
cds &lt;- reduceDimension(cds)
cds &lt;- orderCells(cds)


# 进行差异表达分析
diff_test_res &lt;- differentialGeneTest(cds, fullModelFormulaStr = &quot;~Media&quot;)
sig_genes &lt;- subset(diff_test_res, qval &lt; 0.1)
</code></pre> <div class="line-numbers-wrapper"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br><span class="line-number">6</span><br><span class="line-number">7</span><br><span class="line-number">8</span><br><span class="line-number">9</span><br><span class="line-number">10</span><br><span class="line-number">11</span><br><span class="line-number">12</span><br><span class="line-number">13</span><br><span class="line-number">14</span><br><span class="line-number">15</span><br><span class="line-number">16</span><br><span class="line-number">17</span><br><span class="line-number">18</span><br><span class="line-number">19</span><br><span class="line-number">20</span><br><span class="line-number">21</span><br><span class="line-number">22</span><br><span class="line-number">23</span><br><span class="line-number">24</span><br><span class="line-number">25</span><br><span class="line-number">26</span><br><span class="line-number">27</span><br><span class="line-number">28</span><br><span class="line-number">29</span><br><span class="line-number">30</span><br><span class="line-number">31</span><br><span class="line-number">32</span><br></div></div><p><strong>开始尝试下载、运行</strong></p> <div class="language- line-numbers-mode"><pre class="language-text"><code># 下载（直接下载的包会常有报错，下面是修改过的）
wget -c https://github.com/cole-trapnell-lab/monocle-release/files/10134172/monocle_2.26.0.tar.gz

</code></pre> <div class="line-numbers-wrapper"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br></div></div><p>直接导入数据 -- <code>importCDS(pbmc)</code></p> <p>默认<code>import_all = FALSE</code>，只保留最小数据集；也可以设置参数为，<code>import_all = TRUE</code>，导入全部slot</p> <div class="language- line-numbers-mode"><pre class="language-text"><code>seuobj &lt;-  readRDS('02_anno_marker.rds')
seuobj &lt;- subset(seuobj,idents = c('mPP-Hep','mInt-Hep','mPC-Hep','mNASH-Hep1','mNASH-Hep2'))
</code></pre> <div class="line-numbers-wrapper"><span class="line-number">1</span><br><span class="line-number">2</span><br></div></div><p>构建所需的信息</p> <div class="language- line-numbers-mode"><pre class="language-text"><code># 表达矩阵 # 行-gene，列-barcodes
counts &lt;- as(as.matrix(seuobj@assays[[&quot;RNA&quot;]]@counts), 'sparseMatrix')
counts

# 后面分析运存压力太大，尝试随机选200个细胞进行分析
set.seed(123)
column_names &lt;- colnames(counts)
selected_cell &lt;- sample(column_names, 2000)
counts &lt;- counts[, selected_cell]

counts &lt;- as(as.matrix(counts),'sparseMatrix')

# 基因注释信息
gene_annotation &lt;- data.frame(gene_short_name = rownames(counts), 
                              biotype = rep(&quot;protein_coding&quot;, nrow(counts)))
rownames(gene_annotation)&lt;-rownames(counts)
gene_annotation

# 细胞注释信息
sample_sheet&lt;-as.data.frame(seuobj@meta.data)
sample_sheet&lt;-cbind(rownames(sample_sheet),sample_sheet)
colnames(sample_sheet)[1]&lt;-&quot;cell&quot;
sample_sheet&lt;-as.data.frame(t(subset(t(sample_sheet),select=colnames(counts))))
sample_sheet

# 创建monocle2对象
pd &lt;- new(&quot;AnnotatedDataFrame&quot;, data = sample_sheet)
fd &lt;- new(&quot;AnnotatedDataFrame&quot;, data = gene_annotation)
HSMM &lt;- newCellDataSet(counts,
                       phenoData = pd, 
                       featureData = fd,
                       expressionFamily = negbinomial.size())
</code></pre> <div class="line-numbers-wrapper"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br><span class="line-number">6</span><br><span class="line-number">7</span><br><span class="line-number">8</span><br><span class="line-number">9</span><br><span class="line-number">10</span><br><span class="line-number">11</span><br><span class="line-number">12</span><br><span class="line-number">13</span><br><span class="line-number">14</span><br><span class="line-number">15</span><br><span class="line-number">16</span><br><span class="line-number">17</span><br><span class="line-number">18</span><br><span class="line-number">19</span><br><span class="line-number">20</span><br><span class="line-number">21</span><br><span class="line-number">22</span><br><span class="line-number">23</span><br><span class="line-number">24</span><br><span class="line-number">25</span><br><span class="line-number">26</span><br><span class="line-number">27</span><br><span class="line-number">28</span><br><span class="line-number">29</span><br><span class="line-number">30</span><br><span class="line-number">31</span><br><span class="line-number">32</span><br></div></div><p><strong><code>expressionFamily，有这几个选项：negbinomial.size()，negbinomial()，tobit()，gaussianff()</code></strong></p> <ul><li>negbinomial.size()和negbinomial()：输入的表达矩阵为UMI,一般适用于10x的数据；</li> <li>negbinomial()的结果更准确，但是计算更耗时；一般建议采用negbinomial.size()；稀疏矩阵用negbinomial.size()</li> <li>tobit()：适用于输入的表达矩阵为FPKM或者TPM, 构建monocle2的class时会自动进行log化计算</li> <li>gaussianff()：输入为log化后的FPKM或者TPM。(目前在单细胞数据中，FPKM已不多用，smart-seq2平台数据一般采用TPM)</li></ul> <p>数据的结构格式</p> <p>counts<br> <img src="https://s2.loli.net/2024/04/21/Ac2iLTmR6jdp3Oh.png" alt="image.png|800"></p> <p>sample_sheet<br> <img src="https://s2.loli.net/2024/04/21/sjStEVMBmgHxvlF.png" alt="image.png|800"></p> <p>gene_annotation（name只能是<code>gene_short_name</code>）<br> <img src="https://s2.loli.net/2024/04/21/VCKLStzcvfpiBeN.png" alt="image.png|300"></p> <p>估计大小因子，离散程度</p> <div class="language- line-numbers-mode"><pre class="language-text"><code>cds &lt;- estimateSizeFactors(cds)
cds &lt;- estimateDispersions(cds)
</code></pre> <div class="line-numbers-wrapper"><span class="line-number">1</span><br><span class="line-number">2</span><br></div></div><p>SizeFactor，大小因子/尺度系数，用于标准化细胞间的mRNA差异；不同的样本/实验中，检测到的RNA表达量不一样，所以需要根据每个样本的总量，进行标准化，以便后续不同样本间可以比较。（感觉像是一个“总和”，描述数据的总量大小）；</p> <p>Dispersion，离散度值，用于后续差异分析；（像是一个“极值”，描述数据的范围，离散程度）</p> <p><strong>过滤掉低质量的细胞</strong></p> <div class="language- line-numbers-mode"><pre class="language-text"><code>cds &lt;- detectGenes(cds, min_expr = 0.1) #这一操作会在fData(cds)中添加一列num_cells_expressed

print(head(fData(cds)))
expressed_genes &lt;- row.names(subset(fData(cds),
    num_cells_expressed &gt;= 10)) #过滤掉在小于10个细胞中表达的基因，还剩11095个基因。
</code></pre> <div class="line-numbers-wrapper"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br></div></div><p>接着是<strong>选择定义轨迹的基因，可视化轨迹基因，构建轨迹，可视化轨迹</strong></p> <p>主要步骤是：选择基因，降维数据，伪时间上排序细胞</p> <p>推断单细胞轨迹是一个机器学习的问题。同时对于scRNAseq，数据比较稀疏，数据会很<code>嘈杂</code>，但是不同的基因表达量又包含有关细胞状态的信息，所以需要和研究对象分化轨迹非常相关的基因，这一步的选择会对轨迹的形状有较大的影响。</p> <p>monocle主要是基于<code>关键基因</code>的表达模式，通过学习每个细胞必须经历的基因表达变化的序列，根据模拟时间值，对细胞进行排序，模拟出时间发育过程的动态变化。这个技术是一种在低维空间排布高维数据的降维技术。</p> <p>monocle提供了下面几种选择<code>关键基因</code>的方法</p> <ul><li>选择发育差异表达基因</li> <li>选择分群差异表达基因</li> <li>选择离散程度高的基因</li> <li>自己定义发育标记基因（根据先验知识）</li></ul> <blockquote><p>ps：前三个完全由计算得出，细胞发育轨迹不受人为干预，所以叫无监督分析方法；最后一种人为使用了先验的知识辅助分析，所以叫半监督的方法； -- 监督/半监督，听着本以为很高级...</p></blockquote> <div class="language- line-numbers-mode"><pre class="language-text"><code># 使用seurat 的 variablefeatures 选择的高度变化的基因
ordergene &lt;- VariableFeatures(pbmc)


# 使用seurat 的 findallmarkers 选择不同 clusters之间的差异表达基因
deg.cluster &lt;- FindAllMarkers(pbmc)
ordergene &lt;- subset(deg.cluster,p_val_adj&lt;0.05)$gene


# 使用monocle选择的高变基因
disp_table &lt;- dispersionTable(cds)
ordergene &lt;- subset(disp_table, mean_expression &gt;= 0.1 &amp; dispersion_empirical &gt;= 1 * dispersion_fit)$gene_id


# monocle自己算的轨迹基因（无先验经验，或是为了防止引入偏见）
diff &lt;- differentialGeneTest(cds[expressed_genes,],fullModelFormulaStr=&quot;~cell_type&quot;,cores=1)
deg &lt;- subset(diff, qval &lt; 0.01)
deg &lt;- deg[order(deg$qval, decreasing = F), ]
ordergene &lt;- rownames(deg)


cds &lt;- setOrderingFilter(cds, ordergene) # 把差异基因嵌入数据中
plot_ordering_genes(cds) # 看一看定义轨迹分化的基因
</code></pre> <div class="line-numbers-wrapper"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br><span class="line-number">6</span><br><span class="line-number">7</span><br><span class="line-number">8</span><br><span class="line-number">9</span><br><span class="line-number">10</span><br><span class="line-number">11</span><br><span class="line-number">12</span><br><span class="line-number">13</span><br><span class="line-number">14</span><br><span class="line-number">15</span><br><span class="line-number">16</span><br><span class="line-number">17</span><br><span class="line-number">18</span><br><span class="line-number">19</span><br><span class="line-number">20</span><br><span class="line-number">21</span><br><span class="line-number">22</span><br><span class="line-number">23</span><br></div></div><p>选择用于排序的基因数据一般在2000左右比较合适，太多的话可以选择最靠前的，<code>ordergene &lt;- row.names(deg)[order(deg$qval)][1:400]</code></p> <p>接着数据需要<strong>降维</strong>，将数据从高维投射到低维空间中，相当于“将一颗树上的树干，树枝，叶子，画在纸上”</p> <div class="language- line-numbers-mode"><pre class="language-text"><code>cds &lt;- reduceDimension(cds, max_components = 2, method = 'DDRTree')
</code></pre> <div class="line-numbers-wrapper"><span class="line-number">1</span><br></div></div><p>将细胞在二维的图形，沿着逆时间轨迹排列（可以选择一个为<code>根</code>）</p> <div class="language- line-numbers-mode"><pre class="language-text"><code>cds &lt;- order(cds)
cds &lt;- orderCells(cds, root_state = 5)
</code></pre> <div class="line-numbers-wrapper"><span class="line-number">1</span><br><span class="line-number">2</span><br></div></div><p>下面就是几种可视化效果了</p> <p>根据cds@phenoData@data中的表型信息(metadata)对细胞上色</p> <p>绘制<strong>分化轨迹图</strong></p> <div class="language- line-numbers-mode"><pre class="language-text"><code>plot_cell_trajectory(cds, color_by = &quot;Pseudotime&quot;, size=1, show_backbone=TRUE)
plot_cell_trajectory(cds, color_by = &quot;cell_type&quot;, size=1, show_backbone=TRUE)
plot_cell_trajectory(cds, color_by = &quot;State&quot;,size=1, show_backbone=TRUE)
plot_cell_trajectory(cds, color_by = &quot;seurat_clusters&quot;)

plot_cell_trajectory(cds, color_by = &quot;State&quot;) + 
	facet_wrap(&quot;~State&quot;, nrow = 1) # 根据细胞状态上色的同时，以细胞状态拆分图 
</code></pre> <div class="line-numbers-wrapper"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br><span class="line-number">6</span><br><span class="line-number">7</span><br></div></div><p>绘制<strong>树形分化图</strong></p> <div class="language- line-numbers-mode"><pre class="language-text"><code>p1 &lt;- plot_cell_trajectory(cds, x = 1, y = 2, color_by = &quot;celltype&quot;)

p2 &lt;- plot_complex_cell_trajectory(cds, x = 1, y = 2, color_by = &quot;celltype&quot;)

p1 | p2
</code></pre> <div class="line-numbers-wrapper"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br></div></div><p>还可以沿着时间轴绘制<strong>细胞密度图</strong></p> <div class="language- line-numbers-mode"><pre class="language-text"><code>library(ggpubr)

df &lt;- pData(cds) 
ggplot(df, aes(Pseudotime, colour = cell_type, fill=cell_type)) +
  geom_density(bw=0.5,size=1,alpha = 0.5)+theme_classic2()


# 手动设置颜色的方法
ClusterName_color_panel &lt;- c(
  &quot;Naive CD4 T&quot; = &quot;#DC143C&quot;, &quot;Memory CD4 T&quot; = &quot;#0000FF&quot;, &quot;CD14+ Mono&quot; = &quot;#20B2AA&quot;,
  &quot;B&quot; = &quot;#FFA500&quot;, &quot;CD8 T&quot; = &quot;#9370DB&quot;, &quot;FCGR3A+ Mono&quot; = &quot;#98FB98&quot;,
  &quot;NK&quot; = &quot;#F08080&quot;, &quot;DC&quot; = &quot;#0000FF&quot;, &quot;Platelet&quot; = &quot;#20B2AA&quot;
)
ggplot(df, aes(Pseudotime, colour = cell_type, fill=cell_type)) +
  geom_density(bw=0.5,size=1,alpha = 0.5)+theme_classic2()+ 
  scale_fill_manual(name = &quot;&quot;, values = ClusterName_color_panel)+
  scale_color_manual(name = &quot;&quot;, values = ClusterName_color_panel)
</code></pre> <div class="line-numbers-wrapper"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br><span class="line-number">6</span><br><span class="line-number">7</span><br><span class="line-number">8</span><br><span class="line-number">9</span><br><span class="line-number">10</span><br><span class="line-number">11</span><br><span class="line-number">12</span><br><span class="line-number">13</span><br><span class="line-number">14</span><br><span class="line-number">15</span><br><span class="line-number">16</span><br><span class="line-number">17</span><br></div></div><p>绘制分化轨迹中的<strong>基因表达变化</strong></p> <div class="language- line-numbers-mode"><pre class="language-text"><code>library(patchwork) # 拼图用的


keygenes &lt;- head(ordergene,3) # 一组基因名，还可以设置自己感兴趣的
sub_cds &lt;- HSMM_myo[keygenes,]


p1 &lt;- plot_genes_in_pseudotime(cds_subset = sub_cds, color_by = &quot;State&quot;)
p2 &lt;- plot_genes_in_pseudotime(cds_subset = sub_cds, color_by = &quot;celltype&quot;)
p3 &lt;- plot_genes_in_pseudotime(cds_subset = sub_cds, color_by = &quot;Pseudotime&quot;)
plotc &lt;- p1|p2|p3
ggsave(&quot;Genes_pseudotimeplot.png&quot;, plot = plotc, width = 16, height = 8)


p1 &lt;- plot_genes_jitter(sub_cds, grouping = &quot;State&quot;, color_by = &quot;State&quot;)
p2 &lt;- plot_genes_violin(sub_cds, grouping = &quot;State&quot;, color_by = &quot;State&quot;)
p3 &lt;- plot_genes_in_pseudotime(sub_cds, color_by = &quot;State&quot;)
plotc &lt;- p1|p2|p3
ggsave(&quot;Genes_Jitterplot.png&quot;, plot = plotc, width = 16, height = 8)
</code></pre> <div class="line-numbers-wrapper"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br><span class="line-number">6</span><br><span class="line-number">7</span><br><span class="line-number">8</span><br><span class="line-number">9</span><br><span class="line-number">10</span><br><span class="line-number">11</span><br><span class="line-number">12</span><br><span class="line-number">13</span><br><span class="line-number">14</span><br><span class="line-number">15</span><br><span class="line-number">16</span><br><span class="line-number">17</span><br><span class="line-number">18</span><br><span class="line-number">19</span><br></div></div><p>展示单个基因的<strong>表达量随着轨迹的变化</strong></p> <div class="language- line-numbers-mode"><pre class="language-text"><code>colnames(pData(cds))

pData(cds)$CCL5 = log2( exprs(cds)['CCL5',]+1)
p1=plot_cell_trajectory(cds, color_by = &quot;CCL5&quot;)  + scale_color_gsea()

pData(cds)$S100A4 = log2(exprs(cds)['S100A4',]+1)
p2=plot_cell_trajectory(cds, color_by = &quot;S100A4&quot;)    + scale_color_gsea()

library(patchwork)
p1+p2
</code></pre> <div class="line-numbers-wrapper"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br><span class="line-number">6</span><br><span class="line-number">7</span><br><span class="line-number">8</span><br><span class="line-number">9</span><br><span class="line-number">10</span><br></div></div><p>寻找拟时相关的基因/差异基因</p> <p>三种寻找差异基因的方法：</p> <ol><li>basic differential analysis</li> <li>finding genes that distinguish cell type or state</li> <li>finding genes that change as a function of pseudotime</li></ol> <p>基础差异分析；区别细胞类型或状态的基因；改变轨迹变化的基因</p> <p><code>sm.ns</code>函数指出Monocle应该通过表达式值拟合自然样条曲线，以帮助它将表达式的变化描述为进程的函数。</p> <div class="language- line-numbers-mode"><pre class="language-text"><code>#这里是把排序基因（ordergene）提取出来做回归分析，来找它们是否跟拟时间有显著的关系；如果不设置，就会用所有基因来做它们与拟时间的相关性

Time_diff &lt;- differentialGeneTest(cds[ordergene,], 
	cores = 1, 
	fullModelFormulaStr = &quot;~sm.ns(Pseudotime)&quot;)
Time_diff &lt;- Time_diff[,c(5,2,3,4,1,6,7)] #把gene放前面，也可以不改

Time_genes &lt;- Time_diff %&gt;% pull(gene_short_name) %&gt;% as.character()
p=plot_pseudotime_heatmap(cds[Time_genes,], 
	num_clusters=4, 
	show_rownames=T, 
	return_heatmap=T)
ggsave(&quot;Time_heatmapAll.pdf&quot;, p, width = 5, height = 10)
</code></pre> <div class="line-numbers-wrapper"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br><span class="line-number">6</span><br><span class="line-number">7</span><br><span class="line-number">8</span><br><span class="line-number">9</span><br><span class="line-number">10</span><br><span class="line-number">11</span><br><span class="line-number">12</span><br><span class="line-number">13</span><br></div></div><p>这个p，我一直以为是个图像的对象，结果还能取出来单独分析</p> <div class="language- line-numbers-mode"><pre class="language-text"><code>p$tree_row
clusters &lt;- cutree(p$tree_row, k = 4)
clustering &lt;- data.frame(clusters)
clustering[,1] &lt;- as.character(clustering[,1])
colnames(clustering) &lt;- &quot;Gene_Clusters&quot;
</code></pre> <div class="line-numbers-wrapper"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br></div></div><p>可以把差异基因拿出来，去做富集分析哦！</p> <p>来看看差异基因的热图</p> <div class="language- line-numbers-mode"><pre class="language-text"><code>Time_genes &lt;- top_n(Time_diff, n = 100, desc(qval)) %&gt;% pull(gene_short_name) %&gt;% as.character()
p = plot_pseudotime_heatmap(cds[Time_genes,], num_clusters=4, show_rownames=T, return_heatmap=T)
ggsave(&quot;Time_heatmapTop100.pdf&quot;, p, width = 5, height = 10)
</code></pre> <div class="line-numbers-wrapper"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br></div></div><p>显著基因的排序</p> <div class="language- line-numbers-mode"><pre class="language-text"><code>hp.genes &lt;- p$tree_row$labels[p$tree_row$order]
Time_diff_sig &lt;- Time_diff[hp.genes, c(&quot;gene_short_name&quot;, &quot;pval&quot;, &quot;qval&quot;)]
write.csv(Time_diff_sig, &quot;Time_diff_sig.csv&quot;, row.names = F)

marker_genes &lt;- row.names(subset(fData(cds),
                                 gene_short_name %in% c(&quot;MEF2C&quot;, &quot;MEF2D&quot;, &quot;MYF5&quot;,
                                                        &quot;ANPEP&quot;, &quot;PDGFRA&quot;,&quot;MYOG&quot;,
                                                        &quot;TPM1&quot;,  &quot;TPM2&quot;,  &quot;MYH2&quot;,
                                                        &quot;MYH3&quot;,  &quot;NCAM1&quot;, &quot;TNNT1&quot;,
                                                        &quot;TNNT2&quot;, &quot;TNNC1&quot;, &quot;CDK1&quot;,
                                                        &quot;CDK2&quot;,  &quot;CCNB1&quot;, &quot;CCNB2&quot;,
                                                        &quot;CCND1&quot;, &quot;CCNA1&quot;, &quot;ID1&quot;)))

diff_test_res &lt;- differentialGeneTest(cds[marker_genes,],
                                      fullModelFormulaStr = &quot;~sm.ns(Pseudotime)&quot;)
sig_gene_names &lt;- row.names(subset(diff_test_res, qval &lt; 0.1))
plot_pseudotime_heatmap(cds[sig_gene_names,],
                        num_clusters = 6,
                        cores = 1,
                        show_rownames = T)
</code></pre> <div class="line-numbers-wrapper"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br><span class="line-number">6</span><br><span class="line-number">7</span><br><span class="line-number">8</span><br><span class="line-number">9</span><br><span class="line-number">10</span><br><span class="line-number">11</span><br><span class="line-number">12</span><br><span class="line-number">13</span><br><span class="line-number">14</span><br><span class="line-number">15</span><br><span class="line-number">16</span><br><span class="line-number">17</span><br><span class="line-number">18</span><br><span class="line-number">19</span><br><span class="line-number">20</span><br></div></div><p>针对单细胞的<strong>轨迹分支</strong>，进行分析</p> <p>单细胞轨迹常常包括分支，这些分支的产生是因为细胞执行不同的基因表达程序。monocle2提供一个统计方法：<strong>分支表达式分析建模</strong> -- BEAM，branched expression analysis modeling，用于寻找以依赖于分支方式调控的基因</p> <div class="language- line-numbers-mode"><pre class="language-text"><code>BEAM_res &lt;- BEAM(cds[ordergene,], branch_point = 1, cores = 2) 
#BEAM_res &lt;- BEAM(cds, branch_point = 1, cores = 2) #对全部基因进行排序，运行较慢

BEAM_res &lt;- BEAM_res[order(BEAM_res$qval),]
BEAM_res &lt;- BEAM_res[,c(&quot;gene_short_name&quot;, &quot;pval&quot;, &quot;qval&quot;)]
head(BEAM_res)

plot_genes_branched_heatmap(cds[row.names(subset(BEAM_res,
                                                  qval &lt; 1e-4)),],
                            branch_point = 1, #绘制的是哪个分支
                            num_clusters = 4, #分成几个cluster，根据需要调整
                            cores = 1,
                            use_gene_short_name = T,
                            show_rownames = T)#有632个gene，太多了



#选前100个基因可视化
BEAM_genes &lt;- top_n(BEAM_res, n = 100, desc(qval)) %&gt;% pull(gene_short_name) %&gt;% as.character()
p &lt;- plot_genes_branched_heatmap(cds[BEAM_genes,],  branch_point = 1, 
                                 num_clusters = 3, show_rownames = T, return_heatmap = T)
ggsave(&quot;BEAM_heatmap.pdf&quot;, p$ph_res, width = 6.5, height = 10)

</code></pre> <div class="line-numbers-wrapper"><span class="line-number">1</span><br><span class="line-number">2</span><br><span class="line-number">3</span><br><span class="line-number">4</span><br><span class="line-number">5</span><br><span class="line-number">6</span><br><span class="line-number">7</span><br><span class="line-number">8</span><br><span class="line-number">9</span><br><span class="line-number">10</span><br><span class="line-number">11</span><br><span class="line-number">12</span><br><span class="line-number">13</span><br><span class="line-number">14</span><br><span class="line-number">15</span><br><span class="line-number">16</span><br><span class="line-number">17</span><br><span class="line-number">18</span><br><span class="line-number">19</span><br><span class="line-number">20</span><br><span class="line-number">21</span><br><span class="line-number">22</span><br><span class="line-number">23</span><br></div></div></div></div>  <div class="page-edit"><!----> <div class="tags"><a href="/tags/?tag=%E5%8D%95%E7%BB%86%E8%83%9E" title="标签">#单细胞</a><a href="/tags/?tag=%E6%95%99%E7%A8%8B" title="标签">#教程</a></div> <!----></div> <div class="page-nav-wapper"><div class="page-nav-centre-wrap"><a href="/pages/784c69/" class="page-nav-centre page-nav-centre-prev"><div class="tooltip">Scanpy</div></a> <a href="/pages/893dcf/" class="page-nav-centre page-nav-centre-next"><div class="tooltip">单细胞-轨迹分析</div></a></div> <div class="page-nav"><p class="inner"><span class="prev">
        ←
        <a href="/pages/784c69/" class="prev">Scanpy</a></span> <span class="next"><a href="/pages/893dcf/">单细胞-轨迹分析</a>→
      </span></p></div></div></div> <div class="article-list"><div class="article-title"><a href="/archives/" class="iconfont icon-bi">最近更新</a></div> <div class="article-wrapper"><dl><dd>01</dd> <dt><a href="/pages/90dbe8/"><div>
            APA相关综述
            <!----></div></a> <span class="date">06-27</span></dt></dl><dl><dd>02</dd> <dt><a href="/pages/f4f4b7/"><div>
            胃癌免疫逃逸的决定因素
            <!----></div></a> <span class="date">06-25</span></dt></dl><dl><dd>03</dd> <dt><a href="/pages/0a11dd/"><div>
            胖子CAF
            <!----></div></a> <span class="date">06-21</span></dt></dl> <dl><dd></dd> <dt><a href="/archives/" class="more">更多文章&gt;</a></dt></dl></div></div></main></div> <div class="footer"><div class="icons"><a href="mailto:2590643635@qq.com" title="发邮件" target="_blank" class="iconfont icon-youjian"></a><a href="https://github.com/hongliangfang" title="GitHub" target="_blank" class="iconfont icon-github"></a><a href="https://music.163.com/#/playlist?id=755597173" title="听音乐" target="_blank" class="iconfont icon-erji"></a></div> 
  Theme by
  <a href="https://github.com/xugaoyi/vuepress-theme-vdoing" target="_blank" title="本站主题">Vdoing</a> 
    | Copyright © 2023-2024
    <span>FHL | <a href="https://github.com/xugaoyi/vuepress-theme-vdoing/blob/master/LICENSE" target="_blank">MIT License</a></span></div> <div class="buttons"><div title="返回顶部" class="button blur go-to-top iconfont icon-fanhuidingbu" style="display:none;"></div> <div title="去评论" class="button blur go-to-comment iconfont icon-pinglun" style="display:none;"></div> <div title="主题模式" class="button blur theme-mode-but iconfont icon-zhuti"><ul class="select-box" style="display:none;"><li class="iconfont icon-zidong">
          跟随系统
        </li><li class="iconfont icon-rijianmoshi">
          浅色模式
        </li><li class="iconfont icon-yejianmoshi">
          深色模式
        </li><li class="iconfont icon-yuedu">
          阅读模式
        </li></ul></div></div> <!----> <!----> <!----></div><div class="global-ui"><div></div></div></div>
    <script src="/assets/js/app.bd488e07.js" defer></script><script src="/assets/js/2.bdf5b8cf.js" defer></script><script src="/assets/js/53.4f209254.js" defer></script><script src="/assets/js/3.365ac406.js" defer></script>
  </body>
</html>
